<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.42">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Shane Hogle">
<meta name="dcterms.date" content="2025-07-04">

<title>Calibrate boilprep to Qiagen DNeasy â€“ hambiEvoEnvCoexist</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../../../../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../../../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../../../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../../../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../../../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../../../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../../../../">
<script src="../../../../../site_libs/quarto-html/quarto.js"></script>
<script src="../../../../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../../../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../../../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../../../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../../../../site_libs/quarto-html/quarto-syntax-highlighting-2f5df379a58b258e96c21c0638c20c03.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../../../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../../../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../../../../site_libs/bootstrap/bootstrap-1f23bf93a17969024013c136d185e5ce.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<link href="../../../../../site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet">
<script src="../../../../../site_libs/pagedtable-1.1/js/pagedtable.js"></script>


</head>

<body class="nav-sidebar docked">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item">
      hambiEvoEnvCoexist
      </li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation docked overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="../../../../../">hambiEvoEnvCoexist</a> 
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true">
 <span class="menu-text">1. Monoculture growth</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../../R/monocultures/bioscreen/01_format_growthcurves_batch.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">i) Inspect and format growth curves</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../../R/monocultures/bioscreen/02_plot_analyze_growth.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">ii) Plot growth summaries</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true">
 <span class="menu-text">2. Community growth - experiment design</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../../R/communities/experiment_design/01_pairs.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">i) Species pairs - 96 Well plate layout</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../../R/communities/experiment_design/02_trios.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">ii) Species trios - 96 Well plate layout</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../../R/communities/experiment_design/03_quartets.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">iii) Species quartets - 96 Well plate layout</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true">
 <span class="menu-text">3. Community growth - biomass</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../../R/communities/community_biomass/optical_density/01_format_plot_od.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">i) Process and plot OD data</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="true">
 <span class="menu-text">4. Community growth - amplicon formatting</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../../R/communities/community_composition/amplicon/20240711_BTK_illumina_v3/01_format_rbec_tab.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">i) Batch 2024-07-11</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../../R/communities/community_composition/amplicon/20241024_BTK_illumina_v3/01_format_rbec_tab.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">iii) Batch 2024-10-24</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../../R/communities/community_composition/amplicon/20250127_BTK_illumina_v3/01_format_rbec_tab.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">iv) Batch 2025-01-27</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../../R/communities/community_composition/amplicon/20250502_BTK_illumina_v3/01_format_rbec_tab.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">v) Batch 2025-05-02</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../../R/communities/community_composition/amplicon/01_compile_format.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">vi) Compile pairs, trios, quartets</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" role="navigation" aria-expanded="true">
 <span class="menu-text">5. Community growth - composition</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-5" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../../R/communities/community_composition/01_analysis_pairs.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">i) Species pairs</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../../R/communities/community_composition/02_analysis_trios.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">ii) Species trios</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../../R/communities/community_composition/03_analysis_quartets.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">iii) Species quartets</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../../R/communities/community_composition/04_analysis_octets.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">iv) All strains combined</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../../R/communities/community_composition/05_assembly_rule_violation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">v) Violation of pairwise assembly rule</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#setup" id="toc-setup" class="nav-link active" data-scroll-target="#setup"><span class="header-section-number">1</span> Setup</a>
  <ul class="collapse">
  <li><a href="#libraries" id="toc-libraries" class="nav-link" data-scroll-target="#libraries"><span class="header-section-number">1.1</span> Libraries</a></li>
  <li><a href="#global-variables" id="toc-global-variables" class="nav-link" data-scroll-target="#global-variables"><span class="header-section-number">1.2</span> Global variables</a></li>
  </ul></li>
  <li><a href="#read-data" id="toc-read-data" class="nav-link" data-scroll-target="#read-data"><span class="header-section-number">2</span> Read data</a></li>
  <li><a href="#formatting" id="toc-formatting" class="nav-link" data-scroll-target="#formatting"><span class="header-section-number">3</span> Formatting</a>
  <ul class="collapse">
  <li><a href="#format-positive-control-samples" id="toc-format-positive-control-samples" class="nav-link" data-scroll-target="#format-positive-control-samples"><span class="header-section-number">3.1</span> Format positive control samples</a></li>
  <li><a href="#plot-masterplate-qiagen-and-boil-prep-side-by-side" id="toc-plot-masterplate-qiagen-and-boil-prep-side-by-side" class="nav-link" data-scroll-target="#plot-masterplate-qiagen-and-boil-prep-side-by-side"><span class="header-section-number">3.2</span> Plot masterplate Qiagen and boil-prep side by side</a></li>
  <li><a href="#matrix-of-observed-counts" id="toc-matrix-of-observed-counts" class="nav-link" data-scroll-target="#matrix-of-observed-counts"><span class="header-section-number">3.3</span> Matrix of observed counts</a></li>
  <li><a href="#make-a-matrix-of-true-proportions" id="toc-make-a-matrix-of-true-proportions" class="nav-link" data-scroll-target="#make-a-matrix-of-true-proportions"><span class="header-section-number">3.4</span> Make a matrix of true proportions</a></li>
  </ul></li>
  <li><a href="#metacal-procedure" id="toc-metacal-procedure" class="nav-link" data-scroll-target="#metacal-procedure"><span class="header-section-number">4</span> Metacal procedure</a>
  <ul class="collapse">
  <li><a href="#estimate-bias" id="toc-estimate-bias" class="nav-link" data-scroll-target="#estimate-bias"><span class="header-section-number">4.1</span> Estimate bias</a>
  <ul class="collapse">
  <li><a href="#plot-bias-estimation" id="toc-plot-bias-estimation" class="nav-link" data-scroll-target="#plot-bias-estimation"><span class="header-section-number">4.1.1</span> Plot bias estimation</a></li>
  <li><a href="#plot-metacal-model-fit" id="toc-plot-metacal-model-fit" class="nav-link" data-scroll-target="#plot-metacal-model-fit"><span class="header-section-number">4.1.2</span> Plot metacal model fit</a></li>
  </ul></li>
  <li><a href="#calibrate" id="toc-calibrate" class="nav-link" data-scroll-target="#calibrate"><span class="header-section-number">4.2</span> Calibrate</a></li>
  <li><a href="#run-the-calibrate-function" id="toc-run-the-calibrate-function" class="nav-link" data-scroll-target="#run-the-calibrate-function"><span class="header-section-number">4.3</span> Run the calibrate function</a></li>
  </ul></li>
  <li><a href="#export" id="toc-export" class="nav-link" data-scroll-target="#export"><span class="header-section-number">5</span> Export</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">Calibrate boilprep to Qiagen DNeasy</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Shane Hogle </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">July 4, 2025</p>
    </div>
  </div>
  
    
  </div>
  
<div>
  <div class="abstract">
    <div class="block-title">Abstract</div>
    We will use the <a href="https://mikemc.github.io/metacal/">metacal package</a> for estimating bias and performing calibration in the special case where the bias of all the taxa of interest can be directly measured from the control sample. Since samples extracted by Qiagen and by boil are exactly the same we can estimate scaling factors to produce corrected relative abundance for other samples. Publication: <a href="https://elifesciences.org/articles/46923">Consistent and correctable bias in metagenomic sequencing experiments</a>
  </div>
</div>


</header>


<section id="setup" class="level1" data-number="1">
<h1 data-number="1"><span class="header-section-number">1</span> Setup</h1>
<section id="libraries" class="level2" data-number="1.1">
<h2 data-number="1.1" class="anchored" data-anchor-id="libraries"><span class="header-section-number">1.1</span> Libraries</h2>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="fu">library</span>(here)</span>
<span id="cb1-2"><a href="#cb1-2"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-3"><a href="#cb1-3"></a><span class="fu">library</span>(metacal)</span>
<span id="cb1-4"><a href="#cb1-4"></a><span class="fu">library</span>(scales)</span>
<span id="cb1-5"><a href="#cb1-5"></a><span class="fu">source</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"R"</span>, <span class="st">"utils_generic.R"</span>))</span>
<span id="cb1-6"><a href="#cb1-6"></a><span class="fu">source</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"R"</span>, <span class="st">"communities"</span>, <span class="st">"community_composition"</span>, <span class="st">"amplicon"</span>, <span class="st">"utils_amplicon.R"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="global-variables" class="level2" data-number="1.2">
<h2 data-number="1.2" class="anchored" data-anchor-id="global-variables"><span class="header-section-number">1.2</span> Global variables</h2>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a>data_raw <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"_data_raw"</span>, <span class="st">"communities"</span>, <span class="st">"20240711_BTK_illumina_v3"</span>)</span>
<span id="cb2-2"><a href="#cb2-2"></a>data <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"communities"</span>, <span class="st">"20240711_BTK_illumina_v3"</span>)</span>
<span id="cb2-3"><a href="#cb2-3"></a></span>
<span id="cb2-4"><a href="#cb2-4"></a><span class="co"># make processed data directory if it doesn't exist</span></span>
<span id="cb2-5"><a href="#cb2-5"></a>fs<span class="sc">::</span><span class="fu">dir_create</span>(data)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</section>
<section id="read-data" class="level1" data-number="2">
<h1 data-number="2"><span class="header-section-number">2</span> Read data</h1>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1"></a>pos_ctrl <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(data, <span class="st">"pos_neg_ctrl_counts.tsv"</span>))</span>
<span id="cb3-2"><a href="#cb3-2"></a>samp_pairs <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(data, <span class="st">"pairs_counts.tsv"</span>))</span>
<span id="cb3-3"><a href="#cb3-3"></a>samp_trios <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(data, <span class="st">"trios_counts.tsv"</span>))</span>
<span id="cb3-4"><a href="#cb3-4"></a></span>
<span id="cb3-5"><a href="#cb3-5"></a><span class="co"># metadata</span></span>
<span id="cb3-6"><a href="#cb3-6"></a>mddf <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(data_raw, <span class="st">"20240711_BTK_illumina_v3_metadata.tsv"</span>))</span>
<span id="cb3-7"><a href="#cb3-7"></a>spdf <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(data_raw, <span class="st">"sample_compositions.tsv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="formatting" class="level1" data-number="3">
<h1 data-number="3"><span class="header-section-number">3</span> Formatting</h1>
<section id="format-positive-control-samples" class="level2" data-number="3.1">
<h2 data-number="3.1" class="anchored" data-anchor-id="format-positive-control-samples"><span class="header-section-number">3.1</span> Format positive control samples</h2>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a>pos_ctrl <span class="ot">&lt;-</span> <span class="fu">left_join</span>(pos_ctrl, spdf) <span class="sc">%&gt;%</span> </span>
<span id="cb4-2"><a href="#cb4-2"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(community_type, <span class="st">"^pos"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb4-3"><a href="#cb4-3"></a>  <span class="co"># remove noise from species that aren't really there</span></span>
<span id="cb4-4"><a href="#cb4-4"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(evo_hist)) <span class="sc">%&gt;%</span> </span>
<span id="cb4-5"><a href="#cb4-5"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb4-6"><a href="#cb4-6"></a>  <span class="fu">mutate</span>(<span class="at">f_qg =</span> count_correct<span class="sc">/</span><span class="fu">sum</span>(count_correct)) <span class="sc">%&gt;%</span> </span>
<span id="cb4-7"><a href="#cb4-7"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(sample, strainID, community_id, count_correct, f_qg) <span class="sc">%&gt;%</span> </span>
<span id="cb4-8"><a href="#cb4-8"></a>  <span class="fu">ungroup</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Joining with `by = join_by(strainID, community_id)`</code></pre>
</div>
</div>
</section>
<section id="plot-masterplate-qiagen-and-boil-prep-side-by-side" class="level2" data-number="3.2">
<h2 data-number="3.2" class="anchored" data-anchor-id="plot-masterplate-qiagen-and-boil-prep-side-by-side"><span class="header-section-number">3.2</span> Plot masterplate Qiagen and boil-prep side by side</h2>
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1"></a><span class="fu">left_join</span>(samp_trios, mddf,</span>
<span id="cb6-2"><a href="#cb6-2"></a>          <span class="at">by =</span> <span class="fu">join_by</span>(sample, replicate, strep_conc, transfers, n_species, community_type, plate_well)) <span class="sc">%&gt;%</span> </span>
<span id="cb6-3"><a href="#cb6-3"></a>  <span class="fu">filter</span>(community_type <span class="sc">==</span> <span class="st">"masterplate"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb6-4"><a href="#cb6-4"></a>  <span class="fu">right_join</span>(pos_ctrl, <span class="at">by =</span> <span class="fu">join_by</span>(strainID, community_id)) <span class="sc">%&gt;%</span> </span>
<span id="cb6-5"><a href="#cb6-5"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(community_id, strainID, f_thresh, f_qg) <span class="sc">%&gt;%</span> </span>
<span id="cb6-6"><a href="#cb6-6"></a>  <span class="fu">pivot_longer</span>(<span class="fu">c</span>(f_thresh, f_qg)) <span class="sc">%&gt;%</span> </span>
<span id="cb6-7"><a href="#cb6-7"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb6-8"><a href="#cb6-8"></a>    <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">y =</span> value, <span class="at">x=</span><span class="fu">interaction</span>(name, community_id), <span class="at">fill =</span> strainID), </span>
<span id="cb6-9"><a href="#cb6-9"></a>             <span class="at">color=</span><span class="st">"black"</span>,</span>
<span id="cb6-10"><a href="#cb6-10"></a>             <span class="at">linewidth=</span><span class="fl">0.25</span>, <span class="at">stat=</span><span class="st">"identity"</span>) <span class="sc">+</span></span>
<span id="cb6-11"><a href="#cb6-11"></a>    <span class="fu">scale_y_continuous</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>), <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>), <span class="at">labels =</span> percent) <span class="sc">+</span></span>
<span id="cb6-12"><a href="#cb6-12"></a>    <span class="fu">scale_x_discrete</span>(<span class="at">guide =</span> <span class="fu">guide_axis</span>(<span class="at">angle =</span> <span class="dv">90</span>)) <span class="sc">+</span></span>
<span id="cb6-13"><a href="#cb6-13"></a>    <span class="fu">labs</span>(<span class="at">x=</span><span class="st">""</span>, <span class="at">y=</span><span class="st">"% abundance"</span>) <span class="sc">+</span></span>
<span id="cb6-14"><a href="#cb6-14"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span> </span>
<span id="cb6-15"><a href="#cb6-15"></a>  <span class="fu">theme</span>(<span class="at">panel.grid =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb6-16"><a href="#cb6-16"></a>        <span class="at">legend.position =</span> <span class="st">"bottom"</span>) <span class="sc">+</span></span>
<span id="cb6-17"><a href="#cb6-17"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> hambi_colors)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div id="fig-01" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-01-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="02_correct_boil_to_qiagen_files/figure-html/unnamed-chunk-5-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-01-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;1: Bar plot of masterplate samples extracted by the boilprep method (f) or the Qiagen DNeasy kit (qg). The community trios are appended to the extraction method on the x axs
</figcaption>
</figure>
</div>
</section>
<section id="matrix-of-observed-counts" class="level2" data-number="3.3">
<h2 data-number="3.3" class="anchored" data-anchor-id="matrix-of-observed-counts"><span class="header-section-number">3.3</span> Matrix of observed counts</h2>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1"></a>samp_trios_counts <span class="ot">&lt;-</span> <span class="fu">left_join</span>(samp_trios, mddf,</span>
<span id="cb7-2"><a href="#cb7-2"></a>                               <span class="at">by =</span> <span class="fu">join_by</span>(sample, replicate, strep_conc, </span>
<span id="cb7-3"><a href="#cb7-3"></a>                                            transfers, n_species, community_type, plate_well)) <span class="sc">%&gt;%</span> </span>
<span id="cb7-4"><a href="#cb7-4"></a>  <span class="fu">filter</span>(community_type <span class="sc">==</span> <span class="st">"masterplate"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb7-5"><a href="#cb7-5"></a>  <span class="fu">filter</span>(community_id <span class="sc">%in%</span> <span class="fu">pull</span>(pos_ctrl, community_id)) <span class="sc">%&gt;%</span> </span>
<span id="cb7-6"><a href="#cb7-6"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(community_id, strainID, count_correct_thresh) <span class="sc">%&gt;%</span> </span>
<span id="cb7-7"><a href="#cb7-7"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from=</span><span class="st">"strainID"</span>, <span class="at">values_from=</span><span class="st">"count_correct_thresh"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb7-8"><a href="#cb7-8"></a>  <span class="fu">column_to_rownames</span>(<span class="at">var=</span><span class="st">"community_id"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb7-9"><a href="#cb7-9"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">everything</span>(), <span class="sc">~</span><span class="fu">replace_na</span>(.x, <span class="dv">0</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb7-10"><a href="#cb7-10"></a>  <span class="fu">as.matrix</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="make-a-matrix-of-true-proportions" class="level2" data-number="3.4">
<h2 data-number="3.4" class="anchored" data-anchor-id="make-a-matrix-of-true-proportions"><span class="header-section-number">3.4</span> Make a matrix of true proportions</h2>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1"></a>pos_ctrl_proportions <span class="ot">&lt;-</span> pos_ctrl <span class="sc">%&gt;%</span> </span>
<span id="cb8-2"><a href="#cb8-2"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(community_id, strainID, f_qg) <span class="sc">%&gt;%</span></span>
<span id="cb8-3"><a href="#cb8-3"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from=</span><span class="st">"strainID"</span>, <span class="at">values_from=</span><span class="st">"f_qg"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb8-4"><a href="#cb8-4"></a>  <span class="fu">column_to_rownames</span>(<span class="at">var=</span><span class="st">"community_id"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb8-5"><a href="#cb8-5"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">everything</span>(), <span class="sc">~</span><span class="fu">replace_na</span>(.x, <span class="dv">0</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb8-6"><a href="#cb8-6"></a>  <span class="fu">as.matrix</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</section>
<section id="metacal-procedure" class="level1" data-number="4">
<h1 data-number="4"><span class="header-section-number">4</span> Metacal procedure</h1>
<section id="estimate-bias" class="level2" data-number="4.1">
<h2 data-number="4.1" class="anchored" data-anchor-id="estimate-bias"><span class="header-section-number">4.1</span> Estimate bias</h2>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1"></a><span class="fu">set.seed</span>(<span class="dv">12378</span>)</span>
<span id="cb9-2"><a href="#cb9-2"></a>mc_fit_trios <span class="ot">&lt;-</span> metacal<span class="sc">::</span><span class="fu">estimate_bias</span>(samp_trios_counts, pos_ctrl_proportions, <span class="dv">1</span>, <span class="at">boot=</span><span class="cn">TRUE</span>)</span>
<span id="cb9-3"><a href="#cb9-3"></a>mc_fit_trios_summary <span class="ot">&lt;-</span> <span class="fu">summary</span>(mc_fit_trios)</span>
<span id="cb9-4"><a href="#cb9-4"></a>mc_fit_trios_summary_coef <span class="ot">&lt;-</span>mc_fit_trios_summary[[<span class="st">'coefficients'</span>]]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="plot-bias-estimation" class="level3" data-number="4.1.1">
<h3 data-number="4.1.1" class="anchored" data-anchor-id="plot-bias-estimation"><span class="header-section-number">4.1.1</span> Plot bias estimation</h3>
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1"></a>mc_fit_trios_summary_coef <span class="sc">%&gt;%</span> </span>
<span id="cb10-2"><a href="#cb10-2"></a>  <span class="fu">mutate</span>(<span class="at">taxon =</span> <span class="fu">fct_reorder</span>(taxon, estimate)) <span class="sc">|&gt;</span> </span>
<span id="cb10-3"><a href="#cb10-3"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(taxon, estimate, </span>
<span id="cb10-4"><a href="#cb10-4"></a>             <span class="at">ymin =</span> estimate <span class="sc">/</span> gm_se<span class="sc">^</span><span class="dv">2</span>, <span class="at">ymax =</span> estimate <span class="sc">*</span> gm_se<span class="sc">^</span><span class="dv">2</span>)) <span class="sc">+</span></span>
<span id="cb10-5"><a href="#cb10-5"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">1</span>, <span class="at">color =</span> <span class="st">"grey"</span>) <span class="sc">+</span></span>
<span id="cb10-6"><a href="#cb10-6"></a>  <span class="fu">geom_pointrange</span>(<span class="fu">aes</span>(<span class="at">color =</span> taxon)) <span class="sc">+</span></span>
<span id="cb10-7"><a href="#cb10-7"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> hambi_colors) <span class="sc">+</span></span>
<span id="cb10-8"><a href="#cb10-8"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">""</span>, <span class="at">y =</span> <span class="st">"Bias estimate"</span>, <span class="at">color =</span> <span class="st">""</span>) <span class="sc">+</span></span>
<span id="cb10-9"><a href="#cb10-9"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span> </span>
<span id="cb10-10"><a href="#cb10-10"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb10-11"><a href="#cb10-11"></a>  <span class="fu">theme</span>(<span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb10-12"><a href="#cb10-12"></a>        <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div id="fig-02" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-02-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="02_correct_boil_to_qiagen_files/figure-html/unnamed-chunk-9-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-02-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;2: Bias estimates for each species from the metacal procedure.
</figcaption>
</figure>
</div>
</section>
<section id="plot-metacal-model-fit" class="level3" data-number="4.1.2">
<h3 data-number="4.1.2" class="anchored" data-anchor-id="plot-metacal-model-fit"><span class="header-section-number">4.1.2</span> Plot metacal model fit</h3>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1"></a>a <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">fitted</span>(mc_fit_trios)) <span class="sc">%&gt;%</span> </span>
<span id="cb11-2"><a href="#cb11-2"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"sample"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb11-3"><a href="#cb11-3"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>sample) <span class="sc">%&gt;%</span> </span>
<span id="cb11-4"><a href="#cb11-4"></a>  <span class="fu">mutate</span>(<span class="at">type=</span><span class="st">"Fitted"</span>)</span>
<span id="cb11-5"><a href="#cb11-5"></a></span>
<span id="cb11-6"><a href="#cb11-6"></a>b <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(pos_ctrl_proportions) <span class="sc">%&gt;%</span> </span>
<span id="cb11-7"><a href="#cb11-7"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"sample"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb11-8"><a href="#cb11-8"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>sample) <span class="sc">%&gt;%</span> </span>
<span id="cb11-9"><a href="#cb11-9"></a>  <span class="fu">mutate</span>(<span class="at">type=</span><span class="st">"Actual"</span>)</span>
<span id="cb11-10"><a href="#cb11-10"></a></span>
<span id="cb11-11"><a href="#cb11-11"></a>c <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(samp_trios_counts<span class="sc">/</span><span class="fu">rowSums</span>(samp_trios_counts)) <span class="sc">%&gt;%</span> </span>
<span id="cb11-12"><a href="#cb11-12"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"sample"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb11-13"><a href="#cb11-13"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>sample,  <span class="at">values_to=</span><span class="st">"observed"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1"></a><span class="fu">bind_rows</span>(a,b) <span class="sc">%&gt;%</span> </span>
<span id="cb12-2"><a href="#cb12-2"></a>  <span class="fu">left_join</span>(c, <span class="at">by =</span> <span class="fu">join_by</span>(sample, name)) <span class="sc">%&gt;%</span> </span>
<span id="cb12-3"><a href="#cb12-3"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>observed, <span class="at">y=</span>value, <span class="at">color =</span> name)) <span class="sc">+</span></span>
<span id="cb12-4"><a href="#cb12-4"></a>  <span class="fu">geom_abline</span>(<span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb12-5"><a href="#cb12-5"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb12-6"><a href="#cb12-6"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> hambi_colors) <span class="sc">+</span></span>
<span id="cb12-7"><a href="#cb12-7"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Species proportion in boil-prepped samples"</span>, <span class="at">y =</span> <span class="st">"Species proportion in Qiagen-prepped samples"</span>, <span class="at">color =</span> <span class="st">""</span>) <span class="sc">+</span></span>
<span id="cb12-8"><a href="#cb12-8"></a>  <span class="fu">facet_grid</span>(<span class="sc">~</span>type) <span class="sc">+</span> </span>
<span id="cb12-9"><a href="#cb12-9"></a>  <span class="fu">coord_fixed</span>(<span class="at">xlim =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>), <span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>)) <span class="sc">+</span> </span>
<span id="cb12-10"><a href="#cb12-10"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb12-11"><a href="#cb12-11"></a>  <span class="fu">theme</span>(<span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb12-12"><a href="#cb12-12"></a>        <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div id="fig-03" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-03-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="02_correct_boil_to_qiagen_files/figure-html/unnamed-chunk-11-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-03-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3: Porportion of each species from boil-prepped samples (x-axis) and from the Qiagen DNeasy extracted samples (y-axis) colored by species identity. The left panel shows the relationship between the two extraction procedures before correction with metacal. The right panel shows the relationship after correction. The dashed line is 1:1.
</figcaption>
</figure>
</div>
</section>
</section>
<section id="calibrate" class="level2" data-number="4.2">
<h2 data-number="4.2" class="anchored" data-anchor-id="calibrate"><span class="header-section-number">4.2</span> Calibrate</h2>
<p>Make a matrix of observed counts</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1"></a>trios_2_cal <span class="ot">&lt;-</span> <span class="fu">left_join</span>(samp_trios, mddf, </span>
<span id="cb13-2"><a href="#cb13-2"></a>                         <span class="at">by =</span> <span class="fu">join_by</span>(sample, replicate, strep_conc, transfers, n_species, community_type, plate_well)) <span class="sc">%&gt;%</span> </span>
<span id="cb13-3"><a href="#cb13-3"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(sample, strainID, count_correct_thresh) <span class="sc">%&gt;%</span> </span>
<span id="cb13-4"><a href="#cb13-4"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from=</span><span class="st">"strainID"</span>, <span class="at">values_from=</span><span class="st">"count_correct_thresh"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb13-5"><a href="#cb13-5"></a>  <span class="fu">column_to_rownames</span>(<span class="at">var=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb13-6"><a href="#cb13-6"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">everything</span>(), <span class="sc">~</span><span class="fu">replace_na</span>(.x, <span class="dv">0</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb13-7"><a href="#cb13-7"></a>  <span class="fu">as.matrix</span>()</span>
<span id="cb13-8"><a href="#cb13-8"></a></span>
<span id="cb13-9"><a href="#cb13-9"></a>pairs_2_cal <span class="ot">&lt;-</span> <span class="fu">left_join</span>(samp_pairs, mddf, </span>
<span id="cb13-10"><a href="#cb13-10"></a>                         <span class="at">by =</span> <span class="fu">join_by</span>(sample, replicate, strep_conc, transfers, n_species, community_type, plate_well)) <span class="sc">%&gt;%</span> </span>
<span id="cb13-11"><a href="#cb13-11"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(sample, strainID, count_correct_thresh) <span class="sc">%&gt;%</span> </span>
<span id="cb13-12"><a href="#cb13-12"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from=</span><span class="st">"strainID"</span>, <span class="at">values_from=</span><span class="st">"count_correct_thresh"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb13-13"><a href="#cb13-13"></a>  <span class="fu">column_to_rownames</span>(<span class="at">var=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb13-14"><a href="#cb13-14"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">everything</span>(), <span class="sc">~</span><span class="fu">replace_na</span>(.x, <span class="dv">0</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb13-15"><a href="#cb13-15"></a>  <span class="fu">as.matrix</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="run-the-calibrate-function" class="level2" data-number="4.3">
<h2 data-number="4.3" class="anchored" data-anchor-id="run-the-calibrate-function"><span class="header-section-number">4.3</span> Run the calibrate function</h2>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1"></a><span class="fu">set.seed</span>(<span class="dv">435761</span>)</span>
<span id="cb14-2"><a href="#cb14-2"></a>pairs_calibrated <span class="ot">&lt;-</span> metacal<span class="sc">::</span><span class="fu">calibrate</span>(pairs_2_cal, <span class="fu">coef</span>(mc_fit_trios), <span class="at">margin=</span><span class="dv">1</span>)</span>
<span id="cb14-3"><a href="#cb14-3"></a>trios_calibrated <span class="ot">&lt;-</span> metacal<span class="sc">::</span><span class="fu">calibrate</span>(trios_2_cal, <span class="fu">coef</span>(mc_fit_trios), <span class="at">margin=</span><span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1"></a>pairs_calibrated_l <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(pairs_calibrated) <span class="sc">%&gt;%</span> </span>
<span id="cb15-2"><a href="#cb15-2"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"sample"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb15-3"><a href="#cb15-3"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>sample, <span class="at">names_to =</span> <span class="st">"strainID"</span>, <span class="at">values_to =</span> <span class="st">"f_metacal"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb15-4"><a href="#cb15-4"></a>  <span class="fu">filter</span>(f_metacal <span class="sc">&gt;</span> <span class="dv">0</span>)</span>
<span id="cb15-5"><a href="#cb15-5"></a></span>
<span id="cb15-6"><a href="#cb15-6"></a>trios_calibrated_l <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(trios_calibrated) <span class="sc">%&gt;%</span> </span>
<span id="cb15-7"><a href="#cb15-7"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"sample"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb15-8"><a href="#cb15-8"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>sample, <span class="at">names_to =</span> <span class="st">"strainID"</span>, <span class="at">values_to =</span> <span class="st">"f_metacal"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb15-9"><a href="#cb15-9"></a>  <span class="fu">filter</span>(f_metacal <span class="sc">&gt;</span> <span class="dv">0</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</section>
<section id="export" class="level1" data-number="5">
<h1 data-number="5"><span class="header-section-number">5</span> Export</h1>
<p>Save metacal estimates for later use</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1"></a><span class="fu">write_tsv</span>(pairs_calibrated_l, here<span class="sc">::</span><span class="fu">here</span>(data, <span class="st">"pairs_metacal.tsv"</span>))</span>
<span id="cb16-2"><a href="#cb16-2"></a><span class="fu">write_tsv</span>(trios_calibrated_l, here<span class="sc">::</span><span class="fu">here</span>(data, <span class="st">"trios_metacal.tsv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>


<!-- -->

</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "î§‹";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb17" data-shortcodes="false"><pre class="sourceCode numberSource markdown number-lines code-with-copy"><code class="sourceCode markdown"><span id="cb17-1"><a href="#cb17-1"></a><span class="co">---</span></span>
<span id="cb17-2"><a href="#cb17-2"></a><span class="an">title:</span><span class="co"> "Calibrate boilprep to Qiagen DNeasy"</span></span>
<span id="cb17-3"><a href="#cb17-3"></a><span class="an">author:</span><span class="co"> "Shane Hogle"</span></span>
<span id="cb17-4"><a href="#cb17-4"></a><span class="an">date:</span><span class="co"> today</span></span>
<span id="cb17-5"><a href="#cb17-5"></a><span class="an">link-citations:</span><span class="co"> true</span></span>
<span id="cb17-6"><a href="#cb17-6"></a><span class="an">abstract:</span><span class="co"> "We will use the [metacal package](https://mikemc.github.io/metacal/) for estimating bias and performing calibration in the special case where the bias of all the taxa of interest can be directly measured from the control sample. Since samples extracted by Qiagen and by boil are exactly the same we can estimate scaling factors to produce corrected relative abundance for other samples. Publication: [Consistent and correctable bias in metagenomic sequencing experiments](https://elifesciences.org/articles/46923)"</span></span>
<span id="cb17-7"><a href="#cb17-7"></a><span class="co">---</span></span>
<span id="cb17-8"><a href="#cb17-8"></a></span>
<span id="cb17-9"><a href="#cb17-9"></a><span class="fu"># Setup</span></span>
<span id="cb17-10"><a href="#cb17-10"></a></span>
<span id="cb17-11"><a href="#cb17-11"></a><span class="fu">## Libraries</span></span>
<span id="cb17-12"><a href="#cb17-12"></a></span>
<span id="cb17-15"><a href="#cb17-15"></a><span class="in">```{r}</span></span>
<span id="cb17-16"><a href="#cb17-16"></a><span class="co">#| output: false</span></span>
<span id="cb17-17"><a href="#cb17-17"></a><span class="co">#| warning: false</span></span>
<span id="cb17-18"><a href="#cb17-18"></a><span class="co">#| error: false</span></span>
<span id="cb17-19"><a href="#cb17-19"></a><span class="fu">library</span>(here)</span>
<span id="cb17-20"><a href="#cb17-20"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb17-21"><a href="#cb17-21"></a><span class="fu">library</span>(metacal)</span>
<span id="cb17-22"><a href="#cb17-22"></a><span class="fu">library</span>(scales)</span>
<span id="cb17-23"><a href="#cb17-23"></a><span class="fu">source</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"R"</span>, <span class="st">"utils_generic.R"</span>))</span>
<span id="cb17-24"><a href="#cb17-24"></a><span class="fu">source</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"R"</span>, <span class="st">"communities"</span>, <span class="st">"community_composition"</span>, <span class="st">"amplicon"</span>, <span class="st">"utils_amplicon.R"</span>))</span>
<span id="cb17-25"><a href="#cb17-25"></a><span class="in">```</span></span>
<span id="cb17-26"><a href="#cb17-26"></a></span>
<span id="cb17-27"><a href="#cb17-27"></a><span class="fu">## Global variables</span></span>
<span id="cb17-28"><a href="#cb17-28"></a></span>
<span id="cb17-31"><a href="#cb17-31"></a><span class="in">```{r}</span></span>
<span id="cb17-32"><a href="#cb17-32"></a>data_raw <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"_data_raw"</span>, <span class="st">"communities"</span>, <span class="st">"20240711_BTK_illumina_v3"</span>)</span>
<span id="cb17-33"><a href="#cb17-33"></a>data <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"communities"</span>, <span class="st">"20240711_BTK_illumina_v3"</span>)</span>
<span id="cb17-34"><a href="#cb17-34"></a></span>
<span id="cb17-35"><a href="#cb17-35"></a><span class="co"># make processed data directory if it doesn't exist</span></span>
<span id="cb17-36"><a href="#cb17-36"></a>fs<span class="sc">::</span><span class="fu">dir_create</span>(data)</span>
<span id="cb17-37"><a href="#cb17-37"></a><span class="in">```</span></span>
<span id="cb17-38"><a href="#cb17-38"></a></span>
<span id="cb17-39"><a href="#cb17-39"></a><span class="fu"># Read data</span></span>
<span id="cb17-40"><a href="#cb17-40"></a></span>
<span id="cb17-43"><a href="#cb17-43"></a><span class="in">```{r}</span></span>
<span id="cb17-44"><a href="#cb17-44"></a><span class="co">#| output: false</span></span>
<span id="cb17-45"><a href="#cb17-45"></a><span class="co">#| warning: false</span></span>
<span id="cb17-46"><a href="#cb17-46"></a><span class="co">#| error: false</span></span>
<span id="cb17-47"><a href="#cb17-47"></a>pos_ctrl <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(data, <span class="st">"pos_neg_ctrl_counts.tsv"</span>))</span>
<span id="cb17-48"><a href="#cb17-48"></a>samp_pairs <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(data, <span class="st">"pairs_counts.tsv"</span>))</span>
<span id="cb17-49"><a href="#cb17-49"></a>samp_trios <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(data, <span class="st">"trios_counts.tsv"</span>))</span>
<span id="cb17-50"><a href="#cb17-50"></a></span>
<span id="cb17-51"><a href="#cb17-51"></a><span class="co"># metadata</span></span>
<span id="cb17-52"><a href="#cb17-52"></a>mddf <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(data_raw, <span class="st">"20240711_BTK_illumina_v3_metadata.tsv"</span>))</span>
<span id="cb17-53"><a href="#cb17-53"></a>spdf <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(data_raw, <span class="st">"sample_compositions.tsv"</span>))</span>
<span id="cb17-54"><a href="#cb17-54"></a><span class="in">```</span></span>
<span id="cb17-55"><a href="#cb17-55"></a></span>
<span id="cb17-56"><a href="#cb17-56"></a><span class="fu"># Formatting</span></span>
<span id="cb17-57"><a href="#cb17-57"></a></span>
<span id="cb17-58"><a href="#cb17-58"></a><span class="fu">## Format positive control samples</span></span>
<span id="cb17-59"><a href="#cb17-59"></a></span>
<span id="cb17-62"><a href="#cb17-62"></a><span class="in">```{r}</span></span>
<span id="cb17-63"><a href="#cb17-63"></a>pos_ctrl <span class="ot">&lt;-</span> <span class="fu">left_join</span>(pos_ctrl, spdf) <span class="sc">%&gt;%</span> </span>
<span id="cb17-64"><a href="#cb17-64"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(community_type, <span class="st">"^pos"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb17-65"><a href="#cb17-65"></a>  <span class="co"># remove noise from species that aren't really there</span></span>
<span id="cb17-66"><a href="#cb17-66"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(evo_hist)) <span class="sc">%&gt;%</span> </span>
<span id="cb17-67"><a href="#cb17-67"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb17-68"><a href="#cb17-68"></a>  <span class="fu">mutate</span>(<span class="at">f_qg =</span> count_correct<span class="sc">/</span><span class="fu">sum</span>(count_correct)) <span class="sc">%&gt;%</span> </span>
<span id="cb17-69"><a href="#cb17-69"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(sample, strainID, community_id, count_correct, f_qg) <span class="sc">%&gt;%</span> </span>
<span id="cb17-70"><a href="#cb17-70"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb17-71"><a href="#cb17-71"></a><span class="in">```</span></span>
<span id="cb17-72"><a href="#cb17-72"></a></span>
<span id="cb17-73"><a href="#cb17-73"></a><span class="fu">## Plot masterplate Qiagen and boil-prep side by side</span></span>
<span id="cb17-74"><a href="#cb17-74"></a></span>
<span id="cb17-75"><a href="#cb17-75"></a>::: {#fig-01}</span>
<span id="cb17-78"><a href="#cb17-78"></a><span class="in">```{r}</span></span>
<span id="cb17-79"><a href="#cb17-79"></a><span class="co">#| fig.width: 8</span></span>
<span id="cb17-80"><a href="#cb17-80"></a><span class="co">#| fig.height: 5</span></span>
<span id="cb17-81"><a href="#cb17-81"></a><span class="fu">left_join</span>(samp_trios, mddf,</span>
<span id="cb17-82"><a href="#cb17-82"></a>          <span class="at">by =</span> <span class="fu">join_by</span>(sample, replicate, strep_conc, transfers, n_species, community_type, plate_well)) <span class="sc">%&gt;%</span> </span>
<span id="cb17-83"><a href="#cb17-83"></a>  <span class="fu">filter</span>(community_type <span class="sc">==</span> <span class="st">"masterplate"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb17-84"><a href="#cb17-84"></a>  <span class="fu">right_join</span>(pos_ctrl, <span class="at">by =</span> <span class="fu">join_by</span>(strainID, community_id)) <span class="sc">%&gt;%</span> </span>
<span id="cb17-85"><a href="#cb17-85"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(community_id, strainID, f_thresh, f_qg) <span class="sc">%&gt;%</span> </span>
<span id="cb17-86"><a href="#cb17-86"></a>  <span class="fu">pivot_longer</span>(<span class="fu">c</span>(f_thresh, f_qg)) <span class="sc">%&gt;%</span> </span>
<span id="cb17-87"><a href="#cb17-87"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb17-88"><a href="#cb17-88"></a>    <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">y =</span> value, <span class="at">x=</span><span class="fu">interaction</span>(name, community_id), <span class="at">fill =</span> strainID), </span>
<span id="cb17-89"><a href="#cb17-89"></a>             <span class="at">color=</span><span class="st">"black"</span>,</span>
<span id="cb17-90"><a href="#cb17-90"></a>             <span class="at">linewidth=</span><span class="fl">0.25</span>, <span class="at">stat=</span><span class="st">"identity"</span>) <span class="sc">+</span></span>
<span id="cb17-91"><a href="#cb17-91"></a>    <span class="fu">scale_y_continuous</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>), <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>), <span class="at">labels =</span> percent) <span class="sc">+</span></span>
<span id="cb17-92"><a href="#cb17-92"></a>    <span class="fu">scale_x_discrete</span>(<span class="at">guide =</span> <span class="fu">guide_axis</span>(<span class="at">angle =</span> <span class="dv">90</span>)) <span class="sc">+</span></span>
<span id="cb17-93"><a href="#cb17-93"></a>    <span class="fu">labs</span>(<span class="at">x=</span><span class="st">""</span>, <span class="at">y=</span><span class="st">"% abundance"</span>) <span class="sc">+</span></span>
<span id="cb17-94"><a href="#cb17-94"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span> </span>
<span id="cb17-95"><a href="#cb17-95"></a>  <span class="fu">theme</span>(<span class="at">panel.grid =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb17-96"><a href="#cb17-96"></a>        <span class="at">legend.position =</span> <span class="st">"bottom"</span>) <span class="sc">+</span></span>
<span id="cb17-97"><a href="#cb17-97"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> hambi_colors)</span>
<span id="cb17-98"><a href="#cb17-98"></a><span class="in">```</span></span>
<span id="cb17-99"><a href="#cb17-99"></a>Bar plot of masterplate samples extracted by the boilprep method (f) or the Qiagen DNeasy kit (qg). The community trios are appended to the extraction method on the x axs</span>
<span id="cb17-100"><a href="#cb17-100"></a>:::</span>
<span id="cb17-101"><a href="#cb17-101"></a></span>
<span id="cb17-102"><a href="#cb17-102"></a><span class="fu">## Matrix of observed counts</span></span>
<span id="cb17-103"><a href="#cb17-103"></a></span>
<span id="cb17-106"><a href="#cb17-106"></a><span class="in">```{r}</span></span>
<span id="cb17-107"><a href="#cb17-107"></a>samp_trios_counts <span class="ot">&lt;-</span> <span class="fu">left_join</span>(samp_trios, mddf,</span>
<span id="cb17-108"><a href="#cb17-108"></a>                               <span class="at">by =</span> <span class="fu">join_by</span>(sample, replicate, strep_conc, </span>
<span id="cb17-109"><a href="#cb17-109"></a>                                            transfers, n_species, community_type, plate_well)) <span class="sc">%&gt;%</span> </span>
<span id="cb17-110"><a href="#cb17-110"></a>  <span class="fu">filter</span>(community_type <span class="sc">==</span> <span class="st">"masterplate"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb17-111"><a href="#cb17-111"></a>  <span class="fu">filter</span>(community_id <span class="sc">%in%</span> <span class="fu">pull</span>(pos_ctrl, community_id)) <span class="sc">%&gt;%</span> </span>
<span id="cb17-112"><a href="#cb17-112"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(community_id, strainID, count_correct_thresh) <span class="sc">%&gt;%</span> </span>
<span id="cb17-113"><a href="#cb17-113"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from=</span><span class="st">"strainID"</span>, <span class="at">values_from=</span><span class="st">"count_correct_thresh"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb17-114"><a href="#cb17-114"></a>  <span class="fu">column_to_rownames</span>(<span class="at">var=</span><span class="st">"community_id"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb17-115"><a href="#cb17-115"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">everything</span>(), <span class="sc">~</span><span class="fu">replace_na</span>(.x, <span class="dv">0</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb17-116"><a href="#cb17-116"></a>  <span class="fu">as.matrix</span>()</span>
<span id="cb17-117"><a href="#cb17-117"></a><span class="in">```</span></span>
<span id="cb17-118"><a href="#cb17-118"></a></span>
<span id="cb17-119"><a href="#cb17-119"></a><span class="fu">## Make a matrix of true proportions</span></span>
<span id="cb17-120"><a href="#cb17-120"></a></span>
<span id="cb17-123"><a href="#cb17-123"></a><span class="in">```{r}</span></span>
<span id="cb17-124"><a href="#cb17-124"></a>pos_ctrl_proportions <span class="ot">&lt;-</span> pos_ctrl <span class="sc">%&gt;%</span> </span>
<span id="cb17-125"><a href="#cb17-125"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(community_id, strainID, f_qg) <span class="sc">%&gt;%</span></span>
<span id="cb17-126"><a href="#cb17-126"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from=</span><span class="st">"strainID"</span>, <span class="at">values_from=</span><span class="st">"f_qg"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb17-127"><a href="#cb17-127"></a>  <span class="fu">column_to_rownames</span>(<span class="at">var=</span><span class="st">"community_id"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb17-128"><a href="#cb17-128"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">everything</span>(), <span class="sc">~</span><span class="fu">replace_na</span>(.x, <span class="dv">0</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb17-129"><a href="#cb17-129"></a>  <span class="fu">as.matrix</span>()</span>
<span id="cb17-130"><a href="#cb17-130"></a><span class="in">```</span></span>
<span id="cb17-131"><a href="#cb17-131"></a></span>
<span id="cb17-132"><a href="#cb17-132"></a><span class="fu"># Metacal procedure</span></span>
<span id="cb17-133"><a href="#cb17-133"></a></span>
<span id="cb17-134"><a href="#cb17-134"></a><span class="fu">## Estimate bias</span></span>
<span id="cb17-135"><a href="#cb17-135"></a></span>
<span id="cb17-138"><a href="#cb17-138"></a><span class="in">```{r}</span></span>
<span id="cb17-139"><a href="#cb17-139"></a><span class="fu">set.seed</span>(<span class="dv">12378</span>)</span>
<span id="cb17-140"><a href="#cb17-140"></a>mc_fit_trios <span class="ot">&lt;-</span> metacal<span class="sc">::</span><span class="fu">estimate_bias</span>(samp_trios_counts, pos_ctrl_proportions, <span class="dv">1</span>, <span class="at">boot=</span><span class="cn">TRUE</span>)</span>
<span id="cb17-141"><a href="#cb17-141"></a>mc_fit_trios_summary <span class="ot">&lt;-</span> <span class="fu">summary</span>(mc_fit_trios)</span>
<span id="cb17-142"><a href="#cb17-142"></a>mc_fit_trios_summary_coef <span class="ot">&lt;-</span>mc_fit_trios_summary[[<span class="st">'coefficients'</span>]]</span>
<span id="cb17-143"><a href="#cb17-143"></a><span class="in">```</span></span>
<span id="cb17-144"><a href="#cb17-144"></a></span>
<span id="cb17-145"><a href="#cb17-145"></a><span class="fu">### Plot bias estimation</span></span>
<span id="cb17-146"><a href="#cb17-146"></a></span>
<span id="cb17-147"><a href="#cb17-147"></a>::: {#fig-02}</span>
<span id="cb17-150"><a href="#cb17-150"></a><span class="in">```{r}</span></span>
<span id="cb17-151"><a href="#cb17-151"></a>mc_fit_trios_summary_coef <span class="sc">%&gt;%</span> </span>
<span id="cb17-152"><a href="#cb17-152"></a>  <span class="fu">mutate</span>(<span class="at">taxon =</span> <span class="fu">fct_reorder</span>(taxon, estimate)) <span class="sc">|&gt;</span> </span>
<span id="cb17-153"><a href="#cb17-153"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(taxon, estimate, </span>
<span id="cb17-154"><a href="#cb17-154"></a>             <span class="at">ymin =</span> estimate <span class="sc">/</span> gm_se<span class="sc">^</span><span class="dv">2</span>, <span class="at">ymax =</span> estimate <span class="sc">*</span> gm_se<span class="sc">^</span><span class="dv">2</span>)) <span class="sc">+</span></span>
<span id="cb17-155"><a href="#cb17-155"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">1</span>, <span class="at">color =</span> <span class="st">"grey"</span>) <span class="sc">+</span></span>
<span id="cb17-156"><a href="#cb17-156"></a>  <span class="fu">geom_pointrange</span>(<span class="fu">aes</span>(<span class="at">color =</span> taxon)) <span class="sc">+</span></span>
<span id="cb17-157"><a href="#cb17-157"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> hambi_colors) <span class="sc">+</span></span>
<span id="cb17-158"><a href="#cb17-158"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">""</span>, <span class="at">y =</span> <span class="st">"Bias estimate"</span>, <span class="at">color =</span> <span class="st">""</span>) <span class="sc">+</span></span>
<span id="cb17-159"><a href="#cb17-159"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span> </span>
<span id="cb17-160"><a href="#cb17-160"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb17-161"><a href="#cb17-161"></a>  <span class="fu">theme</span>(<span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb17-162"><a href="#cb17-162"></a>        <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>())</span>
<span id="cb17-163"><a href="#cb17-163"></a><span class="in">```</span></span>
<span id="cb17-164"><a href="#cb17-164"></a>Bias estimates for each species from the metacal procedure.</span>
<span id="cb17-165"><a href="#cb17-165"></a>:::</span>
<span id="cb17-166"><a href="#cb17-166"></a></span>
<span id="cb17-167"><a href="#cb17-167"></a><span class="fu">### Plot metacal model fit</span></span>
<span id="cb17-168"><a href="#cb17-168"></a></span>
<span id="cb17-171"><a href="#cb17-171"></a><span class="in">```{r}</span></span>
<span id="cb17-172"><a href="#cb17-172"></a>a <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">fitted</span>(mc_fit_trios)) <span class="sc">%&gt;%</span> </span>
<span id="cb17-173"><a href="#cb17-173"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"sample"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb17-174"><a href="#cb17-174"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>sample) <span class="sc">%&gt;%</span> </span>
<span id="cb17-175"><a href="#cb17-175"></a>  <span class="fu">mutate</span>(<span class="at">type=</span><span class="st">"Fitted"</span>)</span>
<span id="cb17-176"><a href="#cb17-176"></a></span>
<span id="cb17-177"><a href="#cb17-177"></a>b <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(pos_ctrl_proportions) <span class="sc">%&gt;%</span> </span>
<span id="cb17-178"><a href="#cb17-178"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"sample"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb17-179"><a href="#cb17-179"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>sample) <span class="sc">%&gt;%</span> </span>
<span id="cb17-180"><a href="#cb17-180"></a>  <span class="fu">mutate</span>(<span class="at">type=</span><span class="st">"Actual"</span>)</span>
<span id="cb17-181"><a href="#cb17-181"></a></span>
<span id="cb17-182"><a href="#cb17-182"></a>c <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(samp_trios_counts<span class="sc">/</span><span class="fu">rowSums</span>(samp_trios_counts)) <span class="sc">%&gt;%</span> </span>
<span id="cb17-183"><a href="#cb17-183"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"sample"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb17-184"><a href="#cb17-184"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>sample,  <span class="at">values_to=</span><span class="st">"observed"</span>)</span>
<span id="cb17-185"><a href="#cb17-185"></a><span class="in">```</span></span>
<span id="cb17-186"><a href="#cb17-186"></a></span>
<span id="cb17-187"><a href="#cb17-187"></a></span>
<span id="cb17-188"><a href="#cb17-188"></a>::: {#fig-03}</span>
<span id="cb17-191"><a href="#cb17-191"></a><span class="in">```{r}</span></span>
<span id="cb17-192"><a href="#cb17-192"></a><span class="fu">bind_rows</span>(a,b) <span class="sc">%&gt;%</span> </span>
<span id="cb17-193"><a href="#cb17-193"></a>  <span class="fu">left_join</span>(c, <span class="at">by =</span> <span class="fu">join_by</span>(sample, name)) <span class="sc">%&gt;%</span> </span>
<span id="cb17-194"><a href="#cb17-194"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>observed, <span class="at">y=</span>value, <span class="at">color =</span> name)) <span class="sc">+</span></span>
<span id="cb17-195"><a href="#cb17-195"></a>  <span class="fu">geom_abline</span>(<span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb17-196"><a href="#cb17-196"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb17-197"><a href="#cb17-197"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> hambi_colors) <span class="sc">+</span></span>
<span id="cb17-198"><a href="#cb17-198"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Species proportion in boil-prepped samples"</span>, <span class="at">y =</span> <span class="st">"Species proportion in Qiagen-prepped samples"</span>, <span class="at">color =</span> <span class="st">""</span>) <span class="sc">+</span></span>
<span id="cb17-199"><a href="#cb17-199"></a>  <span class="fu">facet_grid</span>(<span class="sc">~</span>type) <span class="sc">+</span> </span>
<span id="cb17-200"><a href="#cb17-200"></a>  <span class="fu">coord_fixed</span>(<span class="at">xlim =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>), <span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>)) <span class="sc">+</span> </span>
<span id="cb17-201"><a href="#cb17-201"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb17-202"><a href="#cb17-202"></a>  <span class="fu">theme</span>(<span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb17-203"><a href="#cb17-203"></a>        <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>())</span>
<span id="cb17-204"><a href="#cb17-204"></a><span class="in">```</span></span>
<span id="cb17-205"><a href="#cb17-205"></a>Porportion of each species from boil-prepped samples (x-axis) and from the Qiagen DNeasy extracted samples (y-axis) colored by species identity. The left panel shows the relationship between the two extraction procedures before correction with metacal. The right panel shows the relationship after correction. The dashed line is 1:1. </span>
<span id="cb17-206"><a href="#cb17-206"></a>:::</span>
<span id="cb17-207"><a href="#cb17-207"></a></span>
<span id="cb17-208"><a href="#cb17-208"></a></span>
<span id="cb17-209"><a href="#cb17-209"></a><span class="fu">## Calibrate</span></span>
<span id="cb17-210"><a href="#cb17-210"></a></span>
<span id="cb17-211"><a href="#cb17-211"></a>Make a matrix of observed counts</span>
<span id="cb17-214"><a href="#cb17-214"></a><span class="in">```{r}</span></span>
<span id="cb17-215"><a href="#cb17-215"></a>trios_2_cal <span class="ot">&lt;-</span> <span class="fu">left_join</span>(samp_trios, mddf, </span>
<span id="cb17-216"><a href="#cb17-216"></a>                         <span class="at">by =</span> <span class="fu">join_by</span>(sample, replicate, strep_conc, transfers, n_species, community_type, plate_well)) <span class="sc">%&gt;%</span> </span>
<span id="cb17-217"><a href="#cb17-217"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(sample, strainID, count_correct_thresh) <span class="sc">%&gt;%</span> </span>
<span id="cb17-218"><a href="#cb17-218"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from=</span><span class="st">"strainID"</span>, <span class="at">values_from=</span><span class="st">"count_correct_thresh"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb17-219"><a href="#cb17-219"></a>  <span class="fu">column_to_rownames</span>(<span class="at">var=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb17-220"><a href="#cb17-220"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">everything</span>(), <span class="sc">~</span><span class="fu">replace_na</span>(.x, <span class="dv">0</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb17-221"><a href="#cb17-221"></a>  <span class="fu">as.matrix</span>()</span>
<span id="cb17-222"><a href="#cb17-222"></a></span>
<span id="cb17-223"><a href="#cb17-223"></a>pairs_2_cal <span class="ot">&lt;-</span> <span class="fu">left_join</span>(samp_pairs, mddf, </span>
<span id="cb17-224"><a href="#cb17-224"></a>                         <span class="at">by =</span> <span class="fu">join_by</span>(sample, replicate, strep_conc, transfers, n_species, community_type, plate_well)) <span class="sc">%&gt;%</span> </span>
<span id="cb17-225"><a href="#cb17-225"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(sample, strainID, count_correct_thresh) <span class="sc">%&gt;%</span> </span>
<span id="cb17-226"><a href="#cb17-226"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from=</span><span class="st">"strainID"</span>, <span class="at">values_from=</span><span class="st">"count_correct_thresh"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb17-227"><a href="#cb17-227"></a>  <span class="fu">column_to_rownames</span>(<span class="at">var=</span><span class="st">"sample"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb17-228"><a href="#cb17-228"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">everything</span>(), <span class="sc">~</span><span class="fu">replace_na</span>(.x, <span class="dv">0</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb17-229"><a href="#cb17-229"></a>  <span class="fu">as.matrix</span>()</span>
<span id="cb17-230"><a href="#cb17-230"></a><span class="in">```</span></span>
<span id="cb17-231"><a href="#cb17-231"></a></span>
<span id="cb17-232"><a href="#cb17-232"></a><span class="fu">## Run the calibrate function</span></span>
<span id="cb17-233"><a href="#cb17-233"></a></span>
<span id="cb17-236"><a href="#cb17-236"></a><span class="in">```{r}</span></span>
<span id="cb17-237"><a href="#cb17-237"></a><span class="fu">set.seed</span>(<span class="dv">435761</span>)</span>
<span id="cb17-238"><a href="#cb17-238"></a>pairs_calibrated <span class="ot">&lt;-</span> metacal<span class="sc">::</span><span class="fu">calibrate</span>(pairs_2_cal, <span class="fu">coef</span>(mc_fit_trios), <span class="at">margin=</span><span class="dv">1</span>)</span>
<span id="cb17-239"><a href="#cb17-239"></a>trios_calibrated <span class="ot">&lt;-</span> metacal<span class="sc">::</span><span class="fu">calibrate</span>(trios_2_cal, <span class="fu">coef</span>(mc_fit_trios), <span class="at">margin=</span><span class="dv">1</span>)</span>
<span id="cb17-240"><a href="#cb17-240"></a><span class="in">```</span></span>
<span id="cb17-241"><a href="#cb17-241"></a></span>
<span id="cb17-242"><a href="#cb17-242"></a></span>
<span id="cb17-245"><a href="#cb17-245"></a><span class="in">```{r}</span></span>
<span id="cb17-246"><a href="#cb17-246"></a>pairs_calibrated_l <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(pairs_calibrated) <span class="sc">%&gt;%</span> </span>
<span id="cb17-247"><a href="#cb17-247"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"sample"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb17-248"><a href="#cb17-248"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>sample, <span class="at">names_to =</span> <span class="st">"strainID"</span>, <span class="at">values_to =</span> <span class="st">"f_metacal"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb17-249"><a href="#cb17-249"></a>  <span class="fu">filter</span>(f_metacal <span class="sc">&gt;</span> <span class="dv">0</span>)</span>
<span id="cb17-250"><a href="#cb17-250"></a></span>
<span id="cb17-251"><a href="#cb17-251"></a>trios_calibrated_l <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(trios_calibrated) <span class="sc">%&gt;%</span> </span>
<span id="cb17-252"><a href="#cb17-252"></a>  <span class="fu">rownames_to_column</span>(<span class="at">var =</span> <span class="st">"sample"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb17-253"><a href="#cb17-253"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>sample, <span class="at">names_to =</span> <span class="st">"strainID"</span>, <span class="at">values_to =</span> <span class="st">"f_metacal"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb17-254"><a href="#cb17-254"></a>  <span class="fu">filter</span>(f_metacal <span class="sc">&gt;</span> <span class="dv">0</span>)</span>
<span id="cb17-255"><a href="#cb17-255"></a><span class="in">```</span></span>
<span id="cb17-256"><a href="#cb17-256"></a></span>
<span id="cb17-257"><a href="#cb17-257"></a><span class="fu"># Export</span></span>
<span id="cb17-258"><a href="#cb17-258"></a></span>
<span id="cb17-259"><a href="#cb17-259"></a>Save metacal estimates for later use</span>
<span id="cb17-262"><a href="#cb17-262"></a><span class="in">```{r}</span></span>
<span id="cb17-263"><a href="#cb17-263"></a><span class="fu">write_tsv</span>(pairs_calibrated_l, here<span class="sc">::</span><span class="fu">here</span>(data, <span class="st">"pairs_metacal.tsv"</span>))</span>
<span id="cb17-264"><a href="#cb17-264"></a><span class="fu">write_tsv</span>(trios_calibrated_l, here<span class="sc">::</span><span class="fu">here</span>(data, <span class="st">"trios_metacal.tsv"</span>))</span>
<span id="cb17-265"><a href="#cb17-265"></a><span class="in">```</span></span>
</code><button title="Copy to Clipboard" class="code-copy-button" data-in-quarto-modal=""><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->




</body></html>