---
title: "Data anlalysis for the project hambiABBAssembly"
author: "Shane Hogle & Milla Similä"
date: today
bibliography: references.bib
---

This project contains data and analysis from the hambiABBAssembly project.

## Manuscript:

"TBD"

[Preprint available from bioRxiv]()

## Introduction

Past work has shown that the composition of small bacterial communities can be predicted remarkably well from outcomes of competitions between small sets of species [@friedman2017]. Other work has questioned whether such simple qualitative assembly rules hold in more complex situations such as when the species pool shares an evolutionary history [@chang2023] (also see [this blog post](https://sanchezlaboratory.weebly.com/blog/with-a-little-help-of-bacterial-friends-solo-no-puedes-con-amigos-si)). If community assembly dynamics can truly be characterized by additive pairwise processes this would be a great advantage for modelers. It would mean that communities could be described sufficiently using paired species interactions (as in traditional Lotka Volterra). However, if higher-order interactions [@levine2017] are common and important (e.g., species C mediates the interaction between species A and B in a density independent way) then the number of parameters necessary to sufficiently model the system quickly blows up quickly making the entire problem intractable.

Understanding the fundamental processes in community assembly will be key to designing microbial communities with desirable properties for human use and to anticipate the response of communities in nature to change. Two theoretical camps have emerged in contemporary theory; one arguing for pairwise assembly rules and another arguing for the importance of networks of chained pairwise interactions and/or higher order interactions. Evidence exists for both processes occurring in nature. Thus, the question of when/how/why community assembly can be sufficiently described/predicted from additive pairwise processes is relevant and important for gaining a predictive understanding of microbiome dynamics. For example, when will the assembly process emerge as a property of the community, rather than being a straightforward additive process? In particular, do the previous evolutionary history and patterns of coexistence in the species pool determine emergent coexistence? What is the role for different growth conditions (substrate availability, antibacterial compounds, pH, predator presence)? Or does prior adaptive evolution to the environment change the outcomes? Do the "simple" rules governing community assembly fundamentally change as you add more layers of complexity?

The goal here is to shed some additional light on the question of the importance of pairwise competition/emergent coexistence in microbiomes. Our experiment assembles all possible 2, 3, and 4 species sub communities from a 4-species pool under a gradient of antibiotic concentrations and with different evolutionary histories of the four species. We assess the predictability of community assembly outcomes based on subsets of the reduced species assemblies and attempt to understand whether assembly operates in a predominantly pairwise way.

## Experiment overview

### Background:

In prior work from our lab we experimentally evolved all species from the HAMBI synthetic community to increasing concentrations of the aminoglycoside antibiotic streptomycin. Briefly, we would grow each species at the highest streptomycin concentration it could tolerate for a number of growth cycles, then we would transfer into a higher streptomycin. We did this until we reached a streptomycin concentration where we could no longer detect growth. Then these cultures were plated and a single clone was isolated.

These clones are the source "evolved" starting materials for this experiment, where their clonal ancestor is the "ancestral" starting material.

### Traits and physiology of species:

The ancestral and evolved form of each species were grown for 48 hours at 30℃ in 100% R2A growth medium [@reasoner1985a] at concentrations of 0, 4, 8, 16, 32, 64, 128, 256, 512, 1024, 2048, and 4096 μg/ml streptomycin. Growth was followed using optical density at 600 nm in a Bioscreen plate reader. Growth rates and carrying capacities were determined from the high resolution growth curves using AMiGA.

### Species interactions:

The ancestral and evolved forms of each species were grown for 48 hours at 30℃ in 100% R2A growth medium [@reasoner1985a]. Cultures were filtered to remove bacterial biomass and the spent media was stored at 4℃ until use. Next the evolved and ancestral forms of each species were grown in the spent medium of each other evo/anc species for 48 hours at 30℃ and growth was followed using optical density at 600 nm in a Bioscreen plate reader.

For the Burkholder diffusion assay we will add each species/evolutionary form to a soft agar on a petri dish. Then a small volume of each species will be pipetted directly onto the agar and plates will be checked for zones of inhibition.

### Competition/coexistence outcomes (species proportions from communities):

Here we wish to ensure that any equilibrium coexistence or competitive exclusion between species is not multistable - meaning that the outcome is deterministic and truly based on the density-independent competitive abilities of the species and not dependent on their starting density. The multistability problem has been addressed in other studies by growing each species from each community at different starting densities. This is typically done by starting all of the rare species at a low density and one abundant species at a higher density (as measured by optical density). For more than 2 species you cannot reasonably achieve all possible permutations of different densities, but you can start each species from rare, while leaving one remaining species high.

Here we use starting conditions with 10%/90% for pairs, 80%/10%/10% for trios, and 70%/10%/10%/10% for quartets. For pairs, we alternated so that each species started from rare. For trios and quartets, we alternated so that each species was allowed to invade from rare but we did not use all possible staring proportions due to practical and cost limitations. Another important consideration is that we can not distinguish between the ancestral and evolved forms of the same species with amplicon sequencing (or with plate counts) so that our experimental arrangement cannot be fully factorial (e.g., we cannot compete ancestral 1977 against evolved 1977) and thus there are fewer combinations that one would expect from expected scaling of $2^{(8-1)} = 128$

For the coexistence/exclusion experiments experimental communities are inoculated in 500 μl of 100% R2A in deep 96 well plates at a 50 fold dilution at four different streptomycin levels: 0, 16, 64, and 256 μg/ml. Plates were incubated in the dark at 30℃ with shaking at 800 RPM. Serial transfers (50 fold dilution; 10 μl to 490 μl) were conducted every 48 hours for a total of eight growth cycles. For every growth cycle $\mathrm{log_2(50)} \approx 5.6$ generations will have elapsed for a total of about 45 generations over eight cycles.

## Measurements and data types

### Traits and physiology of species:

-   Growth rates and carrying capacity (in optical density units) of each species (both ancestral/evolved) on a gradient of streptomycin concentrations

### Species interactions:

-   Growth rates and carrying capacity (in optical density units) of each species (both ancestral/evolved) on the filtrate of all other species.
-   (NOT YET COMPLETED) Burkholder Diffusion Assays for species pairs to detect presence of interference competition (e.g., antibiotic or toxin production that kills competitors). Note: based on past work with these species we don't expect to detect significant interference competition.

### Competition/coexistence outcomes (species proportions from communities):

-   Community DNA from pairs, trios, and quartets was sampled after eight 48 hour growth cycles
-   16S rRNA amplicon sequencing was performed on samples to get the proportion of each species
-   Optical density was collected at every other growth cycle

## Code

[![DOI](https://zenodo.org/badge/.svg)](https://zenodo.org/badge/latestdoi/) Data and code here is provided under GPL3. Feel free to use or remix as you see fit.

### Project structure

1.  `/R` contains R scripts
2.  `/data` contains data that has been processed in some way for later use
3.  `/_data_raw` contains unprocessed data scraped from compute cluster
4.  `/figs` contains figures generated from R scripts

## Availability

The rendered project site is available at <https://slhogle.github.io/hambiABBAssembly/>. The website has been produced using [Quarto notebooks](https://quarto.org/).

This GitHub repository (<https://github.com/slhogle/hambiABBAssembly>) hosts the code and data for this project. The rendered webpage can be fully recreated using the code.

## Reproducibility

The project uses [`renv`](https://rstudio.github.io/renv/index.html) to create reproducible environment to execute the code in this project. [See here](https://rstudio.github.io/renv/articles/renv.html#collaboration) for a brief overview on collaboration and reproduction of the entire project. To get up and running you can do:

``` r
install.packages("renv")
renv::restore()
```
