<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.42">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Milla Similä &amp; Shane Hogle">
<meta name="dcterms.date" content="2025-07-04">

<title>Formatting Rbec output from 2025-05-02 sequenced by BTK Turku using custom HAMBI Illumina v3 primers – hambiEvoEnvCoexist</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../../../../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../../../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../../../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../../../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../../../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../../../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../../../../">
<script src="../../../../../site_libs/quarto-html/quarto.js"></script>
<script src="../../../../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../../../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../../../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../../../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../../../../site_libs/quarto-html/quarto-syntax-highlighting-2f5df379a58b258e96c21c0638c20c03.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../../../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../../../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../../../../site_libs/bootstrap/bootstrap-1f23bf93a17969024013c136d185e5ce.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<link href="../../../../../site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet">
<script src="../../../../../site_libs/pagedtable-1.1/js/pagedtable.js"></script>


</head>

<body class="nav-sidebar docked">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../../../../R/communities/community_composition/amplicon/20240711_BTK_illumina_v3/01_format_rbec_tab.html">4. Community growth - amplicon formatting</a></li><li class="breadcrumb-item"><a href="../../../../../R/communities/community_composition/amplicon/20250502_BTK_illumina_v3/01_format_rbec_tab.html">v) Batch 2025-05-02</a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation docked overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="../../../../../">hambiEvoEnvCoexist</a> 
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true">
 <span class="menu-text">1. Monoculture growth</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../../R/monocultures/bioscreen/01_format_growthcurves_batch.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">i) Inspect and format growth curves</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../../R/monocultures/bioscreen/02_plot_analyze_growth.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">ii) Plot growth summaries</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true">
 <span class="menu-text">2. Community growth - experiment design</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../../R/communities/experiment_design/01_pairs.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">i) Species pairs - 96 Well plate layout</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../../R/communities/experiment_design/02_trios.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">ii) Species trios - 96 Well plate layout</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../../R/communities/experiment_design/03_quartets.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">iii) Species quartets - 96 Well plate layout</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true">
 <span class="menu-text">3. Community growth - biomass</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../../R/communities/community_biomass/optical_density/01_format_plot_od.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">i) Process and plot OD data</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="true">
 <span class="menu-text">4. Community growth - amplicon formatting</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../../R/communities/community_composition/amplicon/20240711_BTK_illumina_v3/01_format_rbec_tab.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">i) Batch 2024-07-11</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../../R/communities/community_composition/amplicon/20241024_BTK_illumina_v3/01_format_rbec_tab.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">iii) Batch 2024-10-24</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../../R/communities/community_composition/amplicon/20250127_BTK_illumina_v3/01_format_rbec_tab.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">iv) Batch 2025-01-27</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../../R/communities/community_composition/amplicon/20250502_BTK_illumina_v3/01_format_rbec_tab.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">v) Batch 2025-05-02</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../../R/communities/community_composition/amplicon/01_compile_format.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">vi) Compile pairs, trios, quartets</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" role="navigation" aria-expanded="true">
 <span class="menu-text">5. Community growth - composition</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-5" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../../R/communities/community_composition/01_analysis_pairs.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">i) Species pairs</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../../R/communities/community_composition/02_analysis_trios.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">ii) Species trios</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../../R/communities/community_composition/03_analysis_quartets.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">iii) Species quartets</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../../R/communities/community_composition/04_analysis_octets.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">iv) All strains combined</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../../R/communities/community_composition/05_assembly_rule_violation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">v) Violation of pairwise assembly rule</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#setup" id="toc-setup" class="nav-link active" data-scroll-target="#setup"><span class="header-section-number">1</span> Setup</a>
  <ul class="collapse">
  <li><a href="#libraries" id="toc-libraries" class="nav-link" data-scroll-target="#libraries"><span class="header-section-number">1.1</span> Libraries</a></li>
  <li><a href="#global-variables" id="toc-global-variables" class="nav-link" data-scroll-target="#global-variables"><span class="header-section-number">1.2</span> Global variables</a></li>
  </ul></li>
  <li><a href="#read-metadata" id="toc-read-metadata" class="nav-link" data-scroll-target="#read-metadata"><span class="header-section-number">2</span> Read metadata</a></li>
  <li><a href="#read-rbec-raw-counts-tables" id="toc-read-rbec-raw-counts-tables" class="nav-link" data-scroll-target="#read-rbec-raw-counts-tables"><span class="header-section-number">3</span> Read Rbec raw counts tables</a>
  <ul class="collapse">
  <li><a href="#untar-rbec-output-tarball" id="toc-untar-rbec-output-tarball" class="nav-link" data-scroll-target="#untar-rbec-output-tarball"><span class="header-section-number">3.1</span> Untar Rbec output tarball</a></li>
  <li><a href="#setup-directory-structure" id="toc-setup-directory-structure" class="nav-link" data-scroll-target="#setup-directory-structure"><span class="header-section-number">3.2</span> Setup directory structure</a></li>
  <li><a href="#read" id="toc-read" class="nav-link" data-scroll-target="#read"><span class="header-section-number">3.3</span> Read</a></li>
  <li><a href="#clean-up" id="toc-clean-up" class="nav-link" data-scroll-target="#clean-up"><span class="header-section-number">3.4</span> Clean up</a></li>
  </ul></li>
  <li><a href="#format" id="toc-format" class="nav-link" data-scroll-target="#format"><span class="header-section-number">4</span> Format</a></li>
  <li><a href="#analysis" id="toc-analysis" class="nav-link" data-scroll-target="#analysis"><span class="header-section-number">5</span> Analysis</a>
  <ul class="collapse">
  <li><a href="#negative-controls" id="toc-negative-controls" class="nav-link" data-scroll-target="#negative-controls"><span class="header-section-number">5.1</span> Negative controls</a></li>
  <li><a href="#positive-controls" id="toc-positive-controls" class="nav-link" data-scroll-target="#positive-controls"><span class="header-section-number">5.2</span> Positive controls</a></li>
  <li><a href="#misassigned-reads" id="toc-misassigned-reads" class="nav-link" data-scroll-target="#misassigned-reads"><span class="header-section-number">5.3</span> Misassigned reads</a></li>
  <li><a href="#samples-with-few-total-reads" id="toc-samples-with-few-total-reads" class="nav-link" data-scroll-target="#samples-with-few-total-reads"><span class="header-section-number">5.4</span> Samples with few total reads</a></li>
  <li><a href="#filter-to-target-species" id="toc-filter-to-target-species" class="nav-link" data-scroll-target="#filter-to-target-species"><span class="header-section-number">5.5</span> Filter to target species</a></li>
  <li><a href="#master-plates-samples" id="toc-master-plates-samples" class="nav-link" data-scroll-target="#master-plates-samples"><span class="header-section-number">5.6</span> Master plates samples</a></li>
  <li><a href="#experimental-samples" id="toc-experimental-samples" class="nav-link" data-scroll-target="#experimental-samples"><span class="header-section-number">5.7</span> Experimental samples</a></li>
  <li><a href="#plot" id="toc-plot" class="nav-link" data-scroll-target="#plot"><span class="header-section-number">5.8</span> Plot</a>
  <ul class="collapse">
  <li><a href="#pairs-and-trios" id="toc-pairs-and-trios" class="nav-link" data-scroll-target="#pairs-and-trios"><span class="header-section-number">5.8.1</span> Pairs and trios</a></li>
  <li><a href="#sec-octets2" id="toc-sec-octets2" class="nav-link" data-scroll-target="#sec-octets2"><span class="header-section-number">5.8.2</span> Octets</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#export" id="toc-export" class="nav-link" data-scroll-target="#export"><span class="header-section-number">6</span> Export</a>
  <ul class="collapse">
  <li><a href="#pairs" id="toc-pairs" class="nav-link" data-scroll-target="#pairs"><span class="header-section-number">6.1</span> Pairs</a></li>
  <li><a href="#trios" id="toc-trios" class="nav-link" data-scroll-target="#trios"><span class="header-section-number">6.2</span> Trios</a></li>
  <li><a href="#octets" id="toc-octets" class="nav-link" data-scroll-target="#octets"><span class="header-section-number">6.3</span> Octets</a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../../../../R/communities/community_composition/amplicon/20240711_BTK_illumina_v3/01_format_rbec_tab.html">4. Community growth - amplicon formatting</a></li><li class="breadcrumb-item"><a href="../../../../../R/communities/community_composition/amplicon/20250502_BTK_illumina_v3/01_format_rbec_tab.html">v) Batch 2025-05-02</a></li></ol></nav>
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">Formatting Rbec output from 2025-05-02 sequenced by BTK Turku using custom HAMBI Illumina v3 primers</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Milla Similä &amp; Shane Hogle </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">July 4, 2025</p>
    </div>
  </div>
  
    
  </div>
  
<div>
  <div class="abstract">
    <div class="block-title">Abstract</div>
    Results are from samples containing the mixture of 8 strains (both anc/evo form for each species) as well as one single triplet, and all the pairs with 50:50 abundance.
  </div>
</div>


</header>


<section id="setup" class="level1" data-number="1">
<h1 data-number="1"><span class="header-section-number">1</span> Setup</h1>
<section id="libraries" class="level2" data-number="1.1">
<h2 data-number="1.1" class="anchored" data-anchor-id="libraries"><span class="header-section-number">1.1</span> Libraries</h2>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-2"><a href="#cb1-2"></a><span class="fu">library</span>(here)</span>
<span id="cb1-3"><a href="#cb1-3"></a><span class="fu">library</span>(fs)</span>
<span id="cb1-4"><a href="#cb1-4"></a><span class="fu">library</span>(archive)</span>
<span id="cb1-5"><a href="#cb1-5"></a><span class="fu">library</span>(scales)</span>
<span id="cb1-6"><a href="#cb1-6"></a><span class="fu">source</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"R"</span>, <span class="st">"utils_generic.R"</span>))</span>
<span id="cb1-7"><a href="#cb1-7"></a><span class="fu">source</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"R"</span>, <span class="st">"communities"</span>, <span class="st">"community_composition"</span>, <span class="st">"amplicon"</span>, <span class="st">"utils_amplicon.R"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="global-variables" class="level2" data-number="1.2">
<h2 data-number="1.2" class="anchored" data-anchor-id="global-variables"><span class="header-section-number">1.2</span> Global variables</h2>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a>data_raw <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"_data_raw"</span>, <span class="st">"communities"</span>, <span class="st">"20250502_BTK_illumina_v3"</span>)</span>
<span id="cb2-2"><a href="#cb2-2"></a>data <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"communities"</span>, <span class="st">"20250502_BTK_illumina_v3"</span>)</span>
<span id="cb2-3"><a href="#cb2-3"></a>amplicontar <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(data_raw, <span class="st">"rbec_output.tar.gz"</span>)</span>
<span id="cb2-4"><a href="#cb2-4"></a></span>
<span id="cb2-5"><a href="#cb2-5"></a><span class="co"># make processed data directory if it doesn't exist</span></span>
<span id="cb2-6"><a href="#cb2-6"></a>fs<span class="sc">::</span><span class="fu">dir_create</span>(data)</span>
<span id="cb2-7"><a href="#cb2-7"></a></span>
<span id="cb2-8"><a href="#cb2-8"></a><span class="co"># create temporary location to decompress</span></span>
<span id="cb2-9"><a href="#cb2-9"></a>tmpdir <span class="ot">&lt;-</span> fs<span class="sc">::</span><span class="fu">file_temp</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</section>
<section id="read-metadata" class="level1" data-number="2">
<h1 data-number="2"><span class="header-section-number">2</span> Read metadata</h1>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1"></a>mddf <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(data_raw, <span class="st">"20250502_BTK_illumina_v3_metadata.tsv"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb3-2"><a href="#cb3-2"></a>  <span class="fu">mutate</span>(<span class="at">sample =</span> <span class="fu">str_trim</span>(sample))</span>
<span id="cb3-3"><a href="#cb3-3"></a>spdf <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(data_raw, <span class="st">"sample_compositions.tsv"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb3-4"><a href="#cb3-4"></a>  <span class="fu">mutate</span>(<span class="at">sample =</span> <span class="fu">str_trim</span>(sample))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="read-rbec-raw-counts-tables" class="level1" data-number="3">
<h1 data-number="3"><span class="header-section-number">3</span> Read Rbec raw counts tables</h1>
<section id="untar-rbec-output-tarball" class="level2" data-number="3.1">
<h2 data-number="3.1" class="anchored" data-anchor-id="untar-rbec-output-tarball"><span class="header-section-number">3.1</span> Untar Rbec output tarball</h2>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a>archive<span class="sc">::</span><span class="fu">archive_extract</span>(</span>
<span id="cb4-2"><a href="#cb4-2"></a>  amplicontar,</span>
<span id="cb4-3"><a href="#cb4-3"></a>  <span class="at">dir =</span> tmpdir,</span>
<span id="cb4-4"><a href="#cb4-4"></a>  <span class="at">files =</span> <span class="cn">NULL</span>,</span>
<span id="cb4-5"><a href="#cb4-5"></a>  <span class="at">options =</span> <span class="fu">character</span>(),</span>
<span id="cb4-6"><a href="#cb4-6"></a>  <span class="at">strip_components =</span> <span class="dv">0</span>L</span>
<span id="cb4-7"><a href="#cb4-7"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="setup-directory-structure" class="level2" data-number="3.2">
<h2 data-number="3.2" class="anchored" data-anchor-id="setup-directory-structure"><span class="header-section-number">3.2</span> Setup directory structure</h2>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1"></a>tabdir <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(tmpdir, <span class="st">"rbec_output"</span>)</span>
<span id="cb5-2"><a href="#cb5-2"></a>samppaths <span class="ot">&lt;-</span> fs<span class="sc">::</span><span class="fu">dir_ls</span>(tabdir)</span>
<span id="cb5-3"><a href="#cb5-3"></a>sampnames <span class="ot">&lt;-</span> <span class="fu">path_split</span>(samppaths) <span class="sc">%&gt;%</span> </span>
<span id="cb5-4"><a href="#cb5-4"></a>  <span class="fu">map_chr</span>(dplyr<span class="sc">::</span>last)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="read" class="level2" data-number="3.3">
<h2 data-number="3.3" class="anchored" data-anchor-id="read"><span class="header-section-number">3.3</span> Read</h2>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1"></a>straintabs <span class="ot">&lt;-</span> <span class="fu">paste0</span>(samppaths, <span class="st">"/strain_table.txt"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb6-2"><a href="#cb6-2"></a>  <span class="fu">set_names</span>(sampnames) <span class="sc">%&gt;%</span> </span>
<span id="cb6-3"><a href="#cb6-3"></a>  <span class="fu">map_df</span>(</span>
<span id="cb6-4"><a href="#cb6-4"></a>  read_tsv,</span>
<span id="cb6-5"><a href="#cb6-5"></a>  <span class="at">skip =</span> <span class="dv">1</span>,</span>
<span id="cb6-6"><a href="#cb6-6"></a>  <span class="at">col_names =</span> <span class="fu">c</span>(<span class="st">"rRNA16S_locus"</span>,<span class="st">"count"</span>),</span>
<span id="cb6-7"><a href="#cb6-7"></a>  <span class="at">show_col_types =</span> <span class="cn">FALSE</span>, </span>
<span id="cb6-8"><a href="#cb6-8"></a>  <span class="at">.id =</span> <span class="st">"sample"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="clean-up" class="level2" data-number="3.4">
<h2 data-number="3.4" class="anchored" data-anchor-id="clean-up"><span class="header-section-number">3.4</span> Clean up</h2>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1"></a><span class="co"># remove decompressed directory from temp location</span></span>
<span id="cb7-2"><a href="#cb7-2"></a>fs<span class="sc">::</span><span class="fu">dir_delete</span>(tmpdir)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</section>
<section id="format" class="level1" data-number="4">
<h1 data-number="4"><span class="header-section-number">4</span> Format</h1>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1"></a>straintabs_norm <span class="ot">&lt;-</span> straintabs <span class="sc">%&gt;%</span> </span>
<span id="cb8-2"><a href="#cb8-2"></a>  <span class="co"># Calls function that normalize counts by 16S copy number</span></span>
<span id="cb8-3"><a href="#cb8-3"></a>  <span class="fu">normalize_by_copy</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb8-4"><a href="#cb8-4"></a>  <span class="co"># Calls function that completes all combinations of 23 species</span></span>
<span id="cb8-5"><a href="#cb8-5"></a>  <span class="co"># this is important because some species go extinct that should be in the samples</span></span>
<span id="cb8-6"><a href="#cb8-6"></a>  <span class="co"># and we need to have those in the final table</span></span>
<span id="cb8-7"><a href="#cb8-7"></a>  <span class="fu">complete_combos</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb8-8"><a href="#cb8-8"></a>  <span class="co"># removes any extra whitespace on sample names</span></span>
<span id="cb8-9"><a href="#cb8-9"></a>  <span class="fu">mutate</span>(<span class="at">sample =</span> <span class="fu">str_trim</span>(sample))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Make final table</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1"></a><span class="co"># Later we will take advantage of the fact that for species not supposed to be</span></span>
<span id="cb9-2"><a href="#cb9-2"></a><span class="co"># in a sample the prior left_join will have filled the evo_hist category with an</span></span>
<span id="cb9-3"><a href="#cb9-3"></a><span class="co"># NA. We can then filter using this NA value</span></span>
<span id="cb9-4"><a href="#cb9-4"></a>finaltable <span class="ot">&lt;-</span> <span class="fu">left_join</span>(straintabs_norm, mddf, <span class="at">by =</span> <span class="fu">join_by</span>(sample)) <span class="sc">%&gt;%</span> </span>
<span id="cb9-5"><a href="#cb9-5"></a>  <span class="fu">left_join</span>(spdf, <span class="at">by =</span> <span class="fu">join_by</span>(sample, strainID)) <span class="sc">%&gt;%</span> </span>
<span id="cb9-6"><a href="#cb9-6"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb9-7"><a href="#cb9-7"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>genus, <span class="sc">-</span>species) <span class="sc">%&gt;%</span> </span>
<span id="cb9-8"><a href="#cb9-8"></a>  <span class="fu">mutate</span>(<span class="at">f_raw =</span> count_correct<span class="sc">/</span><span class="fu">sum</span>(count_correct),</span>
<span id="cb9-9"><a href="#cb9-9"></a>         <span class="at">sample_tot_raw =</span> <span class="fu">sum</span>(count_correct)) <span class="sc">%&gt;%</span> </span>
<span id="cb9-10"><a href="#cb9-10"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb9-11"><a href="#cb9-11"></a>  <span class="fu">relocate</span>(sample_tot_raw, f_raw, target_f, evo_hist, <span class="at">.after =</span> count_correct)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="analysis" class="level1" data-number="5">
<h1 data-number="5"><span class="header-section-number">5</span> Analysis</h1>
<p>Let’s quickly compare and look the at number of reads from the different kinds of experiment categories. Negative controls are samples from the experiment that contained growth medium but no cells. We included them to make sure that there was not contamination between wells during the experiment. Positive controls contain a mixsture of all 23 species (not sure why this was done), and some wells are just empty but were processed/sequenced anyway for convenience of the person preparing the libraries. Experiment samples are those that came from the actual experiment, and masterplate samples are the defined mixtures we made to inoculate the experiment. We sequenced these masterplates so that we would have T0 information for each experimental treatment.</p>
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1"></a>finaltable <span class="sc">%&gt;%</span> </span>
<span id="cb10-2"><a href="#cb10-2"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(sample, sample_tot_raw, community_type) <span class="sc">%&gt;%</span> </span>
<span id="cb10-3"><a href="#cb10-3"></a>  <span class="fu">distinct</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb10-4"><a href="#cb10-4"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> sample_tot_raw)) <span class="sc">+</span></span>
<span id="cb10-5"><a href="#cb10-5"></a>  <span class="fu">geom_histogram</span>(<span class="at">bins =</span> <span class="dv">20</span>, <span class="fu">aes</span>(<span class="at">fill =</span> community_type)) <span class="sc">+</span></span>
<span id="cb10-6"><a href="#cb10-6"></a>  <span class="fu">scale_x_log10</span>() <span class="sc">+</span> </span>
<span id="cb10-7"><a href="#cb10-7"></a>  <span class="fu">annotation_logticks</span>(<span class="at">sides =</span> <span class="st">"b"</span>, <span class="at">color=</span><span class="st">"grey30"</span>) <span class="sc">+</span></span>
<span id="cb10-8"><a href="#cb10-8"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">"Sample count"</span>, <span class="at">x =</span> <span class="st">"Total reads per sample"</span>)  <span class="sc">+</span></span>
<span id="cb10-9"><a href="#cb10-9"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span> community_type, <span class="at">scales=</span><span class="st">"free_y"</span>, <span class="at">nrow =</span> <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div id="fig-01" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-01-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="01_format_rbec_tab_files/figure-html/unnamed-chunk-10-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-01-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;1: Frequency distribution of total read counts per sample in the experimental samples (red) and the negative controls (blue).
</figcaption>
</figure>
</div>
<p>The smallest number of reads in the experimental samples is 340 and the mean is 1.5133^{4} which is considerably more than the smallest number of reads 99 and the mean 199.6 in the negative controls.</p>
<p>Overall this looks pretty good. From the figure above we can see that there is a clear disjoint between about 1000 reads that separate samples that are negative controls and the experimental samples. There is one experimental sample with fewer than 1000 reads. It is likely that this sample failed or it is from a condition where no species survived.</p>
<section id="negative-controls" class="level2" data-number="5.1">
<h2 data-number="5.1" class="anchored" data-anchor-id="negative-controls"><span class="header-section-number">5.1</span> Negative controls</h2>
<p>Read optical density data for the experiment</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1"></a>OD <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"_data_raw"</span>, <span class="st">"communities"</span>, <span class="st">"optical_density"</span>, <span class="st">"20250425_8sp_5050pairs"</span>, <span class="st">"optical_density_formatted.tsv"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb11-2"><a href="#cb11-2"></a>  <span class="fu">mutate</span>(<span class="at">sample2 =</span> <span class="fu">case_when</span>(<span class="fu">str_detect</span>(sample, <span class="st">"com_"</span>) <span class="sc">~</span> <span class="fu">paste0</span>(<span class="st">"8"</span>, <span class="fu">str_extract</span>(sample, <span class="st">"_</span><span class="sc">\\</span><span class="st">d+$"</span>)),</span>
<span id="cb11-3"><a href="#cb11-3"></a>                             <span class="fu">str_detect</span>(sample, <span class="st">"^3_p"</span>) <span class="sc">~</span> <span class="fu">paste0</span>(<span class="st">"3_c01_"</span>, <span class="fu">str_extract</span>(well, <span class="st">"[:upper:]"</span>), <span class="fu">str_pad</span>(<span class="fu">str_extract</span>(well, <span class="st">"</span><span class="sc">\\</span><span class="st">d+"</span>), <span class="dv">2</span>, <span class="st">"left"</span>, <span class="at">pad =</span> <span class="st">"0"</span>)),</span>
<span id="cb11-4"><a href="#cb11-4"></a>                             <span class="fu">str_detect</span>(sample, <span class="st">"^p</span><span class="sc">\\</span><span class="st">d+"</span>) <span class="sc">~</span> <span class="fu">paste0</span>(<span class="st">"2_"</span>, <span class="fu">str_extract</span>(sample, <span class="st">"</span><span class="sc">\\</span><span class="st">d+"</span>), <span class="st">"_"</span>, <span class="fu">str_extract</span>(well, <span class="st">"[:upper:]"</span>), <span class="fu">str_pad</span>(<span class="fu">str_extract</span>(well, <span class="st">"</span><span class="sc">\\</span><span class="st">d+"</span>), <span class="dv">2</span>, <span class="st">"left"</span>, <span class="at">pad =</span> <span class="st">"0"</span>)),</span>
<span id="cb11-5"><a href="#cb11-5"></a>                             <span class="fu">str_detect</span>(sample, <span class="st">"^p"</span>) <span class="sc">~</span> <span class="fu">str_to_upper</span>(sample)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 960 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: "\t"
chr (2): sample, well
dbl (5): transfers, n_species, strep_conc, replicate, OD

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
</div>
<p>First just checking the negative controls</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1"></a>OD <span class="sc">%&gt;%</span> </span>
<span id="cb13-2"><a href="#cb13-2"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(sample, <span class="st">"neg_control"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb13-3"><a href="#cb13-3"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> OD)) <span class="sc">+</span></span>
<span id="cb13-4"><a href="#cb13-4"></a>  <span class="fu">geom_histogram</span>(<span class="at">bins =</span> <span class="dv">10</span>) <span class="sc">+</span></span>
<span id="cb13-5"><a href="#cb13-5"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fl">0.1</span>, <span class="at">color =</span> <span class="st">"red"</span>) <span class="sc">+</span> </span>
<span id="cb13-6"><a href="#cb13-6"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">"Sample count"</span>, <span class="at">x =</span> <span class="st">"Sample optical density"</span>)  <span class="sc">+</span></span>
<span id="cb13-7"><a href="#cb13-7"></a>  <span class="fu">facet_grid</span>(<span class="sc">~</span> transfers, <span class="at">scales=</span><span class="st">"free_y"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="01_format_rbec_tab_files/figure-html/unnamed-chunk-12-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Negative controls look good. OD in all negative controls is low (&lt;0.07)</p>
</section>
<section id="positive-controls" class="level2" data-number="5.2">
<h2 data-number="5.2" class="anchored" data-anchor-id="positive-controls"><span class="header-section-number">5.2</span> Positive controls</h2>
<p>No positive controls included in this run</p>
</section>
<section id="misassigned-reads" class="level2" data-number="5.3">
<h2 data-number="5.3" class="anchored" data-anchor-id="misassigned-reads"><span class="header-section-number">5.3</span> Misassigned reads</h2>
<p>These libraries were only prepared with samples containing HAMBI_0403, HAMBI_1287, HAMBI_1896, and HAMBI_1977 so any time species other than these show up is just an incorrect assignment by Rbec or index leakage/crosstalk from the positive community controls. Let’s check quickly how many of these there are…</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1"></a>finaltable <span class="sc">%&gt;%</span> </span>
<span id="cb14-2"><a href="#cb14-2"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">str_detect</span>(strainID, <span class="st">"0403|1287|1896|1977"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb14-3"><a href="#cb14-3"></a>  <span class="fu">filter</span>(count_correct <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb14-4"><a href="#cb14-4"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(count_correct))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["sample"],"name":[1],"type":["chr"],"align":["left"]},{"label":["strainID"],"name":[2],"type":["chr"],"align":["left"]},{"label":["count"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["count_correct"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["sample_tot_raw"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["f_raw"],"name":[6],"type":["dbl"],"align":["right"]},{"label":["target_f"],"name":[7],"type":["dbl"],"align":["right"]},{"label":["evo_hist"],"name":[8],"type":["chr"],"align":["left"]},{"label":["n_species"],"name":[9],"type":["dbl"],"align":["right"]},{"label":["community_type"],"name":[10],"type":["chr"],"align":["left"]},{"label":["transfers"],"name":[11],"type":["dbl"],"align":["right"]},{"label":["strep_conc"],"name":[12],"type":["dbl"],"align":["right"]},{"label":["replicate"],"name":[13],"type":["dbl"],"align":["right"]},{"label":["plate_well"],"name":[14],"type":["chr"],"align":["left"]},{"label":["plate_no"],"name":[15],"type":["dbl"],"align":["right"]},{"label":["notes"],"name":[16],"type":["chr"],"align":["left"]},{"label":["microcosm_no"],"name":[17],"type":["chr"],"align":["left"]}],"data":[{"1":"8_08_H01","2":"HAMBI_0006","3":"44","4":"38","5":"23392","6":"1.624487e-03","7":"NA","8":"NA","9":"8","10":"experiment","11":"8","12":"0","13":"1","14":"H01","15":"1","16":"NA","17":"08"},{"1":"2_07","2":"HAMBI_0006","3":"32","4":"30","5":"9833","6":"3.050951e-03","7":"NA","8":"NA","9":"2","10":"master","11":"0","12":"0","13":"1","14":"NA","15":"NA","16":"NA","17":"07"},{"1":"8_16_H02","2":"HAMBI_0006","3":"31","4":"29","5":"23808","6":"1.218078e-03","7":"NA","8":"NA","9":"8","10":"experiment","11":"8","12":"0","13":"1","14":"H02","15":"1","16":"NA","17":"16"},{"1":"8_08_H04","2":"HAMBI_0006","3":"29","4":"28","5":"22090","6":"1.267542e-03","7":"NA","8":"NA","9":"8","10":"experiment","11":"8","12":"16","13":"1","14":"H04","15":"1","16":"NA","17":"08"},{"1":"8_09","2":"HAMBI_0006","3":"30","4":"26","5":"17091","6":"1.521269e-03","7":"NA","8":"NA","9":"8","10":"master","11":"0","12":"0","13":"1","14":"NA","15":"NA","16":"NA","17":"09"},{"1":"2_02","2":"HAMBI_1972","3":"28","4":"20","5":"11933","6":"1.676024e-03","7":"NA","8":"NA","9":"2","10":"master","11":"0","12":"0","13":"1","14":"NA","15":"NA","16":"NA","17":"02"},{"1":"3_c01_H03","2":"HAMBI_0006","3":"19","4":"20","5":"8491","6":"2.355435e-03","7":"NA","8":"NA","9":"3","10":"experiment","11":"8","12":"0","13":"2","14":"H03","15":"1","16":"NA","17":"NA"},{"1":"8_17","2":"HAMBI_0006","3":"24","4":"20","5":"18983","6":"1.053574e-03","7":"NA","8":"NA","9":"8","10":"master","11":"0","12":"0","13":"1","14":"NA","15":"NA","16":"NA","17":"17"},{"1":"8_04","2":"HAMBI_1972","3":"30","4":"19","5":"25884","6":"7.340442e-04","7":"NA","8":"NA","9":"8","10":"master","11":"0","12":"0","13":"1","14":"NA","15":"NA","16":"NA","17":"04"},{"1":"8_16_H05","2":"HAMBI_0006","3":"19","4":"18","5":"28861","6":"6.236790e-04","7":"NA","8":"NA","9":"8","10":"experiment","11":"8","12":"16","13":"1","14":"H05","15":"1","16":"NA","17":"16"},{"1":"8_E03","2":"HAMBI_1972","3":"28","4":"18","5":"150","6":"1.200000e-01","7":"NA","8":"NA","9":"8","10":"neg_control","11":"8","12":"0","13":"NA","14":"E03","15":"1","16":"NA","17":"NA"},{"1":"8_E06","2":"HAMBI_1972","3":"25","4":"15","5":"176","6":"8.522727e-02","7":"NA","8":"NA","9":"8","10":"neg_control","11":"8","12":"16","13":"NA","14":"E06","15":"1","16":"NA","17":"NA"},{"1":"2_09","2":"HAMBI_2164","3":"12","4":"13","5":"9159","6":"1.419369e-03","7":"NA","8":"NA","9":"2","10":"master","11":"0","12":"0","13":"1","14":"NA","15":"NA","16":"NA","17":"09"},{"1":"8_05_E01","2":"HAMBI_1972","3":"23","4":"13","5":"29112","6":"4.465513e-04","7":"NA","8":"NA","9":"8","10":"experiment","11":"8","12":"0","13":"1","14":"E01","15":"1","16":"NA","17":"05"},{"1":"8_14_F02","2":"HAMBI_1972","3":"20","4":"13","5":"20547","6":"6.326958e-04","7":"NA","8":"NA","9":"8","10":"experiment","11":"8","12":"0","13":"1","14":"F02","15":"1","16":"NA","17":"14"},{"1":"8_03","2":"HAMBI_0006","3":"13","4":"12","5":"21558","6":"5.566379e-04","7":"NA","8":"NA","9":"8","10":"master","11":"0","12":"0","13":"1","14":"NA","15":"NA","16":"NA","17":"03"},{"1":"8_06_F01","2":"HAMBI_1972","3":"18","4":"12","5":"20174","6":"5.948250e-04","7":"NA","8":"NA","9":"8","10":"experiment","11":"8","12":"0","13":"1","14":"F01","15":"1","16":"NA","17":"06"},{"1":"8_F06","2":"HAMBI_1972","3":"19","4":"12","5":"132","6":"9.090909e-02","7":"NA","8":"NA","9":"8","10":"neg_control","11":"8","12":"16","13":"NA","14":"F06","15":"1","16":"NA","17":"NA"},{"1":"2_15","2":"HAMBI_0006","3":"14","4":"11","5":"13704","6":"8.026853e-04","7":"NA","8":"NA","9":"2","10":"master","11":"0","12":"0","13":"1","14":"NA","15":"NA","16":"NA","17":"15"},{"1":"2_08","2":"HAMBI_1972","3":"14","4":"9","5":"11079","6":"8.123477e-04","7":"NA","8":"NA","9":"2","10":"master","11":"0","12":"0","13":"1","14":"NA","15":"NA","16":"NA","17":"08"},{"1":"8_11","2":"HAMBI_0006","3":"11","4":"9","5":"29378","6":"3.063517e-04","7":"NA","8":"NA","9":"8","10":"master","11":"0","12":"0","13":"1","14":"NA","15":"NA","16":"NA","17":"11"},{"1":"2_10","2":"HAMBI_2164","3":"7","4":"8","5":"10118","6":"7.906701e-04","7":"NA","8":"NA","9":"2","10":"master","11":"0","12":"0","13":"1","14":"NA","15":"NA","16":"NA","17":"10"},{"1":"8_12","2":"HAMBI_0006","3":"10","4":"8","5":"19372","6":"4.129672e-04","7":"NA","8":"NA","9":"8","10":"master","11":"0","12":"0","13":"1","14":"NA","15":"NA","16":"NA","17":"12"},{"1":"2_01","2":"HAMBI_1972","3":"11","4":"7","5":"15186","6":"4.609509e-04","7":"NA","8":"NA","9":"2","10":"master","11":"0","12":"0","13":"1","14":"NA","15":"NA","16":"NA","17":"01"},{"1":"2_09","2":"HAMBI_1972","3":"13","4":"7","5":"9159","6":"7.642756e-04","7":"NA","8":"NA","9":"2","10":"master","11":"0","12":"0","13":"1","14":"NA","15":"NA","16":"NA","17":"09"},{"1":"8_13_E02","2":"HAMBI_1972","3":"12","4":"7","5":"25635","6":"2.730642e-04","7":"NA","8":"NA","9":"8","10":"experiment","11":"8","12":"0","13":"1","14":"E02","15":"1","16":"NA","17":"13"},{"1":"8_F03","2":"HAMBI_1972","3":"11","4":"7","5":"99","6":"7.070707e-02","7":"NA","8":"NA","9":"8","10":"neg_control","11":"8","12":"0","13":"NA","14":"F03","15":"1","16":"NA","17":"NA"},{"1":"2_01","2":"HAMBI_0006","3":"6","4":"6","5":"15186","6":"3.951008e-04","7":"NA","8":"NA","9":"2","10":"master","11":"0","12":"0","13":"1","14":"NA","15":"NA","16":"NA","17":"01"},{"1":"2_10","2":"HAMBI_1972","3":"10","4":"6","5":"10118","6":"5.930026e-04","7":"NA","8":"NA","9":"2","10":"master","11":"0","12":"0","13":"1","14":"NA","15":"NA","16":"NA","17":"10"},{"1":"8_10","2":"HAMBI_0006","3":"7","4":"6","5":"21754","6":"2.758113e-04","7":"NA","8":"NA","9":"8","10":"master","11":"0","12":"0","13":"1","14":"NA","15":"NA","16":"NA","17":"10"},{"1":"3_c01","2":"HAMBI_1972","3":"11","4":"5","5":"8287","6":"6.033547e-04","7":"NA","8":"NA","9":"3","10":"master","11":"0","12":"0","13":"1","14":"NA","15":"NA","16":"3sp community from well c01 in previous round","17":"NA"},{"1":"8_02","2":"HAMBI_0006","3":"5","4":"5","5":"16918","6":"2.955432e-04","7":"NA","8":"NA","9":"8","10":"master","11":"0","12":"0","13":"1","14":"NA","15":"NA","16":"NA","17":"02"},{"1":"8_03","2":"HAMBI_1972","3":"8","4":"5","5":"21558","6":"2.319325e-04","7":"NA","8":"NA","9":"8","10":"master","11":"0","12":"0","13":"1","14":"NA","15":"NA","16":"NA","17":"03"},{"1":"8_07_G01","2":"HAMBI_1972","3":"8","4":"5","5":"24853","6":"2.011830e-04","7":"NA","8":"NA","9":"8","10":"experiment","11":"8","12":"0","13":"1","14":"G01","15":"1","16":"NA","17":"07"},{"1":"8_15_G02","2":"HAMBI_1972","3":"8","4":"5","5":"20665","6":"2.419550e-04","7":"NA","8":"NA","9":"8","10":"experiment","11":"8","12":"0","13":"1","14":"G02","15":"1","16":"NA","17":"15"},{"1":"2_08","2":"HAMBI_1299","3":"5","4":"4","5":"11079","6":"3.610434e-04","7":"NA","8":"NA","9":"2","10":"master","11":"0","12":"0","13":"1","14":"NA","15":"NA","16":"NA","17":"08"},{"1":"2_08","2":"HAMBI_2164","3":"3","4":"4","5":"11079","6":"3.610434e-04","7":"NA","8":"NA","9":"2","10":"master","11":"0","12":"0","13":"1","14":"NA","15":"NA","16":"NA","17":"08"},{"1":"2_09","2":"HAMBI_1842","3":"3","4":"4","5":"9159","6":"4.367289e-04","7":"NA","8":"NA","9":"2","10":"master","11":"0","12":"0","13":"1","14":"NA","15":"NA","16":"NA","17":"09"},{"1":"2_16","2":"HAMBI_0006","3":"6","4":"4","5":"16202","6":"2.468831e-04","7":"NA","8":"NA","9":"2","10":"master","11":"0","12":"0","13":"1","14":"NA","15":"NA","16":"NA","17":"16"},{"1":"2_24","2":"HAMBI_1972","3":"8","4":"4","5":"8406","6":"4.758506e-04","7":"NA","8":"NA","9":"2","10":"master","11":"0","12":"0","13":"1","14":"NA","15":"NA","16":"NA","17":"24"},{"1":"8_05_E04","2":"HAMBI_1299","3":"5","4":"4","5":"22161","6":"1.804973e-04","7":"NA","8":"NA","9":"8","10":"experiment","11":"8","12":"16","13":"1","14":"E04","15":"1","16":"NA","17":"05"},{"1":"2_17","2":"HAMBI_0006","3":"4","4":"3","5":"16171","6":"1.855173e-04","7":"NA","8":"NA","9":"2","10":"master","11":"0","12":"0","13":"1","14":"NA","15":"NA","16":"NA","17":"17"},{"1":"3_c01_G06","2":"HAMBI_1972","3":"6","4":"3","5":"10107","6":"2.968240e-04","7":"NA","8":"NA","9":"3","10":"experiment","11":"8","12":"16","13":"1","14":"G06","15":"1","16":"NA","17":"NA"},{"1":"3_c01_H06","2":"HAMBI_0006","3":"4","4":"3","5":"10326","6":"2.905288e-04","7":"NA","8":"NA","9":"3","10":"experiment","11":"8","12":"16","13":"2","14":"H06","15":"1","16":"NA","17":"NA"},{"1":"8_05_E04","2":"HAMBI_1972","3":"5","4":"3","5":"22161","6":"1.353730e-04","7":"NA","8":"NA","9":"8","10":"experiment","11":"8","12":"16","13":"1","14":"E04","15":"1","16":"NA","17":"05"},{"1":"3_c01_G03","2":"HAMBI_1972","3":"5","4":"2","5":"8441","6":"2.369388e-04","7":"NA","8":"NA","9":"3","10":"experiment","11":"8","12":"0","13":"1","14":"G03","15":"1","16":"NA","17":"NA"},{"1":"8_01","2":"HAMBI_1972","3":"3","4":"2","5":"21549","6":"9.281173e-05","7":"NA","8":"NA","9":"8","10":"master","11":"0","12":"0","13":"1","14":"NA","15":"NA","16":"NA","17":"01"},{"1":"8_06_F04","2":"HAMBI_1972","3":"3","4":"2","5":"19964","6":"1.001803e-04","7":"NA","8":"NA","9":"8","10":"experiment","11":"8","12":"16","13":"1","14":"F04","15":"1","16":"NA","17":"06"},{"1":"8_07_G01","2":"HAMBI_0006","3":"3","4":"2","5":"24853","6":"8.047318e-05","7":"NA","8":"NA","9":"8","10":"experiment","11":"8","12":"0","13":"1","14":"G01","15":"1","16":"NA","17":"07"},{"1":"8_14_F05","2":"HAMBI_1842","3":"1","4":"2","5":"24839","6":"8.051854e-05","7":"NA","8":"NA","9":"8","10":"experiment","11":"8","12":"16","13":"1","14":"F05","15":"1","16":"NA","17":"14"},{"1":"8_15_G02","2":"HAMBI_0006","3":"2","4":"2","5":"20665","6":"9.678200e-05","7":"NA","8":"NA","9":"8","10":"experiment","11":"8","12":"0","13":"1","14":"G02","15":"1","16":"NA","17":"15"},{"1":"8_15_G02","2":"HAMBI_2443","3":"1","4":"2","5":"20665","6":"9.678200e-05","7":"NA","8":"NA","9":"8","10":"experiment","11":"8","12":"0","13":"1","14":"G02","15":"1","16":"NA","17":"15"},{"1":"8_16_H02","2":"HAMBI_2160","3":"1","4":"2","5":"23808","6":"8.400538e-05","7":"NA","8":"NA","9":"8","10":"experiment","11":"8","12":"0","13":"1","14":"H02","15":"1","16":"NA","17":"16"},{"1":"8_17","2":"HAMBI_2160","3":"1","4":"2","5":"18983","6":"1.053574e-04","7":"NA","8":"NA","9":"8","10":"master","11":"0","12":"0","13":"1","14":"NA","15":"NA","16":"NA","17":"17"},{"1":"8_E03","2":"HAMBI_1842","3":"1","4":"2","5":"150","6":"1.333333e-02","7":"NA","8":"NA","9":"8","10":"neg_control","11":"8","12":"0","13":"NA","14":"E03","15":"1","16":"NA","17":"NA"},{"1":"2_09","2":"HAMBI_1299","3":"1","4":"1","5":"9159","6":"1.091822e-04","7":"NA","8":"NA","9":"2","10":"master","11":"0","12":"0","13":"1","14":"NA","15":"NA","16":"NA","17":"09"},{"1":"3_c01","2":"HAMBI_0006","3":"1","4":"1","5":"8287","6":"1.206709e-04","7":"NA","8":"NA","9":"3","10":"master","11":"0","12":"0","13":"1","14":"NA","15":"NA","16":"3sp community from well c01 in previous round","17":"NA"},{"1":"8_01","2":"HAMBI_0006","3":"1","4":"1","5":"21549","6":"4.640587e-05","7":"NA","8":"NA","9":"8","10":"master","11":"0","12":"0","13":"1","14":"NA","15":"NA","16":"NA","17":"01"},{"1":"8_02","2":"HAMBI_2164","3":"1","4":"1","5":"16918","6":"5.910864e-05","7":"NA","8":"NA","9":"8","10":"master","11":"0","12":"0","13":"1","14":"NA","15":"NA","16":"NA","17":"02"},{"1":"8_03","2":"HAMBI_2164","3":"1","4":"1","5":"21558","6":"4.638649e-05","7":"NA","8":"NA","9":"8","10":"master","11":"0","12":"0","13":"1","14":"NA","15":"NA","16":"NA","17":"03"},{"1":"8_04","2":"HAMBI_0006","3":"1","4":"1","5":"25884","6":"3.863391e-05","7":"NA","8":"NA","9":"8","10":"master","11":"0","12":"0","13":"1","14":"NA","15":"NA","16":"NA","17":"04"},{"1":"8_05_E01","2":"HAMBI_2164","3":"1","4":"1","5":"29112","6":"3.435010e-05","7":"NA","8":"NA","9":"8","10":"experiment","11":"8","12":"0","13":"1","14":"E01","15":"1","16":"NA","17":"05"},{"1":"8_09","2":"HAMBI_2164","3":"1","4":"1","5":"17091","6":"5.851033e-05","7":"NA","8":"NA","9":"8","10":"master","11":"0","12":"0","13":"1","14":"NA","15":"NA","16":"NA","17":"09"},{"1":"8_11","2":"HAMBI_2164","3":"1","4":"1","5":"29378","6":"3.403908e-05","7":"NA","8":"NA","9":"8","10":"master","11":"0","12":"0","13":"1","14":"NA","15":"NA","16":"NA","17":"11"},{"1":"8_13_E05","2":"HAMBI_1972","3":"2","4":"1","5":"26305","6":"3.801559e-05","7":"NA","8":"NA","9":"8","10":"experiment","11":"8","12":"16","13":"1","14":"E05","15":"1","16":"NA","17":"13"},{"1":"8_14_F05","2":"HAMBI_1972","3":"2","4":"1","5":"24839","6":"4.025927e-05","7":"NA","8":"NA","9":"8","10":"experiment","11":"8","12":"16","13":"1","14":"F05","15":"1","16":"NA","17":"14"},{"1":"8_16_H02","2":"HAMBI_2164","3":"1","4":"1","5":"23808","6":"4.200269e-05","7":"NA","8":"NA","9":"8","10":"experiment","11":"8","12":"0","13":"1","14":"H02","15":"1","16":"NA","17":"16"},{"1":"8_16_H05","2":"HAMBI_1972","3":"1","4":"1","5":"28861","6":"3.464883e-05","7":"NA","8":"NA","9":"8","10":"experiment","11":"8","12":"16","13":"1","14":"H05","15":"1","16":"NA","17":"16"},{"1":"8_17","2":"HAMBI_1972","3":"1","4":"1","5":"18983","6":"5.267871e-05","7":"NA","8":"NA","9":"8","10":"master","11":"0","12":"0","13":"1","14":"NA","15":"NA","16":"NA","17":"17"},{"1":"8_E03","2":"HAMBI_1299","3":"1","4":"1","5":"150","6":"6.666667e-03","7":"NA","8":"NA","9":"8","10":"neg_control","11":"8","12":"0","13":"NA","14":"E03","15":"1","16":"NA","17":"NA"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<p>There are only a couple of random reads per experimental sample so we can safely ignore them. Since there was no positive control included here the off-target species must come from missassignments and not index leakage from community control.</p>
</section>
<section id="samples-with-few-total-reads" class="level2" data-number="5.4">
<h2 data-number="5.4" class="anchored" data-anchor-id="samples-with-few-total-reads"><span class="header-section-number">5.4</span> Samples with few total reads</h2>
<p>Some of the experimental pairs had streptomycin concentrations higher than any of the species individually could tolerate. We would expect then that no species would grow successfully in these samples and that the overall biomass would be very low, thus resulting in a low number of recovered reads from these samples.</p>
<p>To look into this. first let’s check which samples have very low OD600 in the endpoint samples. We’ll filter out samples with an OD of less than 0.1 or samples with fewer than 3000 reads. It is generally good practice to exclude samples with low number of reads.</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1"></a><span class="fu">left_join</span>(finaltable, OD, </span>
<span id="cb15-2"><a href="#cb15-2"></a>                 <span class="at">by =</span> <span class="fu">join_by</span>(sample <span class="sc">==</span> sample2)) <span class="sc">%&gt;%</span> </span>
<span id="cb15-3"><a href="#cb15-3"></a>  <span class="fu">filter</span>(transfers.y <span class="sc">==</span> <span class="dv">8</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb15-4"><a href="#cb15-4"></a>  <span class="fu">filter</span>(OD <span class="sc">&lt;</span> <span class="fl">0.1</span> <span class="sc">|</span> sample_tot_raw <span class="sc">&lt;</span> <span class="dv">1000</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb15-5"><a href="#cb15-5"></a>  <span class="fu">distinct</span>(sample, community_type, OD, sample_tot_raw) <span class="sc">%&gt;%</span></span>
<span id="cb15-6"><a href="#cb15-6"></a>  <span class="fu">arrange</span>(sample_tot_raw)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in left_join(finaltable, OD, by = join_by(sample == sample2)): Detected an unexpected many-to-many relationship between `x` and `y`.
ℹ Row 24 of `x` matches multiple rows in `y`.
ℹ Row 97 of `y` matches multiple rows in `x`.
ℹ If a many-to-many relationship is expected, set `relationship =
  "many-to-many"` to silence this warning.</code></pre>
</div>
<div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["sample"],"name":[1],"type":["chr"],"align":["left"]},{"label":["community_type"],"name":[2],"type":["chr"],"align":["left"]},{"label":["OD"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["sample_tot_raw"],"name":[4],"type":["dbl"],"align":["right"]}],"data":[{"1":"2_01_A10","2":"experiment","3":"0.046","4":"340"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1"></a>df <span class="ot">&lt;-</span> <span class="fu">left_join</span>(finaltable, OD, </span>
<span id="cb17-2"><a href="#cb17-2"></a>                 <span class="at">by =</span> <span class="fu">join_by</span>(sample <span class="sc">==</span> sample2)) <span class="sc">%&gt;%</span> </span>
<span id="cb17-3"><a href="#cb17-3"></a>  <span class="fu">filter</span>(transfers.y <span class="sc">==</span> <span class="dv">8</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb17-4"><a href="#cb17-4"></a>  <span class="fu">filter</span>(OD <span class="sc">&lt;</span> <span class="fl">0.1</span> <span class="sc">|</span> sample_tot_raw <span class="sc">&lt;</span> <span class="dv">1000</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb17-5"><a href="#cb17-5"></a>  <span class="fu">distinct</span>(sample, community_type, OD, sample_tot_raw) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in left_join(finaltable, OD, by = join_by(sample == sample2)): Detected an unexpected many-to-many relationship between `x` and `y`.
ℹ Row 24 of `x` matches multiple rows in `y`.
ℹ Row 97 of `y` matches multiple rows in `x`.
ℹ If a many-to-many relationship is expected, set `relationship =
  "many-to-many"` to silence this warning.</code></pre>
</div>
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1"></a>lowread_samps <span class="ot">&lt;-</span> df <span class="sc">%&gt;%</span> </span>
<span id="cb19-2"><a href="#cb19-2"></a>  <span class="fu">filter</span>(sample_tot_raw <span class="sc">&lt;</span> <span class="dv">1000</span> <span class="sc">&amp;</span> community_type <span class="sc">==</span> <span class="st">"experiment"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb19-3"><a href="#cb19-3"></a>  <span class="fu">distinct</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb19-4"><a href="#cb19-4"></a>  <span class="fu">pull</span>(sample)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>There is only one low-read sample. It corresponds to a pair with both ancestral HAMBI_1977 and HAMBI_0403 in 256 ug/ml streptomycin. From monoculture results we know this streptomycin concentration is outside the tolerable range for these two species. Indeed, the OD in this sample at time point 8 is 0.046, which indicates very low biomass at the experiment’s conclusion.</p>
</section>
<section id="filter-to-target-species" class="level2" data-number="5.5">
<h2 data-number="5.5" class="anchored" data-anchor-id="filter-to-target-species"><span class="header-section-number">5.5</span> Filter to target species</h2>
<p>Exclude positive and negative controls and also exclude species that are not in the focal 4</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1"></a>finaltable_exp_mstr <span class="ot">&lt;-</span> finaltable <span class="sc">%&gt;%</span> </span>
<span id="cb20-2"><a href="#cb20-2"></a>  <span class="fu">filter</span>(community_type <span class="sc">%nin%</span> <span class="fu">c</span>(<span class="st">"empty"</span>, <span class="st">"neg_control"</span>, <span class="st">"pos_control"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb20-3"><a href="#cb20-3"></a>  <span class="fu">filter</span>(sample <span class="sc">%nin%</span> lowread_samps) <span class="sc">%&gt;%</span> </span>
<span id="cb20-4"><a href="#cb20-4"></a>  <span class="fu">filter</span>(strainID <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"HAMBI_0403"</span>, <span class="st">"HAMBI_1287"</span>, <span class="st">"HAMBI_1896"</span>, <span class="st">"HAMBI_1977"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb20-5"><a href="#cb20-5"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb20-6"><a href="#cb20-6"></a>  <span class="fu">mutate</span>(<span class="at">f_raw_targetsp =</span> count_correct<span class="sc">/</span><span class="fu">sum</span>(count_correct),</span>
<span id="cb20-7"><a href="#cb20-7"></a>         <span class="at">sample_tot_targetsp =</span> <span class="fu">sum</span>(count_correct)) <span class="sc">%&gt;%</span> </span>
<span id="cb20-8"><a href="#cb20-8"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb20-9"><a href="#cb20-9"></a>  <span class="fu">relocate</span>(<span class="fu">c</span>(sample_tot_targetsp, f_raw_targetsp), <span class="at">.after =</span> f_raw)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="master-plates-samples" class="level2" data-number="5.6">
<h2 data-number="5.6" class="anchored" data-anchor-id="master-plates-samples"><span class="header-section-number">5.6</span> Master plates samples</h2>
<p>Crosstalk on master plates is quite good - usually &lt; 1% frequency.</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1"></a>finaltable_exp_mstr <span class="sc">%&gt;%</span> </span>
<span id="cb21-2"><a href="#cb21-2"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(community_type, <span class="st">"master"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb21-3"><a href="#cb21-3"></a>  <span class="fu">filter</span>(<span class="fu">is.na</span>(evo_hist)) <span class="sc">%&gt;%</span> </span>
<span id="cb21-4"><a href="#cb21-4"></a>  <span class="fu">filter</span>(f_raw_targetsp <span class="sc">&gt;</span> <span class="fl">0.01</span> ) <span class="sc">%&gt;%</span> </span>
<span id="cb21-5"><a href="#cb21-5"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(sample, strainID, f_raw_targetsp) <span class="sc">%&gt;%</span> </span>
<span id="cb21-6"><a href="#cb21-6"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(f_raw_targetsp))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["sample"],"name":[1],"type":["chr"],"align":["left"]},{"label":["strainID"],"name":[2],"type":["chr"],"align":["left"]},{"label":["f_raw_targetsp"],"name":[3],"type":["dbl"],"align":["right"]}],"data":[{"1":"2_12","2":"HAMBI_1896","3":"0.01605314"},{"1":"2_11","2":"HAMBI_1896","3":"0.01418342"},{"1":"2_08","2":"HAMBI_1896","3":"0.01256554"},{"1":"2_18","2":"HAMBI_1896","3":"0.01216276"},{"1":"2_12","2":"HAMBI_1977","3":"0.01107113"},{"1":"2_06","2":"HAMBI_1977","3":"0.01023541"},{"1":"2_07","2":"HAMBI_1896","3":"0.01020096"},{"1":"2_06","2":"HAMBI_1896","3":"0.01010747"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1"></a>finaltable_exp_mstr <span class="sc">%&gt;%</span> </span>
<span id="cb22-2"><a href="#cb22-2"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(community_type, <span class="st">"master"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb22-3"><a href="#cb22-3"></a>  <span class="fu">filter</span>(<span class="fu">is.na</span>(evo_hist)) <span class="sc">%&gt;%</span> </span>
<span id="cb22-4"><a href="#cb22-4"></a>  <span class="fu">contam_histogram</span>(<span class="at">trans=</span><span class="cn">TRUE</span>, <span class="at">x =</span> f_raw_targetsp) <span class="sc">+</span></span>
<span id="cb22-5"><a href="#cb22-5"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Species frequency"</span>, <span class="at">y =</span> <span class="st">"Sample count"</span>) <span class="sc">+</span></span>
<span id="cb22-6"><a href="#cb22-6"></a>  <span class="fu">ggtitle</span>(<span class="st">"Frequency distribution of each species in master plate samples where it should not be"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div id="fig-09" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-09-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="01_format_rbec_tab_files/figure-html/unnamed-chunk-18-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-09-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;2: Frequency distribution of each species in masterplate samples where it <strong>should not</strong> occur and where it exceeds a 3% threshold.
</figcaption>
</figure>
</div>
</section>
<section id="experimental-samples" class="level2" data-number="5.7">
<h2 data-number="5.7" class="anchored" data-anchor-id="experimental-samples"><span class="header-section-number">5.7</span> Experimental samples</h2>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1"></a>finaltable_exp_mstr <span class="sc">%&gt;%</span> </span>
<span id="cb23-2"><a href="#cb23-2"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">str_detect</span>(community_type, <span class="st">"master"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb23-3"><a href="#cb23-3"></a>  <span class="fu">filter</span>(<span class="fu">is.na</span>(evo_hist)) <span class="sc">%&gt;%</span> </span>
<span id="cb23-4"><a href="#cb23-4"></a>  <span class="fu">filter</span>(f_raw_targetsp <span class="sc">&gt;</span> <span class="fl">0.01</span> ) <span class="sc">%&gt;%</span> </span>
<span id="cb23-5"><a href="#cb23-5"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(sample, strainID, f_raw_targetsp) <span class="sc">%&gt;%</span> </span>
<span id="cb23-6"><a href="#cb23-6"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(f_raw_targetsp))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["sample"],"name":[1],"type":["chr"],"align":["left"]},{"label":["strainID"],"name":[2],"type":["chr"],"align":["left"]},{"label":["f_raw_targetsp"],"name":[3],"type":["dbl"],"align":["right"]}],"data":[{"1":"2_17_A06","2":"HAMBI_1896","3":"0.07155678"},{"1":"2_06_F10","2":"HAMBI_1977","3":"0.05984939"},{"1":"2_12_D02","2":"HAMBI_1896","3":"0.04857539"},{"1":"2_11_C02","2":"HAMBI_1896","3":"0.04540368"},{"1":"2_12_D05","2":"HAMBI_1896","3":"0.04256319"},{"1":"2_12_D08","2":"HAMBI_1896","3":"0.04242046"},{"1":"2_11_C08","2":"HAMBI_1896","3":"0.04115768"},{"1":"2_06_F04","2":"HAMBI_1977","3":"0.04095904"},{"1":"2_17_A06","2":"HAMBI_0403","3":"0.03924853"},{"1":"2_09_A02","2":"HAMBI_1977","3":"0.03664728"},{"1":"2_11_C11","2":"HAMBI_1896","3":"0.03590315"},{"1":"2_06_F01","2":"HAMBI_1977","3":"0.03495206"},{"1":"2_12_D11","2":"HAMBI_1896","3":"0.03159151"},{"1":"2_11_C05","2":"HAMBI_1896","3":"0.02775879"},{"1":"2_13_E02","2":"HAMBI_1896","3":"0.02744618"},{"1":"2_01_A07","2":"HAMBI_1896","3":"0.02699118"},{"1":"2_09_A05","2":"HAMBI_1977","3":"0.02691717"},{"1":"2_16_H02","2":"HAMBI_1977","3":"0.02676015"},{"1":"2_10_B02","2":"HAMBI_1977","3":"0.02616490"},{"1":"2_17_A09","2":"HAMBI_1896","3":"0.02458578"},{"1":"2_05_E01","2":"HAMBI_1977","3":"0.02140452"},{"1":"2_06_F07","2":"HAMBI_1977","3":"0.02107112"},{"1":"2_19_C12","2":"HAMBI_1977","3":"0.01898434"},{"1":"2_16_H08","2":"HAMBI_1977","3":"0.01776385"},{"1":"2_16_H05","2":"HAMBI_1977","3":"0.01658204"},{"1":"2_05_E10","2":"HAMBI_1977","3":"0.01631778"},{"1":"2_13_E08","2":"HAMBI_1896","3":"0.01606096"},{"1":"2_09_A05","2":"HAMBI_1287","3":"0.01571674"},{"1":"2_09_A02","2":"HAMBI_1287","3":"0.01486200"},{"1":"2_10_B05","2":"HAMBI_1977","3":"0.01477191"},{"1":"2_16_H11","2":"HAMBI_1977","3":"0.01441077"},{"1":"2_02_B10","2":"HAMBI_1896","3":"0.01401394"},{"1":"2_10_B08","2":"HAMBI_1977","3":"0.01391126"},{"1":"2_02_B07","2":"HAMBI_1896","3":"0.01290554"},{"1":"2_05_E04","2":"HAMBI_1977","3":"0.01261538"},{"1":"2_18_B03","2":"HAMBI_1896","3":"0.01215566"},{"1":"2_07_G07","2":"HAMBI_1287","3":"0.01186624"},{"1":"2_15_G05","2":"HAMBI_1977","3":"0.01180088"},{"1":"2_17_A12","2":"HAMBI_1896","3":"0.01174439"},{"1":"2_17_A03","2":"HAMBI_1896","3":"0.01149897"},{"1":"2_14_F02","2":"HAMBI_1896","3":"0.01065263"},{"1":"2_08_H07","2":"HAMBI_1287","3":"0.01048618"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<p>There are 39 samples (18% of 213 samples) that exceed a 1% crosstalk threshold. 28 samples (13% of 213 samples) have contamination &lt;= 3% and 12 samples (6 % of 213 samples) have &gt; 3% of a species that should not be there. In most cases the species in highly contaminated samples is HAMBI_1896. In general, this level of contamination is quite low and samples with contamination greater than 3% are just crosstalk outliers.</p>
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1"></a>finaltable_exp_mstr <span class="sc">%&gt;%</span> </span>
<span id="cb24-2"><a href="#cb24-2"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(community_type, <span class="st">"experiment"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb24-3"><a href="#cb24-3"></a>  <span class="fu">contaminated_barplot</span>(<span class="at">threshold=</span><span class="fl">0.01</span>, <span class="at">quartet =</span> <span class="cn">FALSE</span>, <span class="at">y=</span>f_raw_targetsp) <span class="sc">+</span></span>
<span id="cb24-4"><a href="#cb24-4"></a>  <span class="fu">ggtitle</span>(<span class="st">"Experiment samples with &gt; 1% contamination"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div id="fig-11" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-11-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="01_format_rbec_tab_files/figure-html/unnamed-chunk-21-1.png" class="img-fluid figure-img" width="864"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-11-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3: Species composition of experiment samples that contain &gt; 3% of a species that shouldn’t be there (shown in grey to black).
</figcaption>
</figure>
</div>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1"></a>contamtrios <span class="ot">&lt;-</span> finaltable_exp_mstr <span class="sc">%&gt;%</span> </span>
<span id="cb25-2"><a href="#cb25-2"></a>  <span class="fu">filter</span>(n_species <span class="sc">==</span> <span class="dv">3</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb25-3"><a href="#cb25-3"></a>  <span class="fu">filter</span>(<span class="fu">is.na</span>(evo_hist)) <span class="sc">%&gt;%</span> </span>
<span id="cb25-4"><a href="#cb25-4"></a>  <span class="fu">filter</span>(f_raw_targetsp <span class="sc">&gt;</span> <span class="fl">0.13</span>  ) <span class="sc">%&gt;%</span> </span>
<span id="cb25-5"><a href="#cb25-5"></a>  <span class="fu">pull</span>(sample)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Only 6 trio samples look really bad (~50 % contamination). Again, I think we should only exclude samples where the contamination is very high (over ~50% of the sample) but those with around 10% or less contaminant I think can be retained, and I will discard the contaminating sequences. Interestingly the lower-level contamination (3% &gt; contamination &gt; 13%) seems to be mostly from from HAMBI_1896. The high level contamination is always in well C01 on 6 different plates ().</p>
</section>
<section id="plot" class="level2" data-number="5.8">
<h2 data-number="5.8" class="anchored" data-anchor-id="plot"><span class="header-section-number">5.8</span> Plot</h2>
<section id="pairs-and-trios" class="level3" data-number="5.8.1">
<h3 data-number="5.8.1" class="anchored" data-anchor-id="pairs-and-trios"><span class="header-section-number">5.8.1</span> Pairs and trios</h3>
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1"></a>finaltable_exp_mstr <span class="sc">%&gt;%</span> </span>
<span id="cb26-2"><a href="#cb26-2"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(community_type, <span class="st">"experiment"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb26-3"><a href="#cb26-3"></a>  <span class="fu">filter</span>(n_species <span class="sc">!=</span> <span class="dv">8</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb26-4"><a href="#cb26-4"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb26-5"><a href="#cb26-5"></a>  <span class="fu">mutate</span>(<span class="at">id =</span> <span class="fu">cur_group_id</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb26-6"><a href="#cb26-6"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb26-7"><a href="#cb26-7"></a>  <span class="fu">mutate</span>(<span class="at">facet =</span> <span class="fu">ntile</span>(id, <span class="dv">5</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb26-8"><a href="#cb26-8"></a>  <span class="co"># to make sure ntile doesnt split up facets</span></span>
<span id="cb26-9"><a href="#cb26-9"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb26-10"><a href="#cb26-10"></a>  <span class="fu">mutate</span>(<span class="at">facet =</span> <span class="fu">first</span>(facet)) <span class="sc">%&gt;%</span> </span>
<span id="cb26-11"><a href="#cb26-11"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb26-12"><a href="#cb26-12"></a>  <span class="fu">contaminated_barplot</span>(<span class="at">threshold =</span> <span class="dv">0</span>, <span class="at">quartet =</span> <span class="cn">TRUE</span>, <span class="at">y=</span>f_raw_targetsp) <span class="sc">+</span></span>
<span id="cb26-13"><a href="#cb26-13"></a>  <span class="fu">ggtitle</span>(<span class="st">"All experimental pairs and trios"</span>) <span class="sc">+</span></span>
<span id="cb26-14"><a href="#cb26-14"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>facet, <span class="at">scales=</span><span class="st">"free_x"</span>, <span class="at">nrow=</span><span class="dv">5</span>) <span class="sc">+</span></span>
<span id="cb26-15"><a href="#cb26-15"></a>  <span class="fu">theme</span>(</span>
<span id="cb26-16"><a href="#cb26-16"></a>    <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb26-17"><a href="#cb26-17"></a>    <span class="at">strip.text.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb26-18"><a href="#cb26-18"></a>    <span class="at">legend.position =</span> <span class="st">"bottom"</span></span>
<span id="cb26-19"><a href="#cb26-19"></a>    )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div id="fig-12" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-12-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="01_format_rbec_tab_files/figure-html/unnamed-chunk-23-1.png" class="img-fluid figure-img" width="960"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-12-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;4: Species composition of all experiment trio samples. Species that should not be present in a sample are shown in grey to black.
</figcaption>
</figure>
</div>
<p>At a birds-eye view this looks pretty good overall.</p>
</section>
<section id="sec-octets2" class="level3" data-number="5.8.2">
<h3 data-number="5.8.2" class="anchored" data-anchor-id="sec-octets2"><span class="header-section-number">5.8.2</span> Octets</h3>
<p>Again, one weakness of this approach is that there is no way to tell from this data whether an octets is contaminated because every octet (by definition) is inoculated with all of the 4 species at different frequencies and with different evolutionary histories. Thus we would expect to see either all 4 species present or a subset of the 4 species in the quartet samples, but we cannot tell if these patterns come from contamination or not. Moving forward we just have to assume that the quartets are OK… However, we’ve found so little contamination in our other experimental samples and all samples were handled in the experiment and processed for sequence library preparation in the same way. I think it is safe to assume that contamination in the quartets is minimal issue.</p>
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1"></a>finaltable_exp_mstr <span class="sc">%&gt;%</span> </span>
<span id="cb27-2"><a href="#cb27-2"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(community_type, <span class="st">"experiment"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb27-3"><a href="#cb27-3"></a>  <span class="fu">filter</span>(n_species <span class="sc">==</span> <span class="dv">8</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb27-4"><a href="#cb27-4"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(sample, strainID, count_correct) <span class="sc">%&gt;%</span> </span>
<span id="cb27-5"><a href="#cb27-5"></a>  <span class="fu">distinct</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb27-6"><a href="#cb27-6"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb27-7"><a href="#cb27-7"></a>  <span class="fu">mutate</span>(<span class="at">f_raw_targetsp =</span> count_correct<span class="sc">/</span><span class="fu">sum</span>(count_correct)) <span class="sc">%&gt;%</span> </span>
<span id="cb27-8"><a href="#cb27-8"></a>  <span class="fu">mutate</span>(<span class="at">id =</span> <span class="fu">cur_group_id</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb27-9"><a href="#cb27-9"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb27-10"><a href="#cb27-10"></a>  <span class="fu">mutate</span>(<span class="at">facet =</span> <span class="fu">ntile</span>(id, <span class="dv">5</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb27-11"><a href="#cb27-11"></a>  <span class="co"># to make sure ntile doesnt split up facets</span></span>
<span id="cb27-12"><a href="#cb27-12"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb27-13"><a href="#cb27-13"></a>  <span class="fu">mutate</span>(<span class="at">facet =</span> <span class="fu">first</span>(facet)) <span class="sc">%&gt;%</span> </span>
<span id="cb27-14"><a href="#cb27-14"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb27-15"><a href="#cb27-15"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb27-16"><a href="#cb27-16"></a>    <span class="fu">geom_col</span>(<span class="fu">aes</span>(<span class="at">x =</span> sample, <span class="at">y =</span> f_raw_targetsp, <span class="at">fill =</span> strainID)) <span class="sc">+</span></span>
<span id="cb27-17"><a href="#cb27-17"></a>    <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">"Percent abundance"</span>, <span class="at">x =</span> <span class="st">""</span>, <span class="at">fill =</span> <span class="st">""</span>) <span class="sc">+</span></span>
<span id="cb27-18"><a href="#cb27-18"></a>    <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"HAMBI_0403"</span> <span class="ot">=</span> <span class="st">"#faa019"</span>,</span>
<span id="cb27-19"><a href="#cb27-19"></a>              <span class="st">"HAMBI_1287"</span> <span class="ot">=</span> <span class="st">"#75afff"</span>,</span>
<span id="cb27-20"><a href="#cb27-20"></a>              <span class="st">"HAMBI_1896"</span> <span class="ot">=</span> <span class="st">"#59cc4e"</span>,</span>
<span id="cb27-21"><a href="#cb27-21"></a>              <span class="st">"HAMBI_1977"</span> <span class="ot">=</span> <span class="st">"#ffd430"</span>)</span>
<span id="cb27-22"><a href="#cb27-22"></a>              ) <span class="sc">+</span></span>
<span id="cb27-23"><a href="#cb27-23"></a>    <span class="fu">scale_y_continuous</span>(<span class="at">labels =</span> percent) <span class="sc">+</span></span>
<span id="cb27-24"><a href="#cb27-24"></a>    <span class="fu">scale_x_discrete</span>(<span class="at">guide =</span> <span class="fu">guide_axis</span>(<span class="at">angle=</span><span class="dv">90</span>)) <span class="sc">+</span></span>
<span id="cb27-25"><a href="#cb27-25"></a>    <span class="fu">theme_bw</span>() <span class="sc">+</span> </span>
<span id="cb27-26"><a href="#cb27-26"></a>    <span class="fu">theme</span>(<span class="at">panel.grid =</span> <span class="fu">element_blank</span>(), <span class="at">legend.position =</span> <span class="st">"right"</span>) <span class="sc">+</span></span>
<span id="cb27-27"><a href="#cb27-27"></a>    <span class="fu">ggtitle</span>(<span class="st">"All experimental octets"</span>) <span class="sc">+</span></span>
<span id="cb27-28"><a href="#cb27-28"></a>    <span class="fu">facet_wrap</span>(<span class="sc">~</span>facet, <span class="at">scales=</span><span class="st">"free_x"</span>, <span class="at">nrow=</span><span class="dv">5</span>) <span class="sc">+</span></span>
<span id="cb27-29"><a href="#cb27-29"></a>    <span class="fu">theme</span>(</span>
<span id="cb27-30"><a href="#cb27-30"></a>    <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb27-31"><a href="#cb27-31"></a>    <span class="at">strip.text.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb27-32"><a href="#cb27-32"></a>    <span class="at">legend.position =</span> <span class="st">"bottom"</span></span>
<span id="cb27-33"><a href="#cb27-33"></a>    )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div id="fig-12" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-12-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="01_format_rbec_tab_files/figure-html/unnamed-chunk-24-1.png" class="img-fluid figure-img" width="960"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-12-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;5: Species composition of all experiment 8 strain samples.
</figcaption>
</figure>
</div>
</section>
</section>
</section>
<section id="export" class="level1" data-number="6">
<h1 data-number="6"><span class="header-section-number">6</span> Export</h1>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1"></a>finaltable_exp_mstr_export <span class="ot">&lt;-</span> finaltable_exp_mstr <span class="sc">%&gt;%</span> </span>
<span id="cb28-2"><a href="#cb28-2"></a>  <span class="co"># first remove samples with low read count (&lt; 3000 reads)</span></span>
<span id="cb28-3"><a href="#cb28-3"></a>  <span class="fu">filter</span>(sample <span class="sc">%nin%</span> lowread_samps) <span class="sc">%&gt;%</span> </span>
<span id="cb28-4"><a href="#cb28-4"></a>  <span class="co"># exclude any remaining counts from species that shouldnt be there using again</span></span>
<span id="cb28-5"><a href="#cb28-5"></a>  <span class="co"># the fact that evo_hist should be NA for these species</span></span>
<span id="cb28-6"><a href="#cb28-6"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(evo_hist)) <span class="sc">%&gt;%</span> </span>
<span id="cb28-7"><a href="#cb28-7"></a>  <span class="co"># because we set 3% as our limit of detection we set read counts of species</span></span>
<span id="cb28-8"><a href="#cb28-8"></a>  <span class="co"># less than 3% to 0</span></span>
<span id="cb28-9"><a href="#cb28-9"></a>  <span class="fu">mutate</span>(<span class="at">count_correct_thresh =</span> <span class="fu">if_else</span>(f_raw_targetsp <span class="sc">&lt;=</span> <span class="fl">0.03</span>, <span class="dv">0</span>, count_correct)) <span class="sc">%&gt;%</span> </span>
<span id="cb28-10"><a href="#cb28-10"></a>  <span class="co"># now calculate a new relative abundance based only on the species that should</span></span>
<span id="cb28-11"><a href="#cb28-11"></a>  <span class="co"># be present and that are &gt; 3% relative abundance</span></span>
<span id="cb28-12"><a href="#cb28-12"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb28-13"><a href="#cb28-13"></a>  <span class="fu">mutate</span>(<span class="at">f_thresh =</span> count_correct_thresh<span class="sc">/</span><span class="fu">sum</span>(count_correct_thresh)) <span class="sc">%&gt;%</span> </span>
<span id="cb28-14"><a href="#cb28-14"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb28-15"><a href="#cb28-15"></a>  <span class="co"># for marking which sequencing batch these came from </span></span>
<span id="cb28-16"><a href="#cb28-16"></a>  <span class="fu">mutate</span>(<span class="at">batch =</span> <span class="st">"run20250502"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb28-17"><a href="#cb28-17"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(sample, strainID, evo_hist, count_correct_thresh, f_thresh,</span>
<span id="cb28-18"><a href="#cb28-18"></a>                <span class="at">target_f_masterplate =</span> target_f, replicate, strep_conc, </span>
<span id="cb28-19"><a href="#cb28-19"></a>                transfers, n_species, community_type, plate_well, batch)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="pairs" class="level2" data-number="6.1">
<h2 data-number="6.1" class="anchored" data-anchor-id="pairs"><span class="header-section-number">6.1</span> Pairs</h2>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1"></a>finaltable_exp_mstr_export <span class="sc">%&gt;%</span></span>
<span id="cb29-2"><a href="#cb29-2"></a>  <span class="fu">filter</span>(n_species <span class="sc">==</span> <span class="dv">2</span>) <span class="sc">%&gt;%</span></span>
<span id="cb29-3"><a href="#cb29-3"></a>  <span class="fu">arrange</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb29-4"><a href="#cb29-4"></a>  <span class="co"># reformating to be compatible with other pair data</span></span>
<span id="cb29-5"><a href="#cb29-5"></a>  <span class="fu">mutate</span>(<span class="at">sample =</span> <span class="fu">if_else</span>(</span>
<span id="cb29-6"><a href="#cb29-6"></a>    community_type <span class="sc">==</span> <span class="st">"master"</span>,</span>
<span id="cb29-7"><a href="#cb29-7"></a>    <span class="fu">paste0</span>(<span class="st">"P"</span>, </span>
<span id="cb29-8"><a href="#cb29-8"></a>           <span class="fu">str_extract</span>(sample, <span class="at">pattern =</span> <span class="st">'(?&lt;=2_)</span><span class="sc">\\</span><span class="st">d{2}'</span>), <span class="st">"_s_0"</span>),</span>
<span id="cb29-9"><a href="#cb29-9"></a>    <span class="fu">paste0</span>(<span class="st">"P"</span>,</span>
<span id="cb29-10"><a href="#cb29-10"></a>           <span class="fu">str_extract</span>(sample, <span class="at">pattern =</span> <span class="st">'(?&lt;=2_)</span><span class="sc">\\</span><span class="st">d{2}'</span>),</span>
<span id="cb29-11"><a href="#cb29-11"></a>           <span class="st">"_"</span>,</span>
<span id="cb29-12"><a href="#cb29-12"></a>           replicate,</span>
<span id="cb29-13"><a href="#cb29-13"></a>           <span class="st">"_"</span>,</span>
<span id="cb29-14"><a href="#cb29-14"></a>           strep_conc</span>
<span id="cb29-15"><a href="#cb29-15"></a>    )</span>
<span id="cb29-16"><a href="#cb29-16"></a>  )) <span class="sc">%&gt;%</span> </span>
<span id="cb29-17"><a href="#cb29-17"></a>  <span class="fu">write_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(data, <span class="st">"pairs_counts.tsv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="trios" class="level2" data-number="6.2">
<h2 data-number="6.2" class="anchored" data-anchor-id="trios"><span class="header-section-number">6.2</span> Trios</h2>
<p>Write the trio samples</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1"></a>finaltable_exp_mstr_export <span class="sc">%&gt;%</span> </span>
<span id="cb30-2"><a href="#cb30-2"></a>  <span class="fu">filter</span>(n_species <span class="sc">==</span> <span class="dv">3</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb30-3"><a href="#cb30-3"></a>  <span class="fu">arrange</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb30-4"><a href="#cb30-4"></a>  <span class="fu">write_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(data, <span class="st">"trios_counts.tsv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="octets" class="level2" data-number="6.3">
<h2 data-number="6.3" class="anchored" data-anchor-id="octets"><span class="header-section-number">6.3</span> Octets</h2>
<p>Write the octet samples</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1"></a>finaltable_exp_mstr_export <span class="sc">%&gt;%</span> </span>
<span id="cb31-2"><a href="#cb31-2"></a>  <span class="fu">filter</span>(n_species <span class="sc">==</span> <span class="dv">8</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb31-3"><a href="#cb31-3"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb31-4"><a href="#cb31-4"></a>  <span class="fu">mutate</span>(<span class="at">max_f_sp =</span> <span class="fu">if_else</span>(target_f_masterplate <span class="sc">==</span> <span class="fu">max</span>(target_f_masterplate), </span>
<span id="cb31-5"><a href="#cb31-5"></a>                            <span class="fu">paste0</span>(<span class="fu">str_to_upper</span>(evo_hist), <span class="st">"_"</span>, <span class="fu">str_extract</span>(strainID, <span class="st">"</span><span class="sc">\\</span><span class="st">d+"</span>)), </span>
<span id="cb31-6"><a href="#cb31-6"></a>                            <span class="cn">NA_character_</span>),</span>
<span id="cb31-7"><a href="#cb31-7"></a>         <span class="at">max_f =</span> <span class="fu">max</span>(target_f_masterplate)) <span class="sc">%&gt;%</span> </span>
<span id="cb31-8"><a href="#cb31-8"></a>  <span class="fu">fill</span>(max_f_sp, <span class="at">.direction =</span> <span class="st">"downup"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb31-9"><a href="#cb31-9"></a>  <span class="fu">group_by</span>(sample, strainID) <span class="sc">%&gt;%</span> </span>
<span id="cb31-10"><a href="#cb31-10"></a>  <span class="fu">mutate</span>(<span class="at">target_f_masterplate =</span> <span class="fu">sum</span>(target_f_masterplate)) <span class="sc">%&gt;%</span> </span>
<span id="cb31-11"><a href="#cb31-11"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb31-12"><a href="#cb31-12"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>evo_hist) <span class="sc">%&gt;%</span> </span>
<span id="cb31-13"><a href="#cb31-13"></a>  <span class="co"># deal with case when all species are equal</span></span>
<span id="cb31-14"><a href="#cb31-14"></a>  <span class="fu">mutate</span>(<span class="at">max_f_sp =</span> <span class="fu">if_else</span>(max_f <span class="sc">==</span> <span class="fl">0.13</span>, <span class="st">"none"</span>, max_f_sp)) <span class="sc">%&gt;%</span> </span>
<span id="cb31-15"><a href="#cb31-15"></a>  <span class="fu">mutate</span>(<span class="at">target_f_masterplate =</span> <span class="fu">if_else</span>(target_f_masterplate <span class="sc">==</span> <span class="fl">0.26</span>, <span class="fl">0.25</span>, target_f_masterplate)) <span class="sc">%&gt;%</span> </span>
<span id="cb31-16"><a href="#cb31-16"></a>  <span class="fu">distinct</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb31-17"><a href="#cb31-17"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb31-18"><a href="#cb31-18"></a>  <span class="fu">mutate</span>(<span class="at">f_thresh =</span> count_correct_thresh<span class="sc">/</span><span class="fu">sum</span>(count_correct_thresh)) <span class="sc">%&gt;%</span> </span>
<span id="cb31-19"><a href="#cb31-19"></a>  <span class="fu">relocate</span>(max_f_sp, max_f, <span class="at">.after =</span> target_f_masterplate) <span class="sc">%&gt;%</span> </span>
<span id="cb31-20"><a href="#cb31-20"></a>  <span class="fu">write_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(data, <span class="st">"octets_counts.tsv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>


<!-- -->

</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb32" data-shortcodes="false"><pre class="sourceCode numberSource markdown number-lines code-with-copy"><code class="sourceCode markdown"><span id="cb32-1"><a href="#cb32-1"></a><span class="co">---</span></span>
<span id="cb32-2"><a href="#cb32-2"></a><span class="an">title:</span><span class="co"> "Formatting Rbec output from 2025-05-02 sequenced by BTK Turku using custom HAMBI Illumina v3 primers"</span></span>
<span id="cb32-3"><a href="#cb32-3"></a><span class="an">author:</span><span class="co"> "Milla Similä &amp; Shane Hogle"</span></span>
<span id="cb32-4"><a href="#cb32-4"></a><span class="an">date:</span><span class="co"> today</span></span>
<span id="cb32-5"><a href="#cb32-5"></a><span class="an">link-citations:</span><span class="co"> true</span></span>
<span id="cb32-6"><a href="#cb32-6"></a><span class="an">abstract:</span><span class="co"> "Results are from samples containing the mixture of 8 strains (both anc/evo form for each species) as well as one single triplet, and all the pairs with 50:50 abundance."</span></span>
<span id="cb32-7"><a href="#cb32-7"></a><span class="co">---</span></span>
<span id="cb32-8"><a href="#cb32-8"></a></span>
<span id="cb32-9"><a href="#cb32-9"></a><span class="fu"># Setup</span></span>
<span id="cb32-10"><a href="#cb32-10"></a></span>
<span id="cb32-11"><a href="#cb32-11"></a><span class="fu">## Libraries</span></span>
<span id="cb32-12"><a href="#cb32-12"></a></span>
<span id="cb32-15"><a href="#cb32-15"></a><span class="in">```{r}</span></span>
<span id="cb32-16"><a href="#cb32-16"></a><span class="co">#| output: false</span></span>
<span id="cb32-17"><a href="#cb32-17"></a><span class="co">#| warning: false</span></span>
<span id="cb32-18"><a href="#cb32-18"></a><span class="co">#| error: false</span></span>
<span id="cb32-19"><a href="#cb32-19"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb32-20"><a href="#cb32-20"></a><span class="fu">library</span>(here)</span>
<span id="cb32-21"><a href="#cb32-21"></a><span class="fu">library</span>(fs)</span>
<span id="cb32-22"><a href="#cb32-22"></a><span class="fu">library</span>(archive)</span>
<span id="cb32-23"><a href="#cb32-23"></a><span class="fu">library</span>(scales)</span>
<span id="cb32-24"><a href="#cb32-24"></a><span class="fu">source</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"R"</span>, <span class="st">"utils_generic.R"</span>))</span>
<span id="cb32-25"><a href="#cb32-25"></a><span class="fu">source</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"R"</span>, <span class="st">"communities"</span>, <span class="st">"community_composition"</span>, <span class="st">"amplicon"</span>, <span class="st">"utils_amplicon.R"</span>))</span>
<span id="cb32-26"><a href="#cb32-26"></a><span class="in">```</span></span>
<span id="cb32-27"><a href="#cb32-27"></a></span>
<span id="cb32-28"><a href="#cb32-28"></a><span class="fu">## Global variables</span></span>
<span id="cb32-29"><a href="#cb32-29"></a></span>
<span id="cb32-32"><a href="#cb32-32"></a><span class="in">```{r}</span></span>
<span id="cb32-33"><a href="#cb32-33"></a><span class="co">#| output: false</span></span>
<span id="cb32-34"><a href="#cb32-34"></a><span class="co">#| warning: false</span></span>
<span id="cb32-35"><a href="#cb32-35"></a><span class="co">#| error: false</span></span>
<span id="cb32-36"><a href="#cb32-36"></a>data_raw <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"_data_raw"</span>, <span class="st">"communities"</span>, <span class="st">"20250502_BTK_illumina_v3"</span>)</span>
<span id="cb32-37"><a href="#cb32-37"></a>data <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"communities"</span>, <span class="st">"20250502_BTK_illumina_v3"</span>)</span>
<span id="cb32-38"><a href="#cb32-38"></a>amplicontar <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(data_raw, <span class="st">"rbec_output.tar.gz"</span>)</span>
<span id="cb32-39"><a href="#cb32-39"></a></span>
<span id="cb32-40"><a href="#cb32-40"></a><span class="co"># make processed data directory if it doesn't exist</span></span>
<span id="cb32-41"><a href="#cb32-41"></a>fs<span class="sc">::</span><span class="fu">dir_create</span>(data)</span>
<span id="cb32-42"><a href="#cb32-42"></a></span>
<span id="cb32-43"><a href="#cb32-43"></a><span class="co"># create temporary location to decompress</span></span>
<span id="cb32-44"><a href="#cb32-44"></a>tmpdir <span class="ot">&lt;-</span> fs<span class="sc">::</span><span class="fu">file_temp</span>()</span>
<span id="cb32-45"><a href="#cb32-45"></a><span class="in">```</span></span>
<span id="cb32-46"><a href="#cb32-46"></a></span>
<span id="cb32-47"><a href="#cb32-47"></a><span class="fu"># Read metadata</span></span>
<span id="cb32-48"><a href="#cb32-48"></a></span>
<span id="cb32-51"><a href="#cb32-51"></a><span class="in">```{r}</span></span>
<span id="cb32-52"><a href="#cb32-52"></a><span class="co">#| output: false</span></span>
<span id="cb32-53"><a href="#cb32-53"></a><span class="co">#| warning: false</span></span>
<span id="cb32-54"><a href="#cb32-54"></a><span class="co">#| error: false</span></span>
<span id="cb32-55"><a href="#cb32-55"></a>mddf <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(data_raw, <span class="st">"20250502_BTK_illumina_v3_metadata.tsv"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb32-56"><a href="#cb32-56"></a>  <span class="fu">mutate</span>(<span class="at">sample =</span> <span class="fu">str_trim</span>(sample))</span>
<span id="cb32-57"><a href="#cb32-57"></a>spdf <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(data_raw, <span class="st">"sample_compositions.tsv"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb32-58"><a href="#cb32-58"></a>  <span class="fu">mutate</span>(<span class="at">sample =</span> <span class="fu">str_trim</span>(sample))</span>
<span id="cb32-59"><a href="#cb32-59"></a><span class="in">```</span></span>
<span id="cb32-60"><a href="#cb32-60"></a></span>
<span id="cb32-61"><a href="#cb32-61"></a><span class="fu"># Read Rbec raw counts tables </span></span>
<span id="cb32-62"><a href="#cb32-62"></a></span>
<span id="cb32-63"><a href="#cb32-63"></a><span class="fu">## Untar Rbec output tarball</span></span>
<span id="cb32-64"><a href="#cb32-64"></a></span>
<span id="cb32-67"><a href="#cb32-67"></a><span class="in">```{r}</span></span>
<span id="cb32-68"><a href="#cb32-68"></a>archive<span class="sc">::</span><span class="fu">archive_extract</span>(</span>
<span id="cb32-69"><a href="#cb32-69"></a>  amplicontar,</span>
<span id="cb32-70"><a href="#cb32-70"></a>  <span class="at">dir =</span> tmpdir,</span>
<span id="cb32-71"><a href="#cb32-71"></a>  <span class="at">files =</span> <span class="cn">NULL</span>,</span>
<span id="cb32-72"><a href="#cb32-72"></a>  <span class="at">options =</span> <span class="fu">character</span>(),</span>
<span id="cb32-73"><a href="#cb32-73"></a>  <span class="at">strip_components =</span> <span class="dv">0</span>L</span>
<span id="cb32-74"><a href="#cb32-74"></a>)</span>
<span id="cb32-75"><a href="#cb32-75"></a><span class="in">```</span></span>
<span id="cb32-76"><a href="#cb32-76"></a></span>
<span id="cb32-77"><a href="#cb32-77"></a><span class="fu">## Setup directory structure</span></span>
<span id="cb32-78"><a href="#cb32-78"></a></span>
<span id="cb32-81"><a href="#cb32-81"></a><span class="in">```{r}</span></span>
<span id="cb32-82"><a href="#cb32-82"></a>tabdir <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(tmpdir, <span class="st">"rbec_output"</span>)</span>
<span id="cb32-83"><a href="#cb32-83"></a>samppaths <span class="ot">&lt;-</span> fs<span class="sc">::</span><span class="fu">dir_ls</span>(tabdir)</span>
<span id="cb32-84"><a href="#cb32-84"></a>sampnames <span class="ot">&lt;-</span> <span class="fu">path_split</span>(samppaths) <span class="sc">%&gt;%</span> </span>
<span id="cb32-85"><a href="#cb32-85"></a>  <span class="fu">map_chr</span>(dplyr<span class="sc">::</span>last)</span>
<span id="cb32-86"><a href="#cb32-86"></a><span class="in">```</span></span>
<span id="cb32-87"><a href="#cb32-87"></a></span>
<span id="cb32-88"><a href="#cb32-88"></a><span class="fu">## Read</span></span>
<span id="cb32-89"><a href="#cb32-89"></a></span>
<span id="cb32-92"><a href="#cb32-92"></a><span class="in">```{r}</span></span>
<span id="cb32-93"><a href="#cb32-93"></a>straintabs <span class="ot">&lt;-</span> <span class="fu">paste0</span>(samppaths, <span class="st">"/strain_table.txt"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb32-94"><a href="#cb32-94"></a>  <span class="fu">set_names</span>(sampnames) <span class="sc">%&gt;%</span> </span>
<span id="cb32-95"><a href="#cb32-95"></a>  <span class="fu">map_df</span>(</span>
<span id="cb32-96"><a href="#cb32-96"></a>  read_tsv,</span>
<span id="cb32-97"><a href="#cb32-97"></a>  <span class="at">skip =</span> <span class="dv">1</span>,</span>
<span id="cb32-98"><a href="#cb32-98"></a>  <span class="at">col_names =</span> <span class="fu">c</span>(<span class="st">"rRNA16S_locus"</span>,<span class="st">"count"</span>),</span>
<span id="cb32-99"><a href="#cb32-99"></a>  <span class="at">show_col_types =</span> <span class="cn">FALSE</span>, </span>
<span id="cb32-100"><a href="#cb32-100"></a>  <span class="at">.id =</span> <span class="st">"sample"</span>)</span>
<span id="cb32-101"><a href="#cb32-101"></a><span class="in">```</span></span>
<span id="cb32-102"><a href="#cb32-102"></a></span>
<span id="cb32-103"><a href="#cb32-103"></a><span class="fu">## Clean up</span></span>
<span id="cb32-104"><a href="#cb32-104"></a></span>
<span id="cb32-107"><a href="#cb32-107"></a><span class="in">```{r}</span></span>
<span id="cb32-108"><a href="#cb32-108"></a><span class="co"># remove decompressed directory from temp location</span></span>
<span id="cb32-109"><a href="#cb32-109"></a>fs<span class="sc">::</span><span class="fu">dir_delete</span>(tmpdir)</span>
<span id="cb32-110"><a href="#cb32-110"></a><span class="in">```</span></span>
<span id="cb32-111"><a href="#cb32-111"></a></span>
<span id="cb32-112"><a href="#cb32-112"></a><span class="fu"># Format</span></span>
<span id="cb32-113"><a href="#cb32-113"></a></span>
<span id="cb32-116"><a href="#cb32-116"></a><span class="in">```{r}</span></span>
<span id="cb32-117"><a href="#cb32-117"></a>straintabs_norm <span class="ot">&lt;-</span> straintabs <span class="sc">%&gt;%</span> </span>
<span id="cb32-118"><a href="#cb32-118"></a>  <span class="co"># Calls function that normalize counts by 16S copy number</span></span>
<span id="cb32-119"><a href="#cb32-119"></a>  <span class="fu">normalize_by_copy</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb32-120"><a href="#cb32-120"></a>  <span class="co"># Calls function that completes all combinations of 23 species</span></span>
<span id="cb32-121"><a href="#cb32-121"></a>  <span class="co"># this is important because some species go extinct that should be in the samples</span></span>
<span id="cb32-122"><a href="#cb32-122"></a>  <span class="co"># and we need to have those in the final table</span></span>
<span id="cb32-123"><a href="#cb32-123"></a>  <span class="fu">complete_combos</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb32-124"><a href="#cb32-124"></a>  <span class="co"># removes any extra whitespace on sample names</span></span>
<span id="cb32-125"><a href="#cb32-125"></a>  <span class="fu">mutate</span>(<span class="at">sample =</span> <span class="fu">str_trim</span>(sample))</span>
<span id="cb32-126"><a href="#cb32-126"></a><span class="in">```</span></span>
<span id="cb32-127"><a href="#cb32-127"></a></span>
<span id="cb32-128"><a href="#cb32-128"></a>Make final table</span>
<span id="cb32-129"><a href="#cb32-129"></a></span>
<span id="cb32-132"><a href="#cb32-132"></a><span class="in">```{r}</span></span>
<span id="cb32-133"><a href="#cb32-133"></a><span class="co"># Later we will take advantage of the fact that for species not supposed to be</span></span>
<span id="cb32-134"><a href="#cb32-134"></a><span class="co"># in a sample the prior left_join will have filled the evo_hist category with an</span></span>
<span id="cb32-135"><a href="#cb32-135"></a><span class="co"># NA. We can then filter using this NA value</span></span>
<span id="cb32-136"><a href="#cb32-136"></a>finaltable <span class="ot">&lt;-</span> <span class="fu">left_join</span>(straintabs_norm, mddf, <span class="at">by =</span> <span class="fu">join_by</span>(sample)) <span class="sc">%&gt;%</span> </span>
<span id="cb32-137"><a href="#cb32-137"></a>  <span class="fu">left_join</span>(spdf, <span class="at">by =</span> <span class="fu">join_by</span>(sample, strainID)) <span class="sc">%&gt;%</span> </span>
<span id="cb32-138"><a href="#cb32-138"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb32-139"><a href="#cb32-139"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>genus, <span class="sc">-</span>species) <span class="sc">%&gt;%</span> </span>
<span id="cb32-140"><a href="#cb32-140"></a>  <span class="fu">mutate</span>(<span class="at">f_raw =</span> count_correct<span class="sc">/</span><span class="fu">sum</span>(count_correct),</span>
<span id="cb32-141"><a href="#cb32-141"></a>         <span class="at">sample_tot_raw =</span> <span class="fu">sum</span>(count_correct)) <span class="sc">%&gt;%</span> </span>
<span id="cb32-142"><a href="#cb32-142"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb32-143"><a href="#cb32-143"></a>  <span class="fu">relocate</span>(sample_tot_raw, f_raw, target_f, evo_hist, <span class="at">.after =</span> count_correct)</span>
<span id="cb32-144"><a href="#cb32-144"></a><span class="in">```</span></span>
<span id="cb32-145"><a href="#cb32-145"></a></span>
<span id="cb32-146"><a href="#cb32-146"></a><span class="fu"># Analysis</span></span>
<span id="cb32-147"><a href="#cb32-147"></a></span>
<span id="cb32-148"><a href="#cb32-148"></a>Let's quickly compare and look the at number of reads from the different kinds of experiment categories. Negative controls are samples from the experiment that contained growth medium but no cells. We included them to make sure that there was not contamination between wells during the experiment. Positive controls contain a mixsture of all 23 species (not sure why this was done), and some wells are just empty but were processed/sequenced anyway for convenience of the person preparing the libraries. Experiment samples are those that came from the actual experiment, and masterplate samples are the defined mixtures we made to inoculate the experiment. We sequenced these masterplates so that we would have T0 information for each experimental treatment.</span>
<span id="cb32-149"><a href="#cb32-149"></a></span>
<span id="cb32-150"><a href="#cb32-150"></a>::: {#fig-01}</span>
<span id="cb32-153"><a href="#cb32-153"></a><span class="in">```{r}</span></span>
<span id="cb32-154"><a href="#cb32-154"></a><span class="co">#| fig.width: 8</span></span>
<span id="cb32-155"><a href="#cb32-155"></a><span class="co">#| fig.height: 5</span></span>
<span id="cb32-156"><a href="#cb32-156"></a>finaltable <span class="sc">%&gt;%</span> </span>
<span id="cb32-157"><a href="#cb32-157"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(sample, sample_tot_raw, community_type) <span class="sc">%&gt;%</span> </span>
<span id="cb32-158"><a href="#cb32-158"></a>  <span class="fu">distinct</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb32-159"><a href="#cb32-159"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> sample_tot_raw)) <span class="sc">+</span></span>
<span id="cb32-160"><a href="#cb32-160"></a>  <span class="fu">geom_histogram</span>(<span class="at">bins =</span> <span class="dv">20</span>, <span class="fu">aes</span>(<span class="at">fill =</span> community_type)) <span class="sc">+</span></span>
<span id="cb32-161"><a href="#cb32-161"></a>  <span class="fu">scale_x_log10</span>() <span class="sc">+</span> </span>
<span id="cb32-162"><a href="#cb32-162"></a>  <span class="fu">annotation_logticks</span>(<span class="at">sides =</span> <span class="st">"b"</span>, <span class="at">color=</span><span class="st">"grey30"</span>) <span class="sc">+</span></span>
<span id="cb32-163"><a href="#cb32-163"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">"Sample count"</span>, <span class="at">x =</span> <span class="st">"Total reads per sample"</span>)  <span class="sc">+</span></span>
<span id="cb32-164"><a href="#cb32-164"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span> community_type, <span class="at">scales=</span><span class="st">"free_y"</span>, <span class="at">nrow =</span> <span class="dv">2</span>)</span>
<span id="cb32-165"><a href="#cb32-165"></a><span class="in">```</span></span>
<span id="cb32-166"><a href="#cb32-166"></a>Frequency distribution of total read counts per sample in the experimental samples (red) and the negative controls (blue).</span>
<span id="cb32-167"><a href="#cb32-167"></a>:::</span>
<span id="cb32-168"><a href="#cb32-168"></a></span>
<span id="cb32-169"><a href="#cb32-169"></a>The smallest number of reads in the experimental samples is <span class="in">`{r} min(finaltable[ grep("^exp", finaltable$community_type), ]$sample_tot_raw)`</span> and the mean is <span class="in">`{r} round(mean(finaltable[ grep("^exp", finaltable$community_type), ]$sample_tot_raw))`</span> which is considerably more than the smallest number of reads <span class="in">`{r} min(finaltable[ grep("^neg", finaltable$community_type), ]$sample_tot_raw)`</span> and the mean <span class="in">`{r} mean(finaltable[ grep("^neg", finaltable$community_type), ]$sample_tot_raw)`</span> in the negative controls.</span>
<span id="cb32-170"><a href="#cb32-170"></a></span>
<span id="cb32-171"><a href="#cb32-171"></a>Overall this looks pretty good. From the figure above we can see that there is a clear disjoint between about 1000 reads that separate samples that are negative controls and the experimental samples. There is one experimental sample with fewer than 1000 reads. It is likely that this sample failed or it is from a condition where no species survived.</span>
<span id="cb32-172"><a href="#cb32-172"></a></span>
<span id="cb32-173"><a href="#cb32-173"></a><span class="fu">## Negative controls</span></span>
<span id="cb32-174"><a href="#cb32-174"></a></span>
<span id="cb32-175"><a href="#cb32-175"></a>Read optical density data for the experiment</span>
<span id="cb32-176"><a href="#cb32-176"></a></span>
<span id="cb32-179"><a href="#cb32-179"></a><span class="in">```{r}</span></span>
<span id="cb32-180"><a href="#cb32-180"></a>OD <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"_data_raw"</span>, <span class="st">"communities"</span>, <span class="st">"optical_density"</span>, <span class="st">"20250425_8sp_5050pairs"</span>, <span class="st">"optical_density_formatted.tsv"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb32-181"><a href="#cb32-181"></a>  <span class="fu">mutate</span>(<span class="at">sample2 =</span> <span class="fu">case_when</span>(<span class="fu">str_detect</span>(sample, <span class="st">"com_"</span>) <span class="sc">~</span> <span class="fu">paste0</span>(<span class="st">"8"</span>, <span class="fu">str_extract</span>(sample, <span class="st">"_</span><span class="sc">\\</span><span class="st">d+$"</span>)),</span>
<span id="cb32-182"><a href="#cb32-182"></a>                             <span class="fu">str_detect</span>(sample, <span class="st">"^3_p"</span>) <span class="sc">~</span> <span class="fu">paste0</span>(<span class="st">"3_c01_"</span>, <span class="fu">str_extract</span>(well, <span class="st">"[:upper:]"</span>), <span class="fu">str_pad</span>(<span class="fu">str_extract</span>(well, <span class="st">"</span><span class="sc">\\</span><span class="st">d+"</span>), <span class="dv">2</span>, <span class="st">"left"</span>, <span class="at">pad =</span> <span class="st">"0"</span>)),</span>
<span id="cb32-183"><a href="#cb32-183"></a>                             <span class="fu">str_detect</span>(sample, <span class="st">"^p</span><span class="sc">\\</span><span class="st">d+"</span>) <span class="sc">~</span> <span class="fu">paste0</span>(<span class="st">"2_"</span>, <span class="fu">str_extract</span>(sample, <span class="st">"</span><span class="sc">\\</span><span class="st">d+"</span>), <span class="st">"_"</span>, <span class="fu">str_extract</span>(well, <span class="st">"[:upper:]"</span>), <span class="fu">str_pad</span>(<span class="fu">str_extract</span>(well, <span class="st">"</span><span class="sc">\\</span><span class="st">d+"</span>), <span class="dv">2</span>, <span class="st">"left"</span>, <span class="at">pad =</span> <span class="st">"0"</span>)),</span>
<span id="cb32-184"><a href="#cb32-184"></a>                             <span class="fu">str_detect</span>(sample, <span class="st">"^p"</span>) <span class="sc">~</span> <span class="fu">str_to_upper</span>(sample)))</span>
<span id="cb32-185"><a href="#cb32-185"></a><span class="in">```</span></span>
<span id="cb32-186"><a href="#cb32-186"></a></span>
<span id="cb32-187"><a href="#cb32-187"></a>First just checking the negative controls</span>
<span id="cb32-188"><a href="#cb32-188"></a></span>
<span id="cb32-191"><a href="#cb32-191"></a><span class="in">```{r}</span></span>
<span id="cb32-192"><a href="#cb32-192"></a><span class="co">#| fig-width: 7</span></span>
<span id="cb32-193"><a href="#cb32-193"></a><span class="co">#| fig-height: 3</span></span>
<span id="cb32-194"><a href="#cb32-194"></a>OD <span class="sc">%&gt;%</span> </span>
<span id="cb32-195"><a href="#cb32-195"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(sample, <span class="st">"neg_control"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb32-196"><a href="#cb32-196"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> OD)) <span class="sc">+</span></span>
<span id="cb32-197"><a href="#cb32-197"></a>  <span class="fu">geom_histogram</span>(<span class="at">bins =</span> <span class="dv">10</span>) <span class="sc">+</span></span>
<span id="cb32-198"><a href="#cb32-198"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fl">0.1</span>, <span class="at">color =</span> <span class="st">"red"</span>) <span class="sc">+</span> </span>
<span id="cb32-199"><a href="#cb32-199"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">"Sample count"</span>, <span class="at">x =</span> <span class="st">"Sample optical density"</span>)  <span class="sc">+</span></span>
<span id="cb32-200"><a href="#cb32-200"></a>  <span class="fu">facet_grid</span>(<span class="sc">~</span> transfers, <span class="at">scales=</span><span class="st">"free_y"</span>)</span>
<span id="cb32-201"><a href="#cb32-201"></a><span class="in">```</span></span>
<span id="cb32-202"><a href="#cb32-202"></a></span>
<span id="cb32-203"><a href="#cb32-203"></a>Negative controls look good. OD in all negative controls is low (&lt;0.07)</span>
<span id="cb32-204"><a href="#cb32-204"></a></span>
<span id="cb32-205"><a href="#cb32-205"></a><span class="fu">## Positive controls</span></span>
<span id="cb32-206"><a href="#cb32-206"></a></span>
<span id="cb32-207"><a href="#cb32-207"></a>No positive controls included in this run</span>
<span id="cb32-208"><a href="#cb32-208"></a></span>
<span id="cb32-209"><a href="#cb32-209"></a><span class="fu">## Misassigned reads</span></span>
<span id="cb32-210"><a href="#cb32-210"></a></span>
<span id="cb32-211"><a href="#cb32-211"></a>These libraries were only prepared with samples containing HAMBI_0403, HAMBI_1287, HAMBI_1896, and HAMBI_1977 so any time species other than these show up is just an incorrect assignment by Rbec or index leakage/crosstalk from the positive community controls. Let's check quickly how many of these there are...</span>
<span id="cb32-212"><a href="#cb32-212"></a></span>
<span id="cb32-215"><a href="#cb32-215"></a><span class="in">```{r}</span></span>
<span id="cb32-216"><a href="#cb32-216"></a>finaltable <span class="sc">%&gt;%</span> </span>
<span id="cb32-217"><a href="#cb32-217"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">str_detect</span>(strainID, <span class="st">"0403|1287|1896|1977"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb32-218"><a href="#cb32-218"></a>  <span class="fu">filter</span>(count_correct <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb32-219"><a href="#cb32-219"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(count_correct))</span>
<span id="cb32-220"><a href="#cb32-220"></a><span class="in">```</span></span>
<span id="cb32-221"><a href="#cb32-221"></a></span>
<span id="cb32-222"><a href="#cb32-222"></a>There are only a couple of random reads per experimental sample so we can safely ignore them. Since there was no positive control included here the off-target species must come from missassignments and not index leakage from community control.</span>
<span id="cb32-223"><a href="#cb32-223"></a></span>
<span id="cb32-224"><a href="#cb32-224"></a><span class="fu">## Samples with few total reads</span></span>
<span id="cb32-225"><a href="#cb32-225"></a></span>
<span id="cb32-226"><a href="#cb32-226"></a>Some of the experimental pairs had streptomycin concentrations higher than any of the species individually could tolerate. We would expect then that no species would grow successfully in these samples and that the overall biomass would be very low, thus resulting in a low number of recovered reads from these samples.</span>
<span id="cb32-227"><a href="#cb32-227"></a></span>
<span id="cb32-228"><a href="#cb32-228"></a>To look into this. first let’s check which samples have very low OD600 in the endpoint samples. We’ll filter out samples with an OD of less than 0.1 or samples with fewer than 3000 reads. It is generally good practice to exclude samples with low number of reads.</span>
<span id="cb32-229"><a href="#cb32-229"></a></span>
<span id="cb32-232"><a href="#cb32-232"></a><span class="in">```{r}</span></span>
<span id="cb32-233"><a href="#cb32-233"></a><span class="fu">left_join</span>(finaltable, OD, </span>
<span id="cb32-234"><a href="#cb32-234"></a>                 <span class="at">by =</span> <span class="fu">join_by</span>(sample <span class="sc">==</span> sample2)) <span class="sc">%&gt;%</span> </span>
<span id="cb32-235"><a href="#cb32-235"></a>  <span class="fu">filter</span>(transfers.y <span class="sc">==</span> <span class="dv">8</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb32-236"><a href="#cb32-236"></a>  <span class="fu">filter</span>(OD <span class="sc">&lt;</span> <span class="fl">0.1</span> <span class="sc">|</span> sample_tot_raw <span class="sc">&lt;</span> <span class="dv">1000</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb32-237"><a href="#cb32-237"></a>  <span class="fu">distinct</span>(sample, community_type, OD, sample_tot_raw) <span class="sc">%&gt;%</span></span>
<span id="cb32-238"><a href="#cb32-238"></a>  <span class="fu">arrange</span>(sample_tot_raw)</span>
<span id="cb32-239"><a href="#cb32-239"></a><span class="in">```</span></span>
<span id="cb32-240"><a href="#cb32-240"></a></span>
<span id="cb32-243"><a href="#cb32-243"></a><span class="in">```{r}</span></span>
<span id="cb32-244"><a href="#cb32-244"></a>df <span class="ot">&lt;-</span> <span class="fu">left_join</span>(finaltable, OD, </span>
<span id="cb32-245"><a href="#cb32-245"></a>                 <span class="at">by =</span> <span class="fu">join_by</span>(sample <span class="sc">==</span> sample2)) <span class="sc">%&gt;%</span> </span>
<span id="cb32-246"><a href="#cb32-246"></a>  <span class="fu">filter</span>(transfers.y <span class="sc">==</span> <span class="dv">8</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb32-247"><a href="#cb32-247"></a>  <span class="fu">filter</span>(OD <span class="sc">&lt;</span> <span class="fl">0.1</span> <span class="sc">|</span> sample_tot_raw <span class="sc">&lt;</span> <span class="dv">1000</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb32-248"><a href="#cb32-248"></a>  <span class="fu">distinct</span>(sample, community_type, OD, sample_tot_raw) </span>
<span id="cb32-249"><a href="#cb32-249"></a></span>
<span id="cb32-250"><a href="#cb32-250"></a>lowread_samps <span class="ot">&lt;-</span> df <span class="sc">%&gt;%</span> </span>
<span id="cb32-251"><a href="#cb32-251"></a>  <span class="fu">filter</span>(sample_tot_raw <span class="sc">&lt;</span> <span class="dv">1000</span> <span class="sc">&amp;</span> community_type <span class="sc">==</span> <span class="st">"experiment"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb32-252"><a href="#cb32-252"></a>  <span class="fu">distinct</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb32-253"><a href="#cb32-253"></a>  <span class="fu">pull</span>(sample)</span>
<span id="cb32-254"><a href="#cb32-254"></a><span class="in">```</span></span>
<span id="cb32-255"><a href="#cb32-255"></a></span>
<span id="cb32-256"><a href="#cb32-256"></a>There is only one low-read sample. It corresponds to a pair with both ancestral HAMBI_1977 and HAMBI_0403 in 256 ug/ml streptomycin. From monoculture results we know this streptomycin concentration is outside the tolerable range for these two species. Indeed, the OD in this sample at time point 8 is 0.046, which indicates very low biomass at the experiment's conclusion.</span>
<span id="cb32-257"><a href="#cb32-257"></a></span>
<span id="cb32-258"><a href="#cb32-258"></a><span class="fu">## Filter to target species</span></span>
<span id="cb32-259"><a href="#cb32-259"></a></span>
<span id="cb32-260"><a href="#cb32-260"></a>Exclude positive and negative controls and also exclude species that are not in the focal 4</span>
<span id="cb32-261"><a href="#cb32-261"></a></span>
<span id="cb32-264"><a href="#cb32-264"></a><span class="in">```{r}</span></span>
<span id="cb32-265"><a href="#cb32-265"></a>finaltable_exp_mstr <span class="ot">&lt;-</span> finaltable <span class="sc">%&gt;%</span> </span>
<span id="cb32-266"><a href="#cb32-266"></a>  <span class="fu">filter</span>(community_type <span class="sc">%nin%</span> <span class="fu">c</span>(<span class="st">"empty"</span>, <span class="st">"neg_control"</span>, <span class="st">"pos_control"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb32-267"><a href="#cb32-267"></a>  <span class="fu">filter</span>(sample <span class="sc">%nin%</span> lowread_samps) <span class="sc">%&gt;%</span> </span>
<span id="cb32-268"><a href="#cb32-268"></a>  <span class="fu">filter</span>(strainID <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"HAMBI_0403"</span>, <span class="st">"HAMBI_1287"</span>, <span class="st">"HAMBI_1896"</span>, <span class="st">"HAMBI_1977"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb32-269"><a href="#cb32-269"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb32-270"><a href="#cb32-270"></a>  <span class="fu">mutate</span>(<span class="at">f_raw_targetsp =</span> count_correct<span class="sc">/</span><span class="fu">sum</span>(count_correct),</span>
<span id="cb32-271"><a href="#cb32-271"></a>         <span class="at">sample_tot_targetsp =</span> <span class="fu">sum</span>(count_correct)) <span class="sc">%&gt;%</span> </span>
<span id="cb32-272"><a href="#cb32-272"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb32-273"><a href="#cb32-273"></a>  <span class="fu">relocate</span>(<span class="fu">c</span>(sample_tot_targetsp, f_raw_targetsp), <span class="at">.after =</span> f_raw)</span>
<span id="cb32-274"><a href="#cb32-274"></a><span class="in">```</span></span>
<span id="cb32-275"><a href="#cb32-275"></a></span>
<span id="cb32-276"><a href="#cb32-276"></a><span class="fu">## Master plates samples</span></span>
<span id="cb32-277"><a href="#cb32-277"></a></span>
<span id="cb32-278"><a href="#cb32-278"></a>Crosstalk on master plates is quite good - usually &lt; 1% frequency.</span>
<span id="cb32-279"><a href="#cb32-279"></a></span>
<span id="cb32-282"><a href="#cb32-282"></a><span class="in">```{r}</span></span>
<span id="cb32-283"><a href="#cb32-283"></a>finaltable_exp_mstr <span class="sc">%&gt;%</span> </span>
<span id="cb32-284"><a href="#cb32-284"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(community_type, <span class="st">"master"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb32-285"><a href="#cb32-285"></a>  <span class="fu">filter</span>(<span class="fu">is.na</span>(evo_hist)) <span class="sc">%&gt;%</span> </span>
<span id="cb32-286"><a href="#cb32-286"></a>  <span class="fu">filter</span>(f_raw_targetsp <span class="sc">&gt;</span> <span class="fl">0.01</span> ) <span class="sc">%&gt;%</span> </span>
<span id="cb32-287"><a href="#cb32-287"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(sample, strainID, f_raw_targetsp) <span class="sc">%&gt;%</span> </span>
<span id="cb32-288"><a href="#cb32-288"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(f_raw_targetsp))</span>
<span id="cb32-289"><a href="#cb32-289"></a><span class="in">```</span></span>
<span id="cb32-290"><a href="#cb32-290"></a></span>
<span id="cb32-291"><a href="#cb32-291"></a>::: {#fig-09}</span>
<span id="cb32-294"><a href="#cb32-294"></a><span class="in">```{r}</span></span>
<span id="cb32-295"><a href="#cb32-295"></a><span class="co">#| fig.width: 8</span></span>
<span id="cb32-296"><a href="#cb32-296"></a><span class="co">#| fig.height: 4</span></span>
<span id="cb32-297"><a href="#cb32-297"></a>finaltable_exp_mstr <span class="sc">%&gt;%</span> </span>
<span id="cb32-298"><a href="#cb32-298"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(community_type, <span class="st">"master"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb32-299"><a href="#cb32-299"></a>  <span class="fu">filter</span>(<span class="fu">is.na</span>(evo_hist)) <span class="sc">%&gt;%</span> </span>
<span id="cb32-300"><a href="#cb32-300"></a>  <span class="fu">contam_histogram</span>(<span class="at">trans=</span><span class="cn">TRUE</span>, <span class="at">x =</span> f_raw_targetsp) <span class="sc">+</span></span>
<span id="cb32-301"><a href="#cb32-301"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Species frequency"</span>, <span class="at">y =</span> <span class="st">"Sample count"</span>) <span class="sc">+</span></span>
<span id="cb32-302"><a href="#cb32-302"></a>  <span class="fu">ggtitle</span>(<span class="st">"Frequency distribution of each species in master plate samples where it should not be"</span>)</span>
<span id="cb32-303"><a href="#cb32-303"></a><span class="in">```</span></span>
<span id="cb32-304"><a href="#cb32-304"></a>Frequency distribution of each species in masterplate samples where it **should not** occur and where it exceeds a 3% threshold.</span>
<span id="cb32-305"><a href="#cb32-305"></a>:::</span>
<span id="cb32-306"><a href="#cb32-306"></a></span>
<span id="cb32-307"><a href="#cb32-307"></a><span class="fu">## Experimental samples</span></span>
<span id="cb32-308"><a href="#cb32-308"></a></span>
<span id="cb32-311"><a href="#cb32-311"></a><span class="in">```{r}</span></span>
<span id="cb32-312"><a href="#cb32-312"></a>finaltable_exp_mstr <span class="sc">%&gt;%</span> </span>
<span id="cb32-313"><a href="#cb32-313"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">str_detect</span>(community_type, <span class="st">"master"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb32-314"><a href="#cb32-314"></a>  <span class="fu">filter</span>(<span class="fu">is.na</span>(evo_hist)) <span class="sc">%&gt;%</span> </span>
<span id="cb32-315"><a href="#cb32-315"></a>  <span class="fu">filter</span>(f_raw_targetsp <span class="sc">&gt;</span> <span class="fl">0.01</span> ) <span class="sc">%&gt;%</span> </span>
<span id="cb32-316"><a href="#cb32-316"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(sample, strainID, f_raw_targetsp) <span class="sc">%&gt;%</span> </span>
<span id="cb32-317"><a href="#cb32-317"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(f_raw_targetsp))</span>
<span id="cb32-318"><a href="#cb32-318"></a><span class="in">```</span></span>
<span id="cb32-319"><a href="#cb32-319"></a></span>
<span id="cb32-322"><a href="#cb32-322"></a><span class="in">```{r}</span></span>
<span id="cb32-323"><a href="#cb32-323"></a><span class="co">#| eval: true</span></span>
<span id="cb32-324"><a href="#cb32-324"></a><span class="co">#| echo: false</span></span>
<span id="cb32-325"><a href="#cb32-325"></a><span class="co">#| output: false</span></span>
<span id="cb32-326"><a href="#cb32-326"></a>myfilter <span class="ot">&lt;-</span> <span class="cf">function</span>(.data, myexpresion){</span>
<span id="cb32-327"><a href="#cb32-327"></a>  .data <span class="sc">%&gt;%</span> </span>
<span id="cb32-328"><a href="#cb32-328"></a>    <span class="fu">filter</span>(<span class="fu">str_detect</span>(community_type, <span class="st">"experiment"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb32-329"><a href="#cb32-329"></a>    <span class="fu">filter</span>(<span class="fu">is.na</span>(evo_hist)) <span class="sc">%&gt;%</span> </span>
<span id="cb32-330"><a href="#cb32-330"></a>    <span class="fu">filter</span>( {{ myexpresion }} ) <span class="sc">%&gt;%</span> </span>
<span id="cb32-331"><a href="#cb32-331"></a>    <span class="fu">distinct</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb32-332"><a href="#cb32-332"></a>    <span class="fu">nrow</span>()</span>
<span id="cb32-333"><a href="#cb32-333"></a>}</span>
<span id="cb32-334"><a href="#cb32-334"></a></span>
<span id="cb32-335"><a href="#cb32-335"></a>thresh <span class="ot">&lt;-</span> <span class="fl">0.03</span></span>
<span id="cb32-336"><a href="#cb32-336"></a>totsamp <span class="ot">&lt;-</span> <span class="fu">nrow</span>(<span class="fu">distinct</span>(finaltable_exp_mstr, sample))</span>
<span id="cb32-337"><a href="#cb32-337"></a>contamsamp <span class="ot">&lt;-</span> <span class="fu">myfilter</span>(finaltable_exp_mstr, f_raw_targetsp <span class="sc">&gt;=</span> <span class="fl">0.01</span>)</span>
<span id="cb32-338"><a href="#cb32-338"></a>contamsamp_lo <span class="ot">&lt;-</span> <span class="fu">myfilter</span>(finaltable_exp_mstr, f_raw_targetsp <span class="sc">&gt;=</span> <span class="fl">0.01</span> <span class="sc">&amp;</span> f_raw_targetsp <span class="sc">&lt;</span> thresh)</span>
<span id="cb32-339"><a href="#cb32-339"></a>contamsamp_hi <span class="ot">&lt;-</span> <span class="fu">myfilter</span>(finaltable_exp_mstr, f_raw_targetsp <span class="sc">&gt;=</span> thresh)</span>
<span id="cb32-340"><a href="#cb32-340"></a><span class="in">```</span></span>
<span id="cb32-341"><a href="#cb32-341"></a></span>
<span id="cb32-342"><a href="#cb32-342"></a>There are <span class="in">`{r} contamsamp`</span> samples (<span class="in">`{r} round(contamsamp/totsamp*100)`</span>% of <span class="in">`{r} totsamp`</span> samples) that exceed a 1% crosstalk threshold. <span class="in">`{r} contamsamp_lo`</span> samples (<span class="in">`{r} round(contamsamp_lo/totsamp*100)`</span>% of <span class="in">`{r} totsamp`</span> samples) have contamination &lt;= <span class="in">`{r} thresh*100`</span>% and <span class="in">`{r} contamsamp_hi`</span> samples (<span class="in">`{r} round(contamsamp_hi/totsamp*100)`</span> % of <span class="in">`{r} totsamp`</span> samples) have &gt; <span class="in">`{r} thresh*100`</span>% of a species that should not be there. In most cases the species in highly contaminated samples is HAMBI_1896. In general, this level of contamination is quite low and samples with contamination greater than <span class="in">`{r} thresh*100`</span>% are just crosstalk outliers.</span>
<span id="cb32-343"><a href="#cb32-343"></a></span>
<span id="cb32-344"><a href="#cb32-344"></a>::: {#fig-11}</span>
<span id="cb32-347"><a href="#cb32-347"></a><span class="in">```{r}</span></span>
<span id="cb32-348"><a href="#cb32-348"></a><span class="co">#| fig.width: 9</span></span>
<span id="cb32-349"><a href="#cb32-349"></a><span class="co">#| fig.height: 4</span></span>
<span id="cb32-350"><a href="#cb32-350"></a>finaltable_exp_mstr <span class="sc">%&gt;%</span> </span>
<span id="cb32-351"><a href="#cb32-351"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(community_type, <span class="st">"experiment"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb32-352"><a href="#cb32-352"></a>  <span class="fu">contaminated_barplot</span>(<span class="at">threshold=</span><span class="fl">0.01</span>, <span class="at">quartet =</span> <span class="cn">FALSE</span>, <span class="at">y=</span>f_raw_targetsp) <span class="sc">+</span></span>
<span id="cb32-353"><a href="#cb32-353"></a>  <span class="fu">ggtitle</span>(<span class="st">"Experiment samples with &gt; 1% contamination"</span>)</span>
<span id="cb32-354"><a href="#cb32-354"></a><span class="in">```</span></span>
<span id="cb32-355"><a href="#cb32-355"></a>Species composition of experiment samples that contain &gt; 3% of a species that shouldn’t be there (shown in grey to black). </span>
<span id="cb32-356"><a href="#cb32-356"></a>:::</span>
<span id="cb32-357"><a href="#cb32-357"></a></span>
<span id="cb32-360"><a href="#cb32-360"></a><span class="in">```{r}</span></span>
<span id="cb32-361"><a href="#cb32-361"></a>contamtrios <span class="ot">&lt;-</span> finaltable_exp_mstr <span class="sc">%&gt;%</span> </span>
<span id="cb32-362"><a href="#cb32-362"></a>  <span class="fu">filter</span>(n_species <span class="sc">==</span> <span class="dv">3</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb32-363"><a href="#cb32-363"></a>  <span class="fu">filter</span>(<span class="fu">is.na</span>(evo_hist)) <span class="sc">%&gt;%</span> </span>
<span id="cb32-364"><a href="#cb32-364"></a>  <span class="fu">filter</span>(f_raw_targetsp <span class="sc">&gt;</span> <span class="fl">0.13</span>  ) <span class="sc">%&gt;%</span> </span>
<span id="cb32-365"><a href="#cb32-365"></a>  <span class="fu">pull</span>(sample)</span>
<span id="cb32-366"><a href="#cb32-366"></a><span class="in">```</span></span>
<span id="cb32-367"><a href="#cb32-367"></a></span>
<span id="cb32-368"><a href="#cb32-368"></a>Only 6 trio samples look really bad (~50 % contamination). Again, I think we should only exclude samples where the contamination is very high (over ~50% of the sample) but those with around 10% or less contaminant I think can be retained, and I will discard the contaminating sequences. Interestingly the lower-level contamination (3% &gt; contamination &gt; 13%) seems to be mostly from from HAMBI_1896. The high level contamination is always in well C01 on 6 different plates (<span class="in">`{r} contamtrios`</span>).</span>
<span id="cb32-369"><a href="#cb32-369"></a></span>
<span id="cb32-370"><a href="#cb32-370"></a><span class="fu">## Plot</span></span>
<span id="cb32-371"><a href="#cb32-371"></a></span>
<span id="cb32-372"><a href="#cb32-372"></a><span class="fu">### Pairs and trios</span></span>
<span id="cb32-373"><a href="#cb32-373"></a></span>
<span id="cb32-374"><a href="#cb32-374"></a>::: {#fig-12}</span>
<span id="cb32-377"><a href="#cb32-377"></a><span class="in">```{r}</span></span>
<span id="cb32-378"><a href="#cb32-378"></a><span class="co">#| fig.width: 10</span></span>
<span id="cb32-379"><a href="#cb32-379"></a><span class="co">#| fig.height: 14</span></span>
<span id="cb32-380"><a href="#cb32-380"></a>finaltable_exp_mstr <span class="sc">%&gt;%</span> </span>
<span id="cb32-381"><a href="#cb32-381"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(community_type, <span class="st">"experiment"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb32-382"><a href="#cb32-382"></a>  <span class="fu">filter</span>(n_species <span class="sc">!=</span> <span class="dv">8</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb32-383"><a href="#cb32-383"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb32-384"><a href="#cb32-384"></a>  <span class="fu">mutate</span>(<span class="at">id =</span> <span class="fu">cur_group_id</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb32-385"><a href="#cb32-385"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb32-386"><a href="#cb32-386"></a>  <span class="fu">mutate</span>(<span class="at">facet =</span> <span class="fu">ntile</span>(id, <span class="dv">5</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb32-387"><a href="#cb32-387"></a>  <span class="co"># to make sure ntile doesnt split up facets</span></span>
<span id="cb32-388"><a href="#cb32-388"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb32-389"><a href="#cb32-389"></a>  <span class="fu">mutate</span>(<span class="at">facet =</span> <span class="fu">first</span>(facet)) <span class="sc">%&gt;%</span> </span>
<span id="cb32-390"><a href="#cb32-390"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb32-391"><a href="#cb32-391"></a>  <span class="fu">contaminated_barplot</span>(<span class="at">threshold =</span> <span class="dv">0</span>, <span class="at">quartet =</span> <span class="cn">TRUE</span>, <span class="at">y=</span>f_raw_targetsp) <span class="sc">+</span></span>
<span id="cb32-392"><a href="#cb32-392"></a>  <span class="fu">ggtitle</span>(<span class="st">"All experimental pairs and trios"</span>) <span class="sc">+</span></span>
<span id="cb32-393"><a href="#cb32-393"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>facet, <span class="at">scales=</span><span class="st">"free_x"</span>, <span class="at">nrow=</span><span class="dv">5</span>) <span class="sc">+</span></span>
<span id="cb32-394"><a href="#cb32-394"></a>  <span class="fu">theme</span>(</span>
<span id="cb32-395"><a href="#cb32-395"></a>    <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb32-396"><a href="#cb32-396"></a>    <span class="at">strip.text.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb32-397"><a href="#cb32-397"></a>    <span class="at">legend.position =</span> <span class="st">"bottom"</span></span>
<span id="cb32-398"><a href="#cb32-398"></a>    )</span>
<span id="cb32-399"><a href="#cb32-399"></a><span class="in">```</span></span>
<span id="cb32-400"><a href="#cb32-400"></a>Species composition of all experiment trio samples. Species that should not be present in a sample are shown in grey to black. </span>
<span id="cb32-401"><a href="#cb32-401"></a>:::</span>
<span id="cb32-402"><a href="#cb32-402"></a></span>
<span id="cb32-403"><a href="#cb32-403"></a>At a birds-eye view this looks pretty good overall.</span>
<span id="cb32-404"><a href="#cb32-404"></a></span>
<span id="cb32-405"><a href="#cb32-405"></a><span class="fu">### Octets {#sec-octets2}</span></span>
<span id="cb32-406"><a href="#cb32-406"></a></span>
<span id="cb32-407"><a href="#cb32-407"></a>Again, one weakness of this approach is that there is no way to tell from this data whether an octets is contaminated because every octet (by definition) is inoculated with all of the 4 species at different frequencies and with different evolutionary histories. Thus we would expect to see either all 4 species present or a subset of the 4 species in the quartet samples, but we cannot tell if these patterns come from contamination or not. Moving forward we just have to assume that the quartets are OK... However, we've found so little contamination in our other experimental samples and all samples were handled in the experiment and processed for sequence library preparation in the same way. I think it is safe to assume that contamination in the quartets is minimal issue. </span>
<span id="cb32-408"><a href="#cb32-408"></a></span>
<span id="cb32-409"><a href="#cb32-409"></a>::: {#fig-12}</span>
<span id="cb32-412"><a href="#cb32-412"></a><span class="in">```{r}</span></span>
<span id="cb32-413"><a href="#cb32-413"></a><span class="co">#| fig.width: 10</span></span>
<span id="cb32-414"><a href="#cb32-414"></a><span class="co">#| fig.height: 14</span></span>
<span id="cb32-415"><a href="#cb32-415"></a>finaltable_exp_mstr <span class="sc">%&gt;%</span> </span>
<span id="cb32-416"><a href="#cb32-416"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(community_type, <span class="st">"experiment"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb32-417"><a href="#cb32-417"></a>  <span class="fu">filter</span>(n_species <span class="sc">==</span> <span class="dv">8</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb32-418"><a href="#cb32-418"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(sample, strainID, count_correct) <span class="sc">%&gt;%</span> </span>
<span id="cb32-419"><a href="#cb32-419"></a>  <span class="fu">distinct</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb32-420"><a href="#cb32-420"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb32-421"><a href="#cb32-421"></a>  <span class="fu">mutate</span>(<span class="at">f_raw_targetsp =</span> count_correct<span class="sc">/</span><span class="fu">sum</span>(count_correct)) <span class="sc">%&gt;%</span> </span>
<span id="cb32-422"><a href="#cb32-422"></a>  <span class="fu">mutate</span>(<span class="at">id =</span> <span class="fu">cur_group_id</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb32-423"><a href="#cb32-423"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb32-424"><a href="#cb32-424"></a>  <span class="fu">mutate</span>(<span class="at">facet =</span> <span class="fu">ntile</span>(id, <span class="dv">5</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb32-425"><a href="#cb32-425"></a>  <span class="co"># to make sure ntile doesnt split up facets</span></span>
<span id="cb32-426"><a href="#cb32-426"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb32-427"><a href="#cb32-427"></a>  <span class="fu">mutate</span>(<span class="at">facet =</span> <span class="fu">first</span>(facet)) <span class="sc">%&gt;%</span> </span>
<span id="cb32-428"><a href="#cb32-428"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb32-429"><a href="#cb32-429"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb32-430"><a href="#cb32-430"></a>    <span class="fu">geom_col</span>(<span class="fu">aes</span>(<span class="at">x =</span> sample, <span class="at">y =</span> f_raw_targetsp, <span class="at">fill =</span> strainID)) <span class="sc">+</span></span>
<span id="cb32-431"><a href="#cb32-431"></a>    <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">"Percent abundance"</span>, <span class="at">x =</span> <span class="st">""</span>, <span class="at">fill =</span> <span class="st">""</span>) <span class="sc">+</span></span>
<span id="cb32-432"><a href="#cb32-432"></a>    <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"HAMBI_0403"</span> <span class="ot">=</span> <span class="st">"#faa019"</span>,</span>
<span id="cb32-433"><a href="#cb32-433"></a>              <span class="st">"HAMBI_1287"</span> <span class="ot">=</span> <span class="st">"#75afff"</span>,</span>
<span id="cb32-434"><a href="#cb32-434"></a>              <span class="st">"HAMBI_1896"</span> <span class="ot">=</span> <span class="st">"#59cc4e"</span>,</span>
<span id="cb32-435"><a href="#cb32-435"></a>              <span class="st">"HAMBI_1977"</span> <span class="ot">=</span> <span class="st">"#ffd430"</span>)</span>
<span id="cb32-436"><a href="#cb32-436"></a>              ) <span class="sc">+</span></span>
<span id="cb32-437"><a href="#cb32-437"></a>    <span class="fu">scale_y_continuous</span>(<span class="at">labels =</span> percent) <span class="sc">+</span></span>
<span id="cb32-438"><a href="#cb32-438"></a>    <span class="fu">scale_x_discrete</span>(<span class="at">guide =</span> <span class="fu">guide_axis</span>(<span class="at">angle=</span><span class="dv">90</span>)) <span class="sc">+</span></span>
<span id="cb32-439"><a href="#cb32-439"></a>    <span class="fu">theme_bw</span>() <span class="sc">+</span> </span>
<span id="cb32-440"><a href="#cb32-440"></a>    <span class="fu">theme</span>(<span class="at">panel.grid =</span> <span class="fu">element_blank</span>(), <span class="at">legend.position =</span> <span class="st">"right"</span>) <span class="sc">+</span></span>
<span id="cb32-441"><a href="#cb32-441"></a>    <span class="fu">ggtitle</span>(<span class="st">"All experimental octets"</span>) <span class="sc">+</span></span>
<span id="cb32-442"><a href="#cb32-442"></a>    <span class="fu">facet_wrap</span>(<span class="sc">~</span>facet, <span class="at">scales=</span><span class="st">"free_x"</span>, <span class="at">nrow=</span><span class="dv">5</span>) <span class="sc">+</span></span>
<span id="cb32-443"><a href="#cb32-443"></a>    <span class="fu">theme</span>(</span>
<span id="cb32-444"><a href="#cb32-444"></a>    <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb32-445"><a href="#cb32-445"></a>    <span class="at">strip.text.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb32-446"><a href="#cb32-446"></a>    <span class="at">legend.position =</span> <span class="st">"bottom"</span></span>
<span id="cb32-447"><a href="#cb32-447"></a>    )</span>
<span id="cb32-448"><a href="#cb32-448"></a><span class="in">```</span></span>
<span id="cb32-449"><a href="#cb32-449"></a>Species composition of all experiment 8 strain samples.</span>
<span id="cb32-450"><a href="#cb32-450"></a>:::</span>
<span id="cb32-451"><a href="#cb32-451"></a></span>
<span id="cb32-452"><a href="#cb32-452"></a><span class="fu"># Export</span></span>
<span id="cb32-453"><a href="#cb32-453"></a></span>
<span id="cb32-456"><a href="#cb32-456"></a><span class="in">```{r}</span></span>
<span id="cb32-457"><a href="#cb32-457"></a>finaltable_exp_mstr_export <span class="ot">&lt;-</span> finaltable_exp_mstr <span class="sc">%&gt;%</span> </span>
<span id="cb32-458"><a href="#cb32-458"></a>  <span class="co"># first remove samples with low read count (&lt; 3000 reads)</span></span>
<span id="cb32-459"><a href="#cb32-459"></a>  <span class="fu">filter</span>(sample <span class="sc">%nin%</span> lowread_samps) <span class="sc">%&gt;%</span> </span>
<span id="cb32-460"><a href="#cb32-460"></a>  <span class="co"># exclude any remaining counts from species that shouldnt be there using again</span></span>
<span id="cb32-461"><a href="#cb32-461"></a>  <span class="co"># the fact that evo_hist should be NA for these species</span></span>
<span id="cb32-462"><a href="#cb32-462"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(evo_hist)) <span class="sc">%&gt;%</span> </span>
<span id="cb32-463"><a href="#cb32-463"></a>  <span class="co"># because we set 3% as our limit of detection we set read counts of species</span></span>
<span id="cb32-464"><a href="#cb32-464"></a>  <span class="co"># less than 3% to 0</span></span>
<span id="cb32-465"><a href="#cb32-465"></a>  <span class="fu">mutate</span>(<span class="at">count_correct_thresh =</span> <span class="fu">if_else</span>(f_raw_targetsp <span class="sc">&lt;=</span> <span class="fl">0.03</span>, <span class="dv">0</span>, count_correct)) <span class="sc">%&gt;%</span> </span>
<span id="cb32-466"><a href="#cb32-466"></a>  <span class="co"># now calculate a new relative abundance based only on the species that should</span></span>
<span id="cb32-467"><a href="#cb32-467"></a>  <span class="co"># be present and that are &gt; 3% relative abundance</span></span>
<span id="cb32-468"><a href="#cb32-468"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb32-469"><a href="#cb32-469"></a>  <span class="fu">mutate</span>(<span class="at">f_thresh =</span> count_correct_thresh<span class="sc">/</span><span class="fu">sum</span>(count_correct_thresh)) <span class="sc">%&gt;%</span> </span>
<span id="cb32-470"><a href="#cb32-470"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb32-471"><a href="#cb32-471"></a>  <span class="co"># for marking which sequencing batch these came from </span></span>
<span id="cb32-472"><a href="#cb32-472"></a>  <span class="fu">mutate</span>(<span class="at">batch =</span> <span class="st">"run20250502"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb32-473"><a href="#cb32-473"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(sample, strainID, evo_hist, count_correct_thresh, f_thresh,</span>
<span id="cb32-474"><a href="#cb32-474"></a>                <span class="at">target_f_masterplate =</span> target_f, replicate, strep_conc, </span>
<span id="cb32-475"><a href="#cb32-475"></a>                transfers, n_species, community_type, plate_well, batch)</span>
<span id="cb32-476"><a href="#cb32-476"></a><span class="in">```</span></span>
<span id="cb32-477"><a href="#cb32-477"></a></span>
<span id="cb32-478"><a href="#cb32-478"></a><span class="fu">## Pairs</span></span>
<span id="cb32-479"><a href="#cb32-479"></a></span>
<span id="cb32-482"><a href="#cb32-482"></a><span class="in">```{r}</span></span>
<span id="cb32-483"><a href="#cb32-483"></a>finaltable_exp_mstr_export <span class="sc">%&gt;%</span></span>
<span id="cb32-484"><a href="#cb32-484"></a>  <span class="fu">filter</span>(n_species <span class="sc">==</span> <span class="dv">2</span>) <span class="sc">%&gt;%</span></span>
<span id="cb32-485"><a href="#cb32-485"></a>  <span class="fu">arrange</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb32-486"><a href="#cb32-486"></a>  <span class="co"># reformating to be compatible with other pair data</span></span>
<span id="cb32-487"><a href="#cb32-487"></a>  <span class="fu">mutate</span>(<span class="at">sample =</span> <span class="fu">if_else</span>(</span>
<span id="cb32-488"><a href="#cb32-488"></a>    community_type <span class="sc">==</span> <span class="st">"master"</span>,</span>
<span id="cb32-489"><a href="#cb32-489"></a>    <span class="fu">paste0</span>(<span class="st">"P"</span>, </span>
<span id="cb32-490"><a href="#cb32-490"></a>           <span class="fu">str_extract</span>(sample, <span class="at">pattern =</span> <span class="st">'(?&lt;=2_)</span><span class="sc">\\</span><span class="st">d{2}'</span>), <span class="st">"_s_0"</span>),</span>
<span id="cb32-491"><a href="#cb32-491"></a>    <span class="fu">paste0</span>(<span class="st">"P"</span>,</span>
<span id="cb32-492"><a href="#cb32-492"></a>           <span class="fu">str_extract</span>(sample, <span class="at">pattern =</span> <span class="st">'(?&lt;=2_)</span><span class="sc">\\</span><span class="st">d{2}'</span>),</span>
<span id="cb32-493"><a href="#cb32-493"></a>           <span class="st">"_"</span>,</span>
<span id="cb32-494"><a href="#cb32-494"></a>           replicate,</span>
<span id="cb32-495"><a href="#cb32-495"></a>           <span class="st">"_"</span>,</span>
<span id="cb32-496"><a href="#cb32-496"></a>           strep_conc</span>
<span id="cb32-497"><a href="#cb32-497"></a>    )</span>
<span id="cb32-498"><a href="#cb32-498"></a>  )) <span class="sc">%&gt;%</span> </span>
<span id="cb32-499"><a href="#cb32-499"></a>  <span class="fu">write_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(data, <span class="st">"pairs_counts.tsv"</span>))</span>
<span id="cb32-500"><a href="#cb32-500"></a><span class="in">```</span></span>
<span id="cb32-501"><a href="#cb32-501"></a></span>
<span id="cb32-502"><a href="#cb32-502"></a><span class="fu">## Trios</span></span>
<span id="cb32-503"><a href="#cb32-503"></a></span>
<span id="cb32-504"><a href="#cb32-504"></a>Write the trio samples</span>
<span id="cb32-505"><a href="#cb32-505"></a></span>
<span id="cb32-508"><a href="#cb32-508"></a><span class="in">```{r}</span></span>
<span id="cb32-509"><a href="#cb32-509"></a>finaltable_exp_mstr_export <span class="sc">%&gt;%</span> </span>
<span id="cb32-510"><a href="#cb32-510"></a>  <span class="fu">filter</span>(n_species <span class="sc">==</span> <span class="dv">3</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb32-511"><a href="#cb32-511"></a>  <span class="fu">arrange</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb32-512"><a href="#cb32-512"></a>  <span class="fu">write_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(data, <span class="st">"trios_counts.tsv"</span>))</span>
<span id="cb32-513"><a href="#cb32-513"></a><span class="in">```</span></span>
<span id="cb32-514"><a href="#cb32-514"></a></span>
<span id="cb32-515"><a href="#cb32-515"></a><span class="fu">## Octets</span></span>
<span id="cb32-516"><a href="#cb32-516"></a></span>
<span id="cb32-517"><a href="#cb32-517"></a>Write the octet samples</span>
<span id="cb32-518"><a href="#cb32-518"></a></span>
<span id="cb32-521"><a href="#cb32-521"></a><span class="in">```{r}</span></span>
<span id="cb32-522"><a href="#cb32-522"></a>finaltable_exp_mstr_export <span class="sc">%&gt;%</span> </span>
<span id="cb32-523"><a href="#cb32-523"></a>  <span class="fu">filter</span>(n_species <span class="sc">==</span> <span class="dv">8</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb32-524"><a href="#cb32-524"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb32-525"><a href="#cb32-525"></a>  <span class="fu">mutate</span>(<span class="at">max_f_sp =</span> <span class="fu">if_else</span>(target_f_masterplate <span class="sc">==</span> <span class="fu">max</span>(target_f_masterplate), </span>
<span id="cb32-526"><a href="#cb32-526"></a>                            <span class="fu">paste0</span>(<span class="fu">str_to_upper</span>(evo_hist), <span class="st">"_"</span>, <span class="fu">str_extract</span>(strainID, <span class="st">"</span><span class="sc">\\</span><span class="st">d+"</span>)), </span>
<span id="cb32-527"><a href="#cb32-527"></a>                            <span class="cn">NA_character_</span>),</span>
<span id="cb32-528"><a href="#cb32-528"></a>         <span class="at">max_f =</span> <span class="fu">max</span>(target_f_masterplate)) <span class="sc">%&gt;%</span> </span>
<span id="cb32-529"><a href="#cb32-529"></a>  <span class="fu">fill</span>(max_f_sp, <span class="at">.direction =</span> <span class="st">"downup"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb32-530"><a href="#cb32-530"></a>  <span class="fu">group_by</span>(sample, strainID) <span class="sc">%&gt;%</span> </span>
<span id="cb32-531"><a href="#cb32-531"></a>  <span class="fu">mutate</span>(<span class="at">target_f_masterplate =</span> <span class="fu">sum</span>(target_f_masterplate)) <span class="sc">%&gt;%</span> </span>
<span id="cb32-532"><a href="#cb32-532"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb32-533"><a href="#cb32-533"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>evo_hist) <span class="sc">%&gt;%</span> </span>
<span id="cb32-534"><a href="#cb32-534"></a>  <span class="co"># deal with case when all species are equal</span></span>
<span id="cb32-535"><a href="#cb32-535"></a>  <span class="fu">mutate</span>(<span class="at">max_f_sp =</span> <span class="fu">if_else</span>(max_f <span class="sc">==</span> <span class="fl">0.13</span>, <span class="st">"none"</span>, max_f_sp)) <span class="sc">%&gt;%</span> </span>
<span id="cb32-536"><a href="#cb32-536"></a>  <span class="fu">mutate</span>(<span class="at">target_f_masterplate =</span> <span class="fu">if_else</span>(target_f_masterplate <span class="sc">==</span> <span class="fl">0.26</span>, <span class="fl">0.25</span>, target_f_masterplate)) <span class="sc">%&gt;%</span> </span>
<span id="cb32-537"><a href="#cb32-537"></a>  <span class="fu">distinct</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb32-538"><a href="#cb32-538"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb32-539"><a href="#cb32-539"></a>  <span class="fu">mutate</span>(<span class="at">f_thresh =</span> count_correct_thresh<span class="sc">/</span><span class="fu">sum</span>(count_correct_thresh)) <span class="sc">%&gt;%</span> </span>
<span id="cb32-540"><a href="#cb32-540"></a>  <span class="fu">relocate</span>(max_f_sp, max_f, <span class="at">.after =</span> target_f_masterplate) <span class="sc">%&gt;%</span> </span>
<span id="cb32-541"><a href="#cb32-541"></a>  <span class="fu">write_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(data, <span class="st">"octets_counts.tsv"</span>))</span>
<span id="cb32-542"><a href="#cb32-542"></a><span class="in">```</span></span>
</code><button title="Copy to Clipboard" class="code-copy-button" data-in-quarto-modal=""><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->




</body></html>