<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.553">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Shane Hogle">
<meta name="dcterms.date" content="2024-07-25">

<title>hambiBottomUp data analysis - Formatting Rbec output</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<link href="../../site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet">
<script src="../../site_libs/pagedtable-1.1/js/pagedtable.js"></script>


</head>

<body class="nav-sidebar docked">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../R/amplicon/01_format_rbec_tab_20240711_BTK_illumina_v3.html">Community composition amplicon</a></li><li class="breadcrumb-item"><a href="../../R/amplicon/01_format_rbec_tab_20240711_BTK_illumina_v3.html">1. Data wrangling and QC - 20240711</a></li></ol></nav>
        <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation docked overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="../../">hambiBottomUp data analysis</a> 
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">
 <span class="menu-text">Community composition amplicon</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../R/amplicon/01_format_rbec_tab_20240711_BTK_illumina_v3.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">1. Data wrangling and QC - 20240711</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#introduction" id="toc-introduction" class="nav-link active" data-scroll-target="#introduction"><span class="header-section-number">1</span> Introduction</a></li>
  <li><a href="#setup" id="toc-setup" class="nav-link" data-scroll-target="#setup"><span class="header-section-number">2</span> Setup</a>
  <ul class="collapse">
  <li><a href="#libraries" id="toc-libraries" class="nav-link" data-scroll-target="#libraries"><span class="header-section-number">2.1</span> Libraries</a></li>
  <li><a href="#global-variables" id="toc-global-variables" class="nav-link" data-scroll-target="#global-variables"><span class="header-section-number">2.2</span> Global variables</a></li>
  <li><a href="#data" id="toc-data" class="nav-link" data-scroll-target="#data"><span class="header-section-number">2.3</span> Data</a></li>
  <li><a href="#functions" id="toc-functions" class="nav-link" data-scroll-target="#functions"><span class="header-section-number">2.4</span> Functions</a></li>
  </ul></li>
  <li><a href="#read-metadata" id="toc-read-metadata" class="nav-link" data-scroll-target="#read-metadata"><span class="header-section-number">3</span> Read metadata</a></li>
  <li><a href="#read-rbec-raw-counts-tables" id="toc-read-rbec-raw-counts-tables" class="nav-link" data-scroll-target="#read-rbec-raw-counts-tables"><span class="header-section-number">4</span> Read Rbec raw counts tables</a>
  <ul class="collapse">
  <li><a href="#untar-rbec-output-tarball" id="toc-untar-rbec-output-tarball" class="nav-link" data-scroll-target="#untar-rbec-output-tarball"><span class="header-section-number">4.1</span> Untar Rbec output tarball</a></li>
  <li><a href="#setup-directory-structure" id="toc-setup-directory-structure" class="nav-link" data-scroll-target="#setup-directory-structure"><span class="header-section-number">4.2</span> Setup directory structure</a></li>
  <li><a href="#read" id="toc-read" class="nav-link" data-scroll-target="#read"><span class="header-section-number">4.3</span> Read</a></li>
  </ul></li>
  <li><a href="#format" id="toc-format" class="nav-link" data-scroll-target="#format"><span class="header-section-number">5</span> Format</a></li>
  <li><a href="#analysis" id="toc-analysis" class="nav-link" data-scroll-target="#analysis"><span class="header-section-number">6</span> Analysis</a>
  <ul class="collapse">
  <li><a href="#negative-controls" id="toc-negative-controls" class="nav-link" data-scroll-target="#negative-controls"><span class="header-section-number">6.1</span> Negative controls</a></li>
  <li><a href="#positive-controls" id="toc-positive-controls" class="nav-link" data-scroll-target="#positive-controls"><span class="header-section-number">6.2</span> Positive controls</a></li>
  <li><a href="#misassigned-reads" id="toc-misassigned-reads" class="nav-link" data-scroll-target="#misassigned-reads"><span class="header-section-number">6.3</span> Misassigned reads</a></li>
  <li><a href="#samples-with-too-few-reads" id="toc-samples-with-too-few-reads" class="nav-link" data-scroll-target="#samples-with-too-few-reads"><span class="header-section-number">6.4</span> Samples with too few reads</a></li>
  <li><a href="#masterplate-samples" id="toc-masterplate-samples" class="nav-link" data-scroll-target="#masterplate-samples"><span class="header-section-number">6.5</span> Masterplate samples</a></li>
  <li><a href="#experimental-samples" id="toc-experimental-samples" class="nav-link" data-scroll-target="#experimental-samples"><span class="header-section-number">6.6</span> Experimental samples</a>
  <ul class="collapse">
  <li><a href="#contaminated-pairs" id="toc-contaminated-pairs" class="nav-link" data-scroll-target="#contaminated-pairs"><span class="header-section-number">6.6.1</span> Contaminated pairs</a></li>
  <li><a href="#contaminated-trios" id="toc-contaminated-trios" class="nav-link" data-scroll-target="#contaminated-trios"><span class="header-section-number">6.6.2</span> Contaminated trios</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#export" id="toc-export" class="nav-link" data-scroll-target="#export"><span class="header-section-number">7</span> Export</a>
  <ul class="collapse">
  <li><a href="#posneg-control-samples" id="toc-posneg-control-samples" class="nav-link" data-scroll-target="#posneg-control-samples"><span class="header-section-number">7.1</span> pos/neg control samples</a></li>
  <li><a href="#pairs" id="toc-pairs" class="nav-link" data-scroll-target="#pairs"><span class="header-section-number">7.2</span> Pairs</a></li>
  <li><a href="#trios" id="toc-trios" class="nav-link" data-scroll-target="#trios"><span class="header-section-number">7.3</span> Trios</a></li>
  </ul></li>
  <li><a href="#clean-up" id="toc-clean-up" class="nav-link" data-scroll-target="#clean-up"><span class="header-section-number">8</span> Clean up</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../R/amplicon/01_format_rbec_tab_20240711_BTK_illumina_v3.html">Community composition amplicon</a></li><li class="breadcrumb-item"><a href="../../R/amplicon/01_format_rbec_tab_20240711_BTK_illumina_v3.html">1. Data wrangling and QC - 20240711</a></li></ol></nav>
<div class="quarto-title">
<h1 class="title">Formatting Rbec output</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Shane Hogle </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">July 25, 2024</p>
    </div>
  </div>
  
    
  </div>
  


</header>


<section id="introduction" class="level1" data-number="1">
<h1 data-number="1"><span class="header-section-number">1</span> Introduction</h1>
<p>Contains results from pairs of all streptomycin concentrations and trios for 0 streptomycin from Milla’s bottom up community assembly experiment</p>
</section>
<section id="setup" class="level1" data-number="2">
<h1 data-number="2"><span class="header-section-number">2</span> Setup</h1>
<section id="libraries" class="level2" data-number="2.1">
<h2 data-number="2.1" class="anchored" data-anchor-id="libraries"><span class="header-section-number">2.1</span> Libraries</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-2"><a href="#cb1-2"></a><span class="fu">library</span>(here)</span>
<span id="cb1-3"><a href="#cb1-3"></a><span class="fu">library</span>(fs)</span>
<span id="cb1-4"><a href="#cb1-4"></a><span class="fu">library</span>(archive)</span>
<span id="cb1-5"><a href="#cb1-5"></a><span class="fu">library</span>(scales)</span>
<span id="cb1-6"><a href="#cb1-6"></a><span class="fu">source</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"R"</span>, <span class="st">"utils_generic.R"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="global-variables" class="level2" data-number="2.2">
<h2 data-number="2.2" class="anchored" data-anchor-id="global-variables"><span class="header-section-number">2.2</span> Global variables</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a>data_raw <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"_data_raw"</span>, <span class="st">"20240711_BTK_illumina_v3"</span>)</span>
<span id="cb2-2"><a href="#cb2-2"></a>data <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"20240711_BTK_illumina_v3"</span>)</span>
<span id="cb2-3"><a href="#cb2-3"></a>amplicontar <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(data_raw, <span class="st">"rbec_output.tar.gz"</span>)</span>
<span id="cb2-4"><a href="#cb2-4"></a></span>
<span id="cb2-5"><a href="#cb2-5"></a><span class="co"># make processed data directory if it doesn't exist</span></span>
<span id="cb2-6"><a href="#cb2-6"></a>fs<span class="sc">::</span><span class="fu">dir_create</span>(data)</span>
<span id="cb2-7"><a href="#cb2-7"></a></span>
<span id="cb2-8"><a href="#cb2-8"></a><span class="co"># create temporary location to decompress</span></span>
<span id="cb2-9"><a href="#cb2-9"></a>tmpdir <span class="ot">&lt;-</span> fs<span class="sc">::</span><span class="fu">file_temp</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="data" class="level2" data-number="2.3">
<h2 data-number="2.3" class="anchored" data-anchor-id="data"><span class="header-section-number">2.3</span> Data</h2>
<p>NOTE! We have commented out the species not included in this experiment</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1"></a>tax_locus_copynum <span class="ot">&lt;-</span> tibble<span class="sc">::</span><span class="fu">tribble</span>(</span>
<span id="cb3-2"><a href="#cb3-2"></a>     <span class="sc">~</span>strainID, <span class="sc">~</span>rRNA16S_cn, <span class="sc">~</span>rRNA16S_locus,             <span class="sc">~</span>genus,        <span class="sc">~</span>species,</span>
<span id="cb3-3"><a href="#cb3-3"></a>  <span class="st">"HAMBI_0006"</span>,          <span class="dv">7</span>L,  <span class="st">"H0006_04757"</span>,      <span class="st">"Pseudomonas"</span>,        <span class="st">"putida"</span>,</span>
<span id="cb3-4"><a href="#cb3-4"></a>  <span class="st">"HAMBI_0097"</span>,          <span class="dv">7</span>L,  <span class="st">"H0097_00044"</span>,    <span class="st">"Acinetobacter"</span>,     <span class="st">"johnsonii"</span>,</span>
<span id="cb3-5"><a href="#cb3-5"></a>  <span class="st">"HAMBI_0097"</span>,          <span class="dv">7</span>L,  <span class="st">"H0097_02759"</span>,    <span class="st">"Acinetobacter"</span>,     <span class="st">"johnsonii"</span>,</span>
<span id="cb3-6"><a href="#cb3-6"></a>  <span class="st">"HAMBI_0097"</span>,          <span class="dv">7</span>L,  <span class="st">"H0097_01762"</span>,    <span class="st">"Acinetobacter"</span>,     <span class="st">"johnsonii"</span>,</span>
<span id="cb3-7"><a href="#cb3-7"></a>  <span class="st">"HAMBI_0105"</span>,          <span class="dv">4</span>L,  <span class="st">"H0105_02306"</span>,    <span class="st">"Agrobacterium"</span>,   <span class="st">"tumefaciens"</span>,</span>
<span id="cb3-8"><a href="#cb3-8"></a>  <span class="st">"HAMBI_0262"</span>,          <span class="dv">3</span>L,  <span class="st">"H0262_00030"</span>,    <span class="st">"Brevundimonas"</span>,       <span class="st">"bullata"</span>,</span>
<span id="cb3-9"><a href="#cb3-9"></a>  <span class="st">"HAMBI_0403"</span>,          <span class="dv">9</span>L,  <span class="st">"H0403_00517"</span>,        <span class="st">"Comamonas"</span>,  <span class="st">"testosteroni"</span>,</span>
<span id="cb3-10"><a href="#cb3-10"></a>  <span class="st">"HAMBI_0403"</span>,          <span class="dv">9</span>L,  <span class="st">"H0403_00522"</span>,        <span class="st">"Comamonas"</span>,  <span class="st">"testosteroni"</span>,</span>
<span id="cb3-11"><a href="#cb3-11"></a>  <span class="st">"HAMBI_1279"</span>,          <span class="dv">7</span>L,  <span class="st">"H1279_03627"</span>,           <span class="st">"Hafnia"</span>,         <span class="st">"alvei"</span>,</span>
<span id="cb3-12"><a href="#cb3-12"></a>  <span class="st">"HAMBI_1279"</span>,          <span class="dv">7</span>L,  <span class="st">"H1279_00125"</span>,           <span class="st">"Hafnia"</span>,         <span class="st">"alvei"</span>,</span>
<span id="cb3-13"><a href="#cb3-13"></a>  <span class="st">"HAMBI_1279"</span>,          <span class="dv">7</span>L,  <span class="st">"H1279_03957"</span>,           <span class="st">"Hafnia"</span>,         <span class="st">"alvei"</span>,</span>
<span id="cb3-14"><a href="#cb3-14"></a>  <span class="st">"HAMBI_1287"</span>,          <span class="dv">7</span>L,  <span class="st">"H1287_03997"</span>,      <span class="st">"Citrobacter"</span>,        <span class="st">"koseri"</span>,</span>
<span id="cb3-15"><a href="#cb3-15"></a>  <span class="st">"HAMBI_1287"</span>,          <span class="dv">7</span>L,  <span class="st">"H1287_03402"</span>,      <span class="st">"Citrobacter"</span>,        <span class="st">"koseri"</span>,</span>
<span id="cb3-16"><a href="#cb3-16"></a>  <span class="st">"HAMBI_1292"</span>,          <span class="dv">7</span>L,  <span class="st">"H1292_03239"</span>,       <span class="st">"Morganella"</span>,      <span class="st">"morganii"</span>,</span>
<span id="cb3-17"><a href="#cb3-17"></a>  <span class="st">"HAMBI_1299"</span>,          <span class="dv">8</span>L,  <span class="st">"H1299_04293"</span>,         <span class="st">"Kluyvera"</span>,    <span class="st">"intermedia"</span>,</span>
<span id="cb3-18"><a href="#cb3-18"></a>  <span class="st">"HAMBI_1299"</span>,          <span class="dv">8</span>L,  <span class="st">"H1299_01283"</span>,         <span class="st">"Kluyvera"</span>,    <span class="st">"intermedia"</span>,</span>
<span id="cb3-19"><a href="#cb3-19"></a>  <span class="st">"HAMBI_1299"</span>,          <span class="dv">8</span>L,  <span class="st">"H1279_03957"</span>,         <span class="st">"Kluyvera"</span>,    <span class="st">"intermedia"</span>,</span>
<span id="cb3-20"><a href="#cb3-20"></a>  <span class="st">"HAMBI_1842"</span>,          <span class="dv">4</span>L,  <span class="st">"H1842_01650"</span>,      <span class="st">"Sphingobium"</span>,    <span class="st">"yanoikuyae"</span>,</span>
<span id="cb3-21"><a href="#cb3-21"></a>  <span class="st">"HAMBI_1896"</span>,          <span class="dv">4</span>L,  <span class="st">"H1896_00963"</span>, <span class="st">"Sphingobacterium"</span>,  <span class="st">"spiritivorum"</span>,</span>
<span id="cb3-22"><a href="#cb3-22"></a>  <span class="st">"HAMBI_1972"</span>,         <span class="dv">10</span>L,  <span class="st">"H1972_00343"</span>,        <span class="st">"Aeromonas"</span>,        <span class="st">"caviae"</span>,</span>
<span id="cb3-23"><a href="#cb3-23"></a>  <span class="st">"HAMBI_1972"</span>,         <span class="dv">10</span>L,  <span class="st">"H1972_03531"</span>,        <span class="st">"Aeromonas"</span>,        <span class="st">"caviae"</span>,</span>
<span id="cb3-24"><a href="#cb3-24"></a>  <span class="st">"HAMBI_1977"</span>,          <span class="dv">5</span>L,  <span class="st">"H1977_00118"</span>,      <span class="st">"Pseudomonas"</span>,  <span class="st">"chlororaphis"</span>,</span>
<span id="cb3-25"><a href="#cb3-25"></a>  <span class="st">"HAMBI_1988"</span>,          <span class="dv">5</span>L,  <span class="st">"H1988_05160"</span>,     <span class="st">"Chitinophaga"</span>,        <span class="st">"sancti"</span>,</span>
<span id="cb3-26"><a href="#cb3-26"></a>  <span class="st">"HAMBI_1988"</span>,          <span class="dv">5</span>L,  <span class="st">"H1988_05152"</span>,     <span class="st">"Chitinophaga"</span>,        <span class="st">"sancti"</span>,</span>
<span id="cb3-27"><a href="#cb3-27"></a>  <span class="st">"HAMBI_1988"</span>,          <span class="dv">5</span>L,  <span class="st">"H1988_05165"</span>,     <span class="st">"Chitinophaga"</span>,        <span class="st">"sancti"</span>,</span>
<span id="cb3-28"><a href="#cb3-28"></a>  <span class="st">"HAMBI_2159"</span>,          <span class="dv">4</span>L,  <span class="st">"H2159_01406"</span>,        <span class="st">"Trinickia"</span>,   <span class="st">"caryophylli"</span>,</span>
<span id="cb3-29"><a href="#cb3-29"></a>  <span class="st">"HAMBI_2159"</span>,          <span class="dv">4</span>L,  <span class="st">"H2159_05851"</span>,        <span class="st">"Trinickia"</span>,   <span class="st">"caryophylli"</span>,</span>
<span id="cb3-30"><a href="#cb3-30"></a>  <span class="st">"HAMBI_2160"</span>,          <span class="dv">3</span>L,  <span class="st">"H2160_00530"</span>,       <span class="st">"Bordetella"</span>,         <span class="st">"avium"</span>,</span>
<span id="cb3-31"><a href="#cb3-31"></a>  <span class="st">"HAMBI_2164"</span>,          <span class="dv">5</span>L,  <span class="st">"H2164_03337"</span>,      <span class="st">"Cupriavidus"</span>,    <span class="st">"oxalaticus"</span>,</span>
<span id="cb3-32"><a href="#cb3-32"></a>  <span class="st">"HAMBI_2443"</span>,          <span class="dv">3</span>L,  <span class="st">"H2443_00128"</span>,       <span class="st">"Paracoccus"</span>, <span class="st">"denitrificans"</span>,</span>
<span id="cb3-33"><a href="#cb3-33"></a>  <span class="st">"HAMBI_2494"</span>,          <span class="dv">4</span>L,  <span class="st">"H2494_03389"</span>, <span class="st">"Paraburkholderia"</span>,   <span class="st">"kururiensis"</span>,</span>
<span id="cb3-34"><a href="#cb3-34"></a>  <span class="st">"HAMBI_2659"</span>,          <span class="dv">4</span>L,  <span class="st">"H2659_00367"</span>, <span class="st">"Stenotrophomonas"</span>,   <span class="st">"maltophilia"</span>,</span>
<span id="cb3-35"><a href="#cb3-35"></a>  <span class="st">"HAMBI_2792"</span>,          <span class="dv">4</span>L,  <span class="st">"H2792_00549"</span>,        <span class="st">"Moraxella"</span>,         <span class="st">"canis"</span>,</span>
<span id="cb3-36"><a href="#cb3-36"></a>  <span class="st">"HAMBI_3031"</span>,          <span class="dv">2</span>L,  <span class="st">"H3031_00830"</span>,         <span class="st">"Niabella"</span>,  <span class="st">"yanshanensis"</span>,</span>
<span id="cb3-37"><a href="#cb3-37"></a>  <span class="st">"HAMBI_3237"</span>,          <span class="dv">6</span>L,  <span class="st">"H3237_00875"</span>,       <span class="st">"Microvirga"</span>,   <span class="st">"lotononidis"</span>,</span>
<span id="cb3-38"><a href="#cb3-38"></a>  <span class="st">"HAMBI_1923"</span>,          <span class="dv">6</span>L,  <span class="st">"H1923_00876"</span>,   <span class="st">"Flavobacterium"</span>,      <span class="st">"odoratum"</span></span>
<span id="cb3-39"><a href="#cb3-39"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="functions" class="level2" data-number="2.4">
<h2 data-number="2.4" class="anchored" data-anchor-id="functions"><span class="header-section-number">2.4</span> Functions</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a><span class="co"># this function </span></span>
<span id="cb4-2"><a href="#cb4-2"></a>normalize_by_copy <span class="ot">&lt;-</span> <span class="cf">function</span>(.data, <span class="at">tlc =</span> tax_locus_copynum){</span>
<span id="cb4-3"><a href="#cb4-3"></a>  .data <span class="sc">%&gt;%</span> </span>
<span id="cb4-4"><a href="#cb4-4"></a>    <span class="co"># join with the copy number data frame. We join by the locus tag so this will add H1279_03957 to HAMBI_1299</span></span>
<span id="cb4-5"><a href="#cb4-5"></a>    dplyr<span class="sc">::</span><span class="fu">left_join</span>(tlc, <span class="at">by =</span> <span class="fu">join_by</span>(rRNA16S_locus)) <span class="sc">%&gt;%</span></span>
<span id="cb4-6"><a href="#cb4-6"></a>    <span class="co"># get total number of mapping reads per species. This aggregates all the difference ASVs per species</span></span>
<span id="cb4-7"><a href="#cb4-7"></a>    dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">count =</span> <span class="fu">sum</span>(count), <span class="at">.by =</span> <span class="fu">c</span>(sample, strainID, rRNA16S_cn)) <span class="sc">%&gt;%</span> </span>
<span id="cb4-8"><a href="#cb4-8"></a>    <span class="co"># group by sample</span></span>
<span id="cb4-9"><a href="#cb4-9"></a>    dplyr<span class="sc">::</span><span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb4-10"><a href="#cb4-10"></a>    <span class="co"># calculate a corrected count which is simply the count divided by copy num for each species</span></span>
<span id="cb4-11"><a href="#cb4-11"></a>    <span class="co"># dividide by the sum of count divided by copy num for whole sample multiplied by the total</span></span>
<span id="cb4-12"><a href="#cb4-12"></a>    <span class="co"># number of mapped reads per sample</span></span>
<span id="cb4-13"><a href="#cb4-13"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">count_correct =</span> <span class="fu">round</span>(<span class="fu">sum</span>(count)<span class="sc">*</span>(count<span class="sc">/</span>rRNA16S_cn)<span class="sc">/</span><span class="fu">sum</span>(count<span class="sc">/</span>rRNA16S_cn))) <span class="sc">%&gt;%</span>  </span>
<span id="cb4-14"><a href="#cb4-14"></a>    dplyr<span class="sc">::</span><span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb4-15"><a href="#cb4-15"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(sample, strainID, count, count_correct)</span>
<span id="cb4-16"><a href="#cb4-16"></a>  }</span>
<span id="cb4-17"><a href="#cb4-17"></a></span>
<span id="cb4-18"><a href="#cb4-18"></a><span class="co"># this function replaces missing species counts with zero</span></span>
<span id="cb4-19"><a href="#cb4-19"></a>completecombos <span class="ot">&lt;-</span> <span class="cf">function</span>(.data, <span class="at">tlc =</span> tax_locus_copynum, <span class="at">countname =</span> count, <span class="at">remove1923 =</span> <span class="cn">TRUE</span>){</span>
<span id="cb4-20"><a href="#cb4-20"></a> </span>
<span id="cb4-21"><a href="#cb4-21"></a>  <span class="co"># get unique strainIDs</span></span>
<span id="cb4-22"><a href="#cb4-22"></a>  strainID <span class="ot">&lt;-</span> <span class="fu">unique</span>(tlc<span class="sc">$</span>strainID)</span>
<span id="cb4-23"><a href="#cb4-23"></a>  <span class="co"># table for assigning genus and species names. Doesn't matter if 1923 is there or not</span></span>
<span id="cb4-24"><a href="#cb4-24"></a>  <span class="co"># because it is filter joined later</span></span>
<span id="cb4-25"><a href="#cb4-25"></a>  tax <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">distinct</span>(dplyr<span class="sc">::</span><span class="fu">select</span>(tlc, strainID, genus, species))</span>
<span id="cb4-26"><a href="#cb4-26"></a>  <span class="cf">if</span> (remove1923) {</span>
<span id="cb4-27"><a href="#cb4-27"></a>    <span class="co"># get unique strainIDs but exclude 1923 if remove1923 is true</span></span>
<span id="cb4-28"><a href="#cb4-28"></a>    strainID <span class="ot">&lt;-</span> strainID[strainID <span class="sc">!=</span> <span class="st">"HAMBI_1923"</span>]</span>
<span id="cb4-29"><a href="#cb4-29"></a>  }</span>
<span id="cb4-30"><a href="#cb4-30"></a>  </span>
<span id="cb4-31"><a href="#cb4-31"></a>  dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(tibble<span class="sc">::</span><span class="fu">tibble</span>(<span class="at">strainID =</span> strainID, <span class="at">sample =</span> <span class="st">"dummy"</span>), .data) <span class="sc">%&gt;%</span> </span>
<span id="cb4-32"><a href="#cb4-32"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>( <span class="st">"{{ countname }}"</span> <span class="sc">:</span><span class="er">=</span> dplyr<span class="sc">::</span><span class="fu">if_else</span>(sample <span class="sc">==</span> <span class="st">"dummy"</span>, <span class="dv">1</span>, {{ countname }})) <span class="sc">%&gt;%</span> </span>
<span id="cb4-33"><a href="#cb4-33"></a>    tidyr<span class="sc">::</span><span class="fu">complete</span>(sample, strainID) <span class="sc">%&gt;%</span> </span>
<span id="cb4-34"><a href="#cb4-34"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(sample <span class="sc">!=</span> <span class="st">"dummy"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb4-35"><a href="#cb4-35"></a>    dplyr<span class="sc">::</span><span class="fu">mutate</span>( <span class="st">"{{ countname }}"</span> <span class="sc">:</span><span class="er">=</span> dplyr<span class="sc">::</span><span class="fu">if_else</span>(<span class="fu">is.na</span>({{ countname }}), <span class="dv">0</span>, {{ countname }})) <span class="sc">%&gt;%</span> </span>
<span id="cb4-36"><a href="#cb4-36"></a>    tidyr<span class="sc">::</span><span class="fu">replace_na</span>(<span class="fu">list</span>(<span class="at">count_correct =</span> <span class="dv">0</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb4-37"><a href="#cb4-37"></a>    dplyr<span class="sc">::</span><span class="fu">left_join</span>(dplyr<span class="sc">::</span><span class="fu">distinct</span>(dplyr<span class="sc">::</span><span class="fu">select</span>(tlc, strainID, genus, species))) <span class="sc">%&gt;%</span> </span>
<span id="cb4-38"><a href="#cb4-38"></a>    dplyr<span class="sc">::</span><span class="fu">relocate</span>(genus, species, <span class="at">.after =</span> strainID)</span>
<span id="cb4-39"><a href="#cb4-39"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="read-metadata" class="level1" data-number="3">
<h1 data-number="3"><span class="header-section-number">3</span> Read metadata</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1"></a>mddf <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(data_raw, <span class="st">"20240711_BTK_illumina_v3_metadata.tsv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 736 Columns: 8
── Column specification ────────────────────────────────────────────────────────
Delimiter: "\t"
chr (4): sample, community_id, community_type, plate_well
dbl (4): n_species, transfers, strep_conc, replicate

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1"></a>spdf <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(data_raw, <span class="st">"sample_compositions.tsv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 384 Columns: 4
── Column specification ────────────────────────────────────────────────────────
Delimiter: "\t"
chr (3): community_id, evo_hist, strainID
dbl (1): target_f

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
</div>
</section>
<section id="read-rbec-raw-counts-tables" class="level1" data-number="4">
<h1 data-number="4"><span class="header-section-number">4</span> Read Rbec raw counts tables</h1>
<section id="untar-rbec-output-tarball" class="level2" data-number="4.1">
<h2 data-number="4.1" class="anchored" data-anchor-id="untar-rbec-output-tarball"><span class="header-section-number">4.1</span> Untar Rbec output tarball</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1"></a>archive<span class="sc">::</span><span class="fu">archive_extract</span>(</span>
<span id="cb9-2"><a href="#cb9-2"></a>  amplicontar,</span>
<span id="cb9-3"><a href="#cb9-3"></a>  <span class="at">dir =</span> tmpdir,</span>
<span id="cb9-4"><a href="#cb9-4"></a>  <span class="at">files =</span> <span class="cn">NULL</span>,</span>
<span id="cb9-5"><a href="#cb9-5"></a>  <span class="at">options =</span> <span class="fu">character</span>(),</span>
<span id="cb9-6"><a href="#cb9-6"></a>  <span class="at">strip_components =</span> <span class="dv">0</span>L</span>
<span id="cb9-7"><a href="#cb9-7"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="setup-directory-structure" class="level2" data-number="4.2">
<h2 data-number="4.2" class="anchored" data-anchor-id="setup-directory-structure"><span class="header-section-number">4.2</span> Setup directory structure</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1"></a>tabdir <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(tmpdir, <span class="st">"rbec_output"</span>)</span>
<span id="cb10-2"><a href="#cb10-2"></a>samppaths <span class="ot">&lt;-</span> fs<span class="sc">::</span><span class="fu">dir_ls</span>(tabdir)</span>
<span id="cb10-3"><a href="#cb10-3"></a>sampnames <span class="ot">&lt;-</span> <span class="fu">path_split</span>(samppaths) <span class="sc">%&gt;%</span> </span>
<span id="cb10-4"><a href="#cb10-4"></a>  <span class="fu">map_chr</span>(dplyr<span class="sc">::</span>last)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="read" class="level2" data-number="4.3">
<h2 data-number="4.3" class="anchored" data-anchor-id="read"><span class="header-section-number">4.3</span> Read</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1"></a>straintabs <span class="ot">&lt;-</span> <span class="fu">paste0</span>(samppaths, <span class="st">"/strain_table.txt"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb11-2"><a href="#cb11-2"></a>  <span class="fu">set_names</span>(sampnames) <span class="sc">%&gt;%</span> </span>
<span id="cb11-3"><a href="#cb11-3"></a>  <span class="fu">map_df</span>(</span>
<span id="cb11-4"><a href="#cb11-4"></a>  read_tsv,</span>
<span id="cb11-5"><a href="#cb11-5"></a>  <span class="at">skip =</span> <span class="dv">1</span>,</span>
<span id="cb11-6"><a href="#cb11-6"></a>  <span class="at">col_names =</span> <span class="fu">c</span>(<span class="st">"rRNA16S_locus"</span>,<span class="st">"count"</span>),</span>
<span id="cb11-7"><a href="#cb11-7"></a>  <span class="at">show_col_types =</span> <span class="cn">FALSE</span>, </span>
<span id="cb11-8"><a href="#cb11-8"></a>  <span class="at">.id =</span> <span class="st">"sample"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb11-9"><a href="#cb11-9"></a>  <span class="co"># naming scheme inconsistent for one sample</span></span>
<span id="cb11-10"><a href="#cb11-10"></a>  <span class="fu">mutate</span>(<span class="at">sample =</span> <span class="fu">if_else</span>(sample <span class="sc">==</span> <span class="st">"P2_s_0"</span>, <span class="st">"P02_s_0"</span>, sample))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="format" class="level1" data-number="5">
<h1 data-number="5"><span class="header-section-number">5</span> Format</h1>
<p>Normalize counts by 16S copy number</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1"></a>straintabs_norm <span class="ot">&lt;-</span> <span class="fu">normalize_by_copy</span>(straintabs)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in dplyr::left_join(., tlc, by = join_by(rRNA16S_locus)): Detected an unexpected many-to-many relationship between `x` and `y`.
ℹ Row 1826 of `x` matches multiple rows in `y`.
ℹ Row 19 of `y` matches multiple rows in `x`.
ℹ If a many-to-many relationship is expected, set `relationship =
  "many-to-many"` to silence this warning.</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1"></a>finaltable <span class="ot">&lt;-</span> <span class="fu">left_join</span>(straintabs_norm, mddf) <span class="sc">%&gt;%</span> </span>
<span id="cb14-2"><a href="#cb14-2"></a>  <span class="fu">left_join</span>(spdf)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Joining with `by = join_by(sample)`
Joining with `by = join_by(strainID, community_id)`</code></pre>
</div>
</div>
</section>
<section id="analysis" class="level1" data-number="6">
<h1 data-number="6"><span class="header-section-number">6</span> Analysis</h1>
<section id="negative-controls" class="level2" data-number="6.1">
<h2 data-number="6.1" class="anchored" data-anchor-id="negative-controls"><span class="header-section-number">6.1</span> Negative controls</h2>
<p>First, we’ll check whether the experimental and plate negative controls look good</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1"></a>finaltable <span class="sc">%&gt;%</span> </span>
<span id="cb16-2"><a href="#cb16-2"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(community_type, <span class="st">"^neg|pos"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb16-3"><a href="#cb16-3"></a>  <span class="fu">summarize</span>(<span class="at">tot =</span> <span class="fu">sum</span>(count_correct), <span class="at">.by =</span> <span class="fu">c</span>(<span class="st">"sample"</span>, <span class="st">"community_id"</span>, <span class="st">"n_species"</span>, <span class="st">"community_type"</span>, <span class="st">"replicate"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["sample"],"name":[1],"type":["chr"],"align":["left"]},{"label":["community_id"],"name":[2],"type":["chr"],"align":["left"]},{"label":["n_species"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["community_type"],"name":[4],"type":["chr"],"align":["left"]},{"label":["replicate"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["tot"],"name":[6],"type":["dbl"],"align":["right"]}],"data":[{"1":"3sp_1_0","2":"3sp","3":"3","4":"pos_ctrl","5":"1","6":"12125"},{"1":"3sp_2_0","2":"3sp","3":"3","4":"pos_ctrl","5":"2","6":"12316"},{"1":"3sp_3_0","2":"3sp","3":"3","4":"pos_ctrl","5":"3","6":"12949"},{"1":"3sp_4_0","2":"3sp","3":"3","4":"pos_ctrl","5":"4","6":"8938"},{"1":"3sp_5_0","2":"3sp","3":"3","4":"pos_ctrl","5":"5","6":"11061"},{"1":"3sp_6_0","2":"3sp","3":"3","4":"pos_ctrl","5":"6","6":"8641"},{"1":"3sp_7_0","2":"3sp","3":"3","4":"pos_ctrl","5":"7","6":"13812"},{"1":"3sp_8_0","2":"3sp","3":"3","4":"pos_ctrl","5":"8","6":"13828"},{"1":"neg_1_0","2":"neg","3":"0","4":"neg_ctrl","5":"1","6":"116"},{"1":"neg_2_0","2":"neg","3":"0","4":"neg_ctrl","5":"2","6":"9018"},{"1":"neg_3_0","2":"neg","3":"0","4":"neg_ctrl","5":"3","6":"9213"},{"1":"neg_4_0","2":"neg","3":"0","4":"neg_ctrl","5":"4","6":"8767"},{"1":"neg_5_0","2":"neg","3":"0","4":"neg_ctrl","5":"5","6":"98"},{"1":"neg_6_0","2":"neg","3":"0","4":"neg_ctrl","5":"6","6":"73"},{"1":"neg_7_0","2":"neg","3":"0","4":"neg_ctrl","5":"7","6":"106"},{"1":"neg_8_0","2":"neg","3":"0","4":"neg_ctrl","5":"8","6":"189"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<p>This looks ok, but there are potentially some problems. Specifically, negative control replicates 2, 3, and 4 all have some contamination. <code>neg_2_0</code> seems to be contaminated with HAMBI_1977, <code>neg_3_0</code> with HAMBI_1287, and <code>neg_4_0</code> with HAMBI_1977. However in the 5 remaining negative controls there is no contamination.</p>
</section>
<section id="positive-controls" class="level2" data-number="6.2">
<h2 data-number="6.2" class="anchored" data-anchor-id="positive-controls"><span class="header-section-number">6.2</span> Positive controls</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1"></a>finaltable <span class="sc">%&gt;%</span> </span>
<span id="cb17-2"><a href="#cb17-2"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(community_type, <span class="st">"^pos"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb17-3"><a href="#cb17-3"></a>  <span class="fu">mutate</span>(<span class="at">total_pos_controls =</span> <span class="fu">n_distinct</span>(sample)) <span class="sc">%&gt;%</span> </span>
<span id="cb17-4"><a href="#cb17-4"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb17-5"><a href="#cb17-5"></a>  <span class="fu">mutate</span>(<span class="at">n_sp_detected =</span> <span class="fu">sum</span>(count <span class="sc">&gt;</span> <span class="dv">0</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb17-6"><a href="#cb17-6"></a>  <span class="fu">distinct</span>(sample, n_sp_detected, total_pos_controls)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["sample"],"name":[1],"type":["chr"],"align":["left"]},{"label":["n_sp_detected"],"name":[2],"type":["int"],"align":["right"]},{"label":["total_pos_controls"],"name":[3],"type":["int"],"align":["right"]}],"data":[{"1":"3sp_1_0","2":"4","3":"8"},{"1":"3sp_2_0","2":"4","3":"8"},{"1":"3sp_3_0","2":"4","3":"8"},{"1":"3sp_4_0","2":"4","3":"8"},{"1":"3sp_5_0","2":"4","3":"8"},{"1":"3sp_6_0","2":"4","3":"8"},{"1":"3sp_7_0","2":"4","3":"8"},{"1":"3sp_8_0","2":"4","3":"8"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<p>This is also good - we detect all 4 species in the positive controls on each plate.</p>
</section>
<section id="misassigned-reads" class="level2" data-number="6.3">
<h2 data-number="6.3" class="anchored" data-anchor-id="misassigned-reads"><span class="header-section-number">6.3</span> Misassigned reads</h2>
<p>These libraries were only prepared with samples from Milla’s 4-species experiment with 403, 1287, 1896, and 1977 so any time species other than these show up is just an incorrect assignment by Rbec. Let’s check quickly how many of these there are…</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1"></a>finaltable <span class="sc">%&gt;%</span> </span>
<span id="cb18-2"><a href="#cb18-2"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">str_detect</span>(strainID, <span class="st">"0403|1287|1896|1977"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["sample"],"name":[1],"type":["chr"],"align":["left"]},{"label":["strainID"],"name":[2],"type":["chr"],"align":["left"]},{"label":["count"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["count_correct"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["community_id"],"name":[5],"type":["chr"],"align":["left"]},{"label":["n_species"],"name":[6],"type":["dbl"],"align":["right"]},{"label":["community_type"],"name":[7],"type":["chr"],"align":["left"]},{"label":["transfers"],"name":[8],"type":["dbl"],"align":["right"]},{"label":["strep_conc"],"name":[9],"type":["dbl"],"align":["right"]},{"label":["replicate"],"name":[10],"type":["dbl"],"align":["right"]},{"label":["plate_well"],"name":[11],"type":["chr"],"align":["left"]},{"label":["evo_hist"],"name":[12],"type":["chr"],"align":["left"]},{"label":["target_f"],"name":[13],"type":["dbl"],"align":["right"]}],"data":[{"1":"P02_1_256","2":"HAMBI_1972","3":"1","4":"0","5":"P02","6":"2","7":"experiment","8":"8","9":"256","10":"1","11":"02B","12":"NA","13":"NA"},{"1":"P22_s_0","2":"HAMBI_0006","3":"1","4":"1","5":"P22","6":"2","7":"masterplate","8":"0","9":"NA","10":"1","11":"02E","12":"NA","13":"NA"},{"1":"P25_2_256","2":"HAMBI_2659","3":"1","4":"1","5":"P25","6":"2","7":"experiment","8":"8","9":"256","10":"2","11":"02C","12":"NA","13":"NA"},{"1":"P26_2_0","2":"HAMBI_1972","3":"1","4":"1","5":"P26","6":"2","7":"experiment","8":"8","9":"0","10":"2","11":"01D","12":"NA","13":"NA"},{"1":"P27_s_0","2":"HAMBI_2659","3":"2","4":"2","5":"P27","6":"2","7":"masterplate","8":"0","9":"NA","10":"1","11":"06A","12":"NA","13":"NA"},{"1":"P27_s_0","2":"HAMBI_0006","3":"1","4":"1","5":"P27","6":"2","7":"masterplate","8":"0","9":"NA","10":"1","11":"06A","12":"NA","13":"NA"},{"1":"P32_1_0","2":"HAMBI_2659","3":"1","4":"1","5":"P32","6":"2","7":"experiment","8":"8","9":"0","10":"1","11":"04D","12":"NA","13":"NA"},{"1":"P35_2_256","2":"HAMBI_1279","3":"1","4":"1","5":"P35","6":"2","7":"experiment","8":"8","9":"256","10":"2","11":"12G","12":"NA","13":"NA"},{"1":"P35_2_256","2":"HAMBI_1299","3":"1","4":"1","5":"P35","6":"2","7":"experiment","8":"8","9":"256","10":"2","11":"12G","12":"NA","13":"NA"},{"1":"P36_1_0","2":"HAMBI_1972","3":"2","4":"1","5":"P36","6":"2","7":"experiment","8":"8","9":"0","10":"1","11":"12H","12":"NA","13":"NA"},{"1":"P36_s_0","2":"HAMBI_2659","3":"1","4":"2","5":"P36","6":"2","7":"masterplate","8":"0","9":"NA","10":"1","11":"05H","12":"NA","13":"NA"},{"1":"P42_s_0","2":"HAMBI_2659","3":"1","4":"1","5":"P42","6":"2","7":"masterplate","8":"0","9":"NA","10":"1","11":"05D","12":"NA","13":"NA"},{"1":"P43_s_0","2":"HAMBI_0006","3":"1","4":"1","5":"P43","6":"2","7":"masterplate","8":"0","9":"NA","10":"1","11":"06D","12":"NA","13":"NA"},{"1":"P46_s_0","2":"HAMBI_0006","3":"1","4":"1","5":"P46","6":"2","7":"masterplate","8":"0","9":"NA","10":"1","11":"05E","12":"NA","13":"NA"},{"1":"P47_2_64","2":"HAMBI_0006","3":"1","4":"1","5":"P47","6":"2","7":"experiment","8":"8","9":"64","10":"2","11":"12C","12":"NA","13":"NA"},{"1":"T01_s_0","2":"HAMBI_1972","3":"1","4":"1","5":"T01","6":"3","7":"masterplate","8":"0","9":"NA","10":"1","11":"01E","12":"NA","13":"NA"},{"1":"T18_1_0","2":"HAMBI_1972","3":"1","4":"1","5":"T18","6":"3","7":"experiment","8":"8","9":"0","10":"1","11":"02B","12":"NA","13":"NA"},{"1":"T23_s_0","2":"HAMBI_1972","3":"5","4":"3","5":"T23","6":"3","7":"masterplate","8":"0","9":"NA","10":"1","11":"06D","12":"NA","13":"NA"},{"1":"T25_1_0","2":"HAMBI_1972","3":"1","4":"1","5":"T25","6":"3","7":"experiment","8":"8","9":"0","10":"1","11":"03E","12":"NA","13":"NA"},{"1":"T28_2_0","2":"HAMBI_3031","3":"1","4":"3","5":"T28","6":"3","7":"experiment","8":"8","9":"0","10":"2","11":"03F","12":"NA","13":"NA"},{"1":"T35_s_0","2":"HAMBI_1972","3":"1","4":"1","5":"T35","6":"3","7":"masterplate","8":"0","9":"NA","10":"1","11":"07D","12":"NA","13":"NA"},{"1":"T56_1_0","2":"HAMBI_0006","3":"1","4":"1","5":"T56","6":"3","7":"experiment","8":"8","9":"0","10":"1","11":"11C","12":"NA","13":"NA"},{"1":"neg_5_0","2":"HAMBI_1972","3":"1","4":"1","5":"neg","6":"0","7":"neg_ctrl","8":"NA","9":"NA","10":"5","11":"08E","12":"NA","13":"NA"},{"1":"neg_7_0","2":"HAMBI_1972","3":"3","4":"2","5":"neg","6":"0","7":"neg_ctrl","8":"NA","9":"NA","10":"7","11":"08G","12":"NA","13":"NA"},{"1":"neg_7_0","2":"HAMBI_1292","3":"1","4":"1","5":"neg","6":"0","7":"neg_ctrl","8":"NA","9":"NA","10":"7","11":"08G","12":"NA","13":"NA"},{"1":"neg_8_0","2":"HAMBI_1972","3":"29","4":"19","5":"neg","6":"0","7":"neg_ctrl","8":"NA","9":"NA","10":"8","11":"08H","12":"NA","13":"NA"},{"1":"neg_8_0","2":"HAMBI_0105","3":"1","4":"2","5":"neg","6":"0","7":"neg_ctrl","8":"NA","9":"NA","10":"8","11":"08H","12":"NA","13":"NA"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<p>There is one or two incorrectly assigned reads here and there but this is just noise. We can safely exclude all reads not mapping to one of the focal species.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1"></a>finaltable <span class="ot">&lt;-</span> finaltable <span class="sc">%&gt;%</span> </span>
<span id="cb19-2"><a href="#cb19-2"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(strainID, <span class="st">"0403|1287|1896|1977"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="samples-with-too-few-reads" class="level2" data-number="6.4">
<h2 data-number="6.4" class="anchored" data-anchor-id="samples-with-too-few-reads"><span class="header-section-number">6.4</span> Samples with too few reads</h2>
<p>Some of the experimental pairs had streptomycin concentrations higher than any of the species individually could tolerate. We would naively expect then that neither species would grow successfully in these samples and that the overall biomass would be very low, thus resulting in a low number of recovered reads from these samples.</p>
<p>To look into this. first let’s check which samples have very low OD600 in the endpoint samples.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1"></a>od <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"_data_raw"</span>, <span class="st">"2024060606_optical_density"</span>, <span class="st">"optical_density_formatted.tsv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 2880 Columns: 6
── Column specification ────────────────────────────────────────────────────────
Delimiter: "\t"
chr (1): community_id
dbl (5): transfers, n_species, strep_conc, replicate, OD

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1"></a>filtids <span class="ot">&lt;-</span> finaltable <span class="sc">%&gt;%</span> </span>
<span id="cb22-2"><a href="#cb22-2"></a>  <span class="fu">summarize</span>(<span class="at">tot =</span> <span class="fu">sum</span>(count_correct), <span class="at">.by =</span> <span class="fu">c</span>(sample, community_id, n_species, transfers, strep_conc, replicate)) <span class="sc">%&gt;%</span></span>
<span id="cb22-3"><a href="#cb22-3"></a>  <span class="fu">left_join</span>(od, <span class="at">by =</span> <span class="fu">join_by</span>(community_id, n_species, transfers, strep_conc, replicate)) <span class="sc">%&gt;%</span> </span>
<span id="cb22-4"><a href="#cb22-4"></a>  <span class="fu">filter</span>(transfers <span class="sc">==</span> <span class="dv">8</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb22-5"><a href="#cb22-5"></a>  <span class="fu">arrange</span>(tot) <span class="sc">%&gt;%</span> </span>
<span id="cb22-6"><a href="#cb22-6"></a>  <span class="fu">arrange</span>(OD) <span class="sc">%&gt;%</span> </span>
<span id="cb22-7"><a href="#cb22-7"></a>  <span class="fu">filter</span>(OD <span class="sc">&lt;</span> <span class="fl">0.1</span> <span class="sc">|</span> tot <span class="sc">&lt;</span> <span class="dv">1000</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb22-8"><a href="#cb22-8"></a>  <span class="fu">pull</span>(sample)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>We’ll filter out samples with an OD of less than 0.1 and also samples with fewer than 1000 reads. It is genearlly good practice to exclude samples with low number of reads.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1"></a>finaltable <span class="ot">&lt;-</span> finaltable <span class="sc">%&gt;%</span> </span>
<span id="cb23-2"><a href="#cb23-2"></a>  <span class="fu">filter</span>(sample <span class="sc">%nin%</span> filtids)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="masterplate-samples" class="level2" data-number="6.5">
<h2 data-number="6.5" class="anchored" data-anchor-id="masterplate-samples"><span class="header-section-number">6.5</span> Masterplate samples</h2>
<p>To set up this experiment, Milla combined the species in the planned proportions on a masterplate. Because this process was time consuming, the masterplate was stored at -80C after construction until the experiment start day when it was taken from the freezer and used to inoculate the experiment. Because Milla knows exactly which strains were added to the master plate and the plates were not allowed to grow, any strains in these samples that are not supposed to be there should be due to Illumina index cross talk and not true contamination. We can get a sense for the average index crosstalk rate from these samples and then draw a threshold of when to exclude likely false positives and when a positive is likely due to contamination.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1"></a>masterplate <span class="ot">&lt;-</span> finaltable <span class="sc">%&gt;%</span></span>
<span id="cb24-2"><a href="#cb24-2"></a>  <span class="fu">filter</span>(community_type <span class="sc">==</span> <span class="st">"masterplate"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb24-3"><a href="#cb24-3"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb24-4"><a href="#cb24-4"></a>  <span class="fu">mutate</span>(<span class="at">f =</span> count_correct<span class="sc">/</span><span class="fu">sum</span>(count_correct)) <span class="sc">%&gt;%</span> </span>
<span id="cb24-5"><a href="#cb24-5"></a>  <span class="fu">ungroup</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Here we focus on masterplate samples with species that should not be there. We exclude species that were inoculated and plot the distribution of percentages of those species</p>
<div id="fig-01" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-01-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1"></a>masterplate <span class="sc">%&gt;%</span> </span>
<span id="cb25-2"><a href="#cb25-2"></a>  <span class="co"># here we take advantage of the fact that for species not supposed to be in a sample</span></span>
<span id="cb25-3"><a href="#cb25-3"></a>  <span class="co"># the prior left_join will have filled the evo_hist category with an NA. We can then filter</span></span>
<span id="cb25-4"><a href="#cb25-4"></a>  <span class="co"># on this NA value</span></span>
<span id="cb25-5"><a href="#cb25-5"></a>  <span class="fu">filter</span>(<span class="fu">is.na</span>(evo_hist)) <span class="sc">%&gt;%</span> </span>
<span id="cb25-6"><a href="#cb25-6"></a>  <span class="fu">filter</span>(f <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb25-7"><a href="#cb25-7"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> f)) <span class="sc">+</span></span>
<span id="cb25-8"><a href="#cb25-8"></a>  <span class="fu">geom_histogram</span>(<span class="fu">aes</span>(<span class="at">fill =</span> strainID), <span class="at">bins =</span> <span class="dv">20</span>) <span class="sc">+</span></span>
<span id="cb25-9"><a href="#cb25-9"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fl">0.01</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb25-10"><a href="#cb25-10"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> hambi_colors) <span class="sc">+</span></span>
<span id="cb25-11"><a href="#cb25-11"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">trans =</span> <span class="st">"log10"</span>,  <span class="at">labels =</span> percent, <span class="at">guide =</span> <span class="fu">guide_axis</span>(<span class="at">angle=</span><span class="dv">90</span>)) <span class="sc">+</span> </span>
<span id="cb25-12"><a href="#cb25-12"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Crosstalk frequency"</span>, <span class="at">y =</span> <span class="st">"Count"</span>) <span class="sc">+</span></span>
<span id="cb25-13"><a href="#cb25-13"></a>  <span class="fu">facet_grid</span>(<span class="sc">~</span>strainID) <span class="sc">+</span> </span>
<span id="cb25-14"><a href="#cb25-14"></a>  <span class="fu">annotation_logticks</span>(<span class="at">sides =</span> <span class="st">"b"</span>, <span class="at">color=</span><span class="st">"grey30"</span>) <span class="sc">+</span></span>
<span id="cb25-15"><a href="#cb25-15"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span> </span>
<span id="cb25-16"><a href="#cb25-16"></a>  <span class="fu">theme</span>(<span class="at">panel.grid =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb25-17"><a href="#cb25-17"></a>        <span class="at">legend.position =</span> <span class="st">"bottom"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="01_format_rbec_tab_20240711_BTK_illumina_v3_files/figure-html/unnamed-chunk-19-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-01-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;1: Frequency distribution of each species in masterplate samples where it <strong>should not</strong> occur. The 1% mark (commonly accepted as an acceptable Illumina cross-talk standard) is demarcated with a dashed line.
</figcaption>
</figure>
</div>
<p>Generally, the cross talk frequency is pretty OK. For 3/4 species it is 1% or less which is more or less what you can expect when you are multiplexing libraries on an Illumina platform. Values greater than 1% are potentially indicative of a different problem, so 1977 requires a bit more investigation.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1"></a>sids <span class="ot">&lt;-</span> masterplate <span class="sc">%&gt;%</span> </span>
<span id="cb26-2"><a href="#cb26-2"></a>  <span class="fu">filter</span>(<span class="fu">is.na</span>(evo_hist)) <span class="sc">%&gt;%</span> </span>
<span id="cb26-3"><a href="#cb26-3"></a>  <span class="fu">filter</span>(f <span class="sc">&gt;</span> <span class="fl">0.01</span> <span class="sc">&amp;</span> strainID <span class="sc">==</span> <span class="st">"HAMBI_1977"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb26-4"><a href="#cb26-4"></a>  <span class="fu">pull</span>(sample) </span>
<span id="cb26-5"><a href="#cb26-5"></a></span>
<span id="cb26-6"><a href="#cb26-6"></a>masterplate <span class="sc">%&gt;%</span> </span>
<span id="cb26-7"><a href="#cb26-7"></a>  <span class="fu">filter</span>(sample <span class="sc">%in%</span> sids) <span class="sc">%&gt;%</span> </span>
<span id="cb26-8"><a href="#cb26-8"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(sample, strainID, count_correct, replicate, evo_hist, target_f, f)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["sample"],"name":[1],"type":["chr"],"align":["left"]},{"label":["strainID"],"name":[2],"type":["chr"],"align":["left"]},{"label":["count_correct"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["replicate"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["evo_hist"],"name":[5],"type":["chr"],"align":["left"]},{"label":["target_f"],"name":[6],"type":["dbl"],"align":["right"]},{"label":["f"],"name":[7],"type":["dbl"],"align":["right"]}],"data":[{"1":"P19_s_0","2":"HAMBI_1287","3":"3135","4":"1","5":"evo","6":"0.95","7":"0.847984853"},{"1":"P19_s_0","2":"HAMBI_1896","3":"481","4":"1","5":"anc","6":"0.05","7":"0.130105491"},{"1":"P19_s_0","2":"HAMBI_1977","3":"58","4":"1","5":"NA","6":"NA","7":"0.015688396"},{"1":"P19_s_0","2":"HAMBI_0403","3":"23","4":"1","5":"NA","6":"NA","7":"0.006221260"},{"1":"P20_s_0","2":"HAMBI_1287","3":"3335","4":"1","5":"evo","6":"0.95","7":"0.842171717"},{"1":"P20_s_0","2":"HAMBI_1896","3":"541","4":"1","5":"evo","6":"0.05","7":"0.136616162"},{"1":"P20_s_0","2":"HAMBI_1977","3":"50","4":"1","5":"NA","6":"NA","7":"0.012626263"},{"1":"P20_s_0","2":"HAMBI_0403","3":"34","4":"1","5":"NA","6":"NA","7":"0.008585859"},{"1":"P30_s_0","2":"HAMBI_1287","3":"4458","4":"1","5":"evo","6":"0.95","7":"0.834518907"},{"1":"P30_s_0","2":"HAMBI_0403","3":"721","4":"1","5":"anc","6":"0.05","7":"0.134968177"},{"1":"P30_s_0","2":"HAMBI_1977","3":"102","4":"1","5":"NA","6":"NA","7":"0.019093972"},{"1":"P30_s_0","2":"HAMBI_1896","3":"61","4":"1","5":"NA","6":"NA","7":"0.011418944"},{"1":"P36_s_0","2":"HAMBI_1287","3":"3969","4":"1","5":"evo","6":"0.95","7":"0.686440678"},{"1":"P36_s_0","2":"HAMBI_0403","3":"1519","4":"1","5":"evo","6":"0.05","7":"0.262711864"},{"1":"P36_s_0","2":"HAMBI_1977","3":"209","4":"1","5":"NA","6":"NA","7":"0.036146662"},{"1":"P36_s_0","2":"HAMBI_1896","3":"85","4":"1","5":"NA","6":"NA","7":"0.014700796"},{"1":"T08_s_0","2":"HAMBI_1287","3":"4684","4":"1","5":"evo","6":"0.90","7":"0.626538256"},{"1":"T08_s_0","2":"HAMBI_0403","3":"1273","4":"1","5":"anc","6":"0.05","7":"0.170278224"},{"1":"T08_s_0","2":"HAMBI_1896","3":"1421","4":"1","5":"anc","6":"0.05","7":"0.190074906"},{"1":"T08_s_0","2":"HAMBI_1977","3":"98","4":"1","5":"NA","6":"NA","7":"0.013108614"},{"1":"T11_s_0","2":"HAMBI_1896","3":"2851","4":"1","5":"anc","6":"0.05","7":"0.502201867"},{"1":"T11_s_0","2":"HAMBI_1287","3":"1313","4":"1","5":"evo","6":"0.90","7":"0.231284129"},{"1":"T11_s_0","2":"HAMBI_0403","3":"1368","4":"1","5":"evo","6":"0.05","7":"0.240972345"},{"1":"T11_s_0","2":"HAMBI_1977","3":"145","4":"1","5":"NA","6":"NA","7":"0.025541659"},{"1":"T15_s_0","2":"HAMBI_1896","3":"6373","4":"1","5":"evo","6":"0.90","7":"0.775209828"},{"1":"T15_s_0","2":"HAMBI_0403","3":"1011","4":"1","5":"anc","6":"0.05","7":"0.122977740"},{"1":"T15_s_0","2":"HAMBI_1287","3":"663","4":"1","5":"anc","6":"0.05","7":"0.080647123"},{"1":"T15_s_0","2":"HAMBI_1977","3":"174","4":"1","5":"NA","6":"NA","7":"0.021165308"},{"1":"T18_s_0","2":"HAMBI_1896","3":"5204","4":"1","5":"evo","6":"0.90","7":"0.844120032"},{"1":"T18_s_0","2":"HAMBI_0403","3":"705","4":"1","5":"evo","6":"0.05","7":"0.114355231"},{"1":"T18_s_0","2":"HAMBI_1977","3":"163","4":"1","5":"NA","6":"NA","7":"0.026439578"},{"1":"T18_s_0","2":"HAMBI_1287","3":"93","4":"1","5":"anc","6":"0.05","7":"0.015085158"},{"1":"T20_s_0","2":"HAMBI_1287","3":"3454","4":"1","5":"evo","6":"0.90","7":"0.469995918"},{"1":"T20_s_0","2":"HAMBI_0403","3":"1436","4":"1","5":"anc","6":"0.05","7":"0.195400735"},{"1":"T20_s_0","2":"HAMBI_1896","3":"2342","4":"1","5":"evo","6":"0.05","7":"0.318682814"},{"1":"T20_s_0","2":"HAMBI_1977","3":"117","4":"1","5":"NA","6":"NA","7":"0.015920533"},{"1":"T21_s_0","2":"HAMBI_1896","3":"6811","4":"1","5":"evo","6":"0.90","7":"0.776890613"},{"1":"T21_s_0","2":"HAMBI_0403","3":"1039","4":"1","5":"anc","6":"0.05","7":"0.118512604"},{"1":"T21_s_0","2":"HAMBI_1287","3":"773","4":"1","5":"evo","6":"0.05","7":"0.088171552"},{"1":"T21_s_0","2":"HAMBI_1977","3":"144","4":"1","5":"NA","6":"NA","7":"0.016425231"},{"1":"T24_s_0","2":"HAMBI_1896","3":"4335","4":"1","5":"evo","6":"0.90","7":"0.784615385"},{"1":"T24_s_0","2":"HAMBI_1287","3":"578","4":"1","5":"evo","6":"0.05","7":"0.104615385"},{"1":"T24_s_0","2":"HAMBI_0403","3":"529","4":"1","5":"evo","6":"0.05","7":"0.095746606"},{"1":"T24_s_0","2":"HAMBI_1977","3":"83","4":"1","5":"NA","6":"NA","7":"0.015022624"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<p>It looks like all the “problematic” samples come from plates 7 and 8 in the library prep. Plate 8 only contains the masterplate samples from trios whereas plate 7 contains both masterplate and experimental samples. <a href="#fig-02" class="quarto-xref">Figure&nbsp;2</a> shows that HAMBI-1977 is very abundant in many of the samples so likely the “leaky” reads come disproportionately from HAMBI-1977 which is why its crosstalk threshold may be higher (<a href="#fig-01" class="quarto-xref">Figure&nbsp;1</a>).</p>
<div id="fig-02" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-02-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1"></a>finaltable <span class="sc">%&gt;%</span> </span>
<span id="cb27-2"><a href="#cb27-2"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb27-3"><a href="#cb27-3"></a>  <span class="fu">mutate</span>(<span class="at">f =</span> count_correct<span class="sc">/</span><span class="fu">sum</span>(count_correct)) <span class="sc">%&gt;%</span> </span>
<span id="cb27-4"><a href="#cb27-4"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb27-5"><a href="#cb27-5"></a>  <span class="co"># here we take advantage of the fact that for species not supposed to be in a sample</span></span>
<span id="cb27-6"><a href="#cb27-6"></a>  <span class="co"># the prior left_join will have filled the evo_hist category with an NA. We can then filter</span></span>
<span id="cb27-7"><a href="#cb27-7"></a>  <span class="co"># on this NA value</span></span>
<span id="cb27-8"><a href="#cb27-8"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(evo_hist)) <span class="sc">%&gt;%</span> </span>
<span id="cb27-9"><a href="#cb27-9"></a>  <span class="fu">filter</span>(f <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb27-10"><a href="#cb27-10"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> f)) <span class="sc">+</span></span>
<span id="cb27-11"><a href="#cb27-11"></a>  <span class="fu">geom_histogram</span>(<span class="fu">aes</span>(<span class="at">fill =</span> strainID), <span class="at">bins =</span> <span class="dv">20</span>) <span class="sc">+</span></span>
<span id="cb27-12"><a href="#cb27-12"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> hambi_colors) <span class="sc">+</span></span>
<span id="cb27-13"><a href="#cb27-13"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">trans =</span> <span class="st">"log10"</span>,  <span class="at">labels =</span> percent, <span class="at">guide =</span> <span class="fu">guide_axis</span>(<span class="at">angle=</span><span class="dv">90</span>)) <span class="sc">+</span> </span>
<span id="cb27-14"><a href="#cb27-14"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Frequency in samples"</span>, <span class="at">y =</span> <span class="st">"Count"</span>) <span class="sc">+</span></span>
<span id="cb27-15"><a href="#cb27-15"></a>  <span class="fu">facet_grid</span>(<span class="sc">~</span>strainID) <span class="sc">+</span> </span>
<span id="cb27-16"><a href="#cb27-16"></a>  <span class="fu">annotation_logticks</span>(<span class="at">sides =</span> <span class="st">"b"</span>, <span class="at">color=</span><span class="st">"grey30"</span>) <span class="sc">+</span></span>
<span id="cb27-17"><a href="#cb27-17"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span> </span>
<span id="cb27-18"><a href="#cb27-18"></a>  <span class="fu">theme</span>(<span class="at">panel.grid =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb27-19"><a href="#cb27-19"></a>        <span class="at">legend.position =</span> <span class="st">"bottom"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="01_format_rbec_tab_20240711_BTK_illumina_v3_files/figure-html/unnamed-chunk-21-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-02-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;2: Frequency distribution of each species in all samples where it <strong>should</strong> occur.
</figcaption>
</figure>
</div>
<p>Anyway, I don’t think this is a problem and that we can move forward with these samples. However, to defined extinction/competitive exclusion we may need to use a higher threshold than 1% (e.g., 3% frequency) because over 3% we can reliably say that a species is present and it is not due to index cross talk.</p>
</section>
<section id="experimental-samples" class="level2" data-number="6.6">
<h2 data-number="6.6" class="anchored" data-anchor-id="experimental-samples"><span class="header-section-number">6.6</span> Experimental samples</h2>
<p>Now we need to see how our experimental samples performed and if there are species present in them that shouldn’t be there</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1"></a>exp_contam <span class="ot">&lt;-</span> finaltable <span class="sc">%&gt;%</span> </span>
<span id="cb28-2"><a href="#cb28-2"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(community_type, <span class="st">"experiment"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb28-3"><a href="#cb28-3"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb28-4"><a href="#cb28-4"></a>  <span class="fu">mutate</span>(<span class="at">f =</span> count_correct<span class="sc">/</span><span class="fu">sum</span>(count_correct)) <span class="sc">%&gt;%</span> </span>
<span id="cb28-5"><a href="#cb28-5"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb28-6"><a href="#cb28-6"></a>  <span class="co"># because we set 3% as our limit of detection we set read counts of species </span></span>
<span id="cb28-7"><a href="#cb28-7"></a>  <span class="co"># less than 1% to 0</span></span>
<span id="cb28-8"><a href="#cb28-8"></a>  <span class="fu">mutate</span>(<span class="at">count_correct_thresh =</span> <span class="fu">if_else</span>(f <span class="sc">&lt;=</span> <span class="fl">0.03</span>, <span class="dv">0</span>, count_correct)) <span class="sc">%&gt;%</span> </span>
<span id="cb28-9"><a href="#cb28-9"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb28-10"><a href="#cb28-10"></a>  <span class="fu">mutate</span>(<span class="at">f =</span> count_correct_thresh<span class="sc">/</span><span class="fu">sum</span>(count_correct_thresh)) <span class="sc">%&gt;%</span> </span>
<span id="cb28-11"><a href="#cb28-11"></a>  <span class="fu">ungroup</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div id="fig-03" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-03-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1"></a>exp_contam <span class="sc">%&gt;%</span> </span>
<span id="cb29-2"><a href="#cb29-2"></a>  <span class="fu">filter</span>(<span class="fu">is.na</span>(evo_hist)) <span class="sc">%&gt;%</span> </span>
<span id="cb29-3"><a href="#cb29-3"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> f)) <span class="sc">+</span></span>
<span id="cb29-4"><a href="#cb29-4"></a>  <span class="fu">geom_histogram</span>(<span class="fu">aes</span>(<span class="at">fill =</span> strainID), <span class="at">bins =</span> <span class="dv">20</span>) <span class="sc">+</span></span>
<span id="cb29-5"><a href="#cb29-5"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> hambi_colors) <span class="sc">+</span></span>
<span id="cb29-6"><a href="#cb29-6"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">trans =</span> <span class="st">"log10"</span>,  <span class="at">labels =</span> percent, <span class="at">guide =</span> <span class="fu">guide_axis</span>(<span class="at">angle=</span><span class="dv">90</span>)) <span class="sc">+</span> </span>
<span id="cb29-7"><a href="#cb29-7"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Frequency in samples"</span>, <span class="at">y =</span> <span class="st">"Count"</span>) <span class="sc">+</span></span>
<span id="cb29-8"><a href="#cb29-8"></a>  <span class="fu">facet_grid</span>(<span class="sc">~</span> strainID) <span class="sc">+</span> </span>
<span id="cb29-9"><a href="#cb29-9"></a>  <span class="fu">annotation_logticks</span>(<span class="at">sides =</span> <span class="st">"b"</span>, <span class="at">color=</span><span class="st">"grey30"</span>) <span class="sc">+</span></span>
<span id="cb29-10"><a href="#cb29-10"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span> </span>
<span id="cb29-11"><a href="#cb29-11"></a>  <span class="fu">theme</span>(<span class="at">panel.grid =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb29-12"><a href="#cb29-12"></a>        <span class="at">legend.position =</span> <span class="st">"bottom"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in scale_x_continuous(trans = "log10", labels = percent, guide =
guide_axis(angle = 90)): log-10 transformation introduced infinite values.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 783 rows containing non-finite outside the scale range
(`stat_bin()`).</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="01_format_rbec_tab_20240711_BTK_illumina_v3_files/figure-html/unnamed-chunk-23-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-03-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3: Frequency distribution of each species in experimental samples where it <strong>should not</strong> occcur.
</figcaption>
</figure>
</div>
<p>It’s not too bad… but 72 of 566 samples (12.7%) have probably been contaminated because they contain a species that shouldn’t be in the sample (using the 3% relative abundance as a robust threshold for presence/absence).</p>
<section id="contaminated-pairs" class="level3" data-number="6.6.1">
<h3 data-number="6.6.1" class="anchored" data-anchor-id="contaminated-pairs"><span class="header-section-number">6.6.1</span> Contaminated pairs</h3>
<p>Taking a closer look at specific experimental samples with contamination.</p>
<p>Here we just select samples that have &gt; 3% of a species that shouldn’t be there and prepare them for plotting.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1"></a>contamsampid <span class="ot">&lt;-</span> exp_contam <span class="sc">%&gt;%</span> </span>
<span id="cb32-2"><a href="#cb32-2"></a>  <span class="fu">filter</span>(f <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb32-3"><a href="#cb32-3"></a>  <span class="fu">filter</span>(<span class="fu">is.na</span>(evo_hist)) <span class="sc">%&gt;%</span> </span>
<span id="cb32-4"><a href="#cb32-4"></a>  <span class="fu">pull</span>(sample)</span>
<span id="cb32-5"><a href="#cb32-5"></a></span>
<span id="cb32-6"><a href="#cb32-6"></a>spcols <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"HAMBI_0403_anc"</span> <span class="ot">=</span> <span class="st">"#faa019"</span>,</span>
<span id="cb32-7"><a href="#cb32-7"></a>            <span class="st">"HAMBI_0403_evo"</span> <span class="ot">=</span> <span class="st">"#bd7811"</span>,</span>
<span id="cb32-8"><a href="#cb32-8"></a>            <span class="st">"HAMBI_1287_anc"</span> <span class="ot">=</span> <span class="st">"#75afff"</span>,</span>
<span id="cb32-9"><a href="#cb32-9"></a>            <span class="st">"HAMBI_1287_evo"</span> <span class="ot">=</span> <span class="st">"#476c9e"</span>,</span>
<span id="cb32-10"><a href="#cb32-10"></a>            <span class="st">"HAMBI_1896_anc"</span> <span class="ot">=</span> <span class="st">"#59cc4e"</span>,</span>
<span id="cb32-11"><a href="#cb32-11"></a>            <span class="st">"HAMBI_1896_evo"</span> <span class="ot">=</span> <span class="st">"#31752a"</span>,</span>
<span id="cb32-12"><a href="#cb32-12"></a>            <span class="st">"HAMBI_1977_anc"</span> <span class="ot">=</span> <span class="st">"#ffd430"</span>,</span>
<span id="cb32-13"><a href="#cb32-13"></a>            <span class="st">"HAMBI_1977_evo"</span> <span class="ot">=</span> <span class="st">"#ab8e1f"</span>,</span>
<span id="cb32-14"><a href="#cb32-14"></a>            </span>
<span id="cb32-15"><a href="#cb32-15"></a>            <span class="st">"HAMBI_0403_NA"</span>  <span class="ot">=</span> <span class="st">"#e6e5e3"</span>,</span>
<span id="cb32-16"><a href="#cb32-16"></a>            <span class="st">"HAMBI_1287_NA"</span>  <span class="ot">=</span> <span class="st">"#bdbcbb"</span>,</span>
<span id="cb32-17"><a href="#cb32-17"></a>            <span class="st">"HAMBI_1896_NA"</span>  <span class="ot">=</span> <span class="st">"#8c8c8b"</span>,</span>
<span id="cb32-18"><a href="#cb32-18"></a>            <span class="st">"HAMBI_1977_NA"</span>  <span class="ot">=</span> <span class="st">"#333333"</span></span>
<span id="cb32-19"><a href="#cb32-19"></a>            )</span>
<span id="cb32-20"><a href="#cb32-20"></a></span>
<span id="cb32-21"><a href="#cb32-21"></a>exp_contam_plot <span class="ot">&lt;-</span> exp_contam <span class="sc">%&gt;%</span></span>
<span id="cb32-22"><a href="#cb32-22"></a>  <span class="fu">filter</span>(sample <span class="sc">%in%</span> contamsampid) <span class="sc">%&gt;%</span> </span>
<span id="cb32-23"><a href="#cb32-23"></a>  <span class="fu">mutate</span>(<span class="at">sp =</span> <span class="fu">paste0</span>(strainID, <span class="st">"_"</span>, evo_hist))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div id="fig-04" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-04-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1"></a>exp_contam_plot <span class="sc">%&gt;%</span> </span>
<span id="cb33-2"><a href="#cb33-2"></a>  <span class="fu">filter</span>(n_species <span class="sc">==</span> <span class="dv">2</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb33-3"><a href="#cb33-3"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb33-4"><a href="#cb33-4"></a>  <span class="fu">geom_col</span>(<span class="fu">aes</span>(<span class="at">x =</span> sample, <span class="at">y=</span>f, <span class="at">fill =</span> sp)) <span class="sc">+</span></span>
<span id="cb33-5"><a href="#cb33-5"></a>  <span class="fu">facet_wrap</span>( <span class="sc">~</span> strep_conc, <span class="at">scales =</span> <span class="st">"free_x"</span>, <span class="at">ncol =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb33-6"><a href="#cb33-6"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">"Abundance"</span>, <span class="at">x =</span> <span class="st">""</span>, <span class="at">fill =</span> <span class="st">""</span>) <span class="sc">+</span></span>
<span id="cb33-7"><a href="#cb33-7"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> spcols) <span class="sc">+</span></span>
<span id="cb33-8"><a href="#cb33-8"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">labels =</span> percent) <span class="sc">+</span></span>
<span id="cb33-9"><a href="#cb33-9"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">guide =</span> <span class="fu">guide_axis</span>(<span class="at">angle=</span><span class="dv">90</span>)) <span class="sc">+</span></span>
<span id="cb33-10"><a href="#cb33-10"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span> </span>
<span id="cb33-11"><a href="#cb33-11"></a>  <span class="fu">theme</span>(<span class="at">panel.grid =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb33-12"><a href="#cb33-12"></a>        <span class="at">legend.position =</span> <span class="st">"right"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="01_format_rbec_tab_20240711_BTK_illumina_v3_files/figure-html/unnamed-chunk-26-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-04-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;4: Species composition of pair samples that contain &gt; 3% of a species that shouldn’t be there (shown in grey to black).
</figcaption>
</figure>
</div>
<p>I think pretty much the only way to deal with this is to inspect manually. Most of the comtaminated samples are in the 0 Streptomycin conditions. I think we should exclude samples where the contamination is very high (over ~50% of the sample) but those with 10% or less contaminant I think can be retained, and I will discard the contaminating sequences.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1"></a>notcontampairs <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"P04_1_0"</span>, <span class="st">"P45_2_0"</span>, <span class="st">"P01_2_64"</span>, <span class="st">"P03_2_64"</span>, </span>
<span id="cb34-2"><a href="#cb34-2"></a>                    <span class="st">"P06_1_256"</span>, <span class="st">"P17_2_256"</span>, <span class="st">"P30_1_256"</span>, <span class="st">"P30_2_256"</span>, </span>
<span id="cb34-3"><a href="#cb34-3"></a>                    <span class="st">"P41_2_256"</span>)</span>
<span id="cb34-4"><a href="#cb34-4"></a></span>
<span id="cb34-5"><a href="#cb34-5"></a>contampairs <span class="ot">&lt;-</span> <span class="fu">setdiff</span>(contamsampid[<span class="fu">grepl</span>(<span class="st">"^P"</span>, contamsampid)], notcontampairs)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="contaminated-trios" class="level3" data-number="6.6.2">
<h3 data-number="6.6.2" class="anchored" data-anchor-id="contaminated-trios"><span class="header-section-number">6.6.2</span> Contaminated trios</h3>
<p>Now we’ll do the same thing and inspect the trios.</p>
<div id="fig-04" class="quarto-figure quarto-figure-center quarto-float anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-04-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1"></a>exp_contam_plot <span class="sc">%&gt;%</span> </span>
<span id="cb35-2"><a href="#cb35-2"></a>  <span class="fu">filter</span>(n_species <span class="sc">==</span> <span class="dv">3</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb35-3"><a href="#cb35-3"></a>  <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb35-4"><a href="#cb35-4"></a>  <span class="fu">geom_col</span>(<span class="fu">aes</span>(<span class="at">x =</span> sample, <span class="at">y=</span>f, <span class="at">fill =</span> sp)) <span class="sc">+</span></span>
<span id="cb35-5"><a href="#cb35-5"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">"Abundance"</span>, <span class="at">x =</span> <span class="st">""</span>, <span class="at">fill =</span> <span class="st">""</span>) <span class="sc">+</span></span>
<span id="cb35-6"><a href="#cb35-6"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> spcols) <span class="sc">+</span></span>
<span id="cb35-7"><a href="#cb35-7"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">labels =</span> percent) <span class="sc">+</span></span>
<span id="cb35-8"><a href="#cb35-8"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">guide =</span> <span class="fu">guide_axis</span>(<span class="at">angle=</span><span class="dv">90</span>)) <span class="sc">+</span></span>
<span id="cb35-9"><a href="#cb35-9"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span> </span>
<span id="cb35-10"><a href="#cb35-10"></a>  <span class="fu">theme</span>(<span class="at">panel.grid =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb35-11"><a href="#cb35-11"></a>        <span class="at">legend.position =</span> <span class="st">"right"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="01_format_rbec_tab_20240711_BTK_illumina_v3_files/figure-html/unnamed-chunk-28-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-04-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;5: Species composition of trio samples that contain &gt; 3% of a species that shouldn’t be there (shown in grey to black).
</figcaption>
</figure>
</div>
<p>Again, I think we should exclude samples where the contamination is very high (over ~50% of the sample) but those with around 10% or less contaminant I think can be retained, and I will discard the contaminating sequences.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1"></a>notcontamtrios <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"T01_2_0"</span>, <span class="st">"T02_2_0"</span>, <span class="st">"T04_2_0"</span>, <span class="st">"T05_2_0"</span>, <span class="st">"T13_2_0"</span>, </span>
<span id="cb36-2"><a href="#cb36-2"></a>                    <span class="st">"T14_2_0"</span>, <span class="st">"T15_1_0"</span>, <span class="st">"T15_2_0"</span>, <span class="st">"T16_1_0"</span>, <span class="st">"T16_2_0"</span>, </span>
<span id="cb36-3"><a href="#cb36-3"></a>                    <span class="st">"T17_2_0"</span>, <span class="st">"T18_2_0"</span>,<span class="st">"T31_2_0"</span>, <span class="st">"T45_1_0"</span>, <span class="st">"T45_2_0"</span>, </span>
<span id="cb36-4"><a href="#cb36-4"></a>                    <span class="st">"T55_1_0"</span>, <span class="st">"T61_1_0"</span>, <span class="st">"T61_2_0"</span>, <span class="st">"T64_1_0"</span>, <span class="st">"T64_2_0"</span>)</span>
<span id="cb36-5"><a href="#cb36-5"></a></span>
<span id="cb36-6"><a href="#cb36-6"></a>contamtrios <span class="ot">&lt;-</span> <span class="fu">setdiff</span>(contamsampid[<span class="fu">grepl</span>(<span class="st">"^T"</span>, contamsampid)], notcontamtrios)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
</section>
<section id="export" class="level1" data-number="7">
<h1 data-number="7"><span class="header-section-number">7</span> Export</h1>
<section id="posneg-control-samples" class="level2" data-number="7.1">
<h2 data-number="7.1" class="anchored" data-anchor-id="posneg-control-samples"><span class="header-section-number">7.1</span> pos/neg control samples</h2>
<p>Separate out the pos/neg control samples</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1"></a>finaltable <span class="sc">%&gt;%</span> </span>
<span id="cb37-2"><a href="#cb37-2"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(community_type, <span class="st">"^pos|^neg"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb37-3"><a href="#cb37-3"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>transfers, <span class="sc">-</span>strep_conc, <span class="sc">-</span>evo_hist, <span class="sc">-</span>target_f, <span class="sc">-</span>count) <span class="sc">%&gt;%</span> </span>
<span id="cb37-4"><a href="#cb37-4"></a>  <span class="fu">write_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(data, <span class="st">"pos_neg_ctrl_counts.tsv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="pairs" class="level2" data-number="7.2">
<h2 data-number="7.2" class="anchored" data-anchor-id="pairs"><span class="header-section-number">7.2</span> Pairs</h2>
<p>separate, format, and write the pair samples</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1"></a>pairs <span class="ot">&lt;-</span> finaltable <span class="sc">%&gt;%</span> </span>
<span id="cb38-2"><a href="#cb38-2"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">str_detect</span>(community_type, <span class="st">"^pos|^neg"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb38-3"><a href="#cb38-3"></a>  <span class="fu">filter</span>(n_species <span class="sc">==</span> <span class="dv">2</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb38-4"><a href="#cb38-4"></a>  <span class="fu">filter</span>(sample <span class="sc">%nin%</span> contampairs) <span class="sc">%&gt;%</span> </span>
<span id="cb38-5"><a href="#cb38-5"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb38-6"><a href="#cb38-6"></a>  <span class="fu">mutate</span>(<span class="at">f =</span> count_correct<span class="sc">/</span><span class="fu">sum</span>(count_correct)) <span class="sc">%&gt;%</span> </span>
<span id="cb38-7"><a href="#cb38-7"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb38-8"><a href="#cb38-8"></a>  <span class="co"># because we set 3% as our limit of detection we set read counts of species </span></span>
<span id="cb38-9"><a href="#cb38-9"></a>  <span class="co"># less than 1% to 0</span></span>
<span id="cb38-10"><a href="#cb38-10"></a>  <span class="fu">mutate</span>(<span class="at">count_correct_thresh =</span> <span class="fu">if_else</span>(f <span class="sc">&lt;=</span> <span class="fl">0.03</span>, <span class="dv">0</span>, count_correct)) <span class="sc">%&gt;%</span> </span>
<span id="cb38-11"><a href="#cb38-11"></a>  <span class="co"># also exclude any remaining counts from species that shouldnt be there</span></span>
<span id="cb38-12"><a href="#cb38-12"></a>  <span class="co"># using again the fact that evo_hist should be NA for these species</span></span>
<span id="cb38-13"><a href="#cb38-13"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(evo_hist)) <span class="sc">%&gt;%</span> </span>
<span id="cb38-14"><a href="#cb38-14"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb38-15"><a href="#cb38-15"></a>  <span class="fu">mutate</span>(<span class="at">f =</span> count_correct_thresh<span class="sc">/</span><span class="fu">sum</span>(count_correct_thresh)) <span class="sc">%&gt;%</span> </span>
<span id="cb38-16"><a href="#cb38-16"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb38-17"><a href="#cb38-17"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(sample, strainID, evo_hist, count_correct, count_correct_thresh, </span>
<span id="cb38-18"><a href="#cb38-18"></a>                f, target_f, replicate, strep_conc, transfers, n_species, community_type, plate_well)</span>
<span id="cb38-19"><a href="#cb38-19"></a></span>
<span id="cb38-20"><a href="#cb38-20"></a><span class="fu">write_tsv</span>(pairs, here<span class="sc">::</span><span class="fu">here</span>(data, <span class="st">"pairs_counts.tsv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="trios" class="level2" data-number="7.3">
<h2 data-number="7.3" class="anchored" data-anchor-id="trios"><span class="header-section-number">7.3</span> Trios</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1"></a>trios <span class="ot">&lt;-</span> finaltable <span class="sc">%&gt;%</span> </span>
<span id="cb39-2"><a href="#cb39-2"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">str_detect</span>(community_type, <span class="st">"^pos|^neg"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb39-3"><a href="#cb39-3"></a>  <span class="fu">filter</span>(n_species <span class="sc">==</span> <span class="dv">3</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb39-4"><a href="#cb39-4"></a>  <span class="fu">filter</span>(sample <span class="sc">%nin%</span> contamtrios) <span class="sc">%&gt;%</span> </span>
<span id="cb39-5"><a href="#cb39-5"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb39-6"><a href="#cb39-6"></a>  <span class="fu">mutate</span>(<span class="at">f =</span> count_correct<span class="sc">/</span><span class="fu">sum</span>(count_correct)) <span class="sc">%&gt;%</span> </span>
<span id="cb39-7"><a href="#cb39-7"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb39-8"><a href="#cb39-8"></a>  <span class="co"># because we set 3% as our limit of detection we set read counts of species </span></span>
<span id="cb39-9"><a href="#cb39-9"></a>  <span class="co"># less than 1% to 0</span></span>
<span id="cb39-10"><a href="#cb39-10"></a>  <span class="fu">mutate</span>(<span class="at">count_correct_thresh =</span> <span class="fu">if_else</span>(f <span class="sc">&lt;=</span> <span class="fl">0.03</span>, <span class="dv">0</span>, count_correct)) <span class="sc">%&gt;%</span> </span>
<span id="cb39-11"><a href="#cb39-11"></a>  <span class="co"># also exclude any remaining counts from species that shouldnt be there</span></span>
<span id="cb39-12"><a href="#cb39-12"></a>  <span class="co"># using again the fact that evo_hist should be NA for these species</span></span>
<span id="cb39-13"><a href="#cb39-13"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(evo_hist)) <span class="sc">%&gt;%</span> </span>
<span id="cb39-14"><a href="#cb39-14"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb39-15"><a href="#cb39-15"></a>  <span class="fu">mutate</span>(<span class="at">f =</span> count_correct_thresh<span class="sc">/</span><span class="fu">sum</span>(count_correct_thresh)) <span class="sc">%&gt;%</span> </span>
<span id="cb39-16"><a href="#cb39-16"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb39-17"><a href="#cb39-17"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(sample, strainID, evo_hist, count_correct, count_correct_thresh, </span>
<span id="cb39-18"><a href="#cb39-18"></a>                f, target_f, replicate, strep_conc, transfers, n_species, community_type, plate_well)</span>
<span id="cb39-19"><a href="#cb39-19"></a></span>
<span id="cb39-20"><a href="#cb39-20"></a><span class="fu">write_tsv</span>(trios, here<span class="sc">::</span><span class="fu">here</span>(data, <span class="st">"trios_counts.tsv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="clean-up" class="level1" data-number="8">
<h1 data-number="8"><span class="header-section-number">8</span> Clean up</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1"></a><span class="co"># remove decompressed coverage directory from temp location</span></span>
<span id="cb40-2"><a href="#cb40-2"></a>fs<span class="sc">::</span><span class="fu">dir_delete</span>(tmpdir)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>