<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.42">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Shane Hogle">
<meta name="dcterms.date" content="2025-03-06">

<title>Analysis of pairwise competition – hambiEvoEnvCoexist</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../../">
<script src="../../../site_libs/quarto-html/quarto.js"></script>
<script src="../../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting-2f5df379a58b258e96c21c0638c20c03.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../../site_libs/bootstrap/bootstrap-1f23bf93a17969024013c136d185e5ce.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<link href="../../../site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet">
<script src="../../../site_libs/pagedtable-1.1/js/pagedtable.js"></script>

  <script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

</head>

<body class="nav-sidebar docked">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../../R/communities/amplicon/02_analysis_pairs.html">5. Community growth - Coexistence/exclusion analysis</a></li><li class="breadcrumb-item"><a href="../../../R/communities/amplicon/02_analysis_pairs.html">i) Species pairs</a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation docked overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="../../../">hambiEvoEnvCoexist</a> 
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true">
 <span class="menu-text">1. Monoculture growth</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../R/monocultures/bioscreen/01_format_growthcurves_batch.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">i) Inspect and format growth curves</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../R/monocultures/bioscreen/02_plot_analyze_growth.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">ii) Plot growth summaries</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true">
 <span class="menu-text">2. Community growth - experiment design</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../R/communities/experiment_design/01_pairs.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">i) Species pairs - 96 Well plate layout</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../R/communities/experiment_design/02_trios.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">ii) Species trios - 96 Well plate layout</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../R/communities/experiment_design/03_quartets.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">iii) Species quartets - 96 Well plate layout</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true">
 <span class="menu-text">3. Community growth - optical density</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../R/communities/optical_density/01_format_plot_od.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">i) Process and plot OD data</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="true">
 <span class="menu-text">4. Community growth - amplicon formatting</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../R/communities/amplicon/20240711_BTK_illumina_v3/01_format_rbec_tab.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">i) Batch 2024-07-11</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../R/communities/amplicon/20240711_BTK_illumina_v3/02_correct_boil_to_qiagen.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">ii) Batch 2024-07-11 calibration</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../R/communities/amplicon/20241024_BTK_illumina_v3/01_format_rbec_tab.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">iii) Batch 2024-10-24</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../R/communities/amplicon/20250127_BTK_illumina_v3/01_format_rbec_tab.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">iv) Batch 2025-01-27</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../R/communities/amplicon/01_compile_format.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">v) Compile pairs, trios, quartets</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" role="navigation" aria-expanded="true">
 <span class="menu-text">5. Community growth - Coexistence/exclusion analysis</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-5" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../R/communities/amplicon/02_analysis_pairs.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">i) Species pairs</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../R/communities/amplicon/03_analysis_trios.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">ii) Species trios</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../R/communities/amplicon/04_analysis_quartets.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">iii) Species quartets</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#introduction" id="toc-introduction" class="nav-link active" data-scroll-target="#introduction"><span class="header-section-number">1</span> Introduction</a></li>
  <li><a href="#setup" id="toc-setup" class="nav-link" data-scroll-target="#setup"><span class="header-section-number">2</span> Setup</a>
  <ul class="collapse">
  <li><a href="#libraries" id="toc-libraries" class="nav-link" data-scroll-target="#libraries"><span class="header-section-number">2.1</span> Libraries</a></li>
  <li><a href="#global-variables" id="toc-global-variables" class="nav-link" data-scroll-target="#global-variables"><span class="header-section-number">2.2</span> Global variables</a></li>
  </ul></li>
  <li><a href="#read-data" id="toc-read-data" class="nav-link" data-scroll-target="#read-data"><span class="header-section-number">3</span> Read data</a>
  <ul class="collapse">
  <li><a href="#species-abundances" id="toc-species-abundances" class="nav-link" data-scroll-target="#species-abundances"><span class="header-section-number">3.1</span> Species abundances</a></li>
  </ul></li>
  <li><a href="#formatting" id="toc-formatting" class="nav-link" data-scroll-target="#formatting"><span class="header-section-number">4</span> Formatting</a>
  <ul class="collapse">
  <li><a href="#transfer-0-starting-proportions" id="toc-transfer-0-starting-proportions" class="nav-link" data-scroll-target="#transfer-0-starting-proportions"><span class="header-section-number">4.1</span> Transfer 0 (starting proportions)</a></li>
  <li><a href="#transfer-8-ending-proportions" id="toc-transfer-8-ending-proportions" class="nav-link" data-scroll-target="#transfer-8-ending-proportions"><span class="header-section-number">4.2</span> Transfer 8 (ending proportions)</a></li>
  </ul></li>
  <li><a href="#define-competition-outcomes" id="toc-define-competition-outcomes" class="nav-link" data-scroll-target="#define-competition-outcomes"><span class="header-section-number">5</span> Define competition outcomes</a>
  <ul class="collapse">
  <li><a href="#binomial-sampling-and-wilcox-test" id="toc-binomial-sampling-and-wilcox-test" class="nav-link" data-scroll-target="#binomial-sampling-and-wilcox-test"><span class="header-section-number">5.1</span> Binomial sampling and Wilcox test</a></li>
  <li><a href="#competition-outcome-rules" id="toc-competition-outcome-rules" class="nav-link" data-scroll-target="#competition-outcome-rules"><span class="header-section-number">5.2</span> Competition outcome rules</a>
  <ul class="collapse">
  <li><a href="#bistability" id="toc-bistability" class="nav-link" data-scroll-target="#bistability"><span class="header-section-number">5.2.1</span> Bistability</a></li>
  <li><a href="#competitive-exclusion" id="toc-competitive-exclusion" class="nav-link" data-scroll-target="#competitive-exclusion"><span class="header-section-number">5.2.2</span> Competitive exclusion</a></li>
  <li><a href="#coexistence" id="toc-coexistence" class="nav-link" data-scroll-target="#coexistence"><span class="header-section-number">5.2.3</span> Coexistence</a></li>
  </ul></li>
  <li><a href="#plotting-pairwise-outpcomes" id="toc-plotting-pairwise-outpcomes" class="nav-link" data-scroll-target="#plotting-pairwise-outpcomes"><span class="header-section-number">5.3</span> Plotting pairwise outpcomes</a></li>
  </ul></li>
  <li><a href="#network" id="toc-network" class="nav-link" data-scroll-target="#network"><span class="header-section-number">6</span> Network</a>
  <ul class="collapse">
  <li><a href="#evolution-and-streptomycin-categories-separate" id="toc-evolution-and-streptomycin-categories-separate" class="nav-link" data-scroll-target="#evolution-and-streptomycin-categories-separate"><span class="header-section-number">6.1</span> Evolution and Streptomycin categories separate</a></li>
  <li><a href="#only-streptomycin-category-separate" id="toc-only-streptomycin-category-separate" class="nav-link" data-scroll-target="#only-streptomycin-category-separate"><span class="header-section-number">6.2</span> Only Streptomycin category separate</a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../../R/communities/amplicon/02_analysis_pairs.html">5. Community growth - Coexistence/exclusion analysis</a></li><li class="breadcrumb-item"><a href="../../../R/communities/amplicon/02_analysis_pairs.html">i) Species pairs</a></li></ol></nav>
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">Analysis of pairwise competition</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Shane Hogle </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">March 6, 2025</p>
    </div>
  </div>
  
    
  </div>
  
<div>
  <div class="abstract">
    <div class="block-title">Abstract</div>
    Here we analyze the community outpcomes from all competing species pairs.
  </div>
</div>


</header>


<section id="introduction" class="level1" data-number="1">
<h1 data-number="1"><span class="header-section-number">1</span> Introduction</h1>
<p>Contains results from pairs of all streptomycin concentrations and trios for 0 streptomycin</p>
</section>
<section id="setup" class="level1" data-number="2">
<h1 data-number="2"><span class="header-section-number">2</span> Setup</h1>
<section id="libraries" class="level2" data-number="2.1">
<h2 data-number="2.1" class="anchored" data-anchor-id="libraries"><span class="header-section-number">2.1</span> Libraries</h2>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-2"><a href="#cb1-2"></a><span class="fu">library</span>(here)</span>
<span id="cb1-3"><a href="#cb1-3"></a><span class="fu">library</span>(fs)</span>
<span id="cb1-4"><a href="#cb1-4"></a><span class="fu">library</span>(scales)</span>
<span id="cb1-5"><a href="#cb1-5"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb1-6"><a href="#cb1-6"></a><span class="fu">library</span>(broom)</span>
<span id="cb1-7"><a href="#cb1-7"></a><span class="fu">library</span>(ggraph)</span>
<span id="cb1-8"><a href="#cb1-8"></a><span class="fu">library</span>(tidygraph)</span>
<span id="cb1-9"><a href="#cb1-9"></a><span class="fu">source</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"R"</span>, <span class="st">"utils_generic.R"</span>))</span>
<span id="cb1-10"><a href="#cb1-10"></a><span class="fu">source</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"R"</span>, <span class="st">"communities"</span>, <span class="st">"amplicon"</span>, <span class="st">"utils_amplicon.R"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="global-variables" class="level2" data-number="2.2">
<h2 data-number="2.2" class="anchored" data-anchor-id="global-variables"><span class="header-section-number">2.2</span> Global variables</h2>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a>data <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"communities"</span>)</span>
<span id="cb2-2"><a href="#cb2-2"></a><span class="co"># make processed data directory if it doesn't exist</span></span>
<span id="cb2-3"><a href="#cb2-3"></a>fs<span class="sc">::</span><span class="fu">dir_create</span>(data)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</section>
<section id="read-data" class="level1" data-number="3">
<h1 data-number="3"><span class="header-section-number">3</span> Read data</h1>
<section id="species-abundances" class="level2" data-number="3.1">
<h2 data-number="3.1" class="anchored" data-anchor-id="species-abundances"><span class="header-section-number">3.1</span> Species abundances</h2>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1"></a>samp_pairs <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(data, <span class="st">"2sps_compiled.tsv"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb3-2"><a href="#cb3-2"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">f =</span> f_thresh)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</section>
<section id="formatting" class="level1" data-number="4">
<h1 data-number="4"><span class="header-section-number">4</span> Formatting</h1>
<section id="transfer-0-starting-proportions" class="level2" data-number="4.1">
<h2 data-number="4.1" class="anchored" data-anchor-id="transfer-0-starting-proportions"><span class="header-section-number">4.1</span> Transfer 0 (starting proportions)</h2>
<p>First get abundances from transfer 0 (masterplate)</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a>samp_pairs_fmt_mp <span class="ot">&lt;-</span> samp_pairs <span class="sc">%&gt;%</span> </span>
<span id="cb4-2"><a href="#cb4-2"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(community_type <span class="sc">==</span> <span class="st">"masterplate"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb4-3"><a href="#cb4-3"></a>  <span class="co"># make a combined evolution and species identifier and extract the community ID</span></span>
<span id="cb4-4"><a href="#cb4-4"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">sp =</span> <span class="fu">paste</span>(<span class="fu">str_to_upper</span>(evo_hist), <span class="fu">str_extract</span>(strainID, <span class="st">"</span><span class="sc">\\</span><span class="st">d+"</span>), <span class="at">sep =</span> <span class="st">"_"</span>),</span>
<span id="cb4-5"><a href="#cb4-5"></a>         <span class="at">community_id =</span> <span class="fu">str_extract</span>(sample, <span class="st">"P</span><span class="sc">\\</span><span class="st">d</span><span class="sc">\\</span><span class="st">d"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb4-6"><a href="#cb4-6"></a>  <span class="co"># this step is important to ensure that when dfs are pivoted wider the </span></span>
<span id="cb4-7"><a href="#cb4-7"></a>  <span class="co"># sp_1 and sp_2 stay consistent </span></span>
<span id="cb4-8"><a href="#cb4-8"></a>  dplyr<span class="sc">::</span><span class="fu">arrange</span>(community_id, sp) <span class="sc">%&gt;%</span></span>
<span id="cb4-9"><a href="#cb4-9"></a>  <span class="co"># this creates an index for each species present in each community, it is needed</span></span>
<span id="cb4-10"><a href="#cb4-10"></a>  <span class="co"># for the pivot to be consistent between the master plate and the samples</span></span>
<span id="cb4-11"><a href="#cb4-11"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(community_id) <span class="sc">%&gt;%</span> </span>
<span id="cb4-12"><a href="#cb4-12"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">n =</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">n</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb4-13"><a href="#cb4-13"></a>  dplyr<span class="sc">::</span><span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb4-14"><a href="#cb4-14"></a>  tidyr<span class="sc">::</span><span class="fu">pivot_wider</span>(<span class="at">id_cols =</span> community_id, <span class="at">values_from =</span> <span class="fu">c</span>(sp, f), <span class="at">names_from =</span> n) <span class="sc">%&gt;%</span> </span>
<span id="cb4-15"><a href="#cb4-15"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">transfer =</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb4-16"><a href="#cb4-16"></a>  tidyr<span class="sc">::</span><span class="fu">expand_grid</span>(<span class="at">strep_conc =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">16</span>, <span class="dv">64</span>, <span class="dv">256</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb4-17"><a href="#cb4-17"></a>  <span class="fu">arrange</span>(sp_1, sp_2, strep_conc)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="transfer-8-ending-proportions" class="level2" data-number="4.2">
<h2 data-number="4.2" class="anchored" data-anchor-id="transfer-8-ending-proportions"><span class="header-section-number">4.2</span> Transfer 8 (ending proportions)</h2>
<p>Format abundances from the experiment, summarizing over replicates. Here we calculate the median frequency across biological replicates using <code>Hmisc::smedian.hilow</code> which computes the sample median and the outer quantiles (0.025 and 0.975).</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1"></a>samp_pairs_fmt_exp <span class="ot">&lt;-</span> samp_pairs <span class="sc">%&gt;%</span> </span>
<span id="cb5-2"><a href="#cb5-2"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(community_type <span class="sc">==</span> <span class="st">"experiment"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb5-3"><a href="#cb5-3"></a>  <span class="co"># make a combined evolution and species identifier and extract the community ID</span></span>
<span id="cb5-4"><a href="#cb5-4"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">sp =</span> <span class="fu">paste</span>(<span class="fu">str_to_upper</span>(evo_hist), <span class="fu">str_extract</span>(strainID, <span class="st">"</span><span class="sc">\\</span><span class="st">d+"</span>), <span class="at">sep =</span> <span class="st">"_"</span>),</span>
<span id="cb5-5"><a href="#cb5-5"></a>         <span class="at">community_id =</span> <span class="fu">str_extract</span>(sample, <span class="st">"P</span><span class="sc">\\</span><span class="st">d</span><span class="sc">\\</span><span class="st">d"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb5-6"><a href="#cb5-6"></a>  <span class="co"># calculate median and 95% CI across replicates</span></span>
<span id="cb5-7"><a href="#cb5-7"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(ggplot2<span class="sc">::</span><span class="fu">median_hilow</span>(f), <span class="at">.by =</span> <span class="fu">c</span>(<span class="st">"community_id"</span>, <span class="st">"sp"</span>, <span class="st">"strep_conc"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb5-8"><a href="#cb5-8"></a>  <span class="co"># rename the y columns as f for compatibility</span></span>
<span id="cb5-9"><a href="#cb5-9"></a>  dplyr<span class="sc">::</span><span class="fu">rename_with</span>(<span class="at">.cols =</span> <span class="fu">starts_with</span>(<span class="st">"y"</span>), \(x) <span class="fu">str_replace</span>(x, <span class="st">"y"</span>, <span class="st">"f"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb5-10"><a href="#cb5-10"></a>  <span class="co"># this step is important to ensure that when dfs are pivoted wider the </span></span>
<span id="cb5-11"><a href="#cb5-11"></a>  <span class="co"># sp_1 and sp_2 stay consistent </span></span>
<span id="cb5-12"><a href="#cb5-12"></a>  dplyr<span class="sc">::</span><span class="fu">arrange</span>(community_id, sp, strep_conc) <span class="sc">%&gt;%</span></span>
<span id="cb5-13"><a href="#cb5-13"></a>  <span class="co"># this creates an index for each species present in each community, it is needed</span></span>
<span id="cb5-14"><a href="#cb5-14"></a>  <span class="co"># for the pivot to be consistent between the master plate and the samples</span></span>
<span id="cb5-15"><a href="#cb5-15"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(community_id, strep_conc) <span class="sc">%&gt;%</span> </span>
<span id="cb5-16"><a href="#cb5-16"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">n =</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">n</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb5-17"><a href="#cb5-17"></a>  dplyr<span class="sc">::</span><span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb5-18"><a href="#cb5-18"></a>  tidyr<span class="sc">::</span><span class="fu">pivot_wider</span>(<span class="at">id_cols =</span> <span class="fu">c</span>(community_id, strep_conc), <span class="at">values_from =</span> <span class="fu">c</span>(sp, f, fmin, fmax), <span class="at">names_from =</span> n) <span class="sc">%&gt;%</span> </span>
<span id="cb5-19"><a href="#cb5-19"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">transfer =</span> <span class="dv">8</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb5-20"><a href="#cb5-20"></a>  <span class="fu">arrange</span>(sp_1, sp_2, strep_conc)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</section>
<section id="define-competition-outcomes" class="level1" data-number="5">
<h1 data-number="5"><span class="header-section-number">5</span> Define competition outcomes</h1>
<section id="binomial-sampling-and-wilcox-test" class="level2" data-number="5.1">
<h2 data-number="5.1" class="anchored" data-anchor-id="binomial-sampling-and-wilcox-test"><span class="header-section-number">5.1</span> Binomial sampling and Wilcox test</h2>
<p>First need to determine which samples significantly decreased/increased from T0 to T8. We don’t have enough biological replicates for to compute a statistic across replicates for <a href="https://www.pnas.org/doi/10.1073/pnas.2302491120">outcome variability</a>. However, we can estimate the mean fraction of A and quantify the inferential uncertainty of the mean by bootstrap resampling. We the used median proportion of species A from two biological replicates of each T8 pair as the probability of success (i.e., drawing species A) from 1000 draws (i.e., sequencing reads) from the binomial distribution. To determine whether the frequency of Species A significantly changed from T0 to T8, the means of the 1000 binomial draws for T0 and T8 were compared using a Wilcoxon rank sum test (N = 2000). Tests with Bonferroni multiple test corrected p values &lt; 1e-5 were considered to represent significantly different T0 and T8 samples.</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1"></a><span class="fu">set.seed</span>(<span class="dv">124341</span>)</span>
<span id="cb6-2"><a href="#cb6-2"></a></span>
<span id="cb6-3"><a href="#cb6-3"></a><span class="co"># first join the T0 and T8 abundances</span></span>
<span id="cb6-4"><a href="#cb6-4"></a>wc_test <span class="ot">&lt;-</span> <span class="fu">left_join</span>(samp_pairs_fmt_exp, samp_pairs_fmt_mp, <span class="at">by =</span> <span class="fu">join_by</span>(community_id, strep_conc, sp_1, sp_2)) <span class="sc">%&gt;%</span> </span>
<span id="cb6-5"><a href="#cb6-5"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">delta_f_1 =</span> f_1.x <span class="sc">-</span> f_1.y) <span class="sc">%&gt;%</span> </span>
<span id="cb6-6"><a href="#cb6-6"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(community_id, strep_conc, sp_1, sp_2, delta_f_1, <span class="at">f_1_8 =</span> f_1.x, <span class="at">f_1_0 =</span> f_1.y, </span>
<span id="cb6-7"><a href="#cb6-7"></a>                <span class="at">f_2_8 =</span> f_2.x, <span class="at">f_2_0 =</span> f_2.y) <span class="sc">%&gt;%</span> </span>
<span id="cb6-8"><a href="#cb6-8"></a>  tidyr<span class="sc">::</span><span class="fu">nest</span>(<span class="at">data =</span> <span class="fu">c</span>(<span class="sc">-</span>community_id, <span class="sc">-</span>strep_conc)) <span class="sc">%&gt;%</span></span>
<span id="cb6-9"><a href="#cb6-9"></a>  <span class="co"># samples 1000 draws from binomial distribution using f_a median as the probability of success</span></span>
<span id="cb6-10"><a href="#cb6-10"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">f_1_0_rs =</span> purrr<span class="sc">::</span><span class="fu">map</span>(data, \(x) <span class="fu">map</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">100</span>, \(i) <span class="fu">sum</span>(<span class="fu">rbinom</span>(<span class="dv">1000</span>, <span class="dv">1</span>, x<span class="sc">$</span>f_1_0))<span class="sc">/</span><span class="dv">1000</span>)),</span>
<span id="cb6-11"><a href="#cb6-11"></a>         <span class="at">f_1_8_rs =</span> purrr<span class="sc">::</span><span class="fu">map</span>(data, \(x) <span class="fu">map</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">100</span>, \(i) <span class="fu">sum</span>(<span class="fu">rbinom</span>(<span class="dv">1000</span>, <span class="dv">1</span>, x<span class="sc">$</span>f_1_8))<span class="sc">/</span><span class="dv">1000</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb6-12"><a href="#cb6-12"></a>  tidyr<span class="sc">::</span><span class="fu">unnest</span>(<span class="at">cols =</span> <span class="fu">c</span>(data, f_1_0_rs, f_1_8_rs)) <span class="sc">%&gt;%</span> </span>
<span id="cb6-13"><a href="#cb6-13"></a>  <span class="co"># nest the samples</span></span>
<span id="cb6-14"><a href="#cb6-14"></a>  tidyr<span class="sc">::</span><span class="fu">nest</span>(<span class="at">bs =</span> <span class="fu">c</span>(f_1_0_rs, f_1_8_rs)) <span class="sc">%&gt;%</span></span>
<span id="cb6-15"><a href="#cb6-15"></a>  <span class="co"># perform the wilcox test</span></span>
<span id="cb6-16"><a href="#cb6-16"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">wc =</span> purrr<span class="sc">::</span><span class="fu">map</span>(bs, \(i) <span class="fu">wilcox.test</span>(<span class="at">x =</span> <span class="fu">as.numeric</span>(i<span class="sc">$</span>f_1_0_rs), <span class="at">y =</span> <span class="fu">as.numeric</span>(i<span class="sc">$</span>f_1_8_rs)))) <span class="sc">%&gt;%</span> </span>
<span id="cb6-17"><a href="#cb6-17"></a>  <span class="co"># tidy-ify the test output</span></span>
<span id="cb6-18"><a href="#cb6-18"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">tidy_wc =</span> purrr<span class="sc">::</span><span class="fu">map</span>(wc, \(x) broom<span class="sc">::</span><span class="fu">tidy</span>(x))) <span class="sc">%&gt;%</span> </span>
<span id="cb6-19"><a href="#cb6-19"></a>  tidyr<span class="sc">::</span><span class="fu">unnest</span>(<span class="at">cols =</span> <span class="fu">c</span>(tidy_wc)) <span class="sc">%&gt;%</span> </span>
<span id="cb6-20"><a href="#cb6-20"></a>  <span class="co"># p-value adjust using bonferroni correction</span></span>
<span id="cb6-21"><a href="#cb6-21"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">p_adjusted =</span> <span class="fu">p.adjust</span>(p.value, <span class="at">method =</span> <span class="st">"bonferroni"</span>, <span class="at">n =</span> <span class="fu">n</span>())) <span class="sc">%&gt;%</span> </span>
<span id="cb6-22"><a href="#cb6-22"></a>  dplyr<span class="sc">::</span><span class="fu">arrange</span>(strep_conc, sp_1, sp_2) <span class="sc">%&gt;%</span> </span>
<span id="cb6-23"><a href="#cb6-23"></a>  <span class="co"># define whether change is significantly positive or negative</span></span>
<span id="cb6-24"><a href="#cb6-24"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">change =</span> dplyr<span class="sc">::</span><span class="fu">case_when</span>(p.value <span class="sc">&gt;</span> <span class="fl">1e-5</span> <span class="sc">~</span> <span class="dv">0</span>,</span>
<span id="cb6-25"><a href="#cb6-25"></a>                             <span class="fu">sign</span>(delta_f_1) <span class="sc">==</span> <span class="sc">-</span><span class="dv">1</span> <span class="sc">&amp;</span> p.value <span class="sc">&lt;=</span> <span class="fl">1e-5</span> <span class="sc">~</span> <span class="sc">-</span><span class="dv">1</span>, </span>
<span id="cb6-26"><a href="#cb6-26"></a>                             <span class="fu">sign</span>(delta_f_1) <span class="sc">==</span> <span class="dv">1</span> <span class="sc">&amp;</span> p.value <span class="sc">&lt;=</span> <span class="fl">1e-5</span> <span class="sc">~</span> <span class="dv">1</span>)) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="competition-outcome-rules" class="level2" data-number="5.2">
<h2 data-number="5.2" class="anchored" data-anchor-id="competition-outcome-rules"><span class="header-section-number">5.2</span> Competition outcome rules</h2>
<p>Here we set up the rules for defining the competition outcomes. In the pairwise competition experiments each species and evolutionary form was competed against all other species and evolutionary forms (excluding evolutionary forms of the same species because these cannot be resolved using amplicon data). Each competitor in a pair <span class="math inline">\(\{A, B\}\)</span> was allowed to invade from rare (<span class="math inline">\(f^{R} = 0.1\)</span>) while the other competitor was common (<span class="math inline">\(f^{C} = 0.9\)</span>) in duplicate for 8 growth cycles resulting in 4 experiments for each competing pair. We used a <span class="math inline">\(f_{extinct} = 0.01\)</span> frequency threshold for deciding whether a species was operationally extinct and a $| _{(t8-t0)} f | $ change over 8 growth cycles to decide whether the magnitude of change over the 8 growth cycles was appreciable. Using the abundance outcomes from both replicates of each the pairwise competition experiments for each pair we classified outcomes as “competitive exclusion”, “coexistence”, “bistability”, or “inconclusive” according to the following criteria:</p>
<section id="bistability" class="level3" data-number="5.2.1">
<h3 data-number="5.2.1" class="anchored" data-anchor-id="bistability"><span class="header-section-number">5.2.1</span> Bistability</h3>
<p>Requires each species in a pair to significantly increase (<span class="math inline">\(\Delta f_{\{A, B\}} \geq 0.05\)</span>) when starting from common <em>and</em> significantly decrease (<span class="math inline">\(\Delta f_{\{A, B\}} \leq -0.05\)</span>) in abundance when invading from rare over the 8 growth cycles.</p>
</section>
<section id="competitive-exclusion" class="level3" data-number="5.2.2">
<h3 data-number="5.2.2" class="anchored" data-anchor-id="competitive-exclusion"><span class="header-section-number">5.2.2</span> Competitive exclusion</h3>
<p>Competitive exclusion means one species of a pair excludes the other to it’s (eventual) extinction. We classified competitive exclusion as “Stable” or “Incomplete” based on the following features:</p>
<section id="stable" class="level4" data-number="5.2.2.1">
<h4 data-number="5.2.2.1" class="anchored" data-anchor-id="stable"><span class="header-section-number">5.2.2.1</span> Stable</h4>
<p>Stable exclusion requires one species in a pair to go extinct (below 1% abundance detection limit threshold) in all 4 experiments performed for the two species. It also requires that the species undergoing extinction should always decrease over the 8 growth cycles by at least 5%. In practice this excludes samples where the extinct species started so rare that it was already effectively extinct.</p>
<p><span class="math display">\[
\begin{align*}
&amp; {f_{\{A, B\}}}_{t8} \leq f_{extinct} \\
&amp; \Delta f_{\{A, B\}} \leq -0.05 \\
\end{align*}
\]</span></p>
</section>
<section id="incomplete" class="level4" data-number="5.2.2.2">
<h4 data-number="5.2.2.2" class="anchored" data-anchor-id="incomplete"><span class="header-section-number">5.2.2.2</span> Incomplete</h4>
<p>Incomplete exclusion results if one of the species at cycle 8 is not extinct but its frequency significantly decreased over 8 growth cycles in all four experiments (Wilcoxon test, P&lt;0.05, N=2000 in each experiment).</p>
<p><span class="math display">\[
\begin{align*}
&amp; {f_{\{A, B\}}}_{t8} &gt; f_{extinct} \\
&amp; \Delta f_{\{A, B\}} \leq -0.05, \space p &lt; 0.05 \\
\end{align*}
\]</span></p>
</section>
</section>
<section id="coexistence" class="level3" data-number="5.2.3">
<h3 data-number="5.2.3" class="anchored" data-anchor-id="coexistence"><span class="header-section-number">5.2.3</span> Coexistence</h3>
<p>We classified a pair as coexisting if both species are able to invade each other from low frequency. We classified competitive exclusion as “Stable” or “Incomplete” based on differing levels of supporting evidence.</p>
<section id="stable-1" class="level4" data-number="5.2.3.1">
<h4 data-number="5.2.3.1" class="anchored" data-anchor-id="stable-1"><span class="header-section-number">5.2.3.1</span> Stable</h4>
<p>We defined “Stable Coexistence” as those species pairs where each species could invade the other from rare (fulfillment of the mutual invasibility criterion). table coexistence results if each species exists outside the extinction threshold and over 8 growth cycles and could significantly increase (Wilcoxon test, P&lt;0.05, N=2000 in each experiment)by at least 5% from rare in both replicates.</p>
<p><span class="math display">\[
\begin{align*}
&amp; {f_{\{A, B\}}^{Rare}}_{t8} &gt; f_{extinct} \\
&amp; \Delta f_{\{A, B\}}^{Rare} \geq 0.05, \space p &lt; 0.05 \\
\end{align*}
\]</span></p>
</section>
<section id="incomplete-1" class="level4" data-number="5.2.3.2">
<h4 data-number="5.2.3.2" class="anchored" data-anchor-id="incomplete-1"><span class="header-section-number">5.2.3.2</span> Incomplete</h4>
<p>We defined “Incomplete coexistence” as those species pairs that were found together after 8 growth cycles but no condition was placed on the significance of the change of each species from T0 to T8. For example, if a species started from low abundance and after 8 growth cycles also ended at low abundance that was not significantly different from the starting abundance.</p>
<p><span class="math display">\[
\begin{align*}
&amp; {f_{\{A, B\}}}_{t8} &gt; f_{extinct} \\
\end{align*}
\]</span></p>
</section>
<section id="inconclusive-outcomes" class="level4" data-number="5.2.3.3">
<h4 data-number="5.2.3.3" class="anchored" data-anchor-id="inconclusive-outcomes"><span class="header-section-number">5.2.3.3</span> Inconclusive outcomes</h4>
<p>This category included pairs that did not meet the requirements to be classified in any of the above described categories.</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1"></a><span class="co"># upper threshold for deciding whether a species goes extinct</span></span>
<span id="cb7-2"><a href="#cb7-2"></a>upper <span class="ot">&lt;-</span> <span class="fl">0.99</span></span>
<span id="cb7-3"><a href="#cb7-3"></a><span class="co"># lower threshold for deciding whether a species goes extinct</span></span>
<span id="cb7-4"><a href="#cb7-4"></a>lower <span class="ot">&lt;-</span> <span class="fl">0.01</span></span>
<span id="cb7-5"><a href="#cb7-5"></a><span class="co"># slope change threshold for deciding whether change from T0 to T8 is important (e.g. requires at least a 5% change of the focal species) </span></span>
<span id="cb7-6"><a href="#cb7-6"></a>slope_thresh <span class="ot">&lt;-</span> <span class="fl">0.05</span></span>
<span id="cb7-7"><a href="#cb7-7"></a></span>
<span id="cb7-8"><a href="#cb7-8"></a>outcomes_classified <span class="ot">&lt;-</span> wc_test <span class="sc">%&gt;%</span> </span>
<span id="cb7-9"><a href="#cb7-9"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(strep_conc, sp_1, sp_2) <span class="sc">%&gt;%</span> </span>
<span id="cb7-10"><a href="#cb7-10"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">outcome =</span> dplyr<span class="sc">::</span><span class="fu">case_when</span>(<span class="co"># Bistability: Requires sp_1 to both increase from T0 high freq and decrease from T0 low freq</span></span>
<span id="cb7-11"><a href="#cb7-11"></a>                             <span class="fu">sum</span>(change <span class="sc">&lt;</span> <span class="dv">0</span> <span class="sc">&amp;</span> f_1_0 <span class="sc">==</span> <span class="fu">min</span>(f_1_0), change <span class="sc">&gt;</span> <span class="dv">0</span> <span class="sc">&amp;</span> f_1_0 <span class="sc">==</span> <span class="fu">max</span>(f_1_0)) <span class="sc">==</span> <span class="dv">2</span> <span class="sc">~</span> <span class="st">"bistable_stable"</span>,</span>
<span id="cb7-12"><a href="#cb7-12"></a>                             <span class="co"># Exclusion: sp_1 increases from both low and high T0 freqs and T8 freq is &gt; than the T0 high freq</span></span>
<span id="cb7-13"><a href="#cb7-13"></a>                             <span class="fu">sum</span>(change) <span class="sc">==</span> <span class="dv">2</span> <span class="sc">&amp;</span> f_1_8 <span class="sc">&gt;=</span> upper <span class="sc">&amp;</span> <span class="fu">abs</span>(delta_f_1) <span class="sc">&gt;</span> slope_thresh <span class="sc">~</span> <span class="st">"exclusion_stable"</span>, </span>
<span id="cb7-14"><a href="#cb7-14"></a>                             <span class="co"># Exclusion: sp_1 decreases from both low and high T0 freqs and T8 freq is &lt; than the T0 low freq</span></span>
<span id="cb7-15"><a href="#cb7-15"></a>                             <span class="fu">sum</span>(change) <span class="sc">==</span> <span class="sc">-</span><span class="dv">2</span> <span class="sc">&amp;</span> f_1_8 <span class="sc">&lt;=</span> lower <span class="sc">&amp;</span> <span class="fu">abs</span>(delta_f_1) <span class="sc">&gt;</span> slope_thresh <span class="sc">~</span> <span class="st">"exclusion_stable"</span>,</span>
<span id="cb7-16"><a href="#cb7-16"></a>                             <span class="co"># Coexistence: sp_1 significantly increases from rare and significantly decreases from abundant</span></span>
<span id="cb7-17"><a href="#cb7-17"></a>                             <span class="fu">sum</span>(change) <span class="sc">==</span> <span class="dv">0</span> <span class="sc">&amp;</span> dplyr<span class="sc">::</span><span class="fu">if_else</span>(f_1_0 <span class="sc">==</span> <span class="fu">pmin</span>(f_1_0, f_2_0), f_1_8 <span class="sc">&gt;</span> f_1_0, f_1_8 <span class="sc">&lt;</span> f_1_0) <span class="sc">~</span> <span class="st">"coexistence_stable"</span>,</span>
<span id="cb7-18"><a href="#cb7-18"></a>                             <span class="co"># Incomplete coexistence: sp_1 is found within extinction thresholds in both experiments, but no condition is</span></span>
<span id="cb7-19"><a href="#cb7-19"></a>                             <span class="co"># placed on the significance of the change from T0 to T8</span></span>
<span id="cb7-20"><a href="#cb7-20"></a>                             <span class="fu">sum</span>(dplyr<span class="sc">::</span><span class="fu">between</span>(f_1_8, lower, upper)) <span class="sc">==</span> <span class="dv">2</span> <span class="sc">~</span> <span class="st">"coexistence_incomplete"</span>,</span>
<span id="cb7-21"><a href="#cb7-21"></a>                             <span class="co"># Incomplete exclusion: sp_1 is outside of the extinction threshold (extinct) for at least one sample and within the the </span></span>
<span id="cb7-22"><a href="#cb7-22"></a>                             <span class="co"># threshold for the other sample.</span></span>
<span id="cb7-23"><a href="#cb7-23"></a>                             <span class="fu">sum</span>(<span class="sc">!</span>dplyr<span class="sc">::</span><span class="fu">between</span>(f_1_8, lower, upper)) <span class="sc">==</span> <span class="dv">1</span> <span class="sc">&amp;</span> <span class="fu">sum</span>(dplyr<span class="sc">::</span><span class="fu">between</span>(f_1_8, lower, upper)) <span class="sc">==</span> <span class="dv">1</span> <span class="sc">~</span> <span class="st">"exclusion_incomplete"</span>,</span>
<span id="cb7-24"><a href="#cb7-24"></a>                             <span class="co"># Inconclusive: samples failing to meet any of the above rules</span></span>
<span id="cb7-25"><a href="#cb7-25"></a>                             <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"inconclusive_"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb7-26"><a href="#cb7-26"></a>  dplyr<span class="sc">::</span><span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb7-27"><a href="#cb7-27"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">type =</span> dplyr<span class="sc">::</span><span class="fu">if_else</span>(stringr<span class="sc">::</span><span class="fu">str_detect</span>(outcome, <span class="st">"incomplete"</span>), <span class="st">"incomplete"</span>, <span class="st">"stable"</span>),</span>
<span id="cb7-28"><a href="#cb7-28"></a>         <span class="at">outcome =</span> stringr<span class="sc">::</span><span class="fu">str_extract</span>(outcome, <span class="st">"(^.+)_.*"</span>, <span class="at">group =</span> <span class="dv">1</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb7-29"><a href="#cb7-29"></a>  dplyr<span class="sc">::</span><span class="fu">relocate</span>(outcome, type, change)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</section>
</section>
<section id="plotting-pairwise-outpcomes" class="level2" data-number="5.3">
<h2 data-number="5.3" class="anchored" data-anchor-id="plotting-pairwise-outpcomes"><span class="header-section-number">5.3</span> Plotting pairwise outpcomes</h2>
<p>Construct final dataframe to be used for plotting</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1"></a>samp_pairs_fmt <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(samp_pairs_fmt_mp, samp_pairs_fmt_exp) <span class="sc">%&gt;%</span></span>
<span id="cb8-2"><a href="#cb8-2"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb8-3"><a href="#cb8-3"></a>    <span class="at">group =</span> <span class="fu">interaction</span>(community_id, strep_conc),</span>
<span id="cb8-4"><a href="#cb8-4"></a>    <span class="at">evo_group =</span> dplyr<span class="sc">::</span><span class="fu">case_when</span>(</span>
<span id="cb8-5"><a href="#cb8-5"></a>      dplyr<span class="sc">::</span><span class="fu">if_all</span>(<span class="fu">c</span>(sp_1, sp_2), \(x) stringr<span class="sc">::</span><span class="fu">str_detect</span>(x, <span class="st">"ANC"</span>)) <span class="sc">~</span> <span class="st">"both_anc"</span>,</span>
<span id="cb8-6"><a href="#cb8-6"></a>      dplyr<span class="sc">::</span><span class="fu">if_all</span>(<span class="fu">c</span>(sp_1, sp_2), \(x) stringr<span class="sc">::</span><span class="fu">str_detect</span>(x, <span class="st">"EVO"</span>)) <span class="sc">~</span> <span class="st">"both_evo"</span>,</span>
<span id="cb8-7"><a href="#cb8-7"></a>      <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"mix"</span></span>
<span id="cb8-8"><a href="#cb8-8"></a>    )</span>
<span id="cb8-9"><a href="#cb8-9"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb8-10"><a href="#cb8-10"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(outcomes_classified,</span>
<span id="cb8-11"><a href="#cb8-11"></a>                   <span class="at">by =</span> dplyr<span class="sc">::</span><span class="fu">join_by</span>(community_id, sp_1, sp_2, strep_conc)) <span class="sc">%&gt;%</span></span>
<span id="cb8-12"><a href="#cb8-12"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb8-13"><a href="#cb8-13"></a>    <span class="at">outcome =</span> <span class="fu">factor</span>(outcome, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"exclusion"</span>, <span class="st">"coexistence"</span>, </span>
<span id="cb8-14"><a href="#cb8-14"></a>                                         <span class="st">"bistable"</span>, <span class="st">"inconclusive"</span>)),</span>
<span id="cb8-15"><a href="#cb8-15"></a>    <span class="at">type =</span> <span class="fu">factor</span>(type, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"stable"</span>, <span class="st">"incomplete"</span>))</span>
<span id="cb8-16"><a href="#cb8-16"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Create different lists of plots for the mixed (i.e.&nbsp;evo competed against anc) conditions</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1"></a>samp_pairs_fmt_plots_split_a <span class="ot">&lt;-</span> samp_pairs_fmt <span class="sc">%&gt;%</span> </span>
<span id="cb9-2"><a href="#cb9-2"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(evo_group <span class="sc">!=</span> <span class="st">"mix"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb9-3"><a href="#cb9-3"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(strep_conc, evo_group) <span class="sc">%&gt;%</span> </span>
<span id="cb9-4"><a href="#cb9-4"></a>  dplyr<span class="sc">::</span><span class="fu">group_split</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb9-5"><a href="#cb9-5"></a>  purrr<span class="sc">::</span><span class="fu">map</span>(pair_plot)</span>
<span id="cb9-6"><a href="#cb9-6"></a></span>
<span id="cb9-7"><a href="#cb9-7"></a>samp_pairs_fmt_plots_split_b <span class="ot">&lt;-</span> samp_pairs_fmt <span class="sc">%&gt;%</span> </span>
<span id="cb9-8"><a href="#cb9-8"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(evo_group <span class="sc">==</span> <span class="st">"mix"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb9-9"><a href="#cb9-9"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(strep_conc) <span class="sc">%&gt;%</span> </span>
<span id="cb9-10"><a href="#cb9-10"></a>  dplyr<span class="sc">::</span><span class="fu">group_split</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb9-11"><a href="#cb9-11"></a>  purrr<span class="sc">::</span><span class="fu">map</span>(pair_plot)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1"></a>fig01 <span class="ot">&lt;-</span> patchwork<span class="sc">::</span><span class="fu">wrap_plots</span>(samp_pairs_fmt_plots_split_a, <span class="at">ncol =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb10-2"><a href="#cb10-2"></a>  patchwork<span class="sc">::</span><span class="fu">plot_annotation</span>(<span class="at">tag_levels =</span> <span class="st">"A"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1"></a><span class="fu">ggsave</span>(</span>
<span id="cb11-2"><a href="#cb11-2"></a>  here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"figs"</span>, <span class="st">"coexistence_pairs.svg"</span>),</span>
<span id="cb11-3"><a href="#cb11-3"></a>  fig01,</span>
<span id="cb11-4"><a href="#cb11-4"></a>  <span class="at">width =</span> <span class="dv">7</span>,</span>
<span id="cb11-5"><a href="#cb11-5"></a>  <span class="at">height =</span> <span class="dv">12</span>,</span>
<span id="cb11-6"><a href="#cb11-6"></a>  <span class="at">units =</span> <span class="st">"in"</span>,</span>
<span id="cb11-7"><a href="#cb11-7"></a>  <span class="at">device =</span> <span class="st">"svg"</span></span>
<span id="cb11-8"><a href="#cb11-8"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div id="fig-01" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-01-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="02_analysis_pairs_files/figure-html/unnamed-chunk-12-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-01-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;1: Outcomes from pairwise cocultures of ancestral (left, A:G) and streptomycin adapted (right, B:H) 0403, 1287, 1896 and 1977 species. Rows in the grid represent differen streptomycin concentrations applied (A:B = 0 µg/ml, C:D = 16 µg/ml, E:F = 64 µg/ml, G:H = 256 µg/ml). Using the rules defined above, red lines show cocultures resulting in exclusion of one of the species, blue lines show cocultures that resulted in coexistence, and purple lines represent inconclusive outcomes due to lack of data or exception to therules. Solid lines show a “definite” outcome and dashed lines show a potentially “incomplete” outcome. Two solid lines of the same color indicate stable coexistence or stable exclusion, a solid and dashed line indicate incomplete coexistence/exclusion, while two dashed lines indicates coexistence without evidence of mutual invasibility. Note some statistical noise has been applied to point positions to prevent overlaps in the plot and aid in visualization. <strong>NOTE:</strong> that in panel E (64 µg/ml streptomycin) both replicates survived with OD ~ 0.15 when ANC_0403 was started from 90% abundance but not when it was started at 10% abundance (OD ~ 0.04). In all replicates of 90%/10% starting abundance ANC_1977 was driven extinct.
</figcaption>
</figure>
</div>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1"></a>fig02 <span class="ot">&lt;-</span> patchwork<span class="sc">::</span><span class="fu">wrap_plots</span>(samp_pairs_fmt_plots_split_b, <span class="at">ncol =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb12-2"><a href="#cb12-2"></a>  patchwork<span class="sc">::</span><span class="fu">plot_annotation</span>(<span class="at">tag_levels =</span> <span class="st">"A"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1"></a><span class="fu">ggsave</span>(</span>
<span id="cb13-2"><a href="#cb13-2"></a>  here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"figs"</span>, <span class="st">"coexistence_pairs_mixed_hist.svg"</span>),</span>
<span id="cb13-3"><a href="#cb13-3"></a>  fig02,</span>
<span id="cb13-4"><a href="#cb13-4"></a>  <span class="at">width =</span> <span class="dv">7</span>,</span>
<span id="cb13-5"><a href="#cb13-5"></a>  <span class="at">height =</span> <span class="dv">7</span>,</span>
<span id="cb13-6"><a href="#cb13-6"></a>  <span class="at">units =</span> <span class="st">"in"</span>,</span>
<span id="cb13-7"><a href="#cb13-7"></a>  <span class="at">device =</span> <span class="st">"svg"</span></span>
<span id="cb13-8"><a href="#cb13-8"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div id="fig-02" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-02-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="02_analysis_pairs_files/figure-html/unnamed-chunk-15-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-02-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;2: Results from co-cultures of mixed ancestal and evolved combinations. Line colors and types are as in <a href="#fig-01" class="quarto-xref">Figure&nbsp;1</a>, with the addition of green showing bistability (i.e., Species A and Species B exclude each other when invading from high abundance, and each are excluded by the other when invading from low abundance).
</figcaption>
</figure>
</div>
</section>
</section>
<section id="network" class="level1" data-number="6">
<h1 data-number="6"><span class="header-section-number">6</span> Network</h1>
<p>Here we will plot the pairwise competition outcomes as a network. <strong>NOTE:</strong> I don’t have time/energy to figure out how to manage all the links in the networks so that they line up properly and don’t overlap so there will need to be some postprocessing in inkscape to move some of the links so they don’t overlap.</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1"></a><span class="fu">library</span>(ggraph)</span>
<span id="cb14-2"><a href="#cb14-2"></a><span class="fu">library</span>(tidygraph)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="evolution-and-streptomycin-categories-separate" class="level2" data-number="6.1">
<h2 data-number="6.1" class="anchored" data-anchor-id="evolution-and-streptomycin-categories-separate"><span class="header-section-number">6.1</span> Evolution and Streptomycin categories separate</h2>
<p>Here we plot a separate graph for each streptomycin concentrations and also by different evolutionary groupings. For example, there is one graph for the competition outcomes of only ancestral species, there is one graph of the outcomes of only evolved species, and there is one graph for the outcomes of mixed competitions where an ancestral species competes against an evolved species.</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1"></a>nodes1 <span class="ot">&lt;-</span> <span class="fu">make_nodes</span>(samp_pairs_fmt, evo_group, strep_conc)</span>
<span id="cb15-2"><a href="#cb15-2"></a>edges1 <span class="ot">&lt;-</span> <span class="fu">make_edges</span>(samp_pairs_fmt, evo_group, strep_conc)</span>
<span id="cb15-3"><a href="#cb15-3"></a></span>
<span id="cb15-4"><a href="#cb15-4"></a>graphs1 <span class="ot">&lt;-</span> <span class="fu">nest</span>(nodes1, <span class="at">sps =</span> <span class="sc">-</span><span class="fu">c</span>(evo_group, strep_conc)) <span class="sc">%&gt;%</span> </span>
<span id="cb15-5"><a href="#cb15-5"></a>  <span class="fu">left_join</span>(<span class="fu">nest</span>(edges1, <span class="at">pairs =</span> <span class="sc">-</span><span class="fu">c</span>(evo_group, strep_conc)),</span>
<span id="cb15-6"><a href="#cb15-6"></a>            <span class="at">by =</span> <span class="fu">join_by</span>(evo_group, strep_conc)) <span class="sc">%&gt;%</span> </span>
<span id="cb15-7"><a href="#cb15-7"></a>  <span class="fu">mutate</span>(<span class="at">network =</span> <span class="fu">map2</span>(sps, pairs, <span class="cf">function</span>(sps, pairs) <span class="fu">tbl_graph</span>(<span class="at">nodes =</span> sps, <span class="at">edges =</span> pairs, <span class="at">directed =</span> T))) <span class="sc">%&gt;%</span> </span>
<span id="cb15-8"><a href="#cb15-8"></a>  <span class="fu">mutate</span>(<span class="at">plot =</span> <span class="fu">map</span>(network, <span class="cf">function</span>(network) <span class="fu">plot_network_hierarchy</span>(network, <span class="at">tune_angle =</span> <span class="fl">1.5</span>, <span class="at">n_rank =</span> <span class="dv">7</span>, <span class="at">n_break =</span> <span class="dv">7</span>)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1"></a>fig03 <span class="ot">&lt;-</span> patchwork<span class="sc">::</span><span class="fu">wrap_plots</span>(graphs1[[<span class="dv">6</span>]], <span class="at">nrow =</span> <span class="dv">3</span>, <span class="at">guides=</span> <span class="st">"collect"</span>) <span class="sc">+</span></span>
<span id="cb16-2"><a href="#cb16-2"></a>  patchwork<span class="sc">::</span><span class="fu">plot_annotation</span>(<span class="at">tag_levels =</span> <span class="st">"A"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1"></a><span class="fu">ggsave</span>(</span>
<span id="cb17-2"><a href="#cb17-2"></a>  here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"figs"</span>, <span class="st">"coexistence_networks_nested_evo_strep.svg"</span>),</span>
<span id="cb17-3"><a href="#cb17-3"></a>  fig03,</span>
<span id="cb17-4"><a href="#cb17-4"></a>  <span class="at">width =</span> <span class="dv">8</span>,</span>
<span id="cb17-5"><a href="#cb17-5"></a>  <span class="at">height =</span> <span class="dv">10</span>,</span>
<span id="cb17-6"><a href="#cb17-6"></a>  <span class="at">units =</span> <span class="st">"in"</span>,</span>
<span id="cb17-7"><a href="#cb17-7"></a>  <span class="at">device =</span> <span class="st">"svg"</span></span>
<span id="cb17-8"><a href="#cb17-8"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div id="fig-03" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-03-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="02_analysis_pairs_files/figure-html/unnamed-chunk-20-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-03-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3: Competitive hierarchy of species pairs separated by evolution grouping (only ancestral pairs, only evolved pairs, and mixed ancestral and evolved pairs) and streptomycin concentration. Subplots A-D are for only ancestral pairs, E-H for only evolved pairs, and I-L for mixed ancestral and evolved pairs. Subplots A, E, I show experiments under no streptomycin, B, F, J 16 µg/ml streptomycin, C, G, K for 64 µg/ml streptomycin, and D, H, L for 256 µg/ml streptomycin. For each evolution/streptomycin grouping, strains are rank ordered on the basis of the number of other strains they exclude, based on data shown in <a href="#fig-01" class="quarto-xref">Figure&nbsp;1</a> and <a href="#fig-02" class="quarto-xref">Figure&nbsp;2</a>. Grey nodes represent strains (denoted by text), red arrows point from winning strain to losing strain, blue arrows indicate coexistence (ignore the arrow heads for blue, coulnd’t figure out how to remove them for only a subset of the edges), and green arrows indicate the presence of bistability (i.e., Species A and Species B exclude each other when invading from high abundance, and each are excluded by the other when invading from low abundance). Line type indicates the degree of certainty of the outcome in each pair: dashed lines represent incomplete exclusion or coexistence without evidence of mutual invasibility, while solid lines indicate complete exclusion or stable coexistence (mutual invasibility).
</figcaption>
</figure>
</div>
</section>
<section id="only-streptomycin-category-separate" class="level2" data-number="6.2">
<h2 data-number="6.2" class="anchored" data-anchor-id="only-streptomycin-category-separate"><span class="header-section-number">6.2</span> Only Streptomycin category separate</h2>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1"></a>nodes2 <span class="ot">&lt;-</span> <span class="fu">make_nodes</span>(samp_pairs_fmt, strep_conc)</span>
<span id="cb18-2"><a href="#cb18-2"></a>edges2 <span class="ot">&lt;-</span> <span class="fu">make_edges</span>(samp_pairs_fmt, strep_conc)</span>
<span id="cb18-3"><a href="#cb18-3"></a></span>
<span id="cb18-4"><a href="#cb18-4"></a>graphs2 <span class="ot">&lt;-</span> <span class="fu">nest</span>(nodes2, <span class="at">sps =</span> <span class="sc">-</span><span class="fu">c</span>(strep_conc)) <span class="sc">%&gt;%</span> </span>
<span id="cb18-5"><a href="#cb18-5"></a>  <span class="fu">left_join</span>(<span class="fu">nest</span>(edges2, <span class="at">pairs =</span> <span class="sc">-</span><span class="fu">c</span>(strep_conc)),</span>
<span id="cb18-6"><a href="#cb18-6"></a>            <span class="at">by =</span> <span class="fu">join_by</span>(strep_conc)) <span class="sc">%&gt;%</span> </span>
<span id="cb18-7"><a href="#cb18-7"></a>  <span class="fu">mutate</span>(<span class="at">network =</span> <span class="fu">map2</span>(sps, pairs, <span class="cf">function</span>(sps, pairs) <span class="fu">tbl_graph</span>(<span class="at">nodes =</span> sps, <span class="at">edges =</span> pairs, <span class="at">directed =</span> T))) <span class="sc">%&gt;%</span> </span>
<span id="cb18-8"><a href="#cb18-8"></a>  <span class="fu">mutate</span>(<span class="at">plot =</span> <span class="fu">map</span>(network, <span class="cf">function</span>(network) <span class="fu">plot_network_hierarchy</span>(network, <span class="at">tune_angle =</span> <span class="fl">1.5</span>, <span class="at">n_rank =</span> <span class="dv">7</span>, <span class="at">n_break =</span> <span class="dv">7</span>)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1"></a>fig04 <span class="ot">&lt;-</span> patchwork<span class="sc">::</span><span class="fu">wrap_plots</span>(graphs2[[<span class="dv">5</span>]], <span class="at">nrow =</span> <span class="dv">1</span>, <span class="at">guides=</span> <span class="st">"collect"</span>) <span class="sc">+</span></span>
<span id="cb19-2"><a href="#cb19-2"></a>  patchwork<span class="sc">::</span><span class="fu">plot_annotation</span>(<span class="at">tag_levels =</span> <span class="st">"A"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1"></a><span class="fu">ggsave</span>(</span>
<span id="cb20-2"><a href="#cb20-2"></a>  here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"figs"</span>, <span class="st">"coexistence_networks_nested_strep.svg"</span>),</span>
<span id="cb20-3"><a href="#cb20-3"></a>  fig04,</span>
<span id="cb20-4"><a href="#cb20-4"></a>  <span class="at">width =</span> <span class="dv">9</span>,</span>
<span id="cb20-5"><a href="#cb20-5"></a>  <span class="at">height =</span> <span class="dv">4</span>,</span>
<span id="cb20-6"><a href="#cb20-6"></a>  <span class="at">units =</span> <span class="st">"in"</span>,</span>
<span id="cb20-7"><a href="#cb20-7"></a>  <span class="at">device =</span> <span class="st">"svg"</span></span>
<span id="cb20-8"><a href="#cb20-8"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div id="fig-04" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-04-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="02_analysis_pairs_files/figure-html/unnamed-chunk-24-1.png" class="img-fluid figure-img" width="864"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-04-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;4: Competitive hierarchy of species pairs separated by streptomycin concentration. Subplot A shows experiments under no streptomycin, B with 16 µg/ml streptomycin, C with 64 µg/ml streptomycin, and D with 256 µg/ml streptomycin. For each evolution/streptomycin grouping, strains are rank ordered on the basis of the number of other strains they exclude, based on data shown in <a href="#fig-01" class="quarto-xref">Figure&nbsp;1</a> and <a href="#fig-02" class="quarto-xref">Figure&nbsp;2</a>. Grey nodes represent strains (denoted by text), red arrows point from winning strain to losing strain, blue arrows indicate coexistence (ignore the arrow heads for blue, coulnd’t figure out how to remove them for only a subset of the edges), and green arrows indicate the presence of bistability (i.e., Species A and Species B exclude each other when invading from high abundance, and each are excluded by the other when invading from low abundance). Line type indicates the degree of certainty of the outcome in each pair: dashed lines represent incomplete exclusion or coexistence without evidence of mutual invasibility, while solid lines indicate complete exclusion or stable coexistence (mutual invasibility).
</figcaption>
</figure>
</div>


<!-- -->

</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb21" data-shortcodes="false"><pre class="sourceCode numberSource markdown number-lines code-with-copy"><code class="sourceCode markdown"><span id="cb21-1"><a href="#cb21-1"></a><span class="co">---</span></span>
<span id="cb21-2"><a href="#cb21-2"></a><span class="an">title:</span><span class="co"> "Analysis of pairwise competition"</span></span>
<span id="cb21-3"><a href="#cb21-3"></a><span class="an">author:</span><span class="co"> "Shane Hogle"</span></span>
<span id="cb21-4"><a href="#cb21-4"></a><span class="an">date:</span><span class="co"> today</span></span>
<span id="cb21-5"><a href="#cb21-5"></a><span class="an">link-citations:</span><span class="co"> true</span></span>
<span id="cb21-6"><a href="#cb21-6"></a><span class="an">abstract:</span><span class="co"> "Here we analyze the community outpcomes from all competing species pairs."</span></span>
<span id="cb21-7"><a href="#cb21-7"></a><span class="co">---</span></span>
<span id="cb21-8"><a href="#cb21-8"></a></span>
<span id="cb21-9"><a href="#cb21-9"></a><span class="fu"># Introduction</span></span>
<span id="cb21-10"><a href="#cb21-10"></a></span>
<span id="cb21-11"><a href="#cb21-11"></a>Contains results from pairs of all streptomycin concentrations and trios for 0 streptomycin</span>
<span id="cb21-12"><a href="#cb21-12"></a></span>
<span id="cb21-13"><a href="#cb21-13"></a><span class="fu"># Setup</span></span>
<span id="cb21-14"><a href="#cb21-14"></a></span>
<span id="cb21-15"><a href="#cb21-15"></a><span class="fu">## Libraries</span></span>
<span id="cb21-16"><a href="#cb21-16"></a></span>
<span id="cb21-19"><a href="#cb21-19"></a><span class="in">```{r}</span></span>
<span id="cb21-20"><a href="#cb21-20"></a><span class="co">#| output: false</span></span>
<span id="cb21-21"><a href="#cb21-21"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb21-22"><a href="#cb21-22"></a><span class="fu">library</span>(here)</span>
<span id="cb21-23"><a href="#cb21-23"></a><span class="fu">library</span>(fs)</span>
<span id="cb21-24"><a href="#cb21-24"></a><span class="fu">library</span>(scales)</span>
<span id="cb21-25"><a href="#cb21-25"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb21-26"><a href="#cb21-26"></a><span class="fu">library</span>(broom)</span>
<span id="cb21-27"><a href="#cb21-27"></a><span class="fu">library</span>(ggraph)</span>
<span id="cb21-28"><a href="#cb21-28"></a><span class="fu">library</span>(tidygraph)</span>
<span id="cb21-29"><a href="#cb21-29"></a><span class="fu">source</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"R"</span>, <span class="st">"utils_generic.R"</span>))</span>
<span id="cb21-30"><a href="#cb21-30"></a><span class="fu">source</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"R"</span>, <span class="st">"communities"</span>, <span class="st">"amplicon"</span>, <span class="st">"utils_amplicon.R"</span>))</span>
<span id="cb21-31"><a href="#cb21-31"></a><span class="in">```</span></span>
<span id="cb21-32"><a href="#cb21-32"></a></span>
<span id="cb21-33"><a href="#cb21-33"></a><span class="fu">## Global variables</span></span>
<span id="cb21-34"><a href="#cb21-34"></a></span>
<span id="cb21-37"><a href="#cb21-37"></a><span class="in">```{r}</span></span>
<span id="cb21-38"><a href="#cb21-38"></a>data <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"communities"</span>)</span>
<span id="cb21-39"><a href="#cb21-39"></a><span class="co"># make processed data directory if it doesn't exist</span></span>
<span id="cb21-40"><a href="#cb21-40"></a>fs<span class="sc">::</span><span class="fu">dir_create</span>(data)</span>
<span id="cb21-41"><a href="#cb21-41"></a><span class="in">```</span></span>
<span id="cb21-42"><a href="#cb21-42"></a></span>
<span id="cb21-43"><a href="#cb21-43"></a><span class="fu"># Read data</span></span>
<span id="cb21-44"><a href="#cb21-44"></a></span>
<span id="cb21-45"><a href="#cb21-45"></a><span class="fu">## Species abundances</span></span>
<span id="cb21-46"><a href="#cb21-46"></a></span>
<span id="cb21-49"><a href="#cb21-49"></a><span class="in">```{r}</span></span>
<span id="cb21-50"><a href="#cb21-50"></a><span class="co">#| output: false</span></span>
<span id="cb21-51"><a href="#cb21-51"></a>samp_pairs <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(data, <span class="st">"2sps_compiled.tsv"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb21-52"><a href="#cb21-52"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">f =</span> f_thresh)</span>
<span id="cb21-53"><a href="#cb21-53"></a><span class="in">```</span></span>
<span id="cb21-54"><a href="#cb21-54"></a></span>
<span id="cb21-55"><a href="#cb21-55"></a><span class="fu"># Formatting</span></span>
<span id="cb21-56"><a href="#cb21-56"></a></span>
<span id="cb21-57"><a href="#cb21-57"></a><span class="fu">## Transfer 0 (starting proportions)</span></span>
<span id="cb21-58"><a href="#cb21-58"></a></span>
<span id="cb21-59"><a href="#cb21-59"></a>First get abundances from transfer 0 (masterplate)</span>
<span id="cb21-60"><a href="#cb21-60"></a></span>
<span id="cb21-63"><a href="#cb21-63"></a><span class="in">```{r}</span></span>
<span id="cb21-64"><a href="#cb21-64"></a>samp_pairs_fmt_mp <span class="ot">&lt;-</span> samp_pairs <span class="sc">%&gt;%</span> </span>
<span id="cb21-65"><a href="#cb21-65"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(community_type <span class="sc">==</span> <span class="st">"masterplate"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb21-66"><a href="#cb21-66"></a>  <span class="co"># make a combined evolution and species identifier and extract the community ID</span></span>
<span id="cb21-67"><a href="#cb21-67"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">sp =</span> <span class="fu">paste</span>(<span class="fu">str_to_upper</span>(evo_hist), <span class="fu">str_extract</span>(strainID, <span class="st">"</span><span class="sc">\\</span><span class="st">d+"</span>), <span class="at">sep =</span> <span class="st">"_"</span>),</span>
<span id="cb21-68"><a href="#cb21-68"></a>         <span class="at">community_id =</span> <span class="fu">str_extract</span>(sample, <span class="st">"P</span><span class="sc">\\</span><span class="st">d</span><span class="sc">\\</span><span class="st">d"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb21-69"><a href="#cb21-69"></a>  <span class="co"># this step is important to ensure that when dfs are pivoted wider the </span></span>
<span id="cb21-70"><a href="#cb21-70"></a>  <span class="co"># sp_1 and sp_2 stay consistent </span></span>
<span id="cb21-71"><a href="#cb21-71"></a>  dplyr<span class="sc">::</span><span class="fu">arrange</span>(community_id, sp) <span class="sc">%&gt;%</span></span>
<span id="cb21-72"><a href="#cb21-72"></a>  <span class="co"># this creates an index for each species present in each community, it is needed</span></span>
<span id="cb21-73"><a href="#cb21-73"></a>  <span class="co"># for the pivot to be consistent between the master plate and the samples</span></span>
<span id="cb21-74"><a href="#cb21-74"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(community_id) <span class="sc">%&gt;%</span> </span>
<span id="cb21-75"><a href="#cb21-75"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">n =</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">n</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb21-76"><a href="#cb21-76"></a>  dplyr<span class="sc">::</span><span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb21-77"><a href="#cb21-77"></a>  tidyr<span class="sc">::</span><span class="fu">pivot_wider</span>(<span class="at">id_cols =</span> community_id, <span class="at">values_from =</span> <span class="fu">c</span>(sp, f), <span class="at">names_from =</span> n) <span class="sc">%&gt;%</span> </span>
<span id="cb21-78"><a href="#cb21-78"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">transfer =</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb21-79"><a href="#cb21-79"></a>  tidyr<span class="sc">::</span><span class="fu">expand_grid</span>(<span class="at">strep_conc =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">16</span>, <span class="dv">64</span>, <span class="dv">256</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb21-80"><a href="#cb21-80"></a>  <span class="fu">arrange</span>(sp_1, sp_2, strep_conc)</span>
<span id="cb21-81"><a href="#cb21-81"></a><span class="in">```</span></span>
<span id="cb21-82"><a href="#cb21-82"></a></span>
<span id="cb21-83"><a href="#cb21-83"></a><span class="fu">## Transfer 8 (ending proportions)</span></span>
<span id="cb21-84"><a href="#cb21-84"></a></span>
<span id="cb21-85"><a href="#cb21-85"></a>Format abundances from the experiment, summarizing over replicates. Here we calculate the median frequency across biological replicates using <span class="in">`Hmisc::smedian.hilow`</span> which computes the sample median and the outer quantiles (0.025 and 0.975).</span>
<span id="cb21-86"><a href="#cb21-86"></a></span>
<span id="cb21-89"><a href="#cb21-89"></a><span class="in">```{r}</span></span>
<span id="cb21-90"><a href="#cb21-90"></a>samp_pairs_fmt_exp <span class="ot">&lt;-</span> samp_pairs <span class="sc">%&gt;%</span> </span>
<span id="cb21-91"><a href="#cb21-91"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(community_type <span class="sc">==</span> <span class="st">"experiment"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb21-92"><a href="#cb21-92"></a>  <span class="co"># make a combined evolution and species identifier and extract the community ID</span></span>
<span id="cb21-93"><a href="#cb21-93"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">sp =</span> <span class="fu">paste</span>(<span class="fu">str_to_upper</span>(evo_hist), <span class="fu">str_extract</span>(strainID, <span class="st">"</span><span class="sc">\\</span><span class="st">d+"</span>), <span class="at">sep =</span> <span class="st">"_"</span>),</span>
<span id="cb21-94"><a href="#cb21-94"></a>         <span class="at">community_id =</span> <span class="fu">str_extract</span>(sample, <span class="st">"P</span><span class="sc">\\</span><span class="st">d</span><span class="sc">\\</span><span class="st">d"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb21-95"><a href="#cb21-95"></a>  <span class="co"># calculate median and 95% CI across replicates</span></span>
<span id="cb21-96"><a href="#cb21-96"></a>  dplyr<span class="sc">::</span><span class="fu">summarize</span>(ggplot2<span class="sc">::</span><span class="fu">median_hilow</span>(f), <span class="at">.by =</span> <span class="fu">c</span>(<span class="st">"community_id"</span>, <span class="st">"sp"</span>, <span class="st">"strep_conc"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb21-97"><a href="#cb21-97"></a>  <span class="co"># rename the y columns as f for compatibility</span></span>
<span id="cb21-98"><a href="#cb21-98"></a>  dplyr<span class="sc">::</span><span class="fu">rename_with</span>(<span class="at">.cols =</span> <span class="fu">starts_with</span>(<span class="st">"y"</span>), \(x) <span class="fu">str_replace</span>(x, <span class="st">"y"</span>, <span class="st">"f"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb21-99"><a href="#cb21-99"></a>  <span class="co"># this step is important to ensure that when dfs are pivoted wider the </span></span>
<span id="cb21-100"><a href="#cb21-100"></a>  <span class="co"># sp_1 and sp_2 stay consistent </span></span>
<span id="cb21-101"><a href="#cb21-101"></a>  dplyr<span class="sc">::</span><span class="fu">arrange</span>(community_id, sp, strep_conc) <span class="sc">%&gt;%</span></span>
<span id="cb21-102"><a href="#cb21-102"></a>  <span class="co"># this creates an index for each species present in each community, it is needed</span></span>
<span id="cb21-103"><a href="#cb21-103"></a>  <span class="co"># for the pivot to be consistent between the master plate and the samples</span></span>
<span id="cb21-104"><a href="#cb21-104"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(community_id, strep_conc) <span class="sc">%&gt;%</span> </span>
<span id="cb21-105"><a href="#cb21-105"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">n =</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">n</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb21-106"><a href="#cb21-106"></a>  dplyr<span class="sc">::</span><span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb21-107"><a href="#cb21-107"></a>  tidyr<span class="sc">::</span><span class="fu">pivot_wider</span>(<span class="at">id_cols =</span> <span class="fu">c</span>(community_id, strep_conc), <span class="at">values_from =</span> <span class="fu">c</span>(sp, f, fmin, fmax), <span class="at">names_from =</span> n) <span class="sc">%&gt;%</span> </span>
<span id="cb21-108"><a href="#cb21-108"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">transfer =</span> <span class="dv">8</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb21-109"><a href="#cb21-109"></a>  <span class="fu">arrange</span>(sp_1, sp_2, strep_conc)</span>
<span id="cb21-110"><a href="#cb21-110"></a><span class="in">```</span></span>
<span id="cb21-111"><a href="#cb21-111"></a></span>
<span id="cb21-112"><a href="#cb21-112"></a><span class="fu"># Define competition outcomes</span></span>
<span id="cb21-113"><a href="#cb21-113"></a></span>
<span id="cb21-114"><a href="#cb21-114"></a><span class="fu">## Binomial sampling and Wilcox test</span></span>
<span id="cb21-115"><a href="#cb21-115"></a></span>
<span id="cb21-116"><a href="#cb21-116"></a>First need to determine which samples significantly decreased/increased from T0 to T8. We don't have enough biological replicates for to compute a statistic across replicates for <span class="co">[</span><span class="ot">outcome variability</span><span class="co">](https://www.pnas.org/doi/10.1073/pnas.2302491120)</span>. However, we can estimate the mean fraction of A and quantify the inferential uncertainty of the mean by bootstrap resampling. We the used median proportion of species A from two biological replicates of each T8 pair as the probability of success (i.e., drawing species A) from 1000 draws (i.e., sequencing reads) from the binomial distribution. To determine whether the frequency of Species A significantly changed from T0 to T8, the means of the 1000 binomial draws for T0 and T8 were compared using a Wilcoxon rank sum test (N = 2000). Tests with Bonferroni multiple test corrected p values &lt; 1e-5 were considered to represent significantly different T0 and T8 samples.  </span>
<span id="cb21-117"><a href="#cb21-117"></a></span>
<span id="cb21-120"><a href="#cb21-120"></a><span class="in">```{r}</span></span>
<span id="cb21-121"><a href="#cb21-121"></a><span class="fu">set.seed</span>(<span class="dv">124341</span>)</span>
<span id="cb21-122"><a href="#cb21-122"></a></span>
<span id="cb21-123"><a href="#cb21-123"></a><span class="co"># first join the T0 and T8 abundances</span></span>
<span id="cb21-124"><a href="#cb21-124"></a>wc_test <span class="ot">&lt;-</span> <span class="fu">left_join</span>(samp_pairs_fmt_exp, samp_pairs_fmt_mp, <span class="at">by =</span> <span class="fu">join_by</span>(community_id, strep_conc, sp_1, sp_2)) <span class="sc">%&gt;%</span> </span>
<span id="cb21-125"><a href="#cb21-125"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">delta_f_1 =</span> f_1.x <span class="sc">-</span> f_1.y) <span class="sc">%&gt;%</span> </span>
<span id="cb21-126"><a href="#cb21-126"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(community_id, strep_conc, sp_1, sp_2, delta_f_1, <span class="at">f_1_8 =</span> f_1.x, <span class="at">f_1_0 =</span> f_1.y, </span>
<span id="cb21-127"><a href="#cb21-127"></a>                <span class="at">f_2_8 =</span> f_2.x, <span class="at">f_2_0 =</span> f_2.y) <span class="sc">%&gt;%</span> </span>
<span id="cb21-128"><a href="#cb21-128"></a>  tidyr<span class="sc">::</span><span class="fu">nest</span>(<span class="at">data =</span> <span class="fu">c</span>(<span class="sc">-</span>community_id, <span class="sc">-</span>strep_conc)) <span class="sc">%&gt;%</span></span>
<span id="cb21-129"><a href="#cb21-129"></a>  <span class="co"># samples 1000 draws from binomial distribution using f_a median as the probability of success</span></span>
<span id="cb21-130"><a href="#cb21-130"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">f_1_0_rs =</span> purrr<span class="sc">::</span><span class="fu">map</span>(data, \(x) <span class="fu">map</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">100</span>, \(i) <span class="fu">sum</span>(<span class="fu">rbinom</span>(<span class="dv">1000</span>, <span class="dv">1</span>, x<span class="sc">$</span>f_1_0))<span class="sc">/</span><span class="dv">1000</span>)),</span>
<span id="cb21-131"><a href="#cb21-131"></a>         <span class="at">f_1_8_rs =</span> purrr<span class="sc">::</span><span class="fu">map</span>(data, \(x) <span class="fu">map</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">100</span>, \(i) <span class="fu">sum</span>(<span class="fu">rbinom</span>(<span class="dv">1000</span>, <span class="dv">1</span>, x<span class="sc">$</span>f_1_8))<span class="sc">/</span><span class="dv">1000</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb21-132"><a href="#cb21-132"></a>  tidyr<span class="sc">::</span><span class="fu">unnest</span>(<span class="at">cols =</span> <span class="fu">c</span>(data, f_1_0_rs, f_1_8_rs)) <span class="sc">%&gt;%</span> </span>
<span id="cb21-133"><a href="#cb21-133"></a>  <span class="co"># nest the samples</span></span>
<span id="cb21-134"><a href="#cb21-134"></a>  tidyr<span class="sc">::</span><span class="fu">nest</span>(<span class="at">bs =</span> <span class="fu">c</span>(f_1_0_rs, f_1_8_rs)) <span class="sc">%&gt;%</span></span>
<span id="cb21-135"><a href="#cb21-135"></a>  <span class="co"># perform the wilcox test</span></span>
<span id="cb21-136"><a href="#cb21-136"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">wc =</span> purrr<span class="sc">::</span><span class="fu">map</span>(bs, \(i) <span class="fu">wilcox.test</span>(<span class="at">x =</span> <span class="fu">as.numeric</span>(i<span class="sc">$</span>f_1_0_rs), <span class="at">y =</span> <span class="fu">as.numeric</span>(i<span class="sc">$</span>f_1_8_rs)))) <span class="sc">%&gt;%</span> </span>
<span id="cb21-137"><a href="#cb21-137"></a>  <span class="co"># tidy-ify the test output</span></span>
<span id="cb21-138"><a href="#cb21-138"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">tidy_wc =</span> purrr<span class="sc">::</span><span class="fu">map</span>(wc, \(x) broom<span class="sc">::</span><span class="fu">tidy</span>(x))) <span class="sc">%&gt;%</span> </span>
<span id="cb21-139"><a href="#cb21-139"></a>  tidyr<span class="sc">::</span><span class="fu">unnest</span>(<span class="at">cols =</span> <span class="fu">c</span>(tidy_wc)) <span class="sc">%&gt;%</span> </span>
<span id="cb21-140"><a href="#cb21-140"></a>  <span class="co"># p-value adjust using bonferroni correction</span></span>
<span id="cb21-141"><a href="#cb21-141"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">p_adjusted =</span> <span class="fu">p.adjust</span>(p.value, <span class="at">method =</span> <span class="st">"bonferroni"</span>, <span class="at">n =</span> <span class="fu">n</span>())) <span class="sc">%&gt;%</span> </span>
<span id="cb21-142"><a href="#cb21-142"></a>  dplyr<span class="sc">::</span><span class="fu">arrange</span>(strep_conc, sp_1, sp_2) <span class="sc">%&gt;%</span> </span>
<span id="cb21-143"><a href="#cb21-143"></a>  <span class="co"># define whether change is significantly positive or negative</span></span>
<span id="cb21-144"><a href="#cb21-144"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">change =</span> dplyr<span class="sc">::</span><span class="fu">case_when</span>(p.value <span class="sc">&gt;</span> <span class="fl">1e-5</span> <span class="sc">~</span> <span class="dv">0</span>,</span>
<span id="cb21-145"><a href="#cb21-145"></a>                             <span class="fu">sign</span>(delta_f_1) <span class="sc">==</span> <span class="sc">-</span><span class="dv">1</span> <span class="sc">&amp;</span> p.value <span class="sc">&lt;=</span> <span class="fl">1e-5</span> <span class="sc">~</span> <span class="sc">-</span><span class="dv">1</span>, </span>
<span id="cb21-146"><a href="#cb21-146"></a>                             <span class="fu">sign</span>(delta_f_1) <span class="sc">==</span> <span class="dv">1</span> <span class="sc">&amp;</span> p.value <span class="sc">&lt;=</span> <span class="fl">1e-5</span> <span class="sc">~</span> <span class="dv">1</span>)) </span>
<span id="cb21-147"><a href="#cb21-147"></a><span class="in">```</span></span>
<span id="cb21-148"><a href="#cb21-148"></a></span>
<span id="cb21-149"><a href="#cb21-149"></a><span class="fu">## Competition outcome rules</span></span>
<span id="cb21-150"><a href="#cb21-150"></a></span>
<span id="cb21-151"><a href="#cb21-151"></a>Here we set up the rules for defining the competition outcomes. In the pairwise competition experiments each species and evolutionary form was competed against all other species and evolutionary forms (excluding evolutionary forms of the same species because these cannot be resolved using amplicon data). Each competitor in a pair $<span class="sc">\{</span>A, B<span class="sc">\}</span>$ was allowed to invade from rare ($f^{R} = 0.1$) while the other competitor was common ($f^{C} = 0.9$) in duplicate for 8 growth cycles resulting in 4 experiments for each competing pair. We used a $f_{extinct} = 0.01$ frequency threshold for deciding whether a species was operationally extinct and a $| \Delta_{(t8-t0)} f | \geq 0.05 $ change over 8 growth cycles to decide whether the magnitude of change over the 8 growth cycles was appreciable. Using the abundance outcomes from both replicates of each the pairwise competition experiments for each pair we classified outcomes as "competitive exclusion", "coexistence", "bistability", or "inconclusive" according to the following criteria:</span>
<span id="cb21-152"><a href="#cb21-152"></a></span>
<span id="cb21-153"><a href="#cb21-153"></a><span class="fu">### Bistability</span></span>
<span id="cb21-154"><a href="#cb21-154"></a></span>
<span id="cb21-155"><a href="#cb21-155"></a>Requires each species in a pair to significantly increase ($\Delta f_{<span class="sc">\{</span>A, B<span class="sc">\}</span>} \geq 0.05$) when starting from common *and* significantly decrease ($\Delta f_{<span class="sc">\{</span>A, B<span class="sc">\}</span>} \leq -0.05$) in abundance when invading from rare over the 8 growth cycles.</span>
<span id="cb21-156"><a href="#cb21-156"></a></span>
<span id="cb21-157"><a href="#cb21-157"></a><span class="fu">### Competitive exclusion</span></span>
<span id="cb21-158"><a href="#cb21-158"></a></span>
<span id="cb21-159"><a href="#cb21-159"></a>Competitive exclusion means one species of a pair excludes the other to it's (eventual) extinction. We classified competitive exclusion as "Stable" or "Incomplete" based on the following features:</span>
<span id="cb21-160"><a href="#cb21-160"></a></span>
<span id="cb21-161"><a href="#cb21-161"></a><span class="fu">#### Stable</span></span>
<span id="cb21-162"><a href="#cb21-162"></a></span>
<span id="cb21-163"><a href="#cb21-163"></a>Stable exclusion requires one species in a pair to go extinct (below 1% abundance detection limit threshold) in all 4 experiments performed for the two species. It also requires that the species undergoing extinction should always decrease over the 8 growth cycles by at least 5%. In practice this excludes samples where the extinct species started so rare that it was already effectively extinct.</span>
<span id="cb21-164"><a href="#cb21-164"></a></span>
<span id="cb21-165"><a href="#cb21-165"></a>$$</span>
<span id="cb21-166"><a href="#cb21-166"></a>\begin{align*} </span>
<span id="cb21-167"><a href="#cb21-167"></a>&amp; {f_{<span class="sc">\{</span>A, B<span class="sc">\}</span>}}_{t8} \leq f_{extinct} <span class="sc">\\</span> </span>
<span id="cb21-168"><a href="#cb21-168"></a>&amp; \Delta f_{<span class="sc">\{</span>A, B<span class="sc">\}</span>} \leq -0.05 <span class="sc">\\</span></span>
<span id="cb21-169"><a href="#cb21-169"></a>\end{align*}</span>
<span id="cb21-170"><a href="#cb21-170"></a>$$</span>
<span id="cb21-171"><a href="#cb21-171"></a></span>
<span id="cb21-172"><a href="#cb21-172"></a><span class="fu">#### Incomplete</span></span>
<span id="cb21-173"><a href="#cb21-173"></a></span>
<span id="cb21-174"><a href="#cb21-174"></a>Incomplete exclusion results if one of the species at cycle 8 is not extinct but its frequency significantly decreased over 8 growth cycles in all four experiments (Wilcoxon test, P&lt;0.05, N=2000 in each experiment).</span>
<span id="cb21-175"><a href="#cb21-175"></a></span>
<span id="cb21-176"><a href="#cb21-176"></a>$$</span>
<span id="cb21-177"><a href="#cb21-177"></a>\begin{align*}</span>
<span id="cb21-178"><a href="#cb21-178"></a>&amp; {f_{<span class="sc">\{</span>A, B<span class="sc">\}</span>}}_{t8} &gt; f_{extinct} <span class="sc">\\</span> </span>
<span id="cb21-179"><a href="#cb21-179"></a>&amp; \Delta f_{<span class="sc">\{</span>A, B<span class="sc">\}</span>} \leq -0.05, \space p &lt; 0.05 <span class="sc">\\</span></span>
<span id="cb21-180"><a href="#cb21-180"></a>\end{align*}</span>
<span id="cb21-181"><a href="#cb21-181"></a>$$</span>
<span id="cb21-182"><a href="#cb21-182"></a>                             </span>
<span id="cb21-183"><a href="#cb21-183"></a><span class="fu">### Coexistence</span></span>
<span id="cb21-184"><a href="#cb21-184"></a></span>
<span id="cb21-185"><a href="#cb21-185"></a>We classified a pair as coexisting if both species are able to invade each other from low frequency. We classified competitive exclusion as "Stable" or "Incomplete" based on differing levels of supporting evidence.</span>
<span id="cb21-186"><a href="#cb21-186"></a></span>
<span id="cb21-187"><a href="#cb21-187"></a><span class="fu">#### Stable</span></span>
<span id="cb21-188"><a href="#cb21-188"></a></span>
<span id="cb21-189"><a href="#cb21-189"></a>We defined "Stable Coexistence" as those species pairs where each species could invade the other from rare (fulfillment of the mutual invasibility criterion). table coexistence results if each species exists outside the extinction threshold and over 8 growth cycles and could significantly increase (Wilcoxon test, P&lt;0.05, N=2000 in each experiment)by at least 5% from rare in both replicates. </span>
<span id="cb21-190"><a href="#cb21-190"></a></span>
<span id="cb21-191"><a href="#cb21-191"></a>$$</span>
<span id="cb21-192"><a href="#cb21-192"></a>\begin{align*}</span>
<span id="cb21-193"><a href="#cb21-193"></a>&amp; {f_{<span class="sc">\{</span>A, B<span class="sc">\}</span>}^{Rare}}_{t8} &gt; f_{extinct} <span class="sc">\\</span> </span>
<span id="cb21-194"><a href="#cb21-194"></a>&amp; \Delta f_{<span class="sc">\{</span>A, B<span class="sc">\}</span>}^{Rare} \geq 0.05, \space p &lt; 0.05 <span class="sc">\\</span></span>
<span id="cb21-195"><a href="#cb21-195"></a>\end{align*}</span>
<span id="cb21-196"><a href="#cb21-196"></a>$$</span>
<span id="cb21-197"><a href="#cb21-197"></a></span>
<span id="cb21-198"><a href="#cb21-198"></a><span class="fu">#### Incomplete</span></span>
<span id="cb21-199"><a href="#cb21-199"></a></span>
<span id="cb21-200"><a href="#cb21-200"></a>We defined "Incomplete coexistence" as those species pairs that were found together after 8 growth cycles but no condition was placed on the significance of the change of each species from T0 to T8. For example, if a species started from low abundance and after 8 growth cycles also ended at low abundance that was not significantly different from the starting abundance.</span>
<span id="cb21-201"><a href="#cb21-201"></a></span>
<span id="cb21-202"><a href="#cb21-202"></a>$$</span>
<span id="cb21-203"><a href="#cb21-203"></a>\begin{align*}</span>
<span id="cb21-204"><a href="#cb21-204"></a>&amp; {f_{<span class="sc">\{</span>A, B<span class="sc">\}</span>}}_{t8} &gt; f_{extinct} <span class="sc">\\</span> </span>
<span id="cb21-205"><a href="#cb21-205"></a>\end{align*}</span>
<span id="cb21-206"><a href="#cb21-206"></a>$$</span>
<span id="cb21-207"><a href="#cb21-207"></a></span>
<span id="cb21-208"><a href="#cb21-208"></a><span class="fu">#### Inconclusive outcomes</span></span>
<span id="cb21-209"><a href="#cb21-209"></a></span>
<span id="cb21-210"><a href="#cb21-210"></a>This category included pairs that did not meet the requirements to be classified in any of the above described categories. </span>
<span id="cb21-211"><a href="#cb21-211"></a></span>
<span id="cb21-214"><a href="#cb21-214"></a><span class="in">```{r}</span></span>
<span id="cb21-215"><a href="#cb21-215"></a><span class="co"># upper threshold for deciding whether a species goes extinct</span></span>
<span id="cb21-216"><a href="#cb21-216"></a>upper <span class="ot">&lt;-</span> <span class="fl">0.99</span></span>
<span id="cb21-217"><a href="#cb21-217"></a><span class="co"># lower threshold for deciding whether a species goes extinct</span></span>
<span id="cb21-218"><a href="#cb21-218"></a>lower <span class="ot">&lt;-</span> <span class="fl">0.01</span></span>
<span id="cb21-219"><a href="#cb21-219"></a><span class="co"># slope change threshold for deciding whether change from T0 to T8 is important (e.g. requires at least a 5% change of the focal species) </span></span>
<span id="cb21-220"><a href="#cb21-220"></a>slope_thresh <span class="ot">&lt;-</span> <span class="fl">0.05</span></span>
<span id="cb21-221"><a href="#cb21-221"></a></span>
<span id="cb21-222"><a href="#cb21-222"></a>outcomes_classified <span class="ot">&lt;-</span> wc_test <span class="sc">%&gt;%</span> </span>
<span id="cb21-223"><a href="#cb21-223"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(strep_conc, sp_1, sp_2) <span class="sc">%&gt;%</span> </span>
<span id="cb21-224"><a href="#cb21-224"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">outcome =</span> dplyr<span class="sc">::</span><span class="fu">case_when</span>(<span class="co"># Bistability: Requires sp_1 to both increase from T0 high freq and decrease from T0 low freq</span></span>
<span id="cb21-225"><a href="#cb21-225"></a>                             <span class="fu">sum</span>(change <span class="sc">&lt;</span> <span class="dv">0</span> <span class="sc">&amp;</span> f_1_0 <span class="sc">==</span> <span class="fu">min</span>(f_1_0), change <span class="sc">&gt;</span> <span class="dv">0</span> <span class="sc">&amp;</span> f_1_0 <span class="sc">==</span> <span class="fu">max</span>(f_1_0)) <span class="sc">==</span> <span class="dv">2</span> <span class="sc">~</span> <span class="st">"bistable_stable"</span>,</span>
<span id="cb21-226"><a href="#cb21-226"></a>                             <span class="co"># Exclusion: sp_1 increases from both low and high T0 freqs and T8 freq is &gt; than the T0 high freq</span></span>
<span id="cb21-227"><a href="#cb21-227"></a>                             <span class="fu">sum</span>(change) <span class="sc">==</span> <span class="dv">2</span> <span class="sc">&amp;</span> f_1_8 <span class="sc">&gt;=</span> upper <span class="sc">&amp;</span> <span class="fu">abs</span>(delta_f_1) <span class="sc">&gt;</span> slope_thresh <span class="sc">~</span> <span class="st">"exclusion_stable"</span>, </span>
<span id="cb21-228"><a href="#cb21-228"></a>                             <span class="co"># Exclusion: sp_1 decreases from both low and high T0 freqs and T8 freq is &lt; than the T0 low freq</span></span>
<span id="cb21-229"><a href="#cb21-229"></a>                             <span class="fu">sum</span>(change) <span class="sc">==</span> <span class="sc">-</span><span class="dv">2</span> <span class="sc">&amp;</span> f_1_8 <span class="sc">&lt;=</span> lower <span class="sc">&amp;</span> <span class="fu">abs</span>(delta_f_1) <span class="sc">&gt;</span> slope_thresh <span class="sc">~</span> <span class="st">"exclusion_stable"</span>,</span>
<span id="cb21-230"><a href="#cb21-230"></a>                             <span class="co"># Coexistence: sp_1 significantly increases from rare and significantly decreases from abundant</span></span>
<span id="cb21-231"><a href="#cb21-231"></a>                             <span class="fu">sum</span>(change) <span class="sc">==</span> <span class="dv">0</span> <span class="sc">&amp;</span> dplyr<span class="sc">::</span><span class="fu">if_else</span>(f_1_0 <span class="sc">==</span> <span class="fu">pmin</span>(f_1_0, f_2_0), f_1_8 <span class="sc">&gt;</span> f_1_0, f_1_8 <span class="sc">&lt;</span> f_1_0) <span class="sc">~</span> <span class="st">"coexistence_stable"</span>,</span>
<span id="cb21-232"><a href="#cb21-232"></a>                             <span class="co"># Incomplete coexistence: sp_1 is found within extinction thresholds in both experiments, but no condition is</span></span>
<span id="cb21-233"><a href="#cb21-233"></a>                             <span class="co"># placed on the significance of the change from T0 to T8</span></span>
<span id="cb21-234"><a href="#cb21-234"></a>                             <span class="fu">sum</span>(dplyr<span class="sc">::</span><span class="fu">between</span>(f_1_8, lower, upper)) <span class="sc">==</span> <span class="dv">2</span> <span class="sc">~</span> <span class="st">"coexistence_incomplete"</span>,</span>
<span id="cb21-235"><a href="#cb21-235"></a>                             <span class="co"># Incomplete exclusion: sp_1 is outside of the extinction threshold (extinct) for at least one sample and within the the </span></span>
<span id="cb21-236"><a href="#cb21-236"></a>                             <span class="co"># threshold for the other sample.</span></span>
<span id="cb21-237"><a href="#cb21-237"></a>                             <span class="fu">sum</span>(<span class="sc">!</span>dplyr<span class="sc">::</span><span class="fu">between</span>(f_1_8, lower, upper)) <span class="sc">==</span> <span class="dv">1</span> <span class="sc">&amp;</span> <span class="fu">sum</span>(dplyr<span class="sc">::</span><span class="fu">between</span>(f_1_8, lower, upper)) <span class="sc">==</span> <span class="dv">1</span> <span class="sc">~</span> <span class="st">"exclusion_incomplete"</span>,</span>
<span id="cb21-238"><a href="#cb21-238"></a>                             <span class="co"># Inconclusive: samples failing to meet any of the above rules</span></span>
<span id="cb21-239"><a href="#cb21-239"></a>                             <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"inconclusive_"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb21-240"><a href="#cb21-240"></a>  dplyr<span class="sc">::</span><span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb21-241"><a href="#cb21-241"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">type =</span> dplyr<span class="sc">::</span><span class="fu">if_else</span>(stringr<span class="sc">::</span><span class="fu">str_detect</span>(outcome, <span class="st">"incomplete"</span>), <span class="st">"incomplete"</span>, <span class="st">"stable"</span>),</span>
<span id="cb21-242"><a href="#cb21-242"></a>         <span class="at">outcome =</span> stringr<span class="sc">::</span><span class="fu">str_extract</span>(outcome, <span class="st">"(^.+)_.*"</span>, <span class="at">group =</span> <span class="dv">1</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb21-243"><a href="#cb21-243"></a>  dplyr<span class="sc">::</span><span class="fu">relocate</span>(outcome, type, change)</span>
<span id="cb21-244"><a href="#cb21-244"></a><span class="in">```</span></span>
<span id="cb21-245"><a href="#cb21-245"></a></span>
<span id="cb21-246"><a href="#cb21-246"></a><span class="fu">## Plotting pairwise outpcomes</span></span>
<span id="cb21-247"><a href="#cb21-247"></a></span>
<span id="cb21-248"><a href="#cb21-248"></a>Construct final dataframe to be used for plotting</span>
<span id="cb21-249"><a href="#cb21-249"></a></span>
<span id="cb21-252"><a href="#cb21-252"></a><span class="in">```{r}</span></span>
<span id="cb21-253"><a href="#cb21-253"></a>samp_pairs_fmt <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">bind_rows</span>(samp_pairs_fmt_mp, samp_pairs_fmt_exp) <span class="sc">%&gt;%</span></span>
<span id="cb21-254"><a href="#cb21-254"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb21-255"><a href="#cb21-255"></a>    <span class="at">group =</span> <span class="fu">interaction</span>(community_id, strep_conc),</span>
<span id="cb21-256"><a href="#cb21-256"></a>    <span class="at">evo_group =</span> dplyr<span class="sc">::</span><span class="fu">case_when</span>(</span>
<span id="cb21-257"><a href="#cb21-257"></a>      dplyr<span class="sc">::</span><span class="fu">if_all</span>(<span class="fu">c</span>(sp_1, sp_2), \(x) stringr<span class="sc">::</span><span class="fu">str_detect</span>(x, <span class="st">"ANC"</span>)) <span class="sc">~</span> <span class="st">"both_anc"</span>,</span>
<span id="cb21-258"><a href="#cb21-258"></a>      dplyr<span class="sc">::</span><span class="fu">if_all</span>(<span class="fu">c</span>(sp_1, sp_2), \(x) stringr<span class="sc">::</span><span class="fu">str_detect</span>(x, <span class="st">"EVO"</span>)) <span class="sc">~</span> <span class="st">"both_evo"</span>,</span>
<span id="cb21-259"><a href="#cb21-259"></a>      <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"mix"</span></span>
<span id="cb21-260"><a href="#cb21-260"></a>    )</span>
<span id="cb21-261"><a href="#cb21-261"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb21-262"><a href="#cb21-262"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(outcomes_classified,</span>
<span id="cb21-263"><a href="#cb21-263"></a>                   <span class="at">by =</span> dplyr<span class="sc">::</span><span class="fu">join_by</span>(community_id, sp_1, sp_2, strep_conc)) <span class="sc">%&gt;%</span></span>
<span id="cb21-264"><a href="#cb21-264"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(</span>
<span id="cb21-265"><a href="#cb21-265"></a>    <span class="at">outcome =</span> <span class="fu">factor</span>(outcome, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"exclusion"</span>, <span class="st">"coexistence"</span>, </span>
<span id="cb21-266"><a href="#cb21-266"></a>                                         <span class="st">"bistable"</span>, <span class="st">"inconclusive"</span>)),</span>
<span id="cb21-267"><a href="#cb21-267"></a>    <span class="at">type =</span> <span class="fu">factor</span>(type, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"stable"</span>, <span class="st">"incomplete"</span>))</span>
<span id="cb21-268"><a href="#cb21-268"></a>  )</span>
<span id="cb21-269"><a href="#cb21-269"></a><span class="in">```</span></span>
<span id="cb21-270"><a href="#cb21-270"></a></span>
<span id="cb21-271"><a href="#cb21-271"></a>Create different lists of plots for the mixed (i.e. evo competed against anc) conditions</span>
<span id="cb21-274"><a href="#cb21-274"></a><span class="in">```{r}</span></span>
<span id="cb21-275"><a href="#cb21-275"></a>samp_pairs_fmt_plots_split_a <span class="ot">&lt;-</span> samp_pairs_fmt <span class="sc">%&gt;%</span> </span>
<span id="cb21-276"><a href="#cb21-276"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(evo_group <span class="sc">!=</span> <span class="st">"mix"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb21-277"><a href="#cb21-277"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(strep_conc, evo_group) <span class="sc">%&gt;%</span> </span>
<span id="cb21-278"><a href="#cb21-278"></a>  dplyr<span class="sc">::</span><span class="fu">group_split</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb21-279"><a href="#cb21-279"></a>  purrr<span class="sc">::</span><span class="fu">map</span>(pair_plot)</span>
<span id="cb21-280"><a href="#cb21-280"></a></span>
<span id="cb21-281"><a href="#cb21-281"></a>samp_pairs_fmt_plots_split_b <span class="ot">&lt;-</span> samp_pairs_fmt <span class="sc">%&gt;%</span> </span>
<span id="cb21-282"><a href="#cb21-282"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(evo_group <span class="sc">==</span> <span class="st">"mix"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb21-283"><a href="#cb21-283"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(strep_conc) <span class="sc">%&gt;%</span> </span>
<span id="cb21-284"><a href="#cb21-284"></a>  dplyr<span class="sc">::</span><span class="fu">group_split</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb21-285"><a href="#cb21-285"></a>  purrr<span class="sc">::</span><span class="fu">map</span>(pair_plot)</span>
<span id="cb21-286"><a href="#cb21-286"></a><span class="in">```</span></span>
<span id="cb21-287"><a href="#cb21-287"></a></span>
<span id="cb21-290"><a href="#cb21-290"></a><span class="in">```{r}</span></span>
<span id="cb21-291"><a href="#cb21-291"></a><span class="co">#| warning: false</span></span>
<span id="cb21-292"><a href="#cb21-292"></a>fig01 <span class="ot">&lt;-</span> patchwork<span class="sc">::</span><span class="fu">wrap_plots</span>(samp_pairs_fmt_plots_split_a, <span class="at">ncol =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb21-293"><a href="#cb21-293"></a>  patchwork<span class="sc">::</span><span class="fu">plot_annotation</span>(<span class="at">tag_levels =</span> <span class="st">"A"</span>)</span>
<span id="cb21-294"><a href="#cb21-294"></a><span class="in">```</span></span>
<span id="cb21-295"><a href="#cb21-295"></a></span>
<span id="cb21-298"><a href="#cb21-298"></a><span class="in">```{r}</span></span>
<span id="cb21-299"><a href="#cb21-299"></a><span class="co">#| eval: false</span></span>
<span id="cb21-300"><a href="#cb21-300"></a><span class="co">#| include: true</span></span>
<span id="cb21-301"><a href="#cb21-301"></a><span class="fu">ggsave</span>(</span>
<span id="cb21-302"><a href="#cb21-302"></a>  here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"figs"</span>, <span class="st">"coexistence_pairs.svg"</span>),</span>
<span id="cb21-303"><a href="#cb21-303"></a>  fig01,</span>
<span id="cb21-304"><a href="#cb21-304"></a>  <span class="at">width =</span> <span class="dv">7</span>,</span>
<span id="cb21-305"><a href="#cb21-305"></a>  <span class="at">height =</span> <span class="dv">12</span>,</span>
<span id="cb21-306"><a href="#cb21-306"></a>  <span class="at">units =</span> <span class="st">"in"</span>,</span>
<span id="cb21-307"><a href="#cb21-307"></a>  <span class="at">device =</span> <span class="st">"svg"</span></span>
<span id="cb21-308"><a href="#cb21-308"></a>)</span>
<span id="cb21-309"><a href="#cb21-309"></a><span class="in">```</span></span>
<span id="cb21-310"><a href="#cb21-310"></a></span>
<span id="cb21-311"><a href="#cb21-311"></a>::: {#fig-01}</span>
<span id="cb21-314"><a href="#cb21-314"></a><span class="in">```{r}</span></span>
<span id="cb21-315"><a href="#cb21-315"></a><span class="co">#| fig-width: 7</span></span>
<span id="cb21-316"><a href="#cb21-316"></a><span class="co">#| fig-height: 12</span></span>
<span id="cb21-317"><a href="#cb21-317"></a><span class="co">#| warning: false</span></span>
<span id="cb21-318"><a href="#cb21-318"></a><span class="co">#| echo: false</span></span>
<span id="cb21-319"><a href="#cb21-319"></a>fig01</span>
<span id="cb21-320"><a href="#cb21-320"></a><span class="in">```</span></span>
<span id="cb21-321"><a href="#cb21-321"></a>Outcomes from pairwise cocultures of ancestral (left, A:G) and streptomycin adapted (right, B:H) 0403, 1287, 1896 and 1977 species. Rows in the grid represent differen streptomycin concentrations applied (A:B = 0 µg/ml, C:D = 16 µg/ml, E:F = 64 µg/ml, G:H = 256 µg/ml). Using the rules defined above, red lines show cocultures resulting in exclusion of one of the species, blue lines show cocultures that resulted in coexistence, and purple lines represent inconclusive outcomes due to lack of data or exception to therules. Solid lines show a "definite" outcome and dashed lines show a potentially "incomplete" outcome. Two solid lines of the same color indicate stable coexistence or stable exclusion, a solid and dashed line indicate incomplete coexistence/exclusion, while two dashed lines indicates coexistence without evidence of mutual invasibility. Note some statistical noise has been applied to point positions to prevent overlaps in the plot and aid in visualization. **NOTE:** that in panel E (64 µg/ml streptomycin) both replicates survived with OD ~ 0.15 when ANC_0403 was started from 90% abundance but not when it was started at 10% abundance (OD ~ 0.04). In all replicates of 90%/10% starting abundance ANC_1977 was driven extinct.</span>
<span id="cb21-322"><a href="#cb21-322"></a>:::</span>
<span id="cb21-323"><a href="#cb21-323"></a></span>
<span id="cb21-326"><a href="#cb21-326"></a><span class="in">```{r}</span></span>
<span id="cb21-327"><a href="#cb21-327"></a><span class="co">#| warning: false</span></span>
<span id="cb21-328"><a href="#cb21-328"></a>fig02 <span class="ot">&lt;-</span> patchwork<span class="sc">::</span><span class="fu">wrap_plots</span>(samp_pairs_fmt_plots_split_b, <span class="at">ncol =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb21-329"><a href="#cb21-329"></a>  patchwork<span class="sc">::</span><span class="fu">plot_annotation</span>(<span class="at">tag_levels =</span> <span class="st">"A"</span>)</span>
<span id="cb21-330"><a href="#cb21-330"></a><span class="in">```</span></span>
<span id="cb21-331"><a href="#cb21-331"></a></span>
<span id="cb21-334"><a href="#cb21-334"></a><span class="in">```{r}</span></span>
<span id="cb21-335"><a href="#cb21-335"></a><span class="co">#| eval: false</span></span>
<span id="cb21-336"><a href="#cb21-336"></a><span class="co">#| include: true</span></span>
<span id="cb21-337"><a href="#cb21-337"></a><span class="fu">ggsave</span>(</span>
<span id="cb21-338"><a href="#cb21-338"></a>  here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"figs"</span>, <span class="st">"coexistence_pairs_mixed_hist.svg"</span>),</span>
<span id="cb21-339"><a href="#cb21-339"></a>  fig02,</span>
<span id="cb21-340"><a href="#cb21-340"></a>  <span class="at">width =</span> <span class="dv">7</span>,</span>
<span id="cb21-341"><a href="#cb21-341"></a>  <span class="at">height =</span> <span class="dv">7</span>,</span>
<span id="cb21-342"><a href="#cb21-342"></a>  <span class="at">units =</span> <span class="st">"in"</span>,</span>
<span id="cb21-343"><a href="#cb21-343"></a>  <span class="at">device =</span> <span class="st">"svg"</span></span>
<span id="cb21-344"><a href="#cb21-344"></a>)</span>
<span id="cb21-345"><a href="#cb21-345"></a><span class="in">```</span></span>
<span id="cb21-346"><a href="#cb21-346"></a></span>
<span id="cb21-347"><a href="#cb21-347"></a>::: {#fig-02}</span>
<span id="cb21-350"><a href="#cb21-350"></a><span class="in">```{r}</span></span>
<span id="cb21-351"><a href="#cb21-351"></a><span class="co">#| fig-width: 7</span></span>
<span id="cb21-352"><a href="#cb21-352"></a><span class="co">#| fig-height: 7</span></span>
<span id="cb21-353"><a href="#cb21-353"></a><span class="co">#| warning: false</span></span>
<span id="cb21-354"><a href="#cb21-354"></a><span class="co">#| echo: false</span></span>
<span id="cb21-355"><a href="#cb21-355"></a>fig02</span>
<span id="cb21-356"><a href="#cb21-356"></a><span class="in">```</span></span>
<span id="cb21-357"><a href="#cb21-357"></a>Results from co-cultures of mixed ancestal and evolved combinations. Line colors and types are as in @fig-01, with the addition of green showing bistability (i.e., Species A and Species B exclude each other when invading from high abundance, and each are excluded by the other when invading from low abundance).</span>
<span id="cb21-358"><a href="#cb21-358"></a>:::</span>
<span id="cb21-359"><a href="#cb21-359"></a></span>
<span id="cb21-360"><a href="#cb21-360"></a><span class="fu"># Network</span></span>
<span id="cb21-361"><a href="#cb21-361"></a></span>
<span id="cb21-362"><a href="#cb21-362"></a>Here we will plot the pairwise competition outcomes as a network. **NOTE:** I don't have time/energy to figure out how to manage all the links in the networks so that they line up properly and don't overlap so there will need to be some postprocessing in inkscape to move some of the links so they don't overlap.</span>
<span id="cb21-363"><a href="#cb21-363"></a></span>
<span id="cb21-366"><a href="#cb21-366"></a><span class="in">```{r}</span></span>
<span id="cb21-367"><a href="#cb21-367"></a><span class="co">#| output: false</span></span>
<span id="cb21-368"><a href="#cb21-368"></a><span class="fu">library</span>(ggraph)</span>
<span id="cb21-369"><a href="#cb21-369"></a><span class="fu">library</span>(tidygraph)</span>
<span id="cb21-370"><a href="#cb21-370"></a><span class="in">```</span></span>
<span id="cb21-371"><a href="#cb21-371"></a></span>
<span id="cb21-372"><a href="#cb21-372"></a><span class="fu">## Evolution and Streptomycin categories separate</span></span>
<span id="cb21-373"><a href="#cb21-373"></a></span>
<span id="cb21-374"><a href="#cb21-374"></a>Here we plot a separate graph for each streptomycin concentrations and also by different evolutionary groupings. For example, there is one graph for the competition outcomes of only ancestral species, there is one graph of the outcomes of only evolved species, and there is one graph for the outcomes of mixed competitions where an ancestral species competes against an evolved species.</span>
<span id="cb21-375"><a href="#cb21-375"></a></span>
<span id="cb21-378"><a href="#cb21-378"></a><span class="in">```{r}</span></span>
<span id="cb21-379"><a href="#cb21-379"></a><span class="co">#| warning: false</span></span>
<span id="cb21-380"><a href="#cb21-380"></a>nodes1 <span class="ot">&lt;-</span> <span class="fu">make_nodes</span>(samp_pairs_fmt, evo_group, strep_conc)</span>
<span id="cb21-381"><a href="#cb21-381"></a>edges1 <span class="ot">&lt;-</span> <span class="fu">make_edges</span>(samp_pairs_fmt, evo_group, strep_conc)</span>
<span id="cb21-382"><a href="#cb21-382"></a></span>
<span id="cb21-383"><a href="#cb21-383"></a>graphs1 <span class="ot">&lt;-</span> <span class="fu">nest</span>(nodes1, <span class="at">sps =</span> <span class="sc">-</span><span class="fu">c</span>(evo_group, strep_conc)) <span class="sc">%&gt;%</span> </span>
<span id="cb21-384"><a href="#cb21-384"></a>  <span class="fu">left_join</span>(<span class="fu">nest</span>(edges1, <span class="at">pairs =</span> <span class="sc">-</span><span class="fu">c</span>(evo_group, strep_conc)),</span>
<span id="cb21-385"><a href="#cb21-385"></a>            <span class="at">by =</span> <span class="fu">join_by</span>(evo_group, strep_conc)) <span class="sc">%&gt;%</span> </span>
<span id="cb21-386"><a href="#cb21-386"></a>  <span class="fu">mutate</span>(<span class="at">network =</span> <span class="fu">map2</span>(sps, pairs, <span class="cf">function</span>(sps, pairs) <span class="fu">tbl_graph</span>(<span class="at">nodes =</span> sps, <span class="at">edges =</span> pairs, <span class="at">directed =</span> T))) <span class="sc">%&gt;%</span> </span>
<span id="cb21-387"><a href="#cb21-387"></a>  <span class="fu">mutate</span>(<span class="at">plot =</span> <span class="fu">map</span>(network, <span class="cf">function</span>(network) <span class="fu">plot_network_hierarchy</span>(network, <span class="at">tune_angle =</span> <span class="fl">1.5</span>, <span class="at">n_rank =</span> <span class="dv">7</span>, <span class="at">n_break =</span> <span class="dv">7</span>)))</span>
<span id="cb21-388"><a href="#cb21-388"></a><span class="in">```</span></span>
<span id="cb21-389"><a href="#cb21-389"></a></span>
<span id="cb21-390"><a href="#cb21-390"></a></span>
<span id="cb21-393"><a href="#cb21-393"></a><span class="in">```{r}</span></span>
<span id="cb21-394"><a href="#cb21-394"></a><span class="co">#| warning: false</span></span>
<span id="cb21-395"><a href="#cb21-395"></a>fig03 <span class="ot">&lt;-</span> patchwork<span class="sc">::</span><span class="fu">wrap_plots</span>(graphs1[[<span class="dv">6</span>]], <span class="at">nrow =</span> <span class="dv">3</span>, <span class="at">guides=</span> <span class="st">"collect"</span>) <span class="sc">+</span></span>
<span id="cb21-396"><a href="#cb21-396"></a>  patchwork<span class="sc">::</span><span class="fu">plot_annotation</span>(<span class="at">tag_levels =</span> <span class="st">"A"</span>)</span>
<span id="cb21-397"><a href="#cb21-397"></a><span class="in">```</span></span>
<span id="cb21-398"><a href="#cb21-398"></a></span>
<span id="cb21-401"><a href="#cb21-401"></a><span class="in">```{r}</span></span>
<span id="cb21-402"><a href="#cb21-402"></a><span class="co">#| eval: false</span></span>
<span id="cb21-403"><a href="#cb21-403"></a><span class="co">#| include: true</span></span>
<span id="cb21-404"><a href="#cb21-404"></a><span class="fu">ggsave</span>(</span>
<span id="cb21-405"><a href="#cb21-405"></a>  here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"figs"</span>, <span class="st">"coexistence_networks_nested_evo_strep.svg"</span>),</span>
<span id="cb21-406"><a href="#cb21-406"></a>  fig03,</span>
<span id="cb21-407"><a href="#cb21-407"></a>  <span class="at">width =</span> <span class="dv">8</span>,</span>
<span id="cb21-408"><a href="#cb21-408"></a>  <span class="at">height =</span> <span class="dv">10</span>,</span>
<span id="cb21-409"><a href="#cb21-409"></a>  <span class="at">units =</span> <span class="st">"in"</span>,</span>
<span id="cb21-410"><a href="#cb21-410"></a>  <span class="at">device =</span> <span class="st">"svg"</span></span>
<span id="cb21-411"><a href="#cb21-411"></a>)</span>
<span id="cb21-412"><a href="#cb21-412"></a><span class="in">```</span></span>
<span id="cb21-413"><a href="#cb21-413"></a></span>
<span id="cb21-414"><a href="#cb21-414"></a>::: {#fig-03}</span>
<span id="cb21-417"><a href="#cb21-417"></a><span class="in">```{r}</span></span>
<span id="cb21-418"><a href="#cb21-418"></a><span class="co">#| fig-width: 8</span></span>
<span id="cb21-419"><a href="#cb21-419"></a><span class="co">#| fig-height: 10</span></span>
<span id="cb21-420"><a href="#cb21-420"></a><span class="co">#| warning: false</span></span>
<span id="cb21-421"><a href="#cb21-421"></a><span class="co">#| echo: false</span></span>
<span id="cb21-422"><a href="#cb21-422"></a>fig03</span>
<span id="cb21-423"><a href="#cb21-423"></a><span class="in">```</span></span>
<span id="cb21-424"><a href="#cb21-424"></a>Competitive hierarchy of species pairs separated by evolution grouping (only ancestral pairs, only evolved pairs, and mixed ancestral and evolved pairs) and streptomycin concentration. Subplots A-D are for only ancestral pairs, E-H for only evolved pairs, and I-L for mixed ancestral and evolved pairs. Subplots A, E, I show experiments under no streptomycin, B, F, J 16 µg/ml streptomycin, C, G, K for 64 µg/ml streptomycin, and D, H, L for 256 µg/ml streptomycin. For each evolution/streptomycin grouping, strains are rank ordered on the basis of the number of other strains they exclude, based on data shown in @fig-01 and @fig-02. Grey nodes represent strains (denoted by text), red arrows point from winning strain to losing strain, blue arrows indicate coexistence (ignore the arrow heads for blue, coulnd't figure out how to remove them for only a subset of the edges), and green arrows indicate the presence of bistability (i.e., Species A and Species B exclude each other when invading from high abundance, and each are excluded by the other when invading from low abundance). Line type indicates the degree of certainty of the outcome in each pair: dashed lines represent incomplete exclusion or coexistence without evidence of mutual invasibility, while solid lines indicate complete exclusion or stable coexistence (mutual invasibility). </span>
<span id="cb21-425"><a href="#cb21-425"></a>:::</span>
<span id="cb21-426"><a href="#cb21-426"></a></span>
<span id="cb21-427"><a href="#cb21-427"></a><span class="fu">## Only Streptomycin category separate</span></span>
<span id="cb21-428"><a href="#cb21-428"></a></span>
<span id="cb21-431"><a href="#cb21-431"></a><span class="in">```{r}</span></span>
<span id="cb21-432"><a href="#cb21-432"></a><span class="co">#| warning: false</span></span>
<span id="cb21-433"><a href="#cb21-433"></a>nodes2 <span class="ot">&lt;-</span> <span class="fu">make_nodes</span>(samp_pairs_fmt, strep_conc)</span>
<span id="cb21-434"><a href="#cb21-434"></a>edges2 <span class="ot">&lt;-</span> <span class="fu">make_edges</span>(samp_pairs_fmt, strep_conc)</span>
<span id="cb21-435"><a href="#cb21-435"></a></span>
<span id="cb21-436"><a href="#cb21-436"></a>graphs2 <span class="ot">&lt;-</span> <span class="fu">nest</span>(nodes2, <span class="at">sps =</span> <span class="sc">-</span><span class="fu">c</span>(strep_conc)) <span class="sc">%&gt;%</span> </span>
<span id="cb21-437"><a href="#cb21-437"></a>  <span class="fu">left_join</span>(<span class="fu">nest</span>(edges2, <span class="at">pairs =</span> <span class="sc">-</span><span class="fu">c</span>(strep_conc)),</span>
<span id="cb21-438"><a href="#cb21-438"></a>            <span class="at">by =</span> <span class="fu">join_by</span>(strep_conc)) <span class="sc">%&gt;%</span> </span>
<span id="cb21-439"><a href="#cb21-439"></a>  <span class="fu">mutate</span>(<span class="at">network =</span> <span class="fu">map2</span>(sps, pairs, <span class="cf">function</span>(sps, pairs) <span class="fu">tbl_graph</span>(<span class="at">nodes =</span> sps, <span class="at">edges =</span> pairs, <span class="at">directed =</span> T))) <span class="sc">%&gt;%</span> </span>
<span id="cb21-440"><a href="#cb21-440"></a>  <span class="fu">mutate</span>(<span class="at">plot =</span> <span class="fu">map</span>(network, <span class="cf">function</span>(network) <span class="fu">plot_network_hierarchy</span>(network, <span class="at">tune_angle =</span> <span class="fl">1.5</span>, <span class="at">n_rank =</span> <span class="dv">7</span>, <span class="at">n_break =</span> <span class="dv">7</span>)))</span>
<span id="cb21-441"><a href="#cb21-441"></a><span class="in">```</span></span>
<span id="cb21-442"><a href="#cb21-442"></a></span>
<span id="cb21-445"><a href="#cb21-445"></a><span class="in">```{r}</span></span>
<span id="cb21-446"><a href="#cb21-446"></a><span class="co">#| warning: false</span></span>
<span id="cb21-447"><a href="#cb21-447"></a>fig04 <span class="ot">&lt;-</span> patchwork<span class="sc">::</span><span class="fu">wrap_plots</span>(graphs2[[<span class="dv">5</span>]], <span class="at">nrow =</span> <span class="dv">1</span>, <span class="at">guides=</span> <span class="st">"collect"</span>) <span class="sc">+</span></span>
<span id="cb21-448"><a href="#cb21-448"></a>  patchwork<span class="sc">::</span><span class="fu">plot_annotation</span>(<span class="at">tag_levels =</span> <span class="st">"A"</span>)</span>
<span id="cb21-449"><a href="#cb21-449"></a><span class="in">```</span></span>
<span id="cb21-450"><a href="#cb21-450"></a></span>
<span id="cb21-453"><a href="#cb21-453"></a><span class="in">```{r}</span></span>
<span id="cb21-454"><a href="#cb21-454"></a><span class="co">#| eval: false</span></span>
<span id="cb21-455"><a href="#cb21-455"></a><span class="co">#| include: true</span></span>
<span id="cb21-456"><a href="#cb21-456"></a><span class="fu">ggsave</span>(</span>
<span id="cb21-457"><a href="#cb21-457"></a>  here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"figs"</span>, <span class="st">"coexistence_networks_nested_strep.svg"</span>),</span>
<span id="cb21-458"><a href="#cb21-458"></a>  fig04,</span>
<span id="cb21-459"><a href="#cb21-459"></a>  <span class="at">width =</span> <span class="dv">9</span>,</span>
<span id="cb21-460"><a href="#cb21-460"></a>  <span class="at">height =</span> <span class="dv">4</span>,</span>
<span id="cb21-461"><a href="#cb21-461"></a>  <span class="at">units =</span> <span class="st">"in"</span>,</span>
<span id="cb21-462"><a href="#cb21-462"></a>  <span class="at">device =</span> <span class="st">"svg"</span></span>
<span id="cb21-463"><a href="#cb21-463"></a>)</span>
<span id="cb21-464"><a href="#cb21-464"></a><span class="in">```</span></span>
<span id="cb21-465"><a href="#cb21-465"></a></span>
<span id="cb21-466"><a href="#cb21-466"></a>::: {#fig-04}</span>
<span id="cb21-469"><a href="#cb21-469"></a><span class="in">```{r}</span></span>
<span id="cb21-470"><a href="#cb21-470"></a><span class="co">#| fig-width: 9</span></span>
<span id="cb21-471"><a href="#cb21-471"></a><span class="co">#| fig-height: 4</span></span>
<span id="cb21-472"><a href="#cb21-472"></a><span class="co">#| warning: false</span></span>
<span id="cb21-473"><a href="#cb21-473"></a><span class="co">#| echo: false</span></span>
<span id="cb21-474"><a href="#cb21-474"></a>fig04</span>
<span id="cb21-475"><a href="#cb21-475"></a><span class="in">```</span></span>
<span id="cb21-476"><a href="#cb21-476"></a>Competitive hierarchy of species pairs separated by streptomycin concentration. Subplot A shows experiments under no streptomycin, B with 16 µg/ml streptomycin, C with 64 µg/ml streptomycin, and D with 256 µg/ml streptomycin. For each evolution/streptomycin grouping, strains are rank ordered on the basis of the number of other strains they exclude, based on data shown in @fig-01 and @fig-02. Grey nodes represent strains (denoted by text), red arrows point from winning strain to losing strain, blue arrows indicate coexistence (ignore the arrow heads for blue, coulnd't figure out how to remove them for only a subset of the edges), and green arrows indicate the presence of bistability (i.e., Species A and Species B exclude each other when invading from high abundance, and each are excluded by the other when invading from low abundance). Line type indicates the degree of certainty of the outcome in each pair: dashed lines represent incomplete exclusion or coexistence without evidence of mutual invasibility, while solid lines indicate complete exclusion or stable coexistence (mutual invasibility). </span>
<span id="cb21-477"><a href="#cb21-477"></a>:::</span>
</code><button title="Copy to Clipboard" class="code-copy-button" data-in-quarto-modal=""><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->




</body></html>