<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.42">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Milla Similä &amp; Shane Hogle">
<meta name="dcterms.date" content="2025-05-24">

<title>Formatting Rbec output from 2024-07-11 sequenced by BTK Turku using custom HAMBI Illumina v3 primers – hambiEvoEnvCoexist</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../../../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../../../">
<script src="../../../../site_libs/quarto-html/quarto.js"></script>
<script src="../../../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../../../site_libs/quarto-html/quarto-syntax-highlighting-2f5df379a58b258e96c21c0638c20c03.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../../../site_libs/bootstrap/bootstrap-1f23bf93a17969024013c136d185e5ce.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<link href="../../../../site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet">
<script src="../../../../site_libs/pagedtable-1.1/js/pagedtable.js"></script>


</head>

<body class="nav-sidebar docked">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../../../R/communities/amplicon/20240711_BTK_illumina_v3/01_format_rbec_tab.html">4. Community growth - amplicon formatting</a></li><li class="breadcrumb-item"><a href="../../../../R/communities/amplicon/20240711_BTK_illumina_v3/01_format_rbec_tab.html">i) Batch 2024-07-11</a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation docked overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="../../../../">hambiEvoEnvCoexist</a> 
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true">
 <span class="menu-text">1. Monoculture growth</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../R/monocultures/bioscreen/01_format_growthcurves_batch.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">i) Inspect and format growth curves</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../R/monocultures/bioscreen/02_plot_analyze_growth.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">ii) Plot growth summaries</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true">
 <span class="menu-text">2. Community growth - experiment design</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../R/communities/experiment_design/01_pairs.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">i) Species pairs - 96 Well plate layout</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../R/communities/experiment_design/02_trios.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">ii) Species trios - 96 Well plate layout</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../R/communities/experiment_design/03_quartets.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">iii) Species quartets - 96 Well plate layout</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true">
 <span class="menu-text">3. Community growth - optical density</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../R/communities/optical_density/01_format_plot_od.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">i) Process and plot OD data</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="true">
 <span class="menu-text">4. Community growth - amplicon formatting</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../R/communities/amplicon/20240711_BTK_illumina_v3/01_format_rbec_tab.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">i) Batch 2024-07-11</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../R/communities/amplicon/20240711_BTK_illumina_v3/02_correct_boil_to_qiagen.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">ii) Batch 2024-07-11 calibration</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../R/communities/amplicon/20241024_BTK_illumina_v3/01_format_rbec_tab.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">iii) Batch 2024-10-24</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../R/communities/amplicon/20250127_BTK_illumina_v3/01_format_rbec_tab.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">iv) Batch 2025-01-27</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../R/communities/amplicon/20250502_BTK_illumina_v3/01_format_rbec_tab.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">v) Batch 2025-05-02</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../R/communities/amplicon/01_compile_format.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">vi) Compile pairs, trios, quartets</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" role="navigation" aria-expanded="true">
 <span class="menu-text">5. Community growth - Coexistence/exclusion analysis</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-5" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../R/communities/amplicon/02_analysis_pairs.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">i) Species pairs</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../R/communities/amplicon/03_analysis_trios.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">ii) Species trios</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../R/communities/amplicon/04_analysis_quartets.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">iii) Species quartets</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../../R/communities/amplicon/05_analysis_octets.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">iv) All strains combined</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#setup" id="toc-setup" class="nav-link active" data-scroll-target="#setup"><span class="header-section-number">1</span> Setup</a>
  <ul class="collapse">
  <li><a href="#libraries" id="toc-libraries" class="nav-link" data-scroll-target="#libraries"><span class="header-section-number">1.1</span> Libraries</a></li>
  <li><a href="#global-variables" id="toc-global-variables" class="nav-link" data-scroll-target="#global-variables"><span class="header-section-number">1.2</span> Global variables</a></li>
  </ul></li>
  <li><a href="#read-metadata" id="toc-read-metadata" class="nav-link" data-scroll-target="#read-metadata"><span class="header-section-number">2</span> Read metadata</a></li>
  <li><a href="#read-rbec-raw-counts-tables" id="toc-read-rbec-raw-counts-tables" class="nav-link" data-scroll-target="#read-rbec-raw-counts-tables"><span class="header-section-number">3</span> Read Rbec raw counts tables</a>
  <ul class="collapse">
  <li><a href="#untar-rbec-output-tarball" id="toc-untar-rbec-output-tarball" class="nav-link" data-scroll-target="#untar-rbec-output-tarball"><span class="header-section-number">3.1</span> Untar Rbec output tarball</a></li>
  <li><a href="#setup-directory-structure" id="toc-setup-directory-structure" class="nav-link" data-scroll-target="#setup-directory-structure"><span class="header-section-number">3.2</span> Setup directory structure</a></li>
  <li><a href="#read" id="toc-read" class="nav-link" data-scroll-target="#read"><span class="header-section-number">3.3</span> Read</a></li>
  <li><a href="#clean-up" id="toc-clean-up" class="nav-link" data-scroll-target="#clean-up"><span class="header-section-number">3.4</span> Clean up</a></li>
  </ul></li>
  <li><a href="#format" id="toc-format" class="nav-link" data-scroll-target="#format"><span class="header-section-number">4</span> Format</a></li>
  <li><a href="#analysis" id="toc-analysis" class="nav-link" data-scroll-target="#analysis"><span class="header-section-number">5</span> Analysis</a>
  <ul class="collapse">
  <li><a href="#negative-controls" id="toc-negative-controls" class="nav-link" data-scroll-target="#negative-controls"><span class="header-section-number">5.1</span> Negative controls</a></li>
  <li><a href="#positive-controls" id="toc-positive-controls" class="nav-link" data-scroll-target="#positive-controls"><span class="header-section-number">5.2</span> Positive controls</a></li>
  <li><a href="#misassigned-reads" id="toc-misassigned-reads" class="nav-link" data-scroll-target="#misassigned-reads"><span class="header-section-number">5.3</span> Misassigned reads</a></li>
  <li><a href="#samples-with-few-total-reads" id="toc-samples-with-few-total-reads" class="nav-link" data-scroll-target="#samples-with-few-total-reads"><span class="header-section-number">5.4</span> Samples with few total reads</a></li>
  <li><a href="#filter-to-target-species" id="toc-filter-to-target-species" class="nav-link" data-scroll-target="#filter-to-target-species"><span class="header-section-number">5.5</span> Filter to target species</a></li>
  <li><a href="#masterplate-samples" id="toc-masterplate-samples" class="nav-link" data-scroll-target="#masterplate-samples"><span class="header-section-number">5.6</span> Masterplate samples</a>
  <ul class="collapse">
  <li><a href="#pairs" id="toc-pairs" class="nav-link" data-scroll-target="#pairs"><span class="header-section-number">5.6.1</span> Pairs</a></li>
  <li><a href="#trios" id="toc-trios" class="nav-link" data-scroll-target="#trios"><span class="header-section-number">5.6.2</span> Trios</a></li>
  <li><a href="#masterplate-summary" id="toc-masterplate-summary" class="nav-link" data-scroll-target="#masterplate-summary"><span class="header-section-number">5.6.3</span> Masterplate Summary</a></li>
  </ul></li>
  <li><a href="#experimental-samples" id="toc-experimental-samples" class="nav-link" data-scroll-target="#experimental-samples"><span class="header-section-number">5.7</span> Experimental samples</a>
  <ul class="collapse">
  <li><a href="#pairs-1" id="toc-pairs-1" class="nav-link" data-scroll-target="#pairs-1"><span class="header-section-number">5.7.1</span> Pairs</a></li>
  <li><a href="#trios-1" id="toc-trios-1" class="nav-link" data-scroll-target="#trios-1"><span class="header-section-number">5.7.2</span> Trios</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#export" id="toc-export" class="nav-link" data-scroll-target="#export"><span class="header-section-number">6</span> Export</a>
  <ul class="collapse">
  <li><a href="#posneg-control-samples" id="toc-posneg-control-samples" class="nav-link" data-scroll-target="#posneg-control-samples"><span class="header-section-number">6.1</span> pos/neg control samples</a></li>
  <li><a href="#pairs-2" id="toc-pairs-2" class="nav-link" data-scroll-target="#pairs-2"><span class="header-section-number">6.2</span> Pairs</a></li>
  <li><a href="#trios-2" id="toc-trios-2" class="nav-link" data-scroll-target="#trios-2"><span class="header-section-number">6.3</span> Trios</a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../../../R/communities/amplicon/20240711_BTK_illumina_v3/01_format_rbec_tab.html">4. Community growth - amplicon formatting</a></li><li class="breadcrumb-item"><a href="../../../../R/communities/amplicon/20240711_BTK_illumina_v3/01_format_rbec_tab.html">i) Batch 2024-07-11</a></li></ol></nav>
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">Formatting Rbec output from 2024-07-11 sequenced by BTK Turku using custom HAMBI Illumina v3 primers</h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Milla Similä &amp; Shane Hogle </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">May 24, 2025</p>
    </div>
  </div>
  
    
  </div>
  
<div>
  <div class="abstract">
    <div class="block-title">Abstract</div>
    Contains results from pairs of all streptomycin concentrations and trios for 0 streptomycin from Milla’s community assembly experiment.
  </div>
</div>


</header>


<section id="setup" class="level1" data-number="1">
<h1 data-number="1"><span class="header-section-number">1</span> Setup</h1>
<section id="libraries" class="level2" data-number="1.1">
<h2 data-number="1.1" class="anchored" data-anchor-id="libraries"><span class="header-section-number">1.1</span> Libraries</h2>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-2"><a href="#cb1-2"></a><span class="fu">library</span>(here)</span>
<span id="cb1-3"><a href="#cb1-3"></a><span class="fu">library</span>(fs)</span>
<span id="cb1-4"><a href="#cb1-4"></a><span class="fu">library</span>(archive)</span>
<span id="cb1-5"><a href="#cb1-5"></a><span class="fu">library</span>(scales)</span>
<span id="cb1-6"><a href="#cb1-6"></a><span class="fu">source</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"R"</span>, <span class="st">"utils_generic.R"</span>))</span>
<span id="cb1-7"><a href="#cb1-7"></a><span class="fu">source</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"R"</span>, <span class="st">"communities"</span>, <span class="st">"amplicon"</span>, <span class="st">"utils_amplicon.R"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="global-variables" class="level2" data-number="1.2">
<h2 data-number="1.2" class="anchored" data-anchor-id="global-variables"><span class="header-section-number">1.2</span> Global variables</h2>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a>data_raw <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"_data_raw"</span>, <span class="st">"communities"</span>, <span class="st">"20240711_BTK_illumina_v3"</span>)</span>
<span id="cb2-2"><a href="#cb2-2"></a>data <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"communities"</span>, <span class="st">"20240711_BTK_illumina_v3"</span>)</span>
<span id="cb2-3"><a href="#cb2-3"></a>amplicontar <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(data_raw, <span class="st">"rbec_output.tar.gz"</span>)</span>
<span id="cb2-4"><a href="#cb2-4"></a></span>
<span id="cb2-5"><a href="#cb2-5"></a><span class="co"># make processed data directory if it doesn't exist</span></span>
<span id="cb2-6"><a href="#cb2-6"></a>fs<span class="sc">::</span><span class="fu">dir_create</span>(data)</span>
<span id="cb2-7"><a href="#cb2-7"></a></span>
<span id="cb2-8"><a href="#cb2-8"></a><span class="co"># create temporary location to decompress</span></span>
<span id="cb2-9"><a href="#cb2-9"></a>tmpdir <span class="ot">&lt;-</span> fs<span class="sc">::</span><span class="fu">file_temp</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</section>
<section id="read-metadata" class="level1" data-number="2">
<h1 data-number="2"><span class="header-section-number">2</span> Read metadata</h1>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1"></a>mddf <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(data_raw, <span class="st">"20240711_BTK_illumina_v3_metadata.tsv"</span>))</span>
<span id="cb3-2"><a href="#cb3-2"></a>spdf <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(data_raw, <span class="st">"sample_compositions.tsv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="read-rbec-raw-counts-tables" class="level1" data-number="3">
<h1 data-number="3"><span class="header-section-number">3</span> Read Rbec raw counts tables</h1>
<section id="untar-rbec-output-tarball" class="level2" data-number="3.1">
<h2 data-number="3.1" class="anchored" data-anchor-id="untar-rbec-output-tarball"><span class="header-section-number">3.1</span> Untar Rbec output tarball</h2>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a>archive<span class="sc">::</span><span class="fu">archive_extract</span>(</span>
<span id="cb4-2"><a href="#cb4-2"></a>  amplicontar,</span>
<span id="cb4-3"><a href="#cb4-3"></a>  <span class="at">dir =</span> tmpdir,</span>
<span id="cb4-4"><a href="#cb4-4"></a>  <span class="at">files =</span> <span class="cn">NULL</span>,</span>
<span id="cb4-5"><a href="#cb4-5"></a>  <span class="at">options =</span> <span class="fu">character</span>(),</span>
<span id="cb4-6"><a href="#cb4-6"></a>  <span class="at">strip_components =</span> <span class="dv">0</span>L</span>
<span id="cb4-7"><a href="#cb4-7"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="setup-directory-structure" class="level2" data-number="3.2">
<h2 data-number="3.2" class="anchored" data-anchor-id="setup-directory-structure"><span class="header-section-number">3.2</span> Setup directory structure</h2>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1"></a>tabdir <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(tmpdir, <span class="st">"rbec_output"</span>)</span>
<span id="cb5-2"><a href="#cb5-2"></a>samppaths <span class="ot">&lt;-</span> fs<span class="sc">::</span><span class="fu">dir_ls</span>(tabdir)</span>
<span id="cb5-3"><a href="#cb5-3"></a>sampnames <span class="ot">&lt;-</span> <span class="fu">path_split</span>(samppaths) <span class="sc">%&gt;%</span> </span>
<span id="cb5-4"><a href="#cb5-4"></a>  <span class="fu">map_chr</span>(dplyr<span class="sc">::</span>last)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="read" class="level2" data-number="3.3">
<h2 data-number="3.3" class="anchored" data-anchor-id="read"><span class="header-section-number">3.3</span> Read</h2>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1"></a>straintabs <span class="ot">&lt;-</span> <span class="fu">paste0</span>(samppaths, <span class="st">"/strain_table.txt"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb6-2"><a href="#cb6-2"></a>  <span class="fu">set_names</span>(sampnames) <span class="sc">%&gt;%</span> </span>
<span id="cb6-3"><a href="#cb6-3"></a>  <span class="fu">map_df</span>(</span>
<span id="cb6-4"><a href="#cb6-4"></a>  read_tsv,</span>
<span id="cb6-5"><a href="#cb6-5"></a>  <span class="at">skip =</span> <span class="dv">1</span>,</span>
<span id="cb6-6"><a href="#cb6-6"></a>  <span class="at">col_names =</span> <span class="fu">c</span>(<span class="st">"rRNA16S_locus"</span>,<span class="st">"count"</span>),</span>
<span id="cb6-7"><a href="#cb6-7"></a>  <span class="at">show_col_types =</span> <span class="cn">FALSE</span>, </span>
<span id="cb6-8"><a href="#cb6-8"></a>  <span class="at">.id =</span> <span class="st">"sample"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb6-9"><a href="#cb6-9"></a>  <span class="co"># naming scheme inconsistent for one sample</span></span>
<span id="cb6-10"><a href="#cb6-10"></a>  <span class="fu">mutate</span>(<span class="at">sample =</span> <span class="fu">if_else</span>(sample <span class="sc">==</span> <span class="st">"P2_s_0"</span>, <span class="st">"P02_s_0"</span>, sample))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="clean-up" class="level2" data-number="3.4">
<h2 data-number="3.4" class="anchored" data-anchor-id="clean-up"><span class="header-section-number">3.4</span> Clean up</h2>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1"></a><span class="co"># remove decompressed directory from temp location</span></span>
<span id="cb7-2"><a href="#cb7-2"></a>fs<span class="sc">::</span><span class="fu">dir_delete</span>(tmpdir)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</section>
<section id="format" class="level1" data-number="4">
<h1 data-number="4"><span class="header-section-number">4</span> Format</h1>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1"></a>straintabs_norm <span class="ot">&lt;-</span> straintabs <span class="sc">%&gt;%</span> </span>
<span id="cb8-2"><a href="#cb8-2"></a>  <span class="co"># Calls function that normalize counts by 16S copy number</span></span>
<span id="cb8-3"><a href="#cb8-3"></a>  <span class="fu">normalize_by_copy</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb8-4"><a href="#cb8-4"></a>  <span class="co"># Calls function that completes all combinations of 23 species</span></span>
<span id="cb8-5"><a href="#cb8-5"></a>  <span class="co"># this is important because some species go extinct that should be in the samples</span></span>
<span id="cb8-6"><a href="#cb8-6"></a>  <span class="co"># and we need to have those in the final table</span></span>
<span id="cb8-7"><a href="#cb8-7"></a>  <span class="fu">complete_combos</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Make final table</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1"></a><span class="co"># Later we will take advantage of the fact that for species not supposed to be</span></span>
<span id="cb9-2"><a href="#cb9-2"></a><span class="co"># in a sample the prior left_join will have filled the evo_hist category with an</span></span>
<span id="cb9-3"><a href="#cb9-3"></a><span class="co"># NA. We can then filter using this NA value</span></span>
<span id="cb9-4"><a href="#cb9-4"></a>finaltable <span class="ot">&lt;-</span> <span class="fu">left_join</span>(straintabs_norm, mddf, <span class="at">by =</span> <span class="fu">join_by</span>(sample)) <span class="sc">%&gt;%</span> </span>
<span id="cb9-5"><a href="#cb9-5"></a>  <span class="fu">left_join</span>(spdf, <span class="at">by =</span> <span class="fu">join_by</span>(community_id, strainID)) <span class="sc">%&gt;%</span> </span>
<span id="cb9-6"><a href="#cb9-6"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb9-7"><a href="#cb9-7"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>genus, <span class="sc">-</span>species) <span class="sc">%&gt;%</span> </span>
<span id="cb9-8"><a href="#cb9-8"></a>  <span class="fu">mutate</span>(<span class="at">f_raw =</span> count_correct<span class="sc">/</span><span class="fu">sum</span>(count_correct),</span>
<span id="cb9-9"><a href="#cb9-9"></a>         <span class="at">sample_tot_raw =</span> <span class="fu">sum</span>(count_correct)) <span class="sc">%&gt;%</span> </span>
<span id="cb9-10"><a href="#cb9-10"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb9-11"><a href="#cb9-11"></a>  <span class="fu">relocate</span>(sample_tot_raw, f_raw, target_f, evo_hist, <span class="at">.after =</span> count_correct)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="analysis" class="level1" data-number="5">
<h1 data-number="5"><span class="header-section-number">5</span> Analysis</h1>
<p>Let’s quickly compare and look the at number of reads from the different kinds of experiment categories. Negative controls are samples from the experiment that contained growth medium but no cells. We included them to make sure that there was not contamination between wells during the experiment. Positive controls here contain defined mixtures of the 4 species used in this experiment. These positive control samples were extracted using Qiagen DNeasy kit, with the idea that we would use them to calibrate the remaining experimental samples which were all extracted using a simplified boil-prep methodology. The workflow in <a href="../../../../R/communities/amplicon/20240711_BTK_illumina_v3/02_correct_boil_to_qiagen.html">this document</a> does this and it shows that the community composition of the same sample extracted using Qiagen or the boil-prep is not drastically different. Thus, we decided to forgo the hassle and expense of doing these calibrations for the remaining samples, but that information is there to demonstrate that our custom extraction proceedure is robust. Experiment samples are those that came from the actual experiment, and masterplate samples are the defined mixtures we made to inoculate the experiment. We sequenced these masterplates so that we would have T0 information for each experimental treatment.</p>
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1"></a>finaltable <span class="sc">%&gt;%</span> </span>
<span id="cb10-2"><a href="#cb10-2"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(sample, sample_tot_raw, community_type) <span class="sc">%&gt;%</span> </span>
<span id="cb10-3"><a href="#cb10-3"></a>  <span class="fu">distinct</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb10-4"><a href="#cb10-4"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> sample_tot_raw)) <span class="sc">+</span></span>
<span id="cb10-5"><a href="#cb10-5"></a>  <span class="fu">geom_histogram</span>(<span class="at">bins =</span> <span class="dv">20</span>, <span class="fu">aes</span>(<span class="at">fill =</span> community_type)) <span class="sc">+</span></span>
<span id="cb10-6"><a href="#cb10-6"></a>  <span class="fu">scale_x_log10</span>() <span class="sc">+</span> </span>
<span id="cb10-7"><a href="#cb10-7"></a>  <span class="fu">annotation_logticks</span>(<span class="at">sides =</span> <span class="st">"b"</span>, <span class="at">color=</span><span class="st">"grey30"</span>) <span class="sc">+</span></span>
<span id="cb10-8"><a href="#cb10-8"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">"Sample count"</span>, <span class="at">x =</span> <span class="st">"Total reads per sample"</span>) <span class="sc">+</span></span>
<span id="cb10-9"><a href="#cb10-9"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span> community_type, <span class="at">scales=</span><span class="st">"free_y"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div id="fig-01" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-01-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="01_format_rbec_tab_files/figure-html/unnamed-chunk-10-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-01-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;1: Frequency distribution of total read counts per sample in the experimental samples (red) masterplates (green), negative controls (blue), and the positive controls (purple).
</figcaption>
</figure>
</div>
<p>The smallest number of reads in the experimental samples is 137 and the smallest in the negative controls is 73. Clearly some of the experimental samples failed to sequence well. The mean number of reads in the experimental samples is 10240.64 and in the negative controls is 3447.5. The mean for the experimental samples is considerably higher than the mean for the negative controls.</p>
<p>Overall this looks really good. The masterplate samples and positive controls all seem to have succeeded. However, there appears to be a negative control or two which has ~ 10000 reads and so it might have been contaminated.</p>
<section id="negative-controls" class="level2" data-number="5.1">
<h2 data-number="5.1" class="anchored" data-anchor-id="negative-controls"><span class="header-section-number">5.1</span> Negative controls</h2>
<p>Lets check what species are contaminating negative controls. Most appear to be low abundance contaminants of species excluded from the experiment. Probably this is cross talk from the positive controls and other samples</p>
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1"></a>finaltable <span class="sc">%&gt;%</span> </span>
<span id="cb11-2"><a href="#cb11-2"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(community_type, <span class="st">"^neg"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb11-3"><a href="#cb11-3"></a>  <span class="fu">filter</span>(f_raw <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb11-4"><a href="#cb11-4"></a>  <span class="fu">contam_histogram</span>(<span class="at">trans=</span><span class="cn">TRUE</span>, <span class="at">x =</span> f_raw) <span class="sc">+</span></span>
<span id="cb11-5"><a href="#cb11-5"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Frequency distribution of each species in the negative controls"</span>, <span class="at">y =</span> <span class="st">"Sample count"</span>) <span class="sc">+</span></span>
<span id="cb11-6"><a href="#cb11-6"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>strainID)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div id="fig-02" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-02-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="01_format_rbec_tab_files/figure-html/unnamed-chunk-11-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-02-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;2: Frequency distribution of all species in the negative controls
</figcaption>
</figure>
</div>
<p>I am not sure where the off-target species are coming from because we did not include any positive community controls here. However, they are all very low abundance and so I think they are just noise. Also it is worth noting that HAMBI_1972 and HAMBI_1977 are the same genus so if some of the read qualities are crappy it could easily be missassigned.</p>
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1"></a>finaltable <span class="sc">%&gt;%</span> </span>
<span id="cb12-2"><a href="#cb12-2"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(community_type, <span class="st">"^neg"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb12-3"><a href="#cb12-3"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(strainID, <span class="st">"0403|1287|1896|1977"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb12-4"><a href="#cb12-4"></a>  <span class="fu">filter</span>(f_raw <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb12-5"><a href="#cb12-5"></a>  <span class="fu">contam_histogram</span>(<span class="at">trans=</span><span class="cn">TRUE</span>, <span class="at">x =</span> f_raw) <span class="sc">+</span></span>
<span id="cb12-6"><a href="#cb12-6"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Frequency distribution of each species in the negative controls"</span>, <span class="at">y =</span> <span class="st">"Sample count"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div id="fig-03" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-03-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="01_format_rbec_tab_files/figure-html/unnamed-chunk-12-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-03-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3: Frequency distribution of only target species in the negative controls
</figcaption>
</figure>
</div>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1"></a>finaltable <span class="sc">%&gt;%</span> </span>
<span id="cb13-2"><a href="#cb13-2"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(community_type, <span class="st">"^neg"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb13-3"><a href="#cb13-3"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(strainID, <span class="st">"0403|1287|1896|1977"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["sample"],"name":[1],"type":["chr"],"align":["left"]},{"label":["strainID"],"name":[2],"type":["chr"],"align":["left"]},{"label":["count"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["count_correct"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["sample_tot_raw"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["f_raw"],"name":[6],"type":["dbl"],"align":["right"]},{"label":["target_f"],"name":[7],"type":["dbl"],"align":["right"]},{"label":["evo_hist"],"name":[8],"type":["chr"],"align":["left"]},{"label":["community_id"],"name":[9],"type":["chr"],"align":["left"]},{"label":["n_species"],"name":[10],"type":["dbl"],"align":["right"]},{"label":["community_type"],"name":[11],"type":["chr"],"align":["left"]},{"label":["transfers"],"name":[12],"type":["dbl"],"align":["right"]},{"label":["strep_conc"],"name":[13],"type":["dbl"],"align":["right"]},{"label":["replicate"],"name":[14],"type":["dbl"],"align":["right"]},{"label":["plate_well"],"name":[15],"type":["chr"],"align":["left"]},{"label":["notes"],"name":[16],"type":["chr"],"align":["left"]}],"data":[{"1":"neg_1_0","2":"HAMBI_0403","3":"34","4":"22","5":"116","6":"0.1896551724","7":"NA","8":"NA","9":"pairs_0_1_G02","10":"0","11":"neg_ctrl","12":"NA","13":"NA","14":"1","15":"08A","16":"These samples were extracted from blank wells on the pairs plate. Goal was to use this to check for contamination during the experiment"},{"1":"neg_1_0","2":"HAMBI_1287","3":"16","4":"14","5":"116","6":"0.1206896552","7":"NA","8":"NA","9":"pairs_0_1_G02","10":"0","11":"neg_ctrl","12":"NA","13":"NA","14":"1","15":"08A","16":"These samples were extracted from blank wells on the pairs plate. Goal was to use this to check for contamination during the experiment"},{"1":"neg_1_0","2":"HAMBI_1896","3":"5","4":"7","5":"116","6":"0.0603448276","7":"NA","8":"NA","9":"pairs_0_1_G02","10":"0","11":"neg_ctrl","12":"NA","13":"NA","14":"1","15":"08A","16":"These samples were extracted from blank wells on the pairs plate. Goal was to use this to check for contamination during the experiment"},{"1":"neg_1_0","2":"HAMBI_1977","3":"61","4":"73","5":"116","6":"0.6293103448","7":"NA","8":"NA","9":"pairs_0_1_G02","10":"0","11":"neg_ctrl","12":"NA","13":"NA","14":"1","15":"08A","16":"These samples were extracted from blank wells on the pairs plate. Goal was to use this to check for contamination during the experiment"},{"1":"neg_2_0","2":"HAMBI_0403","3":"28","4":"16","5":"9018","6":"0.0017742293","7":"NA","8":"NA","9":"pairs_0_1_D05","10":"0","11":"neg_ctrl","12":"NA","13":"NA","14":"2","15":"08B","16":"These samples were extracted from blank wells on the pairs plate. Goal was to use this to check for contamination during the experiment"},{"1":"neg_2_0","2":"HAMBI_1287","3":"7","4":"5","5":"9018","6":"0.0005544467","7":"NA","8":"NA","9":"pairs_0_1_D05","10":"0","11":"neg_ctrl","12":"NA","13":"NA","14":"2","15":"08B","16":"These samples were extracted from blank wells on the pairs plate. Goal was to use this to check for contamination during the experiment"},{"1":"neg_2_0","2":"HAMBI_1896","3":"8","4":"10","5":"9018","6":"0.0011088933","7":"NA","8":"NA","9":"pairs_0_1_D05","10":"0","11":"neg_ctrl","12":"NA","13":"NA","14":"2","15":"08B","16":"These samples were extracted from blank wells on the pairs plate. Goal was to use this to check for contamination during the experiment"},{"1":"neg_2_0","2":"HAMBI_1977","3":"8975","4":"8987","5":"9018","6":"0.9965624307","7":"NA","8":"NA","9":"pairs_0_1_D05","10":"0","11":"neg_ctrl","12":"NA","13":"NA","14":"2","15":"08B","16":"These samples were extracted from blank wells on the pairs plate. Goal was to use this to check for contamination during the experiment"},{"1":"neg_3_0","2":"HAMBI_0403","3":"12","4":"9","5":"9213","6":"0.0009768805","7":"NA","8":"NA","9":"pairs_0_1_G08","10":"0","11":"neg_ctrl","12":"NA","13":"NA","14":"3","15":"08C","16":"These samples were extracted from blank wells on the pairs plate. Goal was to use this to check for contamination during the experiment"},{"1":"neg_3_0","2":"HAMBI_1287","3":"9130","4":"9101","5":"9213","6":"0.9878432650","7":"NA","8":"NA","9":"pairs_0_1_G08","10":"0","11":"neg_ctrl","12":"NA","13":"NA","14":"3","15":"08C","16":"These samples were extracted from blank wells on the pairs plate. Goal was to use this to check for contamination during the experiment"},{"1":"neg_3_0","2":"HAMBI_1896","3":"8","4":"14","5":"9213","6":"0.0015195919","7":"NA","8":"NA","9":"pairs_0_1_G08","10":"0","11":"neg_ctrl","12":"NA","13":"NA","14":"3","15":"08C","16":"These samples were extracted from blank wells on the pairs plate. Goal was to use this to check for contamination during the experiment"},{"1":"neg_3_0","2":"HAMBI_1977","3":"64","4":"89","5":"9213","6":"0.0096602627","7":"NA","8":"NA","9":"pairs_0_1_G08","10":"0","11":"neg_ctrl","12":"NA","13":"NA","14":"3","15":"08C","16":"These samples were extracted from blank wells on the pairs plate. Goal was to use this to check for contamination during the experiment"},{"1":"neg_4_0","2":"HAMBI_0403","3":"0","4":"0","5":"8767","6":"0.0000000000","7":"NA","8":"NA","9":"pairs_0_1_C12","10":"0","11":"neg_ctrl","12":"NA","13":"NA","14":"4","15":"08D","16":"These samples were extracted from blank wells on the pairs plate. Goal was to use this to check for contamination during the experiment"},{"1":"neg_4_0","2":"HAMBI_1287","3":"21","4":"15","5":"8767","6":"0.0017109616","7":"NA","8":"NA","9":"pairs_0_1_C12","10":"0","11":"neg_ctrl","12":"NA","13":"NA","14":"4","15":"08D","16":"These samples were extracted from blank wells on the pairs plate. Goal was to use this to check for contamination during the experiment"},{"1":"neg_4_0","2":"HAMBI_1896","3":"597","4":"734","5":"8767","6":"0.0837230524","7":"NA","8":"NA","9":"pairs_0_1_C12","10":"0","11":"neg_ctrl","12":"NA","13":"NA","14":"4","15":"08D","16":"These samples were extracted from blank wells on the pairs plate. Goal was to use this to check for contamination during the experiment"},{"1":"neg_4_0","2":"HAMBI_1977","3":"8149","4":"8018","5":"8767","6":"0.9145659861","7":"NA","8":"NA","9":"pairs_0_1_C12","10":"0","11":"neg_ctrl","12":"NA","13":"NA","14":"4","15":"08D","16":"These samples were extracted from blank wells on the pairs plate. Goal was to use this to check for contamination during the experiment"},{"1":"neg_5_0","2":"HAMBI_0403","3":"18","4":"11","5":"98","6":"0.1122448980","7":"NA","8":"NA","9":"pairs_0_1_B01","10":"0","11":"neg_ctrl","12":"NA","13":"NA","14":"5","15":"08E","16":"These samples were extracted from blank wells on the pairs plate. Goal was to use this to check for contamination during the experiment"},{"1":"neg_5_0","2":"HAMBI_1287","3":"6","4":"5","5":"98","6":"0.0510204082","7":"NA","8":"NA","9":"pairs_0_1_B01","10":"0","11":"neg_ctrl","12":"NA","13":"NA","14":"5","15":"08E","16":"These samples were extracted from blank wells on the pairs plate. Goal was to use this to check for contamination during the experiment"},{"1":"neg_5_0","2":"HAMBI_1896","3":"18","4":"24","5":"98","6":"0.2448979592","7":"NA","8":"NA","9":"pairs_0_1_B01","10":"0","11":"neg_ctrl","12":"NA","13":"NA","14":"5","15":"08E","16":"These samples were extracted from blank wells on the pairs plate. Goal was to use this to check for contamination during the experiment"},{"1":"neg_5_0","2":"HAMBI_1977","3":"54","4":"57","5":"98","6":"0.5816326531","7":"NA","8":"NA","9":"pairs_0_1_B01","10":"0","11":"neg_ctrl","12":"NA","13":"NA","14":"5","15":"08E","16":"These samples were extracted from blank wells on the pairs plate. Goal was to use this to check for contamination during the experiment"},{"1":"neg_6_0","2":"HAMBI_0403","3":"8","4":"5","5":"73","6":"0.0684931507","7":"NA","8":"NA","9":"pairs_0_1_E04","10":"0","11":"neg_ctrl","12":"NA","13":"NA","14":"6","15":"08F","16":"These samples were extracted from blank wells on the pairs plate. Goal was to use this to check for contamination during the experiment"},{"1":"neg_6_0","2":"HAMBI_1287","3":"3","4":"2","5":"73","6":"0.0273972603","7":"NA","8":"NA","9":"pairs_0_1_E04","10":"0","11":"neg_ctrl","12":"NA","13":"NA","14":"6","15":"08F","16":"These samples were extracted from blank wells on the pairs plate. Goal was to use this to check for contamination during the experiment"},{"1":"neg_6_0","2":"HAMBI_1896","3":"9","4":"12","5":"73","6":"0.1643835616","7":"NA","8":"NA","9":"pairs_0_1_E04","10":"0","11":"neg_ctrl","12":"NA","13":"NA","14":"6","15":"08F","16":"These samples were extracted from blank wells on the pairs plate. Goal was to use this to check for contamination during the experiment"},{"1":"neg_6_0","2":"HAMBI_1977","3":"52","4":"54","5":"73","6":"0.7397260274","7":"NA","8":"NA","9":"pairs_0_1_E04","10":"0","11":"neg_ctrl","12":"NA","13":"NA","14":"6","15":"08F","16":"These samples were extracted from blank wells on the pairs plate. Goal was to use this to check for contamination during the experiment"},{"1":"neg_7_0","2":"HAMBI_0403","3":"19","4":"14","5":"106","6":"0.1320754717","7":"NA","8":"NA","9":"pairs_0_1_H07","10":"0","11":"neg_ctrl","12":"NA","13":"NA","14":"7","15":"08G","16":"These samples were extracted from blank wells on the pairs plate. Goal was to use this to check for contamination during the experiment"},{"1":"neg_7_0","2":"HAMBI_1287","3":"63","4":"60","5":"106","6":"0.5660377358","7":"NA","8":"NA","9":"pairs_0_1_H07","10":"0","11":"neg_ctrl","12":"NA","13":"NA","14":"7","15":"08G","16":"These samples were extracted from blank wells on the pairs plate. Goal was to use this to check for contamination during the experiment"},{"1":"neg_7_0","2":"HAMBI_1896","3":"7","4":"12","5":"106","6":"0.1132075472","7":"NA","8":"NA","9":"pairs_0_1_H07","10":"0","11":"neg_ctrl","12":"NA","13":"NA","14":"7","15":"08G","16":"These samples were extracted from blank wells on the pairs plate. Goal was to use this to check for contamination during the experiment"},{"1":"neg_7_0","2":"HAMBI_1977","3":"13","4":"17","5":"106","6":"0.1603773585","7":"NA","8":"NA","9":"pairs_0_1_H07","10":"0","11":"neg_ctrl","12":"NA","13":"NA","14":"7","15":"08G","16":"These samples were extracted from blank wells on the pairs plate. Goal was to use this to check for contamination during the experiment"},{"1":"neg_8_0","2":"HAMBI_0403","3":"23","4":"17","5":"189","6":"0.0899470899","7":"NA","8":"NA","9":"pairs_0_1_E10","10":"0","11":"neg_ctrl","12":"NA","13":"NA","14":"8","15":"08H","16":"These samples were extracted from blank wells on the pairs plate. Goal was to use this to check for contamination during the experiment"},{"1":"neg_8_0","2":"HAMBI_1287","3":"92","4":"86","5":"189","6":"0.4550264550","7":"NA","8":"NA","9":"pairs_0_1_E10","10":"0","11":"neg_ctrl","12":"NA","13":"NA","14":"8","15":"08H","16":"These samples were extracted from blank wells on the pairs plate. Goal was to use this to check for contamination during the experiment"},{"1":"neg_8_0","2":"HAMBI_1896","3":"25","4":"41","5":"189","6":"0.2169312169","7":"NA","8":"NA","9":"pairs_0_1_E10","10":"0","11":"neg_ctrl","12":"NA","13":"NA","14":"8","15":"08H","16":"These samples were extracted from blank wells on the pairs plate. Goal was to use this to check for contamination during the experiment"},{"1":"neg_8_0","2":"HAMBI_1977","3":"18","4":"24","5":"189","6":"0.1269841270","7":"NA","8":"NA","9":"pairs_0_1_E10","10":"0","11":"neg_ctrl","12":"NA","13":"NA","14":"8","15":"08H","16":"These samples were extracted from blank wells on the pairs plate. Goal was to use this to check for contamination during the experiment"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<p>This looks ok, but there are potentially some problems. Specifically, negative control replicates 2, 3, and 4 all have some contamination. <code>neg_2_0</code> seems to be contaminated with HAMBI_1977, <code>neg_3_0</code> with HAMBI_1287, and <code>neg_4_0</code> with HAMBI_1977. However in the 5 remaining negative controls there is no contamination.</p>
</section>
<section id="positive-controls" class="level2" data-number="5.2">
<h2 data-number="5.2" class="anchored" data-anchor-id="positive-controls"><span class="header-section-number">5.2</span> Positive controls</h2>
<p>The positive controls should each have three species. In all cases the species that shouldn’t be there is very rare</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1"></a>finaltable <span class="sc">%&gt;%</span> </span>
<span id="cb14-2"><a href="#cb14-2"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(community_type, <span class="st">"^pos"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb14-3"><a href="#cb14-3"></a>  <span class="fu">mutate</span>(<span class="at">total_pos_controls =</span> <span class="fu">n_distinct</span>(sample)) <span class="sc">%&gt;%</span> </span>
<span id="cb14-4"><a href="#cb14-4"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb14-5"><a href="#cb14-5"></a>  <span class="fu">mutate</span>(<span class="at">f =</span> <span class="fu">round</span>(f_raw<span class="sc">*</span><span class="dv">100</span>, <span class="dv">1</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb14-6"><a href="#cb14-6"></a>  <span class="fu">mutate</span>(<span class="at">supposed_2_b_there =</span> <span class="fu">if_else</span>(<span class="fu">is.na</span>(evo_hist), <span class="st">"no"</span>, <span class="st">"yes"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb14-7"><a href="#cb14-7"></a>  <span class="fu">relocate</span>(f, supposed_2_b_there) <span class="sc">%&gt;%</span> </span>
<span id="cb14-8"><a href="#cb14-8"></a>  <span class="fu">distinct</span>(f, supposed_2_b_there, sample, strainID, count_correct, n_species, community_type,) <span class="sc">%&gt;%</span> </span>
<span id="cb14-9"><a href="#cb14-9"></a>  <span class="fu">filter</span>(f <span class="sc">&gt;</span> <span class="dv">0</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["f"],"name":[1],"type":["dbl"],"align":["right"]},{"label":["supposed_2_b_there"],"name":[2],"type":["chr"],"align":["left"]},{"label":["sample"],"name":[3],"type":["chr"],"align":["left"]},{"label":["strainID"],"name":[4],"type":["chr"],"align":["left"]},{"label":["count_correct"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["n_species"],"name":[6],"type":["dbl"],"align":["right"]},{"label":["community_type"],"name":[7],"type":["chr"],"align":["left"]}],"data":[{"1":"88.8","2":"yes","3":"3sp_1_0","4":"HAMBI_0403","5":"10766","6":"3","7":"pos_ctrl"},{"1":"5.8","2":"yes","3":"3sp_1_0","4":"HAMBI_1287","5":"698","6":"3","7":"pos_ctrl"},{"1":"5.1","2":"yes","3":"3sp_1_0","4":"HAMBI_1896","5":"620","6":"3","7":"pos_ctrl"},{"1":"0.3","2":"no","3":"3sp_1_0","4":"HAMBI_1977","5":"41","6":"3","7":"pos_ctrl"},{"1":"27.7","2":"yes","3":"3sp_2_0","4":"HAMBI_0403","5":"3406","6":"3","7":"pos_ctrl"},{"1":"9.8","2":"yes","3":"3sp_2_0","4":"HAMBI_1287","5":"1206","6":"3","7":"pos_ctrl"},{"1":"62.3","2":"yes","3":"3sp_2_0","4":"HAMBI_1896","5":"7667","6":"3","7":"pos_ctrl"},{"1":"0.3","2":"no","3":"3sp_2_0","4":"HAMBI_1977","5":"37","6":"3","7":"pos_ctrl"},{"1":"91.1","2":"yes","3":"3sp_3_0","4":"HAMBI_0403","5":"11796","6":"3","7":"pos_ctrl"},{"1":"5.7","2":"yes","3":"3sp_3_0","4":"HAMBI_1287","5":"738","6":"3","7":"pos_ctrl"},{"1":"0.3","2":"no","3":"3sp_3_0","4":"HAMBI_1896","5":"42","6":"3","7":"pos_ctrl"},{"1":"2.9","2":"yes","3":"3sp_3_0","4":"HAMBI_1977","5":"373","6":"3","7":"pos_ctrl"},{"1":"9.3","2":"yes","3":"3sp_4_0","4":"HAMBI_0403","5":"834","6":"3","7":"pos_ctrl"},{"1":"16.1","2":"yes","3":"3sp_4_0","4":"HAMBI_1287","5":"1435","6":"3","7":"pos_ctrl"},{"1":"0.6","2":"no","3":"3sp_4_0","4":"HAMBI_1896","5":"51","6":"3","7":"pos_ctrl"},{"1":"74.0","2":"yes","3":"3sp_4_0","4":"HAMBI_1977","5":"6618","6":"3","7":"pos_ctrl"},{"1":"23.9","2":"yes","3":"3sp_5_0","4":"HAMBI_0403","5":"2643","6":"3","7":"pos_ctrl"},{"1":"0.1","2":"no","3":"3sp_5_0","4":"HAMBI_1287","5":"6","6":"3","7":"pos_ctrl"},{"1":"69.3","2":"yes","3":"3sp_5_0","4":"HAMBI_1896","5":"7661","6":"3","7":"pos_ctrl"},{"1":"6.8","2":"yes","3":"3sp_5_0","4":"HAMBI_1977","5":"751","6":"3","7":"pos_ctrl"},{"1":"33.9","2":"yes","3":"3sp_6_0","4":"HAMBI_0403","5":"2927","6":"3","7":"pos_ctrl"},{"1":"0.1","2":"no","3":"3sp_6_0","4":"HAMBI_1287","5":"8","6":"3","7":"pos_ctrl"},{"1":"12.9","2":"yes","3":"3sp_6_0","4":"HAMBI_1896","5":"1113","6":"3","7":"pos_ctrl"},{"1":"53.2","2":"yes","3":"3sp_6_0","4":"HAMBI_1977","5":"4593","6":"3","7":"pos_ctrl"},{"1":"0.2","2":"no","3":"3sp_7_0","4":"HAMBI_0403","5":"33","6":"3","7":"pos_ctrl"},{"1":"75.9","2":"yes","3":"3sp_7_0","4":"HAMBI_1287","5":"10482","6":"3","7":"pos_ctrl"},{"1":"16.5","2":"yes","3":"3sp_7_0","4":"HAMBI_1896","5":"2278","6":"3","7":"pos_ctrl"},{"1":"7.4","2":"yes","3":"3sp_7_0","4":"HAMBI_1977","5":"1019","6":"3","7":"pos_ctrl"},{"1":"0.2","2":"no","3":"3sp_8_0","4":"HAMBI_0403","5":"31","6":"3","7":"pos_ctrl"},{"1":"11.8","2":"yes","3":"3sp_8_0","4":"HAMBI_1287","5":"1638","6":"3","7":"pos_ctrl"},{"1":"82.4","2":"yes","3":"3sp_8_0","4":"HAMBI_1896","5":"11396","6":"3","7":"pos_ctrl"},{"1":"5.5","2":"yes","3":"3sp_8_0","4":"HAMBI_1977","5":"763","6":"3","7":"pos_ctrl"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<p>This is also good - we detect all 3 species that should be there in the positive controls on each plate.</p>
</section>
<section id="misassigned-reads" class="level2" data-number="5.3">
<h2 data-number="5.3" class="anchored" data-anchor-id="misassigned-reads"><span class="header-section-number">5.3</span> Misassigned reads</h2>
<p>These libraries were only prepared with samples from Milla’s 4-species experiment with 403, 1287, 1896, and 1977 so any time species other than these show up is just an incorrect assignment by Rbec. Let’s check quickly how many of these there are…</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1"></a>finaltable <span class="sc">%&gt;%</span> </span>
<span id="cb15-2"><a href="#cb15-2"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">str_detect</span>(strainID, <span class="st">"0403|1287|1896|1977"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb15-3"><a href="#cb15-3"></a>  <span class="fu">filter</span>(count_correct <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb15-4"><a href="#cb15-4"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(count_correct))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["sample"],"name":[1],"type":["chr"],"align":["left"]},{"label":["strainID"],"name":[2],"type":["chr"],"align":["left"]},{"label":["count"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["count_correct"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["sample_tot_raw"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["f_raw"],"name":[6],"type":["dbl"],"align":["right"]},{"label":["target_f"],"name":[7],"type":["dbl"],"align":["right"]},{"label":["evo_hist"],"name":[8],"type":["chr"],"align":["left"]},{"label":["community_id"],"name":[9],"type":["chr"],"align":["left"]},{"label":["n_species"],"name":[10],"type":["dbl"],"align":["right"]},{"label":["community_type"],"name":[11],"type":["chr"],"align":["left"]},{"label":["transfers"],"name":[12],"type":["dbl"],"align":["right"]},{"label":["strep_conc"],"name":[13],"type":["dbl"],"align":["right"]},{"label":["replicate"],"name":[14],"type":["dbl"],"align":["right"]},{"label":["plate_well"],"name":[15],"type":["chr"],"align":["left"]},{"label":["notes"],"name":[16],"type":["chr"],"align":["left"]}],"data":[{"1":"neg_8_0","2":"HAMBI_1972","3":"29","4":"19","5":"189","6":"1.005291e-01","7":"NA","8":"NA","9":"pairs_0_1_E10","10":"0","11":"neg_ctrl","12":"NA","13":"NA","14":"8","15":"08H","16":"These samples were extracted from blank wells on the pairs plate. Goal was to use this to check for contamination during the experiment"},{"1":"T23_s_0","2":"HAMBI_1972","3":"5","4":"3","5":"9177","6":"3.269042e-04","7":"NA","8":"NA","9":"T23","10":"3","11":"masterplate","12":"0","13":"NA","14":"1","15":"06D","16":"Masterplate samples are those that were constructed ahead of time, then thawed and used in the experiment. They had no time to grow and thus should contain only what we put into them in roughly the proportions we intended"},{"1":"T28_2_0","2":"HAMBI_3031","3":"1","4":"3","5":"6900","6":"4.347826e-04","7":"NA","8":"NA","9":"T28","10":"3","11":"experiment","12":"8","13":"0","14":"2","15":"03F","16":"NA"},{"1":"neg_7_0","2":"HAMBI_1972","3":"3","4":"2","5":"106","6":"1.886792e-02","7":"NA","8":"NA","9":"pairs_0_1_H07","10":"0","11":"neg_ctrl","12":"NA","13":"NA","14":"7","15":"08G","16":"These samples were extracted from blank wells on the pairs plate. Goal was to use this to check for contamination during the experiment"},{"1":"neg_8_0","2":"HAMBI_0105","3":"1","4":"2","5":"189","6":"1.058201e-02","7":"NA","8":"NA","9":"pairs_0_1_E10","10":"0","11":"neg_ctrl","12":"NA","13":"NA","14":"8","15":"08H","16":"These samples were extracted from blank wells on the pairs plate. Goal was to use this to check for contamination during the experiment"},{"1":"P27_s_0","2":"HAMBI_2659","3":"2","4":"2","5":"6444","6":"3.103662e-04","7":"NA","8":"NA","9":"P27","10":"2","11":"masterplate","12":"0","13":"NA","14":"1","15":"06A","16":"Masterplate samples are those that were constructed ahead of time, then thawed and used in the experiment. They had no time to grow and thus should contain only what we put into them in roughly the proportions we intended"},{"1":"P36_s_0","2":"HAMBI_2659","3":"1","4":"2","5":"5784","6":"3.457815e-04","7":"NA","8":"NA","9":"P36","10":"2","11":"masterplate","12":"0","13":"NA","14":"1","15":"05H","16":"Masterplate samples are those that were constructed ahead of time, then thawed and used in the experiment. They had no time to grow and thus should contain only what we put into them in roughly the proportions we intended"},{"1":"neg_5_0","2":"HAMBI_1972","3":"1","4":"1","5":"98","6":"1.020408e-02","7":"NA","8":"NA","9":"pairs_0_1_B01","10":"0","11":"neg_ctrl","12":"NA","13":"NA","14":"5","15":"08E","16":"These samples were extracted from blank wells on the pairs plate. Goal was to use this to check for contamination during the experiment"},{"1":"neg_7_0","2":"HAMBI_1292","3":"1","4":"1","5":"106","6":"9.433962e-03","7":"NA","8":"NA","9":"pairs_0_1_H07","10":"0","11":"neg_ctrl","12":"NA","13":"NA","14":"7","15":"08G","16":"These samples were extracted from blank wells on the pairs plate. Goal was to use this to check for contamination during the experiment"},{"1":"P22_s_0","2":"HAMBI_0006","3":"1","4":"1","5":"5674","6":"1.762425e-04","7":"NA","8":"NA","9":"P22","10":"2","11":"masterplate","12":"0","13":"NA","14":"1","15":"02E","16":"Masterplate samples are those that were constructed ahead of time, then thawed and used in the experiment. They had no time to grow and thus should contain only what we put into them in roughly the proportions we intended"},{"1":"P25_2_256","2":"HAMBI_2659","3":"1","4":"1","5":"407","6":"2.457002e-03","7":"NA","8":"NA","9":"P25","10":"2","11":"experiment","12":"8","13":"256","14":"2","15":"02C","16":"NA"},{"1":"P26_2_0","2":"HAMBI_1972","3":"1","4":"1","5":"9678","6":"1.033271e-04","7":"NA","8":"NA","9":"P26","10":"2","11":"experiment","12":"8","13":"0","14":"2","15":"01D","16":"NA"},{"1":"P27_s_0","2":"HAMBI_0006","3":"1","4":"1","5":"6444","6":"1.551831e-04","7":"NA","8":"NA","9":"P27","10":"2","11":"masterplate","12":"0","13":"NA","14":"1","15":"06A","16":"Masterplate samples are those that were constructed ahead of time, then thawed and used in the experiment. They had no time to grow and thus should contain only what we put into them in roughly the proportions we intended"},{"1":"P32_1_0","2":"HAMBI_2659","3":"1","4":"1","5":"8514","6":"1.174536e-04","7":"NA","8":"NA","9":"P32","10":"2","11":"experiment","12":"8","13":"0","14":"1","15":"04D","16":"NA"},{"1":"P35_2_256","2":"HAMBI_1279","3":"1","4":"1","5":"7089","6":"1.410636e-04","7":"NA","8":"NA","9":"P35","10":"2","11":"experiment","12":"8","13":"256","14":"2","15":"12G","16":"NA"},{"1":"P35_2_256","2":"HAMBI_1299","3":"1","4":"1","5":"7089","6":"1.410636e-04","7":"NA","8":"NA","9":"P35","10":"2","11":"experiment","12":"8","13":"256","14":"2","15":"12G","16":"NA"},{"1":"P36_1_0","2":"HAMBI_1972","3":"2","4":"1","5":"10156","6":"9.846396e-05","7":"NA","8":"NA","9":"P36","10":"2","11":"experiment","12":"8","13":"0","14":"1","15":"12H","16":"NA"},{"1":"P42_s_0","2":"HAMBI_2659","3":"1","4":"1","5":"6672","6":"1.498801e-04","7":"NA","8":"NA","9":"P42","10":"2","11":"masterplate","12":"0","13":"NA","14":"1","15":"05D","16":"Masterplate samples are those that were constructed ahead of time, then thawed and used in the experiment. They had no time to grow and thus should contain only what we put into them in roughly the proportions we intended"},{"1":"P43_s_0","2":"HAMBI_0006","3":"1","4":"1","5":"5658","6":"1.767409e-04","7":"NA","8":"NA","9":"P43","10":"2","11":"masterplate","12":"0","13":"NA","14":"1","15":"06D","16":"Masterplate samples are those that were constructed ahead of time, then thawed and used in the experiment. They had no time to grow and thus should contain only what we put into them in roughly the proportions we intended"},{"1":"P46_s_0","2":"HAMBI_0006","3":"1","4":"1","5":"4846","6":"2.063558e-04","7":"NA","8":"NA","9":"P46","10":"2","11":"masterplate","12":"0","13":"NA","14":"1","15":"05E","16":"Masterplate samples are those that were constructed ahead of time, then thawed and used in the experiment. They had no time to grow and thus should contain only what we put into them in roughly the proportions we intended"},{"1":"P47_2_64","2":"HAMBI_0006","3":"1","4":"1","5":"9604","6":"1.041233e-04","7":"NA","8":"NA","9":"P47","10":"2","11":"experiment","12":"8","13":"64","14":"2","15":"12C","16":"NA"},{"1":"T01_s_0","2":"HAMBI_1972","3":"1","4":"1","5":"8325","6":"1.201201e-04","7":"NA","8":"NA","9":"T01","10":"3","11":"masterplate","12":"0","13":"NA","14":"1","15":"01E","16":"Masterplate samples are those that were constructed ahead of time, then thawed and used in the experiment. They had no time to grow and thus should contain only what we put into them in roughly the proportions we intended"},{"1":"T18_1_0","2":"HAMBI_1972","3":"1","4":"1","5":"11427","6":"8.751203e-05","7":"NA","8":"NA","9":"T18","10":"3","11":"experiment","12":"8","13":"0","14":"1","15":"02B","16":"NA"},{"1":"T25_1_0","2":"HAMBI_1972","3":"1","4":"1","5":"7093","6":"1.409841e-04","7":"NA","8":"NA","9":"T25","10":"3","11":"experiment","12":"8","13":"0","14":"1","15":"03E","16":"NA"},{"1":"T35_s_0","2":"HAMBI_1972","3":"1","4":"1","5":"10098","6":"9.902951e-05","7":"NA","8":"NA","9":"T35","10":"3","11":"masterplate","12":"0","13":"NA","14":"1","15":"07D","16":"Masterplate samples are those that were constructed ahead of time, then thawed and used in the experiment. They had no time to grow and thus should contain only what we put into them in roughly the proportions we intended"},{"1":"T56_1_0","2":"HAMBI_0006","3":"1","4":"1","5":"11451","6":"8.732862e-05","7":"NA","8":"NA","9":"T56","10":"3","11":"experiment","12":"8","13":"0","14":"1","15":"11C","16":"NA"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<p>There is one or two incorrectly assigned reads here and there but this is just noise. We can safely exclude all reads not mapping to one of the focal species.</p>
</section>
<section id="samples-with-few-total-reads" class="level2" data-number="5.4">
<h2 data-number="5.4" class="anchored" data-anchor-id="samples-with-few-total-reads"><span class="header-section-number">5.4</span> Samples with few total reads</h2>
<p>Some of the experimental pairs had streptomycin concentrations higher than any of the species individually could tolerate. We would naively expect then that neither species would grow successfully in these samples and that the overall biomass would be very low, thus resulting in a low number of recovered reads from these samples.</p>
<p>To look into this. first let’s check which samples have very low OD600 in the endpoint samples.</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1"></a>od <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"_data_raw"</span>, <span class="st">"communities"</span>, <span class="st">"optical_density"</span>, <span class="st">"20240606_pairs"</span>, <span class="st">"optical_density_formatted.tsv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>We’ll filter out samples with an OD of less than 0.1 and also samples with fewer than 1000 reads. It is generally good practice to exclude samples with low number of reads.</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1"></a>df <span class="ot">&lt;-</span> <span class="fu">left_join</span>(finaltable, od, </span>
<span id="cb17-2"><a href="#cb17-2"></a>                  <span class="at">by =</span> <span class="fu">join_by</span>(community_id, n_species, transfers, strep_conc, replicate)) <span class="sc">%&gt;%</span> </span>
<span id="cb17-3"><a href="#cb17-3"></a>  <span class="fu">filter</span>(transfers <span class="sc">==</span> <span class="dv">8</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb17-4"><a href="#cb17-4"></a>  <span class="fu">arrange</span>(sample_tot_raw) <span class="sc">%&gt;%</span> </span>
<span id="cb17-5"><a href="#cb17-5"></a>  <span class="fu">arrange</span>(OD) <span class="sc">%&gt;%</span> </span>
<span id="cb17-6"><a href="#cb17-6"></a>  <span class="co"># filter only samples with OD &lt; 0.1 or &lt; 1000 total reads. 1000 read threshold </span></span>
<span id="cb17-7"><a href="#cb17-7"></a>  <span class="co"># was chose based on the first figure in this notebook</span></span>
<span id="cb17-8"><a href="#cb17-8"></a>  <span class="fu">filter</span>(OD <span class="sc">&lt;</span> <span class="fl">0.1</span> <span class="sc">|</span> sample_tot_raw <span class="sc">&lt;</span> <span class="dv">1000</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb17-9"><a href="#cb17-9"></a>  <span class="fu">filter</span>(count_correct <span class="sc">&gt;</span> <span class="dv">0</span>)</span>
<span id="cb17-10"><a href="#cb17-10"></a></span>
<span id="cb17-11"><a href="#cb17-11"></a>lowread_samps <span class="ot">&lt;-</span> df <span class="sc">%&gt;%</span> </span>
<span id="cb17-12"><a href="#cb17-12"></a>  <span class="fu">filter</span>(sample_tot_raw <span class="sc">&lt;</span> <span class="dv">3000</span> <span class="sc">&amp;</span> community_type <span class="sc">==</span> <span class="st">"experiment"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb17-13"><a href="#cb17-13"></a>  <span class="fu">distinct</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb17-14"><a href="#cb17-14"></a>  <span class="fu">pull</span>(sample)</span>
<span id="cb17-15"><a href="#cb17-15"></a>  </span>
<span id="cb17-16"><a href="#cb17-16"></a>df <span class="sc">%&gt;%</span> </span>
<span id="cb17-17"><a href="#cb17-17"></a>  <span class="fu">distinct</span>(sample, community_type, OD, sample_tot_raw)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["sample"],"name":[1],"type":["chr"],"align":["left"]},{"label":["community_type"],"name":[2],"type":["chr"],"align":["left"]},{"label":["OD"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["sample_tot_raw"],"name":[4],"type":["dbl"],"align":["right"]}],"data":[{"1":"P01_1_256","2":"experiment","3":"0.037","4":"208"},{"1":"P25_1_256","2":"experiment","3":"0.039","4":"218"},{"1":"P25_1_64","2":"experiment","3":"0.039","4":"246"},{"1":"P25_2_64","2":"experiment","3":"0.041","4":"378"},{"1":"P01_2_256","2":"experiment","3":"0.042","4":"405"},{"1":"P25_2_256","2":"experiment","3":"0.043","4":"407"},{"1":"P29_1_64","2":"experiment","3":"0.265","4":"168"},{"1":"P11_2_64","2":"experiment","3":"0.270","4":"137"},{"1":"P38_1_0","2":"experiment","3":"0.312","4":"520"},{"1":"T46_1_0","2":"experiment","3":"0.325","4":"551"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<p>There are 10 experimental samples that have less than 1000 reads. 6 of those samples have low optical density which indicates nothing was growing there and that they should have low read number. 4 samples have normal optical density (~ 0.3) and these look to be samples that failed at the library preparation/sequencing step. Regardless, all these 10 samples will be excluded from the downstream analysis.</p>
</section>
<section id="filter-to-target-species" class="level2" data-number="5.5">
<h2 data-number="5.5" class="anchored" data-anchor-id="filter-to-target-species"><span class="header-section-number">5.5</span> Filter to target species</h2>
<p>Exclude positive and negative controls and also exclude species that are not in the focal 4</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1"></a>finaltable_exp_mstr <span class="ot">&lt;-</span> finaltable <span class="sc">%&gt;%</span> </span>
<span id="cb18-2"><a href="#cb18-2"></a>  <span class="fu">filter</span>(community_type <span class="sc">%nin%</span> <span class="fu">c</span>(<span class="st">"empty"</span>, <span class="st">"neg_ctrl"</span>, <span class="st">"pos_ctrl"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb18-3"><a href="#cb18-3"></a>  <span class="co"># also exclude samples with low reads</span></span>
<span id="cb18-4"><a href="#cb18-4"></a>  <span class="fu">filter</span>(sample <span class="sc">%nin%</span> lowread_samps) <span class="sc">%&gt;%</span> </span>
<span id="cb18-5"><a href="#cb18-5"></a>  <span class="fu">filter</span>(strainID <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"HAMBI_0403"</span>, <span class="st">"HAMBI_1287"</span>, <span class="st">"HAMBI_1896"</span>, <span class="st">"HAMBI_1977"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb18-6"><a href="#cb18-6"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb18-7"><a href="#cb18-7"></a>  <span class="fu">mutate</span>(<span class="at">f_raw_targetsp =</span> count_correct<span class="sc">/</span><span class="fu">sum</span>(count_correct),</span>
<span id="cb18-8"><a href="#cb18-8"></a>         <span class="at">sample_tot_targetsp =</span> <span class="fu">sum</span>(count_correct)) <span class="sc">%&gt;%</span> </span>
<span id="cb18-9"><a href="#cb18-9"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb18-10"><a href="#cb18-10"></a>  <span class="fu">relocate</span>(<span class="fu">c</span>(sample_tot_targetsp, f_raw_targetsp), <span class="at">.after =</span> f_raw)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="masterplate-samples" class="level2" data-number="5.6">
<h2 data-number="5.6" class="anchored" data-anchor-id="masterplate-samples"><span class="header-section-number">5.6</span> Masterplate samples</h2>
<p>To set up this experiment, Milla combined the species in the planned proportions on a masterplate. Because this process was time consuming, the masterplate was stored at -80C after construction until the experiment start day when it was taken from the freezer and used to inoculate the experiment. Because Milla knows exactly which strains were added to the master plate and the plates were not allowed to grow, any strains in these samples that are not supposed to be there should be due to Illumina index cross talk and not true contamination. We can get a sense for the average index crosstalk rate from these samples and then draw a threshold of when to exclude likely false positives and when a positive is likely due to contamination.</p>
<section id="pairs" class="level3" data-number="5.6.1">
<h3 data-number="5.6.1" class="anchored" data-anchor-id="pairs"><span class="header-section-number">5.6.1</span> Pairs</h3>
<p>Overall the masterplate pairs look very good. There are no samples where a species that should not be there is present in high abundances. Any contamination (grey to black fill) is well under 5%.</p>
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1"></a>finaltable_exp_mstr <span class="sc">%&gt;%</span></span>
<span id="cb19-2"><a href="#cb19-2"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(community_type, <span class="st">"masterplate"</span>) <span class="sc">&amp;</span> n_species <span class="sc">==</span> <span class="dv">2</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb19-3"><a href="#cb19-3"></a>  <span class="fu">contaminated_barplot</span>(<span class="at">threshold =</span> <span class="dv">0</span>, <span class="at">quartet =</span> <span class="cn">FALSE</span>, <span class="at">y=</span>f_raw_targetsp) <span class="sc">+</span></span>
<span id="cb19-4"><a href="#cb19-4"></a>  <span class="fu">ggtitle</span>(<span class="st">"All Masterplate pairs"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div id="fig-04" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-04-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="01_format_rbec_tab_files/figure-html/unnamed-chunk-19-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-04-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;4: Species composition of all masterplate pairs. Any species that shouldn’t be there (&gt; 0%) is shown in grey to black.
</figcaption>
</figure>
</div>
<p>There are no samples with more than 4% of a species that shouldn’t be there</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1"></a>finaltable_exp_mstr <span class="sc">%&gt;%</span> </span>
<span id="cb20-2"><a href="#cb20-2"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(community_type, <span class="st">"masterplate"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb20-3"><a href="#cb20-3"></a>  <span class="fu">filter</span>(n_species <span class="sc">==</span> <span class="st">"2"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb20-4"><a href="#cb20-4"></a>  <span class="fu">filter</span>(<span class="fu">is.na</span>(evo_hist)) <span class="sc">%&gt;%</span> </span>
<span id="cb20-5"><a href="#cb20-5"></a>  <span class="fu">filter</span>(f_raw_targetsp <span class="sc">&gt;</span> <span class="fl">0.04</span> ) <span class="sc">%&gt;%</span> </span>
<span id="cb20-6"><a href="#cb20-6"></a>  <span class="fu">distinct</span>(sample)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["sample"],"name":[1],"type":["chr"],"align":["left"]}],"data":[],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1"></a>finaltable_exp_mstr <span class="sc">%&gt;%</span></span>
<span id="cb21-2"><a href="#cb21-2"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(community_type, <span class="st">"masterplate"</span>) <span class="sc">&amp;</span> n_species <span class="sc">==</span> <span class="dv">2</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb21-3"><a href="#cb21-3"></a>  <span class="fu">filter</span>(<span class="fu">is.na</span>(evo_hist)) <span class="sc">%&gt;%</span> </span>
<span id="cb21-4"><a href="#cb21-4"></a>  <span class="fu">filter</span>(f_raw_targetsp <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb21-5"><a href="#cb21-5"></a>  <span class="fu">contam_histogram</span>(<span class="at">trans =</span> <span class="cn">TRUE</span>, <span class="at">x =</span> f_raw_targetsp) <span class="sc">+</span></span>
<span id="cb21-6"><a href="#cb21-6"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Species frequency"</span>, <span class="at">y =</span> <span class="st">"Sample count"</span>) <span class="sc">+</span></span>
<span id="cb21-7"><a href="#cb21-7"></a>  <span class="fu">ggtitle</span>(<span class="st">"Frequency distribution of misassigned species in masterplate pairs"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div id="fig-05" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-05-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="01_format_rbec_tab_files/figure-html/unnamed-chunk-21-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-05-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;5: Frequency distribution of each species in masterplate pair samples where it <strong>should not</strong> occur. The 1% mark (commonly accepted as an acceptable Illumina cross-talk standard) is demarcated with a dashed line.
</figcaption>
</figure>
</div>
</section>
<section id="trios" class="level3" data-number="5.6.2">
<h3 data-number="5.6.2" class="anchored" data-anchor-id="trios"><span class="header-section-number">5.6.2</span> Trios</h3>
<p>Trio masterplates also look very good. There are some samples where a species that shouldn’t be there is &gt; 3%, but these are always &lt; 5% and there are no samples with major contamination. Importantly these all match the proportions that we used to start the experiment</p>
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1"></a>finaltable_exp_mstr <span class="sc">%&gt;%</span></span>
<span id="cb22-2"><a href="#cb22-2"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(community_type, <span class="st">"masterplate"</span>) <span class="sc">&amp;</span> n_species <span class="sc">==</span> <span class="dv">3</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb22-3"><a href="#cb22-3"></a>  <span class="co"># this is a dumb hack so that I don't have to plot all bars on the same axis</span></span>
<span id="cb22-4"><a href="#cb22-4"></a>  <span class="co"># but I can wrap them using a meaningless facet</span></span>
<span id="cb22-5"><a href="#cb22-5"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb22-6"><a href="#cb22-6"></a>  <span class="fu">mutate</span>(<span class="at">id =</span> <span class="fu">cur_group_id</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb22-7"><a href="#cb22-7"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb22-8"><a href="#cb22-8"></a>  <span class="fu">mutate</span>(<span class="at">facet =</span> <span class="fu">ntile</span>(id, <span class="dv">4</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb22-9"><a href="#cb22-9"></a>  <span class="fu">contaminated_barplot</span>(<span class="at">threshold =</span> <span class="dv">0</span>, <span class="at">quartet =</span> <span class="cn">FALSE</span>, <span class="at">y=</span>f_raw_targetsp) <span class="sc">+</span></span>
<span id="cb22-10"><a href="#cb22-10"></a>  <span class="fu">ggtitle</span>(<span class="st">"All Masterplate trios"</span>) <span class="sc">+</span></span>
<span id="cb22-11"><a href="#cb22-11"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>facet, <span class="at">scales=</span><span class="st">"free_x"</span>, <span class="at">nrow=</span><span class="dv">4</span>) <span class="sc">+</span></span>
<span id="cb22-12"><a href="#cb22-12"></a>  <span class="fu">theme</span>(</span>
<span id="cb22-13"><a href="#cb22-13"></a>    <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb22-14"><a href="#cb22-14"></a>    <span class="at">strip.text.x =</span> <span class="fu">element_blank</span>()</span>
<span id="cb22-15"><a href="#cb22-15"></a>    )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div id="fig-06" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-06-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="01_format_rbec_tab_files/figure-html/unnamed-chunk-22-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-06-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;6: Species composition of all masterplate trios. Any species that shouldn’t be there (&gt; 0%) is shown in grey to black. The samples are split over four rows to aid in visualization - the specific subplots are meaningless.
</figcaption>
</figure>
</div>
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1"></a>finaltable_exp_mstr <span class="sc">%&gt;%</span></span>
<span id="cb23-2"><a href="#cb23-2"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(community_type, <span class="st">"masterplate"</span>) <span class="sc">&amp;</span> n_species <span class="sc">==</span> <span class="dv">3</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb23-3"><a href="#cb23-3"></a>  <span class="fu">filter</span>(<span class="fu">is.na</span>(evo_hist)) <span class="sc">%&gt;%</span> </span>
<span id="cb23-4"><a href="#cb23-4"></a>  <span class="fu">filter</span>(f_raw_targetsp <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb23-5"><a href="#cb23-5"></a>  <span class="fu">contam_histogram</span>(<span class="at">trans =</span> <span class="cn">TRUE</span>, <span class="at">x =</span> f_raw_targetsp) <span class="sc">+</span></span>
<span id="cb23-6"><a href="#cb23-6"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Species frequency"</span>, <span class="at">y =</span> <span class="st">"Sample count"</span>) <span class="sc">+</span></span>
<span id="cb23-7"><a href="#cb23-7"></a>  <span class="fu">ggtitle</span>(<span class="st">"Frequency distribution of misassigned species in masterplate trios"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div id="fig-07" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-07-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="01_format_rbec_tab_files/figure-html/unnamed-chunk-23-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-07-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;7: Frequency distribution of each species in masterplate trio samples where it <strong>should not</strong> occur. The 1% mark (commonly accepted as an acceptable Illumina cross-talk standard) is demarcated with a dashed line.
</figcaption>
</figure>
</div>
<p>There are no samples with more than 3% of a species that shouldn’t be there</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1"></a>finaltable_exp_mstr <span class="sc">%&gt;%</span> </span>
<span id="cb24-2"><a href="#cb24-2"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(community_type, <span class="st">"masterplate"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb24-3"><a href="#cb24-3"></a>  <span class="fu">filter</span>(n_species <span class="sc">==</span> <span class="st">"3"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb24-4"><a href="#cb24-4"></a>  <span class="fu">filter</span>(<span class="fu">is.na</span>(evo_hist)) <span class="sc">%&gt;%</span> </span>
<span id="cb24-5"><a href="#cb24-5"></a>  <span class="fu">filter</span>(f_raw_targetsp <span class="sc">&gt;</span> <span class="fl">0.03</span> ) <span class="sc">%&gt;%</span> </span>
<span id="cb24-6"><a href="#cb24-6"></a>  <span class="fu">distinct</span>(sample)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["sample"],"name":[1],"type":["chr"],"align":["left"]}],"data":[],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
</section>
<section id="masterplate-summary" class="level3" data-number="5.6.3">
<h3 data-number="5.6.3" class="anchored" data-anchor-id="masterplate-summary"><span class="header-section-number">5.6.3</span> Masterplate Summary</h3>
<p>Generally, the cross talk frequency is really good. For 3/4 species it is 1% or less which is more or less what you can expect when you are multiplexing libraries on an Illumina platform. Values greater than 1% are potentially indicative of a different problem, so 1977 requires a bit more investigation.</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1"></a>finaltable_exp_mstr <span class="sc">%&gt;%</span></span>
<span id="cb25-2"><a href="#cb25-2"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(community_type, <span class="st">"masterplate"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb25-3"><a href="#cb25-3"></a>  <span class="fu">filter</span>(<span class="fu">is.na</span>(evo_hist)) <span class="sc">%&gt;%</span> </span>
<span id="cb25-4"><a href="#cb25-4"></a>  <span class="fu">filter</span>(f_raw_targetsp <span class="sc">&gt;</span> <span class="fl">0.01</span> <span class="sc">&amp;</span> strainID <span class="sc">==</span> <span class="st">"HAMBI_1977"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb25-5"><a href="#cb25-5"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(f_raw_targetsp))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["sample"],"name":[1],"type":["chr"],"align":["left"]},{"label":["strainID"],"name":[2],"type":["chr"],"align":["left"]},{"label":["count"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["count_correct"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["sample_tot_raw"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["f_raw"],"name":[6],"type":["dbl"],"align":["right"]},{"label":["sample_tot_targetsp"],"name":[7],"type":["dbl"],"align":["right"]},{"label":["f_raw_targetsp"],"name":[8],"type":["dbl"],"align":["right"]},{"label":["target_f"],"name":[9],"type":["dbl"],"align":["right"]},{"label":["evo_hist"],"name":[10],"type":["chr"],"align":["left"]},{"label":["community_id"],"name":[11],"type":["chr"],"align":["left"]},{"label":["n_species"],"name":[12],"type":["dbl"],"align":["right"]},{"label":["community_type"],"name":[13],"type":["chr"],"align":["left"]},{"label":["transfers"],"name":[14],"type":["dbl"],"align":["right"]},{"label":["strep_conc"],"name":[15],"type":["dbl"],"align":["right"]},{"label":["replicate"],"name":[16],"type":["dbl"],"align":["right"]},{"label":["plate_well"],"name":[17],"type":["chr"],"align":["left"]},{"label":["notes"],"name":[18],"type":["chr"],"align":["left"]}],"data":[{"1":"P36_s_0","2":"HAMBI_1977","3":"141","4":"209","5":"5784","6":"0.03613416","7":"5782","8":"0.03614666","9":"NA","10":"NA","11":"P36","12":"2","13":"masterplate","14":"0","15":"NA","16":"1","17":"05H","18":"Masterplate samples are those that were constructed ahead of time, then thawed and used in the experiment. They had no time to grow and thus should contain only what we put into them in roughly the proportions we intended"},{"1":"T18_s_0","2":"HAMBI_1977","3":"176","4":"163","5":"6165","6":"0.02643958","7":"6165","8":"0.02643958","9":"NA","10":"NA","11":"T18","12":"3","13":"masterplate","14":"0","15":"NA","16":"1","17":"02B","18":"Masterplate samples are those that were constructed ahead of time, then thawed and used in the experiment. They had no time to grow and thus should contain only what we put into them in roughly the proportions we intended"},{"1":"T11_s_0","2":"HAMBI_1977","3":"122","4":"145","5":"5677","6":"0.02554166","7":"5677","8":"0.02554166","9":"NA","10":"NA","11":"T11","12":"3","13":"masterplate","14":"0","15":"NA","16":"1","17":"05D","18":"Masterplate samples are those that were constructed ahead of time, then thawed and used in the experiment. They had no time to grow and thus should contain only what we put into them in roughly the proportions we intended"},{"1":"T15_s_0","2":"HAMBI_1977","3":"178","4":"174","5":"8221","6":"0.02116531","7":"8221","8":"0.02116531","9":"NA","10":"NA","11":"T15","12":"3","13":"masterplate","14":"0","15":"NA","16":"1","17":"02A","18":"Masterplate samples are those that were constructed ahead of time, then thawed and used in the experiment. They had no time to grow and thus should contain only what we put into them in roughly the proportions we intended"},{"1":"P30_s_0","2":"HAMBI_1977","3":"71","4":"102","5":"5342","6":"0.01909397","7":"5342","8":"0.01909397","9":"NA","10":"NA","11":"P30","12":"2","13":"masterplate","14":"0","15":"NA","16":"1","17":"05G","18":"Masterplate samples are those that were constructed ahead of time, then thawed and used in the experiment. They had no time to grow and thus should contain only what we put into them in roughly the proportions we intended"},{"1":"T21_s_0","2":"HAMBI_1977","3":"148","4":"144","5":"8767","6":"0.01642523","7":"8767","8":"0.01642523","9":"NA","10":"NA","11":"T21","12":"3","13":"masterplate","14":"0","15":"NA","16":"1","17":"02C","18":"Masterplate samples are those that were constructed ahead of time, then thawed and used in the experiment. They had no time to grow and thus should contain only what we put into them in roughly the proportions we intended"},{"1":"T20_s_0","2":"HAMBI_1977","3":"91","4":"117","5":"7349","6":"0.01592053","7":"7349","8":"0.01592053","9":"NA","10":"NA","11":"T20","12":"3","13":"masterplate","14":"0","15":"NA","16":"1","17":"06C","18":"Masterplate samples are those that were constructed ahead of time, then thawed and used in the experiment. They had no time to grow and thus should contain only what we put into them in roughly the proportions we intended"},{"1":"P19_s_0","2":"HAMBI_1977","3":"44","4":"58","5":"3697","6":"0.01568840","7":"3697","8":"0.01568840","9":"NA","10":"NA","11":"P19","12":"2","13":"masterplate","14":"0","15":"NA","16":"1","17":"03D","18":"Masterplate samples are those that were constructed ahead of time, then thawed and used in the experiment. They had no time to grow and thus should contain only what we put into them in roughly the proportions we intended"},{"1":"T24_s_0","2":"HAMBI_1977","3":"86","4":"83","5":"5525","6":"0.01502262","7":"5525","8":"0.01502262","9":"NA","10":"NA","11":"T24","12":"3","13":"masterplate","14":"0","15":"NA","16":"1","17":"02D","18":"Masterplate samples are those that were constructed ahead of time, then thawed and used in the experiment. They had no time to grow and thus should contain only what we put into them in roughly the proportions we intended"},{"1":"T08_s_0","2":"HAMBI_1977","3":"73","4":"98","5":"7476","6":"0.01310861","7":"7476","8":"0.01310861","9":"NA","10":"NA","11":"T08","12":"3","13":"masterplate","14":"0","15":"NA","16":"1","17":"05C","18":"Masterplate samples are those that were constructed ahead of time, then thawed and used in the experiment. They had no time to grow and thus should contain only what we put into them in roughly the proportions we intended"},{"1":"P20_s_0","2":"HAMBI_1977","3":"38","4":"50","5":"3960","6":"0.01262626","7":"3960","8":"0.01262626","9":"NA","10":"NA","11":"P20","12":"2","13":"masterplate","14":"0","15":"NA","16":"1","17":"03H","18":"Masterplate samples are those that were constructed ahead of time, then thawed and used in the experiment. They had no time to grow and thus should contain only what we put into them in roughly the proportions we intended"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<p>It looks like all the “problematic” samples come from plates 7 and 8 in the library prep. Plate 8 only contains the masterplate samples from trios whereas plate 7 contains both masterplate and experimental samples. In <a href="#fig-02" class="quarto-xref">Figure&nbsp;2</a> it shows that HAMBI-1977 is very abundant in many of the samples so likely the “leaky” reads come disproportionately from HAMBI-1977 which is why its crosstalk threshold may be higher (<a href="#fig-01" class="quarto-xref">Figure&nbsp;1</a>). Anyway, I don’t think this is a problem and that we can move forward with these samples. However, to define extinction/competitive exclusion we may need to use a higher threshold than 1% (e.g., 3% frequency) because over 3% we can reliably say that a species is present and it is not due to index cross talk.</p>
</section>
</section>
<section id="experimental-samples" class="level2" data-number="5.7">
<h2 data-number="5.7" class="anchored" data-anchor-id="experimental-samples"><span class="header-section-number">5.7</span> Experimental samples</h2>
<p>We may need to use a higher threshold than 1% (e.g., 3% frequency) because over 3% we can reliably say that a species is present and it is not due to index cross talk. Below I basically empirically chose 8% as a reasonable threshold value that includes the maximum number of samples with low-level contamination and exclude samples with very high contamination.</p>
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1"></a>finaltable_exp_mstr <span class="sc">%&gt;%</span> </span>
<span id="cb26-2"><a href="#cb26-2"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(community_type, <span class="st">"experiment"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb26-3"><a href="#cb26-3"></a>  <span class="fu">filter</span>(<span class="fu">is.na</span>(evo_hist)) <span class="sc">%&gt;%</span> </span>
<span id="cb26-4"><a href="#cb26-4"></a>  <span class="fu">filter</span>(f_raw <span class="sc">&gt;</span> <span class="fl">0.03</span> ) <span class="sc">%&gt;%</span> </span>
<span id="cb26-5"><a href="#cb26-5"></a>  <span class="fu">contam_histogram</span>(<span class="at">trans=</span><span class="cn">TRUE</span>, <span class="at">x =</span> f_raw_targetsp) <span class="sc">+</span></span>
<span id="cb26-6"><a href="#cb26-6"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Species frequency"</span>, <span class="at">y =</span> <span class="st">"Sample count"</span>) <span class="sc">+</span></span>
<span id="cb26-7"><a href="#cb26-7"></a>  <span class="fu">ggtitle</span>(<span class="st">"Frequency distribution of each species exceeding the 3% threshold"</span>) <span class="sc">+</span></span>
<span id="cb26-8"><a href="#cb26-8"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fl">0.08</span>, <span class="at">color =</span> <span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb26-9"><a href="#cb26-9"></a>  <span class="fu">facet_grid</span>(n_species<span class="sc">~</span>strainID)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div id="fig-08" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-08-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="01_format_rbec_tab_files/figure-html/unnamed-chunk-26-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-08-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;8: Frequency distribution of each species in experimental samples where it <strong>should not</strong> occur and where it exceeds a 3% threshold. Grid columns are the different species and grid rows represent whether the sample was an experiment pair (2) or trio (3). The red vertical line is at 8%.
</figcaption>
</figure>
</div>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1"></a>finaltable_exp_mstr <span class="sc">%&gt;%</span> </span>
<span id="cb27-2"><a href="#cb27-2"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(community_type, <span class="st">"experiment"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb27-3"><a href="#cb27-3"></a>  <span class="fu">filter</span>(<span class="fu">is.na</span>(evo_hist)) <span class="sc">%&gt;%</span> </span>
<span id="cb27-4"><a href="#cb27-4"></a>  <span class="fu">filter</span>(f_raw_targetsp <span class="sc">&gt;</span> <span class="fl">0.03</span> ) <span class="sc">%&gt;%</span> </span>
<span id="cb27-5"><a href="#cb27-5"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(f_raw_targetsp))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div data-pagedtable="false">
  <script data-pagedtable-source="" type="application/json">
{"columns":[{"label":["sample"],"name":[1],"type":["chr"],"align":["left"]},{"label":["strainID"],"name":[2],"type":["chr"],"align":["left"]},{"label":["count"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["count_correct"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["sample_tot_raw"],"name":[5],"type":["dbl"],"align":["right"]},{"label":["f_raw"],"name":[6],"type":["dbl"],"align":["right"]},{"label":["sample_tot_targetsp"],"name":[7],"type":["dbl"],"align":["right"]},{"label":["f_raw_targetsp"],"name":[8],"type":["dbl"],"align":["right"]},{"label":["target_f"],"name":[9],"type":["dbl"],"align":["right"]},{"label":["evo_hist"],"name":[10],"type":["chr"],"align":["left"]},{"label":["community_id"],"name":[11],"type":["chr"],"align":["left"]},{"label":["n_species"],"name":[12],"type":["dbl"],"align":["right"]},{"label":["community_type"],"name":[13],"type":["chr"],"align":["left"]},{"label":["transfers"],"name":[14],"type":["dbl"],"align":["right"]},{"label":["strep_conc"],"name":[15],"type":["dbl"],"align":["right"]},{"label":["replicate"],"name":[16],"type":["dbl"],"align":["right"]},{"label":["plate_well"],"name":[17],"type":["chr"],"align":["left"]},{"label":["notes"],"name":[18],"type":["chr"],"align":["left"]}],"data":[{"1":"P30_2_0","2":"HAMBI_1977","3":"9835","4":"10144","5":"11075","6":"0.91593679","7":"11075","8":"0.91593679","9":"NA","10":"NA","11":"P30","12":"2","13":"experiment","14":"8","15":"0","16":"2","17":"09H","18":"NA"},{"1":"P06_2_0","2":"HAMBI_1977","3":"9572","4":"10056","5":"11305","6":"0.88951791","7":"11305","8":"0.88951791","9":"NA","10":"NA","11":"P06","12":"2","13":"experiment","14":"8","15":"0","16":"2","17":"09F","18":"NA"},{"1":"P06_1_0","2":"HAMBI_1977","3":"6742","4":"7286","5":"8779","6":"0.82993507","7":"8779","8":"0.82993507","9":"NA","10":"NA","11":"P06","12":"2","13":"experiment","14":"8","15":"0","16":"1","17":"10F","18":"NA"},{"1":"T24_1_0","2":"HAMBI_1977","3":"5981","4":"5984","5":"7270","6":"0.82310867","7":"7270","8":"0.82310867","9":"NA","10":"NA","11":"T24","12":"3","13":"experiment","14":"8","15":"0","16":"1","17":"02D","18":"NA"},{"1":"T23_1_0","2":"HAMBI_1977","3":"6437","4":"6439","5":"7838","6":"0.82151059","7":"7838","8":"0.82151059","9":"NA","10":"NA","11":"T23","12":"3","13":"experiment","14":"8","15":"0","16":"1","17":"06D","18":"NA"},{"1":"T23_2_0","2":"HAMBI_1977","3":"10996","4":"11082","5":"13511","6":"0.82022056","7":"13511","8":"0.82022056","9":"NA","10":"NA","11":"T23","12":"3","13":"experiment","14":"8","15":"0","16":"2","17":"06D","18":"NA"},{"1":"T20_1_0","2":"HAMBI_1977","3":"5434","4":"5387","5":"6586","6":"0.81794716","7":"6586","8":"0.81794716","9":"NA","10":"NA","11":"T20","12":"3","13":"experiment","14":"8","15":"0","16":"1","17":"06C","18":"NA"},{"1":"T21_1_0","2":"HAMBI_1977","3":"6202","4":"6233","5":"7675","6":"0.81211726","7":"7675","8":"0.81211726","9":"NA","10":"NA","11":"T21","12":"3","13":"experiment","14":"8","15":"0","16":"1","17":"02C","18":"NA"},{"1":"P44_2_0","2":"HAMBI_1977","3":"11884","4":"11982","5":"15018","6":"0.79784259","7":"15018","8":"0.79784259","9":"NA","10":"NA","11":"P44","12":"2","13":"experiment","14":"8","15":"0","16":"2","17":"07H","18":"NA"},{"1":"T19_1_0","2":"HAMBI_1977","3":"6772","4":"6731","5":"8487","6":"0.79309532","7":"8487","8":"0.79309532","9":"NA","10":"NA","11":"T19","12":"3","13":"experiment","14":"8","15":"0","16":"1","17":"08E","18":"NA"},{"1":"P44_1_0","2":"HAMBI_1977","3":"12069","4":"12119","5":"15352","6":"0.78940855","7":"15352","8":"0.78940855","9":"NA","10":"NA","11":"P44","12":"2","13":"experiment","14":"8","15":"0","16":"1","17":"08H","18":"NA"},{"1":"P20_2_0","2":"HAMBI_1977","3":"7896","4":"7953","5":"10168","6":"0.78215972","7":"10168","8":"0.78215972","9":"NA","10":"NA","11":"P20","12":"2","13":"experiment","14":"8","15":"0","16":"2","17":"07F","18":"NA"},{"1":"T20_2_0","2":"HAMBI_1977","3":"10207","4":"10247","5":"13191","6":"0.77681753","7":"13191","8":"0.77681753","9":"NA","10":"NA","11":"T20","12":"3","13":"experiment","14":"8","15":"0","16":"2","17":"06C","18":"NA"},{"1":"T24_2_0","2":"HAMBI_1977","3":"8138","4":"8255","5":"10710","6":"0.77077498","7":"10710","8":"0.77077498","9":"NA","10":"NA","11":"T24","12":"3","13":"experiment","14":"8","15":"0","16":"2","17":"02D","18":"NA"},{"1":"P20_1_0","2":"HAMBI_1977","3":"8104","4":"8248","5":"10823","6":"0.76208075","7":"10823","8":"0.76208075","9":"NA","10":"NA","11":"P20","12":"2","13":"experiment","14":"8","15":"0","16":"1","17":"08F","18":"NA"},{"1":"T19_2_0","2":"HAMBI_1977","3":"9127","4":"9363","5":"12464","6":"0.75120347","7":"12464","8":"0.75120347","9":"NA","10":"NA","11":"T19","12":"3","13":"experiment","14":"8","15":"0","16":"2","17":"08E","18":"NA"},{"1":"P19_1_0","2":"HAMBI_1977","3":"9773","4":"9315","5":"12498","6":"0.74531925","7":"12498","8":"0.74531925","9":"NA","10":"NA","11":"P19","12":"2","13":"experiment","14":"8","15":"0","16":"1","17":"07E","18":"NA"},{"1":"T21_2_0","2":"HAMBI_1977","3":"8809","4":"9051","5":"12315","6":"0.73495737","7":"12315","8":"0.73495737","9":"NA","10":"NA","11":"T21","12":"3","13":"experiment","14":"8","15":"0","16":"2","17":"02C","18":"NA"},{"1":"T09_1_0","2":"HAMBI_1977","3":"11580","4":"11348","5":"15853","6":"0.71582666","7":"15853","8":"0.71582666","9":"NA","10":"NA","11":"T09","12":"3","13":"experiment","14":"8","15":"0","16":"1","17":"01C","18":"NA"},{"1":"T11_1_0","2":"HAMBI_1977","3":"7301","4":"7118","5":"9973","6":"0.71372706","7":"9973","8":"0.71372706","9":"NA","10":"NA","11":"T11","12":"3","13":"experiment","14":"8","15":"0","16":"1","17":"05D","18":"NA"},{"1":"T11_2_0","2":"HAMBI_1977","3":"10050","4":"9859","5":"13931","6":"0.70770225","7":"13931","8":"0.70770225","9":"NA","10":"NA","11":"T11","12":"3","13":"experiment","14":"8","15":"0","16":"2","17":"05D","18":"NA"},{"1":"P43_1_0","2":"HAMBI_1977","3":"12086","4":"11928","5":"16943","6":"0.70400755","7":"16943","8":"0.70400755","9":"NA","10":"NA","11":"P43","12":"2","13":"experiment","14":"8","15":"0","16":"1","17":"07G","18":"NA"},{"1":"T12_2_0","2":"HAMBI_1977","3":"9583","4":"9183","5":"13088","6":"0.70163509","7":"13088","8":"0.70163509","9":"NA","10":"NA","11":"T12","12":"3","13":"experiment","14":"8","15":"0","16":"2","17":"01D","18":"NA"},{"1":"T12_1_0","2":"HAMBI_1977","3":"8165","4":"8015","5":"11532","6":"0.69502255","7":"11532","8":"0.69502255","9":"NA","10":"NA","11":"T12","12":"3","13":"experiment","14":"8","15":"0","16":"1","17":"01D","18":"NA"},{"1":"T07_1_0","2":"HAMBI_1977","3":"7698","4":"7660","5":"11032","6":"0.69434373","7":"11032","8":"0.69434373","9":"NA","10":"NA","11":"T07","12":"3","13":"experiment","14":"8","15":"0","16":"1","17":"07E","18":"NA"},{"1":"T08_1_0","2":"HAMBI_1977","3":"8136","4":"7923","5":"11445","6":"0.69226737","7":"11445","8":"0.69226737","9":"NA","10":"NA","11":"T08","12":"3","13":"experiment","14":"8","15":"0","16":"1","17":"05C","18":"NA"},{"1":"P43_2_0","2":"HAMBI_1977","3":"5222","4":"5060","5":"7502","6":"0.67448680","7":"7502","8":"0.67448680","9":"NA","10":"NA","11":"P43","12":"2","13":"experiment","14":"8","15":"0","16":"2","17":"08G","18":"NA"},{"1":"T08_2_0","2":"HAMBI_1977","3":"9673","4":"9471","5":"14207","6":"0.66664320","7":"14207","8":"0.66664320","9":"NA","10":"NA","11":"T08","12":"3","13":"experiment","14":"8","15":"0","16":"2","17":"05C","18":"NA"},{"1":"P07_2_0","2":"HAMBI_1287","3":"6838","4":"6554","5":"9937","6":"0.65955520","7":"9937","8":"0.65955520","9":"NA","10":"NA","11":"P07","12":"2","13":"experiment","14":"8","15":"0","16":"2","17":"04A","18":"NA"},{"1":"T09_2_0","2":"HAMBI_1977","3":"8400","4":"8283","5":"12596","6":"0.65758971","7":"12596","8":"0.65758971","9":"NA","10":"NA","11":"T09","12":"3","13":"experiment","14":"8","15":"0","16":"2","17":"01C","18":"NA"},{"1":"P36_2_0","2":"HAMBI_1977","3":"4324","4":"5139","5":"7821","6":"0.65707710","7":"7821","8":"0.65707710","9":"NA","10":"NA","11":"P36","12":"2","13":"experiment","14":"8","15":"0","16":"2","17":"11H","18":"NA"},{"1":"P19_2_0","2":"HAMBI_1977","3":"6502","4":"6470","5":"9922","6":"0.65208627","7":"9922","8":"0.65208627","9":"NA","10":"NA","11":"P19","12":"2","13":"experiment","14":"8","15":"0","16":"2","17":"08E","18":"NA"},{"1":"P07_1_0","2":"HAMBI_1287","3":"6635","4":"6387","5":"9923","6":"0.64365615","7":"9923","8":"0.64365615","9":"NA","10":"NA","11":"P07","12":"2","13":"experiment","14":"8","15":"0","16":"1","17":"03A","18":"NA"},{"1":"T07_2_0","2":"HAMBI_1977","3":"9179","4":"9213","5":"14505","6":"0.63516029","7":"14505","8":"0.63516029","9":"NA","10":"NA","11":"T07","12":"3","13":"experiment","14":"8","15":"0","16":"2","17":"07E","18":"NA"},{"1":"T22_2_0","2":"HAMBI_1977","3":"5287","4":"5572","5":"9161","6":"0.60823054","7":"9161","8":"0.60823054","9":"NA","10":"NA","11":"T22","12":"3","13":"experiment","14":"8","15":"0","16":"2","17":"08F","18":"NA"},{"1":"P07_1_64","2":"HAMBI_1287","3":"5303","4":"5881","5":"9897","6":"0.59422047","7":"9897","8":"0.59422047","9":"NA","10":"NA","11":"P07","12":"2","13":"experiment","14":"8","15":"64","16":"1","17":"03A","18":"NA"},{"1":"P12_2_0","2":"HAMBI_1977","3":"4233","4":"5255","5":"8881","6":"0.59171264","7":"8881","8":"0.59171264","9":"NA","10":"NA","11":"P12","12":"2","13":"experiment","14":"8","15":"0","16":"2","17":"11F","18":"NA"},{"1":"T22_1_0","2":"HAMBI_1977","3":"3815","4":"4359","5":"7434","6":"0.58635997","7":"7434","8":"0.58635997","9":"NA","10":"NA","11":"T22","12":"3","13":"experiment","14":"8","15":"0","16":"1","17":"08F","18":"NA"},{"1":"T10_2_0","2":"HAMBI_1977","3":"5345","4":"5341","5":"9219","6":"0.57934700","7":"9219","8":"0.57934700","9":"NA","10":"NA","11":"T10","12":"3","13":"experiment","14":"8","15":"0","16":"2","17":"07F","18":"NA"},{"1":"T58_2_0","2":"HAMBI_1287","3":"5721","4":"5016","5":"8760","6":"0.57260274","7":"8760","8":"0.57260274","9":"NA","10":"NA","11":"T58","12":"3","13":"experiment","14":"8","15":"0","16":"2","17":"11F","18":"NA"},{"1":"P07_2_16","2":"HAMBI_1287","3":"3601","4":"4039","5":"7404","6":"0.54551594","7":"7404","8":"0.54551594","9":"NA","10":"NA","11":"P07","12":"2","13":"experiment","14":"8","15":"16","16":"2","17":"04A","18":"NA"},{"1":"T10_1_0","2":"HAMBI_1977","3":"3273","4":"3558","5":"7325","6":"0.48573379","7":"7325","8":"0.48573379","9":"NA","10":"NA","11":"T10","12":"3","13":"experiment","14":"8","15":"0","16":"1","17":"07F","18":"NA"},{"1":"T58_1_0","2":"HAMBI_1287","3":"3427","4":"3023","5":"6464","6":"0.46766708","7":"6464","8":"0.46766708","9":"NA","10":"NA","11":"T58","12":"3","13":"experiment","14":"8","15":"0","16":"1","17":"11F","18":"NA"},{"1":"T64_2_0","2":"HAMBI_1287","3":"1930","4":"1452","5":"9154","6":"0.15861918","7":"9154","8":"0.15861918","9":"NA","10":"NA","11":"T64","12":"3","13":"experiment","14":"8","15":"0","16":"2","17":"06F","18":"NA"},{"1":"T61_2_0","2":"HAMBI_1287","3":"2202","4":"1685","5":"11402","6":"0.14778109","7":"11402","8":"0.14778109","9":"NA","10":"NA","11":"T61","12":"3","13":"experiment","14":"8","15":"0","16":"2","17":"06E","18":"NA"},{"1":"T31_2_0","2":"HAMBI_1896","3":"1720","4":"2166","5":"15161","6":"0.14286657","7":"15161","8":"0.14286657","9":"NA","10":"NA","11":"T31","12":"3","13":"experiment","14":"8","15":"0","16":"2","17":"09E","18":"NA"},{"1":"T64_1_0","2":"HAMBI_1287","3":"782","4":"577","5":"4458","6":"0.12943024","7":"4458","8":"0.12943024","9":"NA","10":"NA","11":"T64","12":"3","13":"experiment","14":"8","15":"0","16":"1","17":"06F","18":"NA"},{"1":"T61_1_0","2":"HAMBI_1287","3":"1544","4":"1165","5":"10674","6":"0.10914371","7":"10674","8":"0.10914371","9":"NA","10":"NA","11":"T61","12":"3","13":"experiment","14":"8","15":"0","16":"1","17":"06E","18":"NA"},{"1":"T45_2_0","2":"HAMBI_1896","3":"738","4":"1113","5":"11621","6":"0.09577489","7":"11621","8":"0.09577489","9":"NA","10":"NA","11":"T45","12":"3","13":"experiment","14":"8","15":"0","16":"2","17":"04C","18":"NA"},{"1":"T45_1_0","2":"HAMBI_1896","3":"434","4":"741","5":"8317","6":"0.08909463","7":"8317","8":"0.08909463","9":"NA","10":"NA","11":"T45","12":"3","13":"experiment","14":"8","15":"0","16":"1","17":"04C","18":"NA"},{"1":"P17_2_256","2":"HAMBI_1896","3":"140","4":"238","5":"3683","6":"0.06462123","7":"3683","8":"0.06462123","9":"NA","10":"NA","11":"P17","12":"2","13":"experiment","14":"8","15":"256","16":"2","17":"08A","18":"NA"},{"1":"T13_2_0","2":"HAMBI_1977","3":"433","4":"579","5":"10374","6":"0.05581261","7":"10374","8":"0.05581261","9":"NA","10":"NA","11":"T13","12":"3","13":"experiment","14":"8","15":"0","16":"2","17":"02E","18":"NA"},{"1":"T14_2_0","2":"HAMBI_1977","3":"400","4":"534","5":"10297","6":"0.05185976","7":"10297","8":"0.05185976","9":"NA","10":"NA","11":"T14","12":"3","13":"experiment","14":"8","15":"0","16":"2","17":"06A","18":"NA"},{"1":"P30_2_64","2":"HAMBI_1896","3":"64","4":"109","5":"2128","6":"0.05122180","7":"2128","8":"0.05122180","9":"NA","10":"NA","11":"P30","12":"2","13":"experiment","14":"8","15":"64","16":"2","17":"09H","18":"NA"},{"1":"P06_1_256","2":"HAMBI_1896","3":"79","4":"135","5":"2847","6":"0.04741834","7":"2847","8":"0.04741834","9":"NA","10":"NA","11":"P06","12":"2","13":"experiment","14":"8","15":"256","16":"1","17":"10F","18":"NA"},{"1":"T18_2_0","2":"HAMBI_1977","3":"343","4":"502","5":"10882","6":"0.04613123","7":"10882","8":"0.04613123","9":"NA","10":"NA","11":"T18","12":"3","13":"experiment","14":"8","15":"0","16":"2","17":"02B","18":"NA"},{"1":"T16_2_0","2":"HAMBI_1977","3":"313","4":"462","5":"10213","6":"0.04523646","7":"10213","8":"0.04523646","9":"NA","10":"NA","11":"T16","12":"3","13":"experiment","14":"8","15":"0","16":"2","17":"02F","18":"NA"},{"1":"P45_2_0","2":"HAMBI_1287","3":"74","4":"49","5":"1107","6":"0.04426378","7":"1107","8":"0.04426378","9":"NA","10":"NA","11":"P45","12":"2","13":"experiment","14":"8","15":"0","16":"2","17":"10C","18":"NA"},{"1":"T15_2_0","2":"HAMBI_1977","3":"400","4":"520","5":"12330","6":"0.04217356","7":"12330","8":"0.04217356","9":"NA","10":"NA","11":"T15","12":"3","13":"experiment","14":"8","15":"0","16":"2","17":"02A","18":"NA"},{"1":"T04_2_0","2":"HAMBI_1977","3":"234","4":"342","5":"8159","6":"0.04191690","7":"8159","8":"0.04191690","9":"NA","10":"NA","11":"T04","12":"3","13":"experiment","14":"8","15":"0","16":"2","17":"01F","18":"NA"},{"1":"P01_2_64","2":"HAMBI_1896","3":"179","4":"393","5":"9693","6":"0.04054472","7":"9693","8":"0.04054472","9":"NA","10":"NA","11":"P01","12":"2","13":"experiment","14":"8","15":"64","16":"2","17":"02A","18":"NA"},{"1":"T17_2_0","2":"HAMBI_1977","3":"283","4":"382","5":"9533","6":"0.04007133","7":"9533","8":"0.04007133","9":"NA","10":"NA","11":"T17","12":"3","13":"experiment","14":"8","15":"0","16":"2","17":"06B","18":"NA"},{"1":"T05_2_0","2":"HAMBI_1977","3":"339","4":"410","5":"10693","6":"0.03834284","7":"10693","8":"0.03834284","9":"NA","10":"NA","11":"T05","12":"3","13":"experiment","14":"8","15":"0","16":"2","17":"05B","18":"NA"},{"1":"T15_1_0","2":"HAMBI_1977","3":"197","4":"258","5":"6819","6":"0.03783546","7":"6819","8":"0.03783546","9":"NA","10":"NA","11":"T15","12":"3","13":"experiment","14":"8","15":"0","16":"1","17":"02A","18":"NA"},{"1":"T55_1_0","2":"HAMBI_1287","3":"529","4":"379","5":"10087","6":"0.03757311","7":"10087","8":"0.03757311","9":"NA","10":"NA","11":"T55","12":"3","13":"experiment","14":"8","15":"0","16":"1","17":"11E","18":"NA"},{"1":"P30_1_256","2":"HAMBI_1896","3":"130","4":"224","5":"6059","6":"0.03696980","7":"6059","8":"0.03696980","9":"NA","10":"NA","11":"P30","12":"2","13":"experiment","14":"8","15":"256","16":"1","17":"10H","18":"NA"},{"1":"T16_1_0","2":"HAMBI_1977","3":"134","4":"179","5":"4881","6":"0.03667281","7":"4881","8":"0.03667281","9":"NA","10":"NA","11":"T16","12":"3","13":"experiment","14":"8","15":"0","16":"1","17":"02F","18":"NA"},{"1":"T02_2_0","2":"HAMBI_1977","3":"463","4":"542","5":"15590","6":"0.03476588","7":"15590","8":"0.03476588","9":"NA","10":"NA","11":"T02","12":"3","13":"experiment","14":"8","15":"0","16":"2","17":"05A","18":"NA"},{"1":"P30_2_256","2":"HAMBI_1896","3":"143","4":"246","5":"7137","6":"0.03446826","7":"7137","8":"0.03446826","9":"NA","10":"NA","11":"P30","12":"2","13":"experiment","14":"8","15":"256","16":"2","17":"09H","18":"NA"},{"1":"T01_2_0","2":"HAMBI_1977","3":"350","4":"477","5":"14103","6":"0.03382259","7":"14103","8":"0.03382259","9":"NA","10":"NA","11":"T01","12":"3","13":"experiment","14":"8","15":"0","16":"2","17":"01E","18":"NA"},{"1":"P41_2_256","2":"HAMBI_1896","3":"132","4":"228","5":"7140","6":"0.03193277","7":"7140","8":"0.03193277","9":"NA","10":"NA","11":"P41","12":"2","13":"experiment","14":"8","15":"256","16":"2","17":"08C","18":"NA"},{"1":"P04_1_0","2":"HAMBI_1977","3":"200","4":"253","5":"8369","6":"0.03023061","7":"8369","8":"0.03023061","9":"NA","10":"NA","11":"P04","12":"2","13":"experiment","14":"8","15":"0","16":"1","17":"02F","18":"NA"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<p>There are 73 samples (10% of 710 samples) that exceed a 3% crosstalk threshold. 23 samples (3% of 710 samples) have contamination &lt;= 8% and 50 samples (7 % of 710 samples) have &gt; 8% of a species that should not be there. In most cases the species in highly contaminated samples is HAMBI_1977. I think these 50 samples are probably at very high risk for legitimate contamination. It is likely that the rest of the 23 samples with &lt;= 8% contamination are not contaminated but just crosstalk outliers.</p>
<section id="pairs-1" class="level3" data-number="5.7.1">
<h3 data-number="5.7.1" class="anchored" data-anchor-id="pairs-1"><span class="header-section-number">5.7.1</span> Pairs</h3>
<p>Here taking a closer look at the pairs from the experiment that appear to be contaminated.</p>
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1"></a>finaltable_exp_mstr <span class="sc">%&gt;%</span> </span>
<span id="cb28-2"><a href="#cb28-2"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(community_type, <span class="st">"experiment"</span>) <span class="sc">&amp;</span> n_species <span class="sc">==</span> <span class="dv">2</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb28-3"><a href="#cb28-3"></a>  <span class="fu">contaminated_barplot</span>(<span class="at">threshold =</span> <span class="fl">0.16</span>,  <span class="at">quartet =</span> <span class="cn">FALSE</span>, <span class="at">y=</span>f_raw_targetsp) <span class="sc">+</span></span>
<span id="cb28-4"><a href="#cb28-4"></a>  <span class="fu">ggtitle</span>(<span class="st">"Experiment pairs with &gt;= 15% contamination"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div id="fig-09" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-09-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="01_format_rbec_tab_files/figure-html/unnamed-chunk-29-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-09-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;9: Species composition of pairs experimental samples that contain &gt;= 15% of a species that shouldn’t be there (shown in grey to black).
</figcaption>
</figure>
</div>
<p>Most of the comtaminated samples are in the 0 Streptomycin conditions. I think we should exclude samples where the contamination is very high (over ~50% of the sample) but those with 8% or less contaminant I think can be retained, and I will discard the contaminating sequences.</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1"></a>contampairs <span class="ot">&lt;-</span> finaltable_exp_mstr <span class="sc">%&gt;%</span> </span>
<span id="cb29-2"><a href="#cb29-2"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(community_type, <span class="st">"experiment"</span>) <span class="sc">&amp;</span> n_species <span class="sc">==</span> <span class="dv">2</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb29-3"><a href="#cb29-3"></a>  <span class="fu">filter</span>(<span class="fu">is.na</span>(evo_hist)) <span class="sc">%&gt;%</span> </span>
<span id="cb29-4"><a href="#cb29-4"></a>  <span class="fu">filter</span>(f_raw_targetsp <span class="sc">&gt;</span> <span class="fl">0.16</span> ) <span class="sc">%&gt;%</span> </span>
<span id="cb29-5"><a href="#cb29-5"></a>  <span class="fu">pull</span>(sample)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="trios-1" class="level3" data-number="5.7.2">
<h3 data-number="5.7.2" class="anchored" data-anchor-id="trios-1"><span class="header-section-number">5.7.2</span> Trios</h3>
<p>Here just taking a closer look at the trios from the experiment that appear to be contaminated.</p>
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1"></a>finaltable_exp_mstr <span class="sc">%&gt;%</span> </span>
<span id="cb30-2"><a href="#cb30-2"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(community_type, <span class="st">"experiment"</span>) <span class="sc">&amp;</span> n_species <span class="sc">==</span> <span class="dv">3</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb30-3"><a href="#cb30-3"></a>  <span class="fu">contaminated_barplot</span>(<span class="at">threshold=</span><span class="fl">0.16</span>,  <span class="at">quartet =</span> <span class="cn">FALSE</span>, <span class="at">y=</span>f_raw_targetsp) <span class="sc">+</span></span>
<span id="cb30-4"><a href="#cb30-4"></a>  <span class="fu">ggtitle</span>(<span class="st">"Experiment trios with &gt;= 15% contamination"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div id="fig-10" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-10-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="01_format_rbec_tab_files/figure-html/unnamed-chunk-31-1.png" class="img-fluid figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-10-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;10: Species composition of experiment trio samples that contain &gt;= 15% of a species that shouldn’t be there (shown in grey to black).
</figcaption>
</figure>
</div>
<p>Again, I think we should exclude samples where the contamination is very high (over ~50% of the sample) but those with around less than 15% contaminant I think can be retained, and I will discard the contaminating sequences.</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1"></a>contamtrios <span class="ot">&lt;-</span> finaltable_exp_mstr <span class="sc">%&gt;%</span> </span>
<span id="cb31-2"><a href="#cb31-2"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(community_type, <span class="st">"experiment"</span>) <span class="sc">&amp;</span> n_species <span class="sc">==</span> <span class="dv">3</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb31-3"><a href="#cb31-3"></a>  <span class="fu">filter</span>(<span class="fu">is.na</span>(evo_hist)) <span class="sc">%&gt;%</span> </span>
<span id="cb31-4"><a href="#cb31-4"></a>  <span class="fu">filter</span>(f_raw_targetsp <span class="sc">&gt;</span> <span class="fl">0.16</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb31-5"><a href="#cb31-5"></a>  <span class="fu">pull</span>(sample)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
</section>
</section>
<section id="export" class="level1" data-number="6">
<h1 data-number="6"><span class="header-section-number">6</span> Export</h1>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1"></a>finaltable_exp_mstr_export <span class="ot">&lt;-</span> finaltable_exp_mstr <span class="sc">%&gt;%</span> </span>
<span id="cb32-2"><a href="#cb32-2"></a>  <span class="co"># first remove samples with low read count (&lt; 3000 reads)</span></span>
<span id="cb32-3"><a href="#cb32-3"></a>  <span class="fu">filter</span>(sample <span class="sc">%nin%</span> lowread_samps) <span class="sc">%&gt;%</span> </span>
<span id="cb32-4"><a href="#cb32-4"></a>  <span class="co"># next remove masterplate pair samples that were highly contaminated</span></span>
<span id="cb32-5"><a href="#cb32-5"></a>  <span class="fu">filter</span>(sample <span class="sc">%nin%</span> contampairs) <span class="sc">%&gt;%</span> </span>
<span id="cb32-6"><a href="#cb32-6"></a>  <span class="co"># next remove trio samples that were highly contaminated</span></span>
<span id="cb32-7"><a href="#cb32-7"></a>  <span class="fu">filter</span>(sample <span class="sc">%nin%</span> contamtrios) <span class="sc">%&gt;%</span> </span>
<span id="cb32-8"><a href="#cb32-8"></a>  <span class="co"># exclude any remaining counts from species that shouldnt be there using again</span></span>
<span id="cb32-9"><a href="#cb32-9"></a>  <span class="co"># the fact that evo_hist should be NA for these species</span></span>
<span id="cb32-10"><a href="#cb32-10"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(evo_hist)) <span class="sc">%&gt;%</span> </span>
<span id="cb32-11"><a href="#cb32-11"></a>  <span class="co"># because we set 3% as our limit of detection we set read counts of species</span></span>
<span id="cb32-12"><a href="#cb32-12"></a>  <span class="co"># less than 1% to 0</span></span>
<span id="cb32-13"><a href="#cb32-13"></a>  <span class="fu">mutate</span>(<span class="at">count_correct_thresh =</span> <span class="fu">if_else</span>(f_raw_targetsp <span class="sc">&lt;=</span> <span class="fl">0.03</span>, <span class="dv">0</span>, count_correct)) <span class="sc">%&gt;%</span> </span>
<span id="cb32-14"><a href="#cb32-14"></a>  <span class="co"># now calculate a new relative abundance based only on the species that should</span></span>
<span id="cb32-15"><a href="#cb32-15"></a>  <span class="co"># be present and that are &gt; 3% relative abundance</span></span>
<span id="cb32-16"><a href="#cb32-16"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb32-17"><a href="#cb32-17"></a>  <span class="fu">mutate</span>(<span class="at">f_thresh =</span> count_correct_thresh<span class="sc">/</span><span class="fu">sum</span>(count_correct_thresh)) <span class="sc">%&gt;%</span> </span>
<span id="cb32-18"><a href="#cb32-18"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb32-19"><a href="#cb32-19"></a>  <span class="co"># for marking which sequencing batch these came from </span></span>
<span id="cb32-20"><a href="#cb32-20"></a>  <span class="fu">mutate</span>(<span class="at">batch =</span> <span class="st">"run20240711"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb32-21"><a href="#cb32-21"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(sample, strainID, evo_hist, count_correct_thresh, f_thresh,</span>
<span id="cb32-22"><a href="#cb32-22"></a>                <span class="at">target_f_masterplate =</span> target_f, replicate, strep_conc, </span>
<span id="cb32-23"><a href="#cb32-23"></a>                transfers, n_species, community_type, plate_well, batch)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<section id="posneg-control-samples" class="level2" data-number="6.1">
<h2 data-number="6.1" class="anchored" data-anchor-id="posneg-control-samples"><span class="header-section-number">6.1</span> pos/neg control samples</h2>
<p>Separate out the pos/neg control samples</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1"></a>finaltable <span class="sc">%&gt;%</span> </span>
<span id="cb33-2"><a href="#cb33-2"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(community_type, <span class="st">"^pos|^neg"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb33-3"><a href="#cb33-3"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>transfers, <span class="sc">-</span>strep_conc, <span class="sc">-</span>evo_hist, <span class="sc">-</span>target_f, <span class="sc">-</span>count) <span class="sc">%&gt;%</span> </span>
<span id="cb33-4"><a href="#cb33-4"></a>  <span class="fu">mutate</span>(<span class="at">batch =</span> <span class="st">"run20240711"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb33-5"><a href="#cb33-5"></a>  <span class="fu">write_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(data, <span class="st">"pos_neg_ctrl_counts.tsv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="pairs-2" class="level2" data-number="6.2">
<h2 data-number="6.2" class="anchored" data-anchor-id="pairs-2"><span class="header-section-number">6.2</span> Pairs</h2>
<p>Write the pair samples</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1"></a>finaltable_exp_mstr_export <span class="sc">%&gt;%</span> </span>
<span id="cb34-2"><a href="#cb34-2"></a>  <span class="fu">filter</span>(n_species <span class="sc">==</span> <span class="dv">2</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb34-3"><a href="#cb34-3"></a>  <span class="fu">arrange</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb34-4"><a href="#cb34-4"></a>  <span class="fu">write_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(data, <span class="st">"pairs_counts.tsv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="trios-2" class="level2" data-number="6.3">
<h2 data-number="6.3" class="anchored" data-anchor-id="trios-2"><span class="header-section-number">6.3</span> Trios</h2>
<p>Write the trio samples</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Show/hide code</summary>
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1"></a>finaltable_exp_mstr_export <span class="sc">%&gt;%</span> </span>
<span id="cb35-2"><a href="#cb35-2"></a>  <span class="fu">filter</span>(n_species <span class="sc">==</span> <span class="dv">3</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb35-3"><a href="#cb35-3"></a>  <span class="fu">arrange</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb35-4"><a href="#cb35-4"></a>  <span class="fu">write_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(data, <span class="st">"trios_counts.tsv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>


<!-- -->

</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb36" data-shortcodes="false"><pre class="sourceCode numberSource markdown number-lines code-with-copy"><code class="sourceCode markdown"><span id="cb36-1"><a href="#cb36-1"></a><span class="co">---</span></span>
<span id="cb36-2"><a href="#cb36-2"></a><span class="an">title:</span><span class="co"> "Formatting Rbec output from  2024-07-11 sequenced by BTK Turku using custom HAMBI Illumina v3 primers"</span></span>
<span id="cb36-3"><a href="#cb36-3"></a><span class="an">author:</span><span class="co"> "Milla Similä &amp; Shane Hogle"</span></span>
<span id="cb36-4"><a href="#cb36-4"></a><span class="an">date:</span><span class="co"> today</span></span>
<span id="cb36-5"><a href="#cb36-5"></a><span class="an">link-citations:</span><span class="co"> true</span></span>
<span id="cb36-6"><a href="#cb36-6"></a><span class="an">abstract:</span><span class="co"> "Contains results from pairs of all streptomycin concentrations and trios for 0 streptomycin from Milla's community assembly experiment."</span></span>
<span id="cb36-7"><a href="#cb36-7"></a><span class="co">---</span></span>
<span id="cb36-8"><a href="#cb36-8"></a></span>
<span id="cb36-9"><a href="#cb36-9"></a><span class="fu"># Setup</span></span>
<span id="cb36-10"><a href="#cb36-10"></a></span>
<span id="cb36-11"><a href="#cb36-11"></a><span class="fu">## Libraries</span></span>
<span id="cb36-12"><a href="#cb36-12"></a></span>
<span id="cb36-15"><a href="#cb36-15"></a><span class="in">```{r}</span></span>
<span id="cb36-16"><a href="#cb36-16"></a><span class="co">#| output: false</span></span>
<span id="cb36-17"><a href="#cb36-17"></a><span class="co">#| warning: false</span></span>
<span id="cb36-18"><a href="#cb36-18"></a><span class="co">#| error: false</span></span>
<span id="cb36-19"><a href="#cb36-19"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb36-20"><a href="#cb36-20"></a><span class="fu">library</span>(here)</span>
<span id="cb36-21"><a href="#cb36-21"></a><span class="fu">library</span>(fs)</span>
<span id="cb36-22"><a href="#cb36-22"></a><span class="fu">library</span>(archive)</span>
<span id="cb36-23"><a href="#cb36-23"></a><span class="fu">library</span>(scales)</span>
<span id="cb36-24"><a href="#cb36-24"></a><span class="fu">source</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"R"</span>, <span class="st">"utils_generic.R"</span>))</span>
<span id="cb36-25"><a href="#cb36-25"></a><span class="fu">source</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"R"</span>, <span class="st">"communities"</span>, <span class="st">"amplicon"</span>, <span class="st">"utils_amplicon.R"</span>))</span>
<span id="cb36-26"><a href="#cb36-26"></a><span class="in">```</span></span>
<span id="cb36-27"><a href="#cb36-27"></a></span>
<span id="cb36-28"><a href="#cb36-28"></a><span class="fu">## Global variables</span></span>
<span id="cb36-29"><a href="#cb36-29"></a></span>
<span id="cb36-32"><a href="#cb36-32"></a><span class="in">```{r}</span></span>
<span id="cb36-33"><a href="#cb36-33"></a>data_raw <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"_data_raw"</span>, <span class="st">"communities"</span>, <span class="st">"20240711_BTK_illumina_v3"</span>)</span>
<span id="cb36-34"><a href="#cb36-34"></a>data <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"communities"</span>, <span class="st">"20240711_BTK_illumina_v3"</span>)</span>
<span id="cb36-35"><a href="#cb36-35"></a>amplicontar <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(data_raw, <span class="st">"rbec_output.tar.gz"</span>)</span>
<span id="cb36-36"><a href="#cb36-36"></a></span>
<span id="cb36-37"><a href="#cb36-37"></a><span class="co"># make processed data directory if it doesn't exist</span></span>
<span id="cb36-38"><a href="#cb36-38"></a>fs<span class="sc">::</span><span class="fu">dir_create</span>(data)</span>
<span id="cb36-39"><a href="#cb36-39"></a></span>
<span id="cb36-40"><a href="#cb36-40"></a><span class="co"># create temporary location to decompress</span></span>
<span id="cb36-41"><a href="#cb36-41"></a>tmpdir <span class="ot">&lt;-</span> fs<span class="sc">::</span><span class="fu">file_temp</span>()</span>
<span id="cb36-42"><a href="#cb36-42"></a><span class="in">```</span></span>
<span id="cb36-43"><a href="#cb36-43"></a></span>
<span id="cb36-44"><a href="#cb36-44"></a><span class="fu"># Read metadata</span></span>
<span id="cb36-45"><a href="#cb36-45"></a></span>
<span id="cb36-48"><a href="#cb36-48"></a><span class="in">```{r}</span></span>
<span id="cb36-49"><a href="#cb36-49"></a><span class="co">#| output: false</span></span>
<span id="cb36-50"><a href="#cb36-50"></a><span class="co">#| warning: false</span></span>
<span id="cb36-51"><a href="#cb36-51"></a><span class="co">#| error: false</span></span>
<span id="cb36-52"><a href="#cb36-52"></a>mddf <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(data_raw, <span class="st">"20240711_BTK_illumina_v3_metadata.tsv"</span>))</span>
<span id="cb36-53"><a href="#cb36-53"></a>spdf <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(data_raw, <span class="st">"sample_compositions.tsv"</span>))</span>
<span id="cb36-54"><a href="#cb36-54"></a><span class="in">```</span></span>
<span id="cb36-55"><a href="#cb36-55"></a></span>
<span id="cb36-56"><a href="#cb36-56"></a><span class="fu"># Read Rbec raw counts tables </span></span>
<span id="cb36-57"><a href="#cb36-57"></a></span>
<span id="cb36-58"><a href="#cb36-58"></a><span class="fu">## Untar Rbec output tarball</span></span>
<span id="cb36-59"><a href="#cb36-59"></a></span>
<span id="cb36-62"><a href="#cb36-62"></a><span class="in">```{r}</span></span>
<span id="cb36-63"><a href="#cb36-63"></a>archive<span class="sc">::</span><span class="fu">archive_extract</span>(</span>
<span id="cb36-64"><a href="#cb36-64"></a>  amplicontar,</span>
<span id="cb36-65"><a href="#cb36-65"></a>  <span class="at">dir =</span> tmpdir,</span>
<span id="cb36-66"><a href="#cb36-66"></a>  <span class="at">files =</span> <span class="cn">NULL</span>,</span>
<span id="cb36-67"><a href="#cb36-67"></a>  <span class="at">options =</span> <span class="fu">character</span>(),</span>
<span id="cb36-68"><a href="#cb36-68"></a>  <span class="at">strip_components =</span> <span class="dv">0</span>L</span>
<span id="cb36-69"><a href="#cb36-69"></a>)</span>
<span id="cb36-70"><a href="#cb36-70"></a><span class="in">```</span></span>
<span id="cb36-71"><a href="#cb36-71"></a></span>
<span id="cb36-72"><a href="#cb36-72"></a><span class="fu">## Setup directory structure</span></span>
<span id="cb36-73"><a href="#cb36-73"></a></span>
<span id="cb36-76"><a href="#cb36-76"></a><span class="in">```{r}</span></span>
<span id="cb36-77"><a href="#cb36-77"></a>tabdir <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(tmpdir, <span class="st">"rbec_output"</span>)</span>
<span id="cb36-78"><a href="#cb36-78"></a>samppaths <span class="ot">&lt;-</span> fs<span class="sc">::</span><span class="fu">dir_ls</span>(tabdir)</span>
<span id="cb36-79"><a href="#cb36-79"></a>sampnames <span class="ot">&lt;-</span> <span class="fu">path_split</span>(samppaths) <span class="sc">%&gt;%</span> </span>
<span id="cb36-80"><a href="#cb36-80"></a>  <span class="fu">map_chr</span>(dplyr<span class="sc">::</span>last)</span>
<span id="cb36-81"><a href="#cb36-81"></a><span class="in">```</span></span>
<span id="cb36-82"><a href="#cb36-82"></a></span>
<span id="cb36-83"><a href="#cb36-83"></a><span class="fu">## Read</span></span>
<span id="cb36-84"><a href="#cb36-84"></a></span>
<span id="cb36-87"><a href="#cb36-87"></a><span class="in">```{r}</span></span>
<span id="cb36-88"><a href="#cb36-88"></a><span class="co">#| output: false</span></span>
<span id="cb36-89"><a href="#cb36-89"></a><span class="co">#| warning: false</span></span>
<span id="cb36-90"><a href="#cb36-90"></a><span class="co">#| error: false</span></span>
<span id="cb36-91"><a href="#cb36-91"></a>straintabs <span class="ot">&lt;-</span> <span class="fu">paste0</span>(samppaths, <span class="st">"/strain_table.txt"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb36-92"><a href="#cb36-92"></a>  <span class="fu">set_names</span>(sampnames) <span class="sc">%&gt;%</span> </span>
<span id="cb36-93"><a href="#cb36-93"></a>  <span class="fu">map_df</span>(</span>
<span id="cb36-94"><a href="#cb36-94"></a>  read_tsv,</span>
<span id="cb36-95"><a href="#cb36-95"></a>  <span class="at">skip =</span> <span class="dv">1</span>,</span>
<span id="cb36-96"><a href="#cb36-96"></a>  <span class="at">col_names =</span> <span class="fu">c</span>(<span class="st">"rRNA16S_locus"</span>,<span class="st">"count"</span>),</span>
<span id="cb36-97"><a href="#cb36-97"></a>  <span class="at">show_col_types =</span> <span class="cn">FALSE</span>, </span>
<span id="cb36-98"><a href="#cb36-98"></a>  <span class="at">.id =</span> <span class="st">"sample"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb36-99"><a href="#cb36-99"></a>  <span class="co"># naming scheme inconsistent for one sample</span></span>
<span id="cb36-100"><a href="#cb36-100"></a>  <span class="fu">mutate</span>(<span class="at">sample =</span> <span class="fu">if_else</span>(sample <span class="sc">==</span> <span class="st">"P2_s_0"</span>, <span class="st">"P02_s_0"</span>, sample))</span>
<span id="cb36-101"><a href="#cb36-101"></a><span class="in">```</span></span>
<span id="cb36-102"><a href="#cb36-102"></a></span>
<span id="cb36-103"><a href="#cb36-103"></a><span class="fu">## Clean up</span></span>
<span id="cb36-104"><a href="#cb36-104"></a></span>
<span id="cb36-107"><a href="#cb36-107"></a><span class="in">```{r}</span></span>
<span id="cb36-108"><a href="#cb36-108"></a><span class="co"># remove decompressed directory from temp location</span></span>
<span id="cb36-109"><a href="#cb36-109"></a>fs<span class="sc">::</span><span class="fu">dir_delete</span>(tmpdir)</span>
<span id="cb36-110"><a href="#cb36-110"></a><span class="in">```</span></span>
<span id="cb36-111"><a href="#cb36-111"></a></span>
<span id="cb36-112"><a href="#cb36-112"></a><span class="fu"># Format</span></span>
<span id="cb36-113"><a href="#cb36-113"></a></span>
<span id="cb36-116"><a href="#cb36-116"></a><span class="in">```{r}</span></span>
<span id="cb36-117"><a href="#cb36-117"></a>straintabs_norm <span class="ot">&lt;-</span> straintabs <span class="sc">%&gt;%</span> </span>
<span id="cb36-118"><a href="#cb36-118"></a>  <span class="co"># Calls function that normalize counts by 16S copy number</span></span>
<span id="cb36-119"><a href="#cb36-119"></a>  <span class="fu">normalize_by_copy</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb36-120"><a href="#cb36-120"></a>  <span class="co"># Calls function that completes all combinations of 23 species</span></span>
<span id="cb36-121"><a href="#cb36-121"></a>  <span class="co"># this is important because some species go extinct that should be in the samples</span></span>
<span id="cb36-122"><a href="#cb36-122"></a>  <span class="co"># and we need to have those in the final table</span></span>
<span id="cb36-123"><a href="#cb36-123"></a>  <span class="fu">complete_combos</span>()</span>
<span id="cb36-124"><a href="#cb36-124"></a><span class="in">```</span></span>
<span id="cb36-125"><a href="#cb36-125"></a></span>
<span id="cb36-126"><a href="#cb36-126"></a>Make final table</span>
<span id="cb36-127"><a href="#cb36-127"></a></span>
<span id="cb36-130"><a href="#cb36-130"></a><span class="in">```{r}</span></span>
<span id="cb36-131"><a href="#cb36-131"></a><span class="co"># Later we will take advantage of the fact that for species not supposed to be</span></span>
<span id="cb36-132"><a href="#cb36-132"></a><span class="co"># in a sample the prior left_join will have filled the evo_hist category with an</span></span>
<span id="cb36-133"><a href="#cb36-133"></a><span class="co"># NA. We can then filter using this NA value</span></span>
<span id="cb36-134"><a href="#cb36-134"></a>finaltable <span class="ot">&lt;-</span> <span class="fu">left_join</span>(straintabs_norm, mddf, <span class="at">by =</span> <span class="fu">join_by</span>(sample)) <span class="sc">%&gt;%</span> </span>
<span id="cb36-135"><a href="#cb36-135"></a>  <span class="fu">left_join</span>(spdf, <span class="at">by =</span> <span class="fu">join_by</span>(community_id, strainID)) <span class="sc">%&gt;%</span> </span>
<span id="cb36-136"><a href="#cb36-136"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb36-137"><a href="#cb36-137"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>genus, <span class="sc">-</span>species) <span class="sc">%&gt;%</span> </span>
<span id="cb36-138"><a href="#cb36-138"></a>  <span class="fu">mutate</span>(<span class="at">f_raw =</span> count_correct<span class="sc">/</span><span class="fu">sum</span>(count_correct),</span>
<span id="cb36-139"><a href="#cb36-139"></a>         <span class="at">sample_tot_raw =</span> <span class="fu">sum</span>(count_correct)) <span class="sc">%&gt;%</span> </span>
<span id="cb36-140"><a href="#cb36-140"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb36-141"><a href="#cb36-141"></a>  <span class="fu">relocate</span>(sample_tot_raw, f_raw, target_f, evo_hist, <span class="at">.after =</span> count_correct)</span>
<span id="cb36-142"><a href="#cb36-142"></a><span class="in">```</span></span>
<span id="cb36-143"><a href="#cb36-143"></a></span>
<span id="cb36-144"><a href="#cb36-144"></a><span class="fu"># Analysis</span></span>
<span id="cb36-145"><a href="#cb36-145"></a></span>
<span id="cb36-146"><a href="#cb36-146"></a>Let's quickly compare and look the at number of reads from the different kinds of experiment categories. Negative controls are samples from the experiment that contained growth medium but no cells. We included them to make sure that there was not contamination between wells during the experiment. Positive controls here contain defined mixtures of the 4 species used in this experiment. These positive control samples were extracted using Qiagen DNeasy kit, with the idea that we would use them to calibrate the remaining experimental samples which were all extracted using a simplified boil-prep methodology. The workflow in <span class="co">[</span><span class="ot">this document</span><span class="co">](02_correct_boil_to_qiagen.qmd)</span> does this and it shows that the community composition of the same sample extracted using Qiagen or the boil-prep is not drastically different. Thus, we decided to forgo the hassle and expense of doing these calibrations for the remaining samples, but that information is there to demonstrate that our custom extraction proceedure is robust. Experiment samples are those that came from the actual experiment, and masterplate samples are the defined mixtures we made to inoculate the experiment. We sequenced these masterplates so that we would have T0 information for each experimental treatment.</span>
<span id="cb36-147"><a href="#cb36-147"></a></span>
<span id="cb36-148"><a href="#cb36-148"></a>::: {#fig-01}</span>
<span id="cb36-151"><a href="#cb36-151"></a><span class="in">```{r}</span></span>
<span id="cb36-152"><a href="#cb36-152"></a><span class="co">#| fig.width: 8</span></span>
<span id="cb36-153"><a href="#cb36-153"></a><span class="co">#| fig.height: 5</span></span>
<span id="cb36-154"><a href="#cb36-154"></a>finaltable <span class="sc">%&gt;%</span> </span>
<span id="cb36-155"><a href="#cb36-155"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(sample, sample_tot_raw, community_type) <span class="sc">%&gt;%</span> </span>
<span id="cb36-156"><a href="#cb36-156"></a>  <span class="fu">distinct</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb36-157"><a href="#cb36-157"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> sample_tot_raw)) <span class="sc">+</span></span>
<span id="cb36-158"><a href="#cb36-158"></a>  <span class="fu">geom_histogram</span>(<span class="at">bins =</span> <span class="dv">20</span>, <span class="fu">aes</span>(<span class="at">fill =</span> community_type)) <span class="sc">+</span></span>
<span id="cb36-159"><a href="#cb36-159"></a>  <span class="fu">scale_x_log10</span>() <span class="sc">+</span> </span>
<span id="cb36-160"><a href="#cb36-160"></a>  <span class="fu">annotation_logticks</span>(<span class="at">sides =</span> <span class="st">"b"</span>, <span class="at">color=</span><span class="st">"grey30"</span>) <span class="sc">+</span></span>
<span id="cb36-161"><a href="#cb36-161"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">"Sample count"</span>, <span class="at">x =</span> <span class="st">"Total reads per sample"</span>) <span class="sc">+</span></span>
<span id="cb36-162"><a href="#cb36-162"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span> community_type, <span class="at">scales=</span><span class="st">"free_y"</span>)</span>
<span id="cb36-163"><a href="#cb36-163"></a><span class="in">```</span></span>
<span id="cb36-164"><a href="#cb36-164"></a>Frequency distribution of total read counts per sample in the experimental samples (red) masterplates (green), negative controls (blue), and the positive controls (purple).</span>
<span id="cb36-165"><a href="#cb36-165"></a>:::</span>
<span id="cb36-166"><a href="#cb36-166"></a></span>
<span id="cb36-167"><a href="#cb36-167"></a>The smallest number of reads in the experimental samples is <span class="in">`{r} min(finaltable[ grep("^exp", finaltable$community_type), ]$sample_tot_raw)`</span> and the smallest in the negative controls is <span class="in">`{r} min(finaltable[ grep("^neg", finaltable$community_type), ]$sample_tot_raw)`</span>. Clearly some of the experimental samples failed to sequence well. The mean number of reads in the experimental samples is <span class="in">`{r} format(mean(finaltable[ grep("^exp", finaltable$community_type), ]$sample_tot_raw), scientific=F)`</span> and in the negative controls is <span class="in">`{r} mean(finaltable[ grep("^neg", finaltable$community_type), ]$sample_tot_raw)`</span>. The mean for the experimental samples is considerably higher than the mean for the negative controls.</span>
<span id="cb36-168"><a href="#cb36-168"></a></span>
<span id="cb36-169"><a href="#cb36-169"></a>Overall this looks really good. The masterplate samples and positive controls all seem to have succeeded. However, there appears to be a negative control or two which has ~ 10000 reads and so it might have been contaminated. </span>
<span id="cb36-170"><a href="#cb36-170"></a></span>
<span id="cb36-171"><a href="#cb36-171"></a><span class="fu">## Negative controls</span></span>
<span id="cb36-172"><a href="#cb36-172"></a></span>
<span id="cb36-173"><a href="#cb36-173"></a>Lets check what species are contaminating negative controls. Most appear to be low abundance contaminants of species excluded from the experiment. Probably this is cross talk from the positive controls and other samples</span>
<span id="cb36-174"><a href="#cb36-174"></a></span>
<span id="cb36-175"><a href="#cb36-175"></a>::: {#fig-02}</span>
<span id="cb36-178"><a href="#cb36-178"></a><span class="in">```{r}</span></span>
<span id="cb36-179"><a href="#cb36-179"></a><span class="co">#| fig.width: 8</span></span>
<span id="cb36-180"><a href="#cb36-180"></a><span class="co">#| fig.height: 7</span></span>
<span id="cb36-181"><a href="#cb36-181"></a><span class="co">#| output: true</span></span>
<span id="cb36-182"><a href="#cb36-182"></a><span class="co">#| warning: false</span></span>
<span id="cb36-183"><a href="#cb36-183"></a><span class="co">#| error: false</span></span>
<span id="cb36-184"><a href="#cb36-184"></a>finaltable <span class="sc">%&gt;%</span> </span>
<span id="cb36-185"><a href="#cb36-185"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(community_type, <span class="st">"^neg"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb36-186"><a href="#cb36-186"></a>  <span class="fu">filter</span>(f_raw <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb36-187"><a href="#cb36-187"></a>  <span class="fu">contam_histogram</span>(<span class="at">trans=</span><span class="cn">TRUE</span>, <span class="at">x =</span> f_raw) <span class="sc">+</span></span>
<span id="cb36-188"><a href="#cb36-188"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Frequency distribution of each species in the negative controls"</span>, <span class="at">y =</span> <span class="st">"Sample count"</span>) <span class="sc">+</span></span>
<span id="cb36-189"><a href="#cb36-189"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>strainID)</span>
<span id="cb36-190"><a href="#cb36-190"></a><span class="in">```</span></span>
<span id="cb36-191"><a href="#cb36-191"></a>Frequency distribution of all species in the negative controls</span>
<span id="cb36-192"><a href="#cb36-192"></a>:::</span>
<span id="cb36-193"><a href="#cb36-193"></a></span>
<span id="cb36-194"><a href="#cb36-194"></a>I am not sure where the off-target species are coming from because we did not include any positive community controls here. However, they are all very low abundance and so I think they are just noise. Also it is worth noting that HAMBI_1972 and HAMBI_1977 are the same genus so if some of the read qualities are crappy it could easily be missassigned. </span>
<span id="cb36-195"><a href="#cb36-195"></a></span>
<span id="cb36-196"><a href="#cb36-196"></a>::: {#fig-03}</span>
<span id="cb36-199"><a href="#cb36-199"></a><span class="in">```{r}</span></span>
<span id="cb36-200"><a href="#cb36-200"></a><span class="co">#| fig.width: 8</span></span>
<span id="cb36-201"><a href="#cb36-201"></a><span class="co">#| fig.height: 3</span></span>
<span id="cb36-202"><a href="#cb36-202"></a>finaltable <span class="sc">%&gt;%</span> </span>
<span id="cb36-203"><a href="#cb36-203"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(community_type, <span class="st">"^neg"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb36-204"><a href="#cb36-204"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(strainID, <span class="st">"0403|1287|1896|1977"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb36-205"><a href="#cb36-205"></a>  <span class="fu">filter</span>(f_raw <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb36-206"><a href="#cb36-206"></a>  <span class="fu">contam_histogram</span>(<span class="at">trans=</span><span class="cn">TRUE</span>, <span class="at">x =</span> f_raw) <span class="sc">+</span></span>
<span id="cb36-207"><a href="#cb36-207"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Frequency distribution of each species in the negative controls"</span>, <span class="at">y =</span> <span class="st">"Sample count"</span>)</span>
<span id="cb36-208"><a href="#cb36-208"></a><span class="in">```</span></span>
<span id="cb36-209"><a href="#cb36-209"></a>Frequency distribution of only target species in the negative controls</span>
<span id="cb36-210"><a href="#cb36-210"></a>:::</span>
<span id="cb36-211"><a href="#cb36-211"></a></span>
<span id="cb36-214"><a href="#cb36-214"></a><span class="in">```{r}</span></span>
<span id="cb36-215"><a href="#cb36-215"></a>finaltable <span class="sc">%&gt;%</span> </span>
<span id="cb36-216"><a href="#cb36-216"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(community_type, <span class="st">"^neg"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb36-217"><a href="#cb36-217"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(strainID, <span class="st">"0403|1287|1896|1977"</span>))</span>
<span id="cb36-218"><a href="#cb36-218"></a><span class="in">```</span></span>
<span id="cb36-219"><a href="#cb36-219"></a></span>
<span id="cb36-220"><a href="#cb36-220"></a>This looks ok, but there are potentially some problems. Specifically, negative control replicates 2, 3, and 4 all have some contamination. <span class="in">`neg_2_0`</span> seems to be contaminated with HAMBI_1977, <span class="in">`neg_3_0`</span> with HAMBI_1287, and <span class="in">`neg_4_0`</span> with HAMBI_1977. However in the 5 remaining negative controls there is no contamination.</span>
<span id="cb36-221"><a href="#cb36-221"></a></span>
<span id="cb36-222"><a href="#cb36-222"></a><span class="fu">## Positive controls</span></span>
<span id="cb36-223"><a href="#cb36-223"></a></span>
<span id="cb36-224"><a href="#cb36-224"></a>The positive controls should each have three species. In all cases the species that shouldn't be there is very rare</span>
<span id="cb36-225"><a href="#cb36-225"></a></span>
<span id="cb36-228"><a href="#cb36-228"></a><span class="in">```{r}</span></span>
<span id="cb36-229"><a href="#cb36-229"></a>finaltable <span class="sc">%&gt;%</span> </span>
<span id="cb36-230"><a href="#cb36-230"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(community_type, <span class="st">"^pos"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb36-231"><a href="#cb36-231"></a>  <span class="fu">mutate</span>(<span class="at">total_pos_controls =</span> <span class="fu">n_distinct</span>(sample)) <span class="sc">%&gt;%</span> </span>
<span id="cb36-232"><a href="#cb36-232"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb36-233"><a href="#cb36-233"></a>  <span class="fu">mutate</span>(<span class="at">f =</span> <span class="fu">round</span>(f_raw<span class="sc">*</span><span class="dv">100</span>, <span class="dv">1</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb36-234"><a href="#cb36-234"></a>  <span class="fu">mutate</span>(<span class="at">supposed_2_b_there =</span> <span class="fu">if_else</span>(<span class="fu">is.na</span>(evo_hist), <span class="st">"no"</span>, <span class="st">"yes"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb36-235"><a href="#cb36-235"></a>  <span class="fu">relocate</span>(f, supposed_2_b_there) <span class="sc">%&gt;%</span> </span>
<span id="cb36-236"><a href="#cb36-236"></a>  <span class="fu">distinct</span>(f, supposed_2_b_there, sample, strainID, count_correct, n_species, community_type,) <span class="sc">%&gt;%</span> </span>
<span id="cb36-237"><a href="#cb36-237"></a>  <span class="fu">filter</span>(f <span class="sc">&gt;</span> <span class="dv">0</span>)</span>
<span id="cb36-238"><a href="#cb36-238"></a><span class="in">```</span></span>
<span id="cb36-239"><a href="#cb36-239"></a></span>
<span id="cb36-240"><a href="#cb36-240"></a>This is also good - we detect all 3 species that should be there in the positive controls on each plate.</span>
<span id="cb36-241"><a href="#cb36-241"></a></span>
<span id="cb36-242"><a href="#cb36-242"></a><span class="fu">## Misassigned reads</span></span>
<span id="cb36-243"><a href="#cb36-243"></a></span>
<span id="cb36-244"><a href="#cb36-244"></a>These libraries were only prepared with samples from Milla's 4-species experiment with 403, 1287, 1896, and 1977 so any time species other than these show up is just an incorrect assignment by Rbec. Let's check quickly how many of these there are...</span>
<span id="cb36-245"><a href="#cb36-245"></a></span>
<span id="cb36-248"><a href="#cb36-248"></a><span class="in">```{r}</span></span>
<span id="cb36-249"><a href="#cb36-249"></a>finaltable <span class="sc">%&gt;%</span> </span>
<span id="cb36-250"><a href="#cb36-250"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">str_detect</span>(strainID, <span class="st">"0403|1287|1896|1977"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb36-251"><a href="#cb36-251"></a>  <span class="fu">filter</span>(count_correct <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb36-252"><a href="#cb36-252"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(count_correct))</span>
<span id="cb36-253"><a href="#cb36-253"></a><span class="in">```</span></span>
<span id="cb36-254"><a href="#cb36-254"></a></span>
<span id="cb36-255"><a href="#cb36-255"></a>There is one or two incorrectly assigned reads here and there but this is just noise. We can safely exclude all reads not mapping to one of the focal species.</span>
<span id="cb36-256"><a href="#cb36-256"></a></span>
<span id="cb36-257"><a href="#cb36-257"></a><span class="fu">## Samples with few total reads</span></span>
<span id="cb36-258"><a href="#cb36-258"></a></span>
<span id="cb36-259"><a href="#cb36-259"></a>Some of the experimental pairs had streptomycin concentrations higher than any of the species individually could tolerate. We would naively expect then that neither species would grow successfully in these samples and that the overall biomass would be very low, thus resulting in a low number of recovered reads from these samples.</span>
<span id="cb36-260"><a href="#cb36-260"></a></span>
<span id="cb36-261"><a href="#cb36-261"></a>To look into this. first let's check which samples have very low OD600 in the endpoint samples.</span>
<span id="cb36-262"><a href="#cb36-262"></a></span>
<span id="cb36-265"><a href="#cb36-265"></a><span class="in">```{r}</span></span>
<span id="cb36-266"><a href="#cb36-266"></a><span class="co">#| output: false</span></span>
<span id="cb36-267"><a href="#cb36-267"></a><span class="co">#| warning: false</span></span>
<span id="cb36-268"><a href="#cb36-268"></a><span class="co">#| error: false</span></span>
<span id="cb36-269"><a href="#cb36-269"></a>od <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"_data_raw"</span>, <span class="st">"communities"</span>, <span class="st">"optical_density"</span>, <span class="st">"20240606_pairs"</span>, <span class="st">"optical_density_formatted.tsv"</span>))</span>
<span id="cb36-270"><a href="#cb36-270"></a><span class="in">```</span></span>
<span id="cb36-271"><a href="#cb36-271"></a></span>
<span id="cb36-272"><a href="#cb36-272"></a>We'll filter out samples with an OD of less than 0.1 and also samples with fewer than 1000 reads. It is generally good practice to exclude samples with low number of reads.</span>
<span id="cb36-273"><a href="#cb36-273"></a></span>
<span id="cb36-276"><a href="#cb36-276"></a><span class="in">```{r}</span></span>
<span id="cb36-277"><a href="#cb36-277"></a>df <span class="ot">&lt;-</span> <span class="fu">left_join</span>(finaltable, od, </span>
<span id="cb36-278"><a href="#cb36-278"></a>                  <span class="at">by =</span> <span class="fu">join_by</span>(community_id, n_species, transfers, strep_conc, replicate)) <span class="sc">%&gt;%</span> </span>
<span id="cb36-279"><a href="#cb36-279"></a>  <span class="fu">filter</span>(transfers <span class="sc">==</span> <span class="dv">8</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb36-280"><a href="#cb36-280"></a>  <span class="fu">arrange</span>(sample_tot_raw) <span class="sc">%&gt;%</span> </span>
<span id="cb36-281"><a href="#cb36-281"></a>  <span class="fu">arrange</span>(OD) <span class="sc">%&gt;%</span> </span>
<span id="cb36-282"><a href="#cb36-282"></a>  <span class="co"># filter only samples with OD &lt; 0.1 or &lt; 1000 total reads. 1000 read threshold </span></span>
<span id="cb36-283"><a href="#cb36-283"></a>  <span class="co"># was chose based on the first figure in this notebook</span></span>
<span id="cb36-284"><a href="#cb36-284"></a>  <span class="fu">filter</span>(OD <span class="sc">&lt;</span> <span class="fl">0.1</span> <span class="sc">|</span> sample_tot_raw <span class="sc">&lt;</span> <span class="dv">1000</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb36-285"><a href="#cb36-285"></a>  <span class="fu">filter</span>(count_correct <span class="sc">&gt;</span> <span class="dv">0</span>)</span>
<span id="cb36-286"><a href="#cb36-286"></a></span>
<span id="cb36-287"><a href="#cb36-287"></a>lowread_samps <span class="ot">&lt;-</span> df <span class="sc">%&gt;%</span> </span>
<span id="cb36-288"><a href="#cb36-288"></a>  <span class="fu">filter</span>(sample_tot_raw <span class="sc">&lt;</span> <span class="dv">3000</span> <span class="sc">&amp;</span> community_type <span class="sc">==</span> <span class="st">"experiment"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb36-289"><a href="#cb36-289"></a>  <span class="fu">distinct</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb36-290"><a href="#cb36-290"></a>  <span class="fu">pull</span>(sample)</span>
<span id="cb36-291"><a href="#cb36-291"></a>  </span>
<span id="cb36-292"><a href="#cb36-292"></a>df <span class="sc">%&gt;%</span> </span>
<span id="cb36-293"><a href="#cb36-293"></a>  <span class="fu">distinct</span>(sample, community_type, OD, sample_tot_raw)</span>
<span id="cb36-294"><a href="#cb36-294"></a><span class="in">```</span></span>
<span id="cb36-295"><a href="#cb36-295"></a></span>
<span id="cb36-296"><a href="#cb36-296"></a>There are 10 experimental samples that have less than 1000 reads. 6 of those samples have low optical density which indicates nothing was growing there and that they should have low read number. 4 samples have normal optical density (~ 0.3) and these look to be samples that failed at the library preparation/sequencing step. Regardless, all these 10 samples will be excluded from the downstream analysis.</span>
<span id="cb36-297"><a href="#cb36-297"></a></span>
<span id="cb36-298"><a href="#cb36-298"></a><span class="fu">## Filter to target species</span></span>
<span id="cb36-299"><a href="#cb36-299"></a></span>
<span id="cb36-300"><a href="#cb36-300"></a>Exclude positive and negative controls and also exclude species that are not in the focal 4</span>
<span id="cb36-301"><a href="#cb36-301"></a></span>
<span id="cb36-304"><a href="#cb36-304"></a><span class="in">```{r}</span></span>
<span id="cb36-305"><a href="#cb36-305"></a>finaltable_exp_mstr <span class="ot">&lt;-</span> finaltable <span class="sc">%&gt;%</span> </span>
<span id="cb36-306"><a href="#cb36-306"></a>  <span class="fu">filter</span>(community_type <span class="sc">%nin%</span> <span class="fu">c</span>(<span class="st">"empty"</span>, <span class="st">"neg_ctrl"</span>, <span class="st">"pos_ctrl"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb36-307"><a href="#cb36-307"></a>  <span class="co"># also exclude samples with low reads</span></span>
<span id="cb36-308"><a href="#cb36-308"></a>  <span class="fu">filter</span>(sample <span class="sc">%nin%</span> lowread_samps) <span class="sc">%&gt;%</span> </span>
<span id="cb36-309"><a href="#cb36-309"></a>  <span class="fu">filter</span>(strainID <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"HAMBI_0403"</span>, <span class="st">"HAMBI_1287"</span>, <span class="st">"HAMBI_1896"</span>, <span class="st">"HAMBI_1977"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb36-310"><a href="#cb36-310"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb36-311"><a href="#cb36-311"></a>  <span class="fu">mutate</span>(<span class="at">f_raw_targetsp =</span> count_correct<span class="sc">/</span><span class="fu">sum</span>(count_correct),</span>
<span id="cb36-312"><a href="#cb36-312"></a>         <span class="at">sample_tot_targetsp =</span> <span class="fu">sum</span>(count_correct)) <span class="sc">%&gt;%</span> </span>
<span id="cb36-313"><a href="#cb36-313"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb36-314"><a href="#cb36-314"></a>  <span class="fu">relocate</span>(<span class="fu">c</span>(sample_tot_targetsp, f_raw_targetsp), <span class="at">.after =</span> f_raw)</span>
<span id="cb36-315"><a href="#cb36-315"></a><span class="in">```</span></span>
<span id="cb36-316"><a href="#cb36-316"></a></span>
<span id="cb36-317"><a href="#cb36-317"></a><span class="fu">## Masterplate samples</span></span>
<span id="cb36-318"><a href="#cb36-318"></a></span>
<span id="cb36-319"><a href="#cb36-319"></a>To set up this experiment, Milla combined the species in the planned proportions on a masterplate. Because this process was time consuming, the masterplate was stored at -80C after construction until the experiment start day when it was taken from the freezer and used to inoculate the experiment. Because Milla knows exactly which strains were added to the master plate and the plates were not allowed to grow, any strains in these samples that are not supposed to be there should be due to Illumina index cross talk and not true contamination. We can get a sense for the average index crosstalk rate from these samples and then draw a threshold of when to exclude likely false positives and when a positive is likely due to contamination.</span>
<span id="cb36-320"><a href="#cb36-320"></a></span>
<span id="cb36-321"><a href="#cb36-321"></a><span class="fu">### Pairs</span></span>
<span id="cb36-322"><a href="#cb36-322"></a></span>
<span id="cb36-323"><a href="#cb36-323"></a>Overall the masterplate pairs look very good. There are no samples where a species that should not be there is present in high abundances. Any contamination </span>
<span id="cb36-324"><a href="#cb36-324"></a>(grey to black fill) is well under 5%.</span>
<span id="cb36-325"><a href="#cb36-325"></a></span>
<span id="cb36-326"><a href="#cb36-326"></a>::: {#fig-04}</span>
<span id="cb36-329"><a href="#cb36-329"></a><span class="in">```{r}</span></span>
<span id="cb36-330"><a href="#cb36-330"></a><span class="co">#| fig.width: 8</span></span>
<span id="cb36-331"><a href="#cb36-331"></a><span class="co">#| fig.height: 5</span></span>
<span id="cb36-332"><a href="#cb36-332"></a>finaltable_exp_mstr <span class="sc">%&gt;%</span></span>
<span id="cb36-333"><a href="#cb36-333"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(community_type, <span class="st">"masterplate"</span>) <span class="sc">&amp;</span> n_species <span class="sc">==</span> <span class="dv">2</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb36-334"><a href="#cb36-334"></a>  <span class="fu">contaminated_barplot</span>(<span class="at">threshold =</span> <span class="dv">0</span>, <span class="at">quartet =</span> <span class="cn">FALSE</span>, <span class="at">y=</span>f_raw_targetsp) <span class="sc">+</span></span>
<span id="cb36-335"><a href="#cb36-335"></a>  <span class="fu">ggtitle</span>(<span class="st">"All Masterplate pairs"</span>)</span>
<span id="cb36-336"><a href="#cb36-336"></a><span class="in">```</span></span>
<span id="cb36-337"><a href="#cb36-337"></a>Species composition of all masterplate pairs. Any species that shouldn’t be there (&gt; 0%) is shown in grey to black.</span>
<span id="cb36-338"><a href="#cb36-338"></a>:::</span>
<span id="cb36-339"><a href="#cb36-339"></a></span>
<span id="cb36-340"><a href="#cb36-340"></a>There are no samples with more than 4% of a species that shouldn't be there</span>
<span id="cb36-343"><a href="#cb36-343"></a><span class="in">```{r}</span></span>
<span id="cb36-344"><a href="#cb36-344"></a>finaltable_exp_mstr <span class="sc">%&gt;%</span> </span>
<span id="cb36-345"><a href="#cb36-345"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(community_type, <span class="st">"masterplate"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb36-346"><a href="#cb36-346"></a>  <span class="fu">filter</span>(n_species <span class="sc">==</span> <span class="st">"2"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb36-347"><a href="#cb36-347"></a>  <span class="fu">filter</span>(<span class="fu">is.na</span>(evo_hist)) <span class="sc">%&gt;%</span> </span>
<span id="cb36-348"><a href="#cb36-348"></a>  <span class="fu">filter</span>(f_raw_targetsp <span class="sc">&gt;</span> <span class="fl">0.04</span> ) <span class="sc">%&gt;%</span> </span>
<span id="cb36-349"><a href="#cb36-349"></a>  <span class="fu">distinct</span>(sample)</span>
<span id="cb36-350"><a href="#cb36-350"></a><span class="in">```</span></span>
<span id="cb36-351"><a href="#cb36-351"></a></span>
<span id="cb36-352"><a href="#cb36-352"></a>::: {#fig-05}</span>
<span id="cb36-355"><a href="#cb36-355"></a><span class="in">```{r}</span></span>
<span id="cb36-356"><a href="#cb36-356"></a><span class="co">#| fig.width: 8</span></span>
<span id="cb36-357"><a href="#cb36-357"></a><span class="co">#| fig.height: 4</span></span>
<span id="cb36-358"><a href="#cb36-358"></a>finaltable_exp_mstr <span class="sc">%&gt;%</span></span>
<span id="cb36-359"><a href="#cb36-359"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(community_type, <span class="st">"masterplate"</span>) <span class="sc">&amp;</span> n_species <span class="sc">==</span> <span class="dv">2</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb36-360"><a href="#cb36-360"></a>  <span class="fu">filter</span>(<span class="fu">is.na</span>(evo_hist)) <span class="sc">%&gt;%</span> </span>
<span id="cb36-361"><a href="#cb36-361"></a>  <span class="fu">filter</span>(f_raw_targetsp <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb36-362"><a href="#cb36-362"></a>  <span class="fu">contam_histogram</span>(<span class="at">trans =</span> <span class="cn">TRUE</span>, <span class="at">x =</span> f_raw_targetsp) <span class="sc">+</span></span>
<span id="cb36-363"><a href="#cb36-363"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Species frequency"</span>, <span class="at">y =</span> <span class="st">"Sample count"</span>) <span class="sc">+</span></span>
<span id="cb36-364"><a href="#cb36-364"></a>  <span class="fu">ggtitle</span>(<span class="st">"Frequency distribution of misassigned species in masterplate pairs"</span>)</span>
<span id="cb36-365"><a href="#cb36-365"></a><span class="in">```</span></span>
<span id="cb36-366"><a href="#cb36-366"></a>Frequency distribution of each species in masterplate pair samples where it **should not** occur. The 1% mark (commonly accepted as an acceptable Illumina cross-talk standard) is demarcated with a dashed line. </span>
<span id="cb36-367"><a href="#cb36-367"></a>:::</span>
<span id="cb36-368"><a href="#cb36-368"></a></span>
<span id="cb36-369"><a href="#cb36-369"></a><span class="fu">### Trios</span></span>
<span id="cb36-370"><a href="#cb36-370"></a></span>
<span id="cb36-371"><a href="#cb36-371"></a>Trio masterplates also look very good. There are some samples where a species that shouldn't be there is &gt; 3%, but these are always &lt; 5% and there are no samples with major contamination. Importantly these all match the proportions that we used to start the experiment</span>
<span id="cb36-372"><a href="#cb36-372"></a></span>
<span id="cb36-373"><a href="#cb36-373"></a>::: {#fig-06}</span>
<span id="cb36-376"><a href="#cb36-376"></a><span class="in">```{r}</span></span>
<span id="cb36-377"><a href="#cb36-377"></a><span class="co">#| fig.width: 8</span></span>
<span id="cb36-378"><a href="#cb36-378"></a><span class="co">#| fig.height: 10</span></span>
<span id="cb36-379"><a href="#cb36-379"></a>finaltable_exp_mstr <span class="sc">%&gt;%</span></span>
<span id="cb36-380"><a href="#cb36-380"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(community_type, <span class="st">"masterplate"</span>) <span class="sc">&amp;</span> n_species <span class="sc">==</span> <span class="dv">3</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb36-381"><a href="#cb36-381"></a>  <span class="co"># this is a dumb hack so that I don't have to plot all bars on the same axis</span></span>
<span id="cb36-382"><a href="#cb36-382"></a>  <span class="co"># but I can wrap them using a meaningless facet</span></span>
<span id="cb36-383"><a href="#cb36-383"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb36-384"><a href="#cb36-384"></a>  <span class="fu">mutate</span>(<span class="at">id =</span> <span class="fu">cur_group_id</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb36-385"><a href="#cb36-385"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb36-386"><a href="#cb36-386"></a>  <span class="fu">mutate</span>(<span class="at">facet =</span> <span class="fu">ntile</span>(id, <span class="dv">4</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb36-387"><a href="#cb36-387"></a>  <span class="fu">contaminated_barplot</span>(<span class="at">threshold =</span> <span class="dv">0</span>, <span class="at">quartet =</span> <span class="cn">FALSE</span>, <span class="at">y=</span>f_raw_targetsp) <span class="sc">+</span></span>
<span id="cb36-388"><a href="#cb36-388"></a>  <span class="fu">ggtitle</span>(<span class="st">"All Masterplate trios"</span>) <span class="sc">+</span></span>
<span id="cb36-389"><a href="#cb36-389"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>facet, <span class="at">scales=</span><span class="st">"free_x"</span>, <span class="at">nrow=</span><span class="dv">4</span>) <span class="sc">+</span></span>
<span id="cb36-390"><a href="#cb36-390"></a>  <span class="fu">theme</span>(</span>
<span id="cb36-391"><a href="#cb36-391"></a>    <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb36-392"><a href="#cb36-392"></a>    <span class="at">strip.text.x =</span> <span class="fu">element_blank</span>()</span>
<span id="cb36-393"><a href="#cb36-393"></a>    )</span>
<span id="cb36-394"><a href="#cb36-394"></a><span class="in">```</span></span>
<span id="cb36-395"><a href="#cb36-395"></a>Species composition of all masterplate trios. Any species that shouldn’t be there (&gt; 0%) is shown in grey to black. The samples are split over four rows to aid in visualization - the specific subplots are meaningless.</span>
<span id="cb36-396"><a href="#cb36-396"></a>:::</span>
<span id="cb36-397"><a href="#cb36-397"></a></span>
<span id="cb36-398"><a href="#cb36-398"></a>::: {#fig-07}</span>
<span id="cb36-401"><a href="#cb36-401"></a><span class="in">```{r}</span></span>
<span id="cb36-402"><a href="#cb36-402"></a><span class="co">#| fig.width: 8</span></span>
<span id="cb36-403"><a href="#cb36-403"></a><span class="co">#| fig.height: 4</span></span>
<span id="cb36-404"><a href="#cb36-404"></a>finaltable_exp_mstr <span class="sc">%&gt;%</span></span>
<span id="cb36-405"><a href="#cb36-405"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(community_type, <span class="st">"masterplate"</span>) <span class="sc">&amp;</span> n_species <span class="sc">==</span> <span class="dv">3</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb36-406"><a href="#cb36-406"></a>  <span class="fu">filter</span>(<span class="fu">is.na</span>(evo_hist)) <span class="sc">%&gt;%</span> </span>
<span id="cb36-407"><a href="#cb36-407"></a>  <span class="fu">filter</span>(f_raw_targetsp <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb36-408"><a href="#cb36-408"></a>  <span class="fu">contam_histogram</span>(<span class="at">trans =</span> <span class="cn">TRUE</span>, <span class="at">x =</span> f_raw_targetsp) <span class="sc">+</span></span>
<span id="cb36-409"><a href="#cb36-409"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Species frequency"</span>, <span class="at">y =</span> <span class="st">"Sample count"</span>) <span class="sc">+</span></span>
<span id="cb36-410"><a href="#cb36-410"></a>  <span class="fu">ggtitle</span>(<span class="st">"Frequency distribution of misassigned species in masterplate trios"</span>)</span>
<span id="cb36-411"><a href="#cb36-411"></a><span class="in">```</span></span>
<span id="cb36-412"><a href="#cb36-412"></a>Frequency distribution of each species in masterplate trio samples where it **should not** occur. The 1% mark (commonly accepted as an acceptable Illumina cross-talk standard) is demarcated with a dashed line. </span>
<span id="cb36-413"><a href="#cb36-413"></a>:::</span>
<span id="cb36-414"><a href="#cb36-414"></a></span>
<span id="cb36-415"><a href="#cb36-415"></a>There are no samples with more than 3% of a species that shouldn't be there</span>
<span id="cb36-418"><a href="#cb36-418"></a><span class="in">```{r}</span></span>
<span id="cb36-419"><a href="#cb36-419"></a>finaltable_exp_mstr <span class="sc">%&gt;%</span> </span>
<span id="cb36-420"><a href="#cb36-420"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(community_type, <span class="st">"masterplate"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb36-421"><a href="#cb36-421"></a>  <span class="fu">filter</span>(n_species <span class="sc">==</span> <span class="st">"3"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb36-422"><a href="#cb36-422"></a>  <span class="fu">filter</span>(<span class="fu">is.na</span>(evo_hist)) <span class="sc">%&gt;%</span> </span>
<span id="cb36-423"><a href="#cb36-423"></a>  <span class="fu">filter</span>(f_raw_targetsp <span class="sc">&gt;</span> <span class="fl">0.03</span> ) <span class="sc">%&gt;%</span> </span>
<span id="cb36-424"><a href="#cb36-424"></a>  <span class="fu">distinct</span>(sample)</span>
<span id="cb36-425"><a href="#cb36-425"></a><span class="in">```</span></span>
<span id="cb36-426"><a href="#cb36-426"></a></span>
<span id="cb36-427"><a href="#cb36-427"></a><span class="fu">### Masterplate Summary</span></span>
<span id="cb36-428"><a href="#cb36-428"></a></span>
<span id="cb36-429"><a href="#cb36-429"></a>Generally, the cross talk frequency is really good. For 3/4 species it is 1% or less which is more or less what you can expect when you are multiplexing libraries on an Illumina platform. Values greater than 1% are potentially indicative of a different problem, so 1977 requires a bit more investigation.</span>
<span id="cb36-430"><a href="#cb36-430"></a></span>
<span id="cb36-433"><a href="#cb36-433"></a><span class="in">```{r}</span></span>
<span id="cb36-434"><a href="#cb36-434"></a>finaltable_exp_mstr <span class="sc">%&gt;%</span></span>
<span id="cb36-435"><a href="#cb36-435"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(community_type, <span class="st">"masterplate"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb36-436"><a href="#cb36-436"></a>  <span class="fu">filter</span>(<span class="fu">is.na</span>(evo_hist)) <span class="sc">%&gt;%</span> </span>
<span id="cb36-437"><a href="#cb36-437"></a>  <span class="fu">filter</span>(f_raw_targetsp <span class="sc">&gt;</span> <span class="fl">0.01</span> <span class="sc">&amp;</span> strainID <span class="sc">==</span> <span class="st">"HAMBI_1977"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb36-438"><a href="#cb36-438"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(f_raw_targetsp))</span>
<span id="cb36-439"><a href="#cb36-439"></a><span class="in">```</span></span>
<span id="cb36-440"><a href="#cb36-440"></a></span>
<span id="cb36-441"><a href="#cb36-441"></a>It looks like all the "problematic" samples come from plates 7 and 8 in the library prep. Plate 8 only contains the masterplate samples from trios whereas plate 7 contains both masterplate and experimental samples. In @fig-02 it shows that HAMBI-1977 is very abundant in many of the samples so likely the "leaky" reads come disproportionately from HAMBI-1977 which is why its crosstalk threshold may be higher (@Fig-01). Anyway, I don't think this is a problem and that we can move forward with these samples. However, to define extinction/competitive exclusion we may need to use a higher threshold than 1% (e.g., 3% frequency) because over 3% we can reliably say that a species is present and it is not due to index cross talk.</span>
<span id="cb36-442"><a href="#cb36-442"></a></span>
<span id="cb36-443"><a href="#cb36-443"></a><span class="fu">## Experimental samples</span></span>
<span id="cb36-444"><a href="#cb36-444"></a></span>
<span id="cb36-445"><a href="#cb36-445"></a>We may need to use a higher threshold than 1% (e.g., 3% frequency) because over 3% we can reliably say that a species is present and it is not due to index cross talk. Below I basically empirically chose 8% as a reasonable threshold value that includes the maximum number of samples with low-level contamination and exclude samples with very high contamination.</span>
<span id="cb36-446"><a href="#cb36-446"></a></span>
<span id="cb36-447"><a href="#cb36-447"></a>::: {#fig-08}</span>
<span id="cb36-450"><a href="#cb36-450"></a><span class="in">```{r}</span></span>
<span id="cb36-451"><a href="#cb36-451"></a><span class="co">#| fig.width: 8</span></span>
<span id="cb36-452"><a href="#cb36-452"></a><span class="co">#| fig.height: 5</span></span>
<span id="cb36-453"><a href="#cb36-453"></a>finaltable_exp_mstr <span class="sc">%&gt;%</span> </span>
<span id="cb36-454"><a href="#cb36-454"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(community_type, <span class="st">"experiment"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb36-455"><a href="#cb36-455"></a>  <span class="fu">filter</span>(<span class="fu">is.na</span>(evo_hist)) <span class="sc">%&gt;%</span> </span>
<span id="cb36-456"><a href="#cb36-456"></a>  <span class="fu">filter</span>(f_raw <span class="sc">&gt;</span> <span class="fl">0.03</span> ) <span class="sc">%&gt;%</span> </span>
<span id="cb36-457"><a href="#cb36-457"></a>  <span class="fu">contam_histogram</span>(<span class="at">trans=</span><span class="cn">TRUE</span>, <span class="at">x =</span> f_raw_targetsp) <span class="sc">+</span></span>
<span id="cb36-458"><a href="#cb36-458"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Species frequency"</span>, <span class="at">y =</span> <span class="st">"Sample count"</span>) <span class="sc">+</span></span>
<span id="cb36-459"><a href="#cb36-459"></a>  <span class="fu">ggtitle</span>(<span class="st">"Frequency distribution of each species exceeding the 3% threshold"</span>) <span class="sc">+</span></span>
<span id="cb36-460"><a href="#cb36-460"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fl">0.08</span>, <span class="at">color =</span> <span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb36-461"><a href="#cb36-461"></a>  <span class="fu">facet_grid</span>(n_species<span class="sc">~</span>strainID)</span>
<span id="cb36-462"><a href="#cb36-462"></a><span class="in">```</span></span>
<span id="cb36-463"><a href="#cb36-463"></a>Frequency distribution of each species in experimental samples where it **should not** occur and where it exceeds a 3% threshold. Grid columns are the different species and grid rows represent whether the sample was an experiment pair (2) or trio (3). The red vertical line is at 8%.</span>
<span id="cb36-464"><a href="#cb36-464"></a>:::</span>
<span id="cb36-465"><a href="#cb36-465"></a></span>
<span id="cb36-468"><a href="#cb36-468"></a><span class="in">```{r}</span></span>
<span id="cb36-469"><a href="#cb36-469"></a>finaltable_exp_mstr <span class="sc">%&gt;%</span> </span>
<span id="cb36-470"><a href="#cb36-470"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(community_type, <span class="st">"experiment"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb36-471"><a href="#cb36-471"></a>  <span class="fu">filter</span>(<span class="fu">is.na</span>(evo_hist)) <span class="sc">%&gt;%</span> </span>
<span id="cb36-472"><a href="#cb36-472"></a>  <span class="fu">filter</span>(f_raw_targetsp <span class="sc">&gt;</span> <span class="fl">0.03</span> ) <span class="sc">%&gt;%</span> </span>
<span id="cb36-473"><a href="#cb36-473"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(f_raw_targetsp))</span>
<span id="cb36-474"><a href="#cb36-474"></a><span class="in">```</span></span>
<span id="cb36-475"><a href="#cb36-475"></a></span>
<span id="cb36-478"><a href="#cb36-478"></a><span class="in">```{r}</span></span>
<span id="cb36-479"><a href="#cb36-479"></a><span class="co">#| eval: true</span></span>
<span id="cb36-480"><a href="#cb36-480"></a><span class="co">#| echo: false</span></span>
<span id="cb36-481"><a href="#cb36-481"></a><span class="co">#| output: false</span></span>
<span id="cb36-482"><a href="#cb36-482"></a>myfilter <span class="ot">&lt;-</span> <span class="cf">function</span>(.data, myexpresion){</span>
<span id="cb36-483"><a href="#cb36-483"></a>  .data <span class="sc">%&gt;%</span> </span>
<span id="cb36-484"><a href="#cb36-484"></a>    <span class="fu">filter</span>(<span class="fu">is.na</span>(evo_hist)) <span class="sc">%&gt;%</span> </span>
<span id="cb36-485"><a href="#cb36-485"></a>    <span class="fu">filter</span>( {{ myexpresion }} ) <span class="sc">%&gt;%</span> </span>
<span id="cb36-486"><a href="#cb36-486"></a>    <span class="fu">distinct</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb36-487"><a href="#cb36-487"></a>    <span class="fu">nrow</span>()</span>
<span id="cb36-488"><a href="#cb36-488"></a>}</span>
<span id="cb36-489"><a href="#cb36-489"></a></span>
<span id="cb36-490"><a href="#cb36-490"></a>thresh <span class="ot">&lt;-</span> <span class="fl">0.08</span></span>
<span id="cb36-491"><a href="#cb36-491"></a>totsamp <span class="ot">&lt;-</span> <span class="fu">nrow</span>(<span class="fu">distinct</span>(finaltable_exp_mstr, sample))</span>
<span id="cb36-492"><a href="#cb36-492"></a>contamsamp <span class="ot">&lt;-</span> <span class="fu">myfilter</span>(finaltable_exp_mstr, f_raw_targetsp <span class="sc">&gt;=</span> <span class="fl">0.03</span>)</span>
<span id="cb36-493"><a href="#cb36-493"></a>contamsamp_lo <span class="ot">&lt;-</span> <span class="fu">myfilter</span>(finaltable_exp_mstr, f_raw_targetsp <span class="sc">&gt;=</span> <span class="fl">0.03</span> <span class="sc">&amp;</span> f_raw_targetsp <span class="sc">&lt;</span> thresh)</span>
<span id="cb36-494"><a href="#cb36-494"></a>contamsamp_hi <span class="ot">&lt;-</span> <span class="fu">myfilter</span>(finaltable_exp_mstr, f_raw_targetsp <span class="sc">&gt;=</span> thresh)</span>
<span id="cb36-495"><a href="#cb36-495"></a><span class="in">```</span></span>
<span id="cb36-496"><a href="#cb36-496"></a></span>
<span id="cb36-497"><a href="#cb36-497"></a>There are <span class="in">`{r} contamsamp`</span> samples (<span class="in">`{r} round(contamsamp/totsamp*100)`</span>% of <span class="in">`{r} totsamp`</span> samples) that exceed a 3% crosstalk threshold. <span class="in">`{r} contamsamp_lo`</span> samples (<span class="in">`{r} round(contamsamp_lo/totsamp*100)`</span>% of <span class="in">`{r} totsamp`</span> samples) have contamination &lt;= <span class="in">`{r} thresh*100`</span>% and <span class="in">`{r} contamsamp_hi`</span> samples (<span class="in">`{r} round(contamsamp_hi/totsamp*100)`</span> % of <span class="in">`{r} totsamp`</span> samples) have &gt; <span class="in">`{r} thresh*100`</span>% of a species that should not be there. In most cases the species in highly contaminated samples is HAMBI_1977. I think these <span class="in">`{r} contamsamp_hi`</span> samples are probably at very high risk for legitimate contamination. It is likely that the rest of the <span class="in">`{r} contamsamp_lo`</span> samples with &lt;= <span class="in">`{r} thresh*100`</span>%  contamination are not contaminated but just crosstalk outliers.</span>
<span id="cb36-498"><a href="#cb36-498"></a></span>
<span id="cb36-499"><a href="#cb36-499"></a><span class="fu">### Pairs</span></span>
<span id="cb36-500"><a href="#cb36-500"></a></span>
<span id="cb36-501"><a href="#cb36-501"></a>Here taking a closer look at the pairs from the experiment that appear to be contaminated.</span>
<span id="cb36-502"><a href="#cb36-502"></a></span>
<span id="cb36-503"><a href="#cb36-503"></a>::: {#fig-09}</span>
<span id="cb36-506"><a href="#cb36-506"></a><span class="in">```{r}</span></span>
<span id="cb36-507"><a href="#cb36-507"></a><span class="co">#| fig.width: 8</span></span>
<span id="cb36-508"><a href="#cb36-508"></a><span class="co">#| fig.height: 5</span></span>
<span id="cb36-509"><a href="#cb36-509"></a>finaltable_exp_mstr <span class="sc">%&gt;%</span> </span>
<span id="cb36-510"><a href="#cb36-510"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(community_type, <span class="st">"experiment"</span>) <span class="sc">&amp;</span> n_species <span class="sc">==</span> <span class="dv">2</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb36-511"><a href="#cb36-511"></a>  <span class="fu">contaminated_barplot</span>(<span class="at">threshold =</span> <span class="fl">0.16</span>,  <span class="at">quartet =</span> <span class="cn">FALSE</span>, <span class="at">y=</span>f_raw_targetsp) <span class="sc">+</span></span>
<span id="cb36-512"><a href="#cb36-512"></a>  <span class="fu">ggtitle</span>(<span class="st">"Experiment pairs with &gt;= 15% contamination"</span>)</span>
<span id="cb36-513"><a href="#cb36-513"></a><span class="in">```</span></span>
<span id="cb36-514"><a href="#cb36-514"></a>Species composition of pairs experimental samples that contain &gt;= 15% of a species that shouldn’t be there (shown in grey to black). </span>
<span id="cb36-515"><a href="#cb36-515"></a>:::</span>
<span id="cb36-516"><a href="#cb36-516"></a></span>
<span id="cb36-517"><a href="#cb36-517"></a>Most of the comtaminated samples are in the 0 Streptomycin conditions. I think we should exclude samples where the contamination is very high (over ~50% of the sample) but those with 8% or less contaminant I think can be retained, and I will discard the contaminating sequences.</span>
<span id="cb36-518"><a href="#cb36-518"></a></span>
<span id="cb36-521"><a href="#cb36-521"></a><span class="in">```{r}</span></span>
<span id="cb36-522"><a href="#cb36-522"></a>contampairs <span class="ot">&lt;-</span> finaltable_exp_mstr <span class="sc">%&gt;%</span> </span>
<span id="cb36-523"><a href="#cb36-523"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(community_type, <span class="st">"experiment"</span>) <span class="sc">&amp;</span> n_species <span class="sc">==</span> <span class="dv">2</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb36-524"><a href="#cb36-524"></a>  <span class="fu">filter</span>(<span class="fu">is.na</span>(evo_hist)) <span class="sc">%&gt;%</span> </span>
<span id="cb36-525"><a href="#cb36-525"></a>  <span class="fu">filter</span>(f_raw_targetsp <span class="sc">&gt;</span> <span class="fl">0.16</span> ) <span class="sc">%&gt;%</span> </span>
<span id="cb36-526"><a href="#cb36-526"></a>  <span class="fu">pull</span>(sample)</span>
<span id="cb36-527"><a href="#cb36-527"></a><span class="in">```</span></span>
<span id="cb36-528"><a href="#cb36-528"></a></span>
<span id="cb36-529"><a href="#cb36-529"></a><span class="fu">### Trios</span></span>
<span id="cb36-530"><a href="#cb36-530"></a></span>
<span id="cb36-531"><a href="#cb36-531"></a>Here just taking a closer look at the trios from the experiment that appear to be contaminated.</span>
<span id="cb36-532"><a href="#cb36-532"></a></span>
<span id="cb36-533"><a href="#cb36-533"></a>::: {#fig-10}</span>
<span id="cb36-536"><a href="#cb36-536"></a><span class="in">```{r}</span></span>
<span id="cb36-537"><a href="#cb36-537"></a><span class="co">#| fig.width: 8</span></span>
<span id="cb36-538"><a href="#cb36-538"></a><span class="co">#| fig.height: 5</span></span>
<span id="cb36-539"><a href="#cb36-539"></a>finaltable_exp_mstr <span class="sc">%&gt;%</span> </span>
<span id="cb36-540"><a href="#cb36-540"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(community_type, <span class="st">"experiment"</span>) <span class="sc">&amp;</span> n_species <span class="sc">==</span> <span class="dv">3</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb36-541"><a href="#cb36-541"></a>  <span class="fu">contaminated_barplot</span>(<span class="at">threshold=</span><span class="fl">0.16</span>,  <span class="at">quartet =</span> <span class="cn">FALSE</span>, <span class="at">y=</span>f_raw_targetsp) <span class="sc">+</span></span>
<span id="cb36-542"><a href="#cb36-542"></a>  <span class="fu">ggtitle</span>(<span class="st">"Experiment trios with &gt;= 15% contamination"</span>)</span>
<span id="cb36-543"><a href="#cb36-543"></a><span class="in">```</span></span>
<span id="cb36-544"><a href="#cb36-544"></a>Species composition of experiment trio samples that contain &gt;= 15% of a species that shouldn’t be there (shown in grey to black). </span>
<span id="cb36-545"><a href="#cb36-545"></a>:::</span>
<span id="cb36-546"><a href="#cb36-546"></a></span>
<span id="cb36-547"><a href="#cb36-547"></a>Again, I think we should exclude samples where the contamination is very high (over ~50% of the sample) but those with around less than 15% contaminant I think can be retained, and I will discard the contaminating sequences.</span>
<span id="cb36-548"><a href="#cb36-548"></a></span>
<span id="cb36-551"><a href="#cb36-551"></a><span class="in">```{r}</span></span>
<span id="cb36-552"><a href="#cb36-552"></a>contamtrios <span class="ot">&lt;-</span> finaltable_exp_mstr <span class="sc">%&gt;%</span> </span>
<span id="cb36-553"><a href="#cb36-553"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(community_type, <span class="st">"experiment"</span>) <span class="sc">&amp;</span> n_species <span class="sc">==</span> <span class="dv">3</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb36-554"><a href="#cb36-554"></a>  <span class="fu">filter</span>(<span class="fu">is.na</span>(evo_hist)) <span class="sc">%&gt;%</span> </span>
<span id="cb36-555"><a href="#cb36-555"></a>  <span class="fu">filter</span>(f_raw_targetsp <span class="sc">&gt;</span> <span class="fl">0.16</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb36-556"><a href="#cb36-556"></a>  <span class="fu">pull</span>(sample)</span>
<span id="cb36-557"><a href="#cb36-557"></a><span class="in">```</span></span>
<span id="cb36-558"><a href="#cb36-558"></a></span>
<span id="cb36-559"><a href="#cb36-559"></a><span class="fu"># Export</span></span>
<span id="cb36-560"><a href="#cb36-560"></a></span>
<span id="cb36-563"><a href="#cb36-563"></a><span class="in">```{r}</span></span>
<span id="cb36-564"><a href="#cb36-564"></a>finaltable_exp_mstr_export <span class="ot">&lt;-</span> finaltable_exp_mstr <span class="sc">%&gt;%</span> </span>
<span id="cb36-565"><a href="#cb36-565"></a>  <span class="co"># first remove samples with low read count (&lt; 3000 reads)</span></span>
<span id="cb36-566"><a href="#cb36-566"></a>  <span class="fu">filter</span>(sample <span class="sc">%nin%</span> lowread_samps) <span class="sc">%&gt;%</span> </span>
<span id="cb36-567"><a href="#cb36-567"></a>  <span class="co"># next remove masterplate pair samples that were highly contaminated</span></span>
<span id="cb36-568"><a href="#cb36-568"></a>  <span class="fu">filter</span>(sample <span class="sc">%nin%</span> contampairs) <span class="sc">%&gt;%</span> </span>
<span id="cb36-569"><a href="#cb36-569"></a>  <span class="co"># next remove trio samples that were highly contaminated</span></span>
<span id="cb36-570"><a href="#cb36-570"></a>  <span class="fu">filter</span>(sample <span class="sc">%nin%</span> contamtrios) <span class="sc">%&gt;%</span> </span>
<span id="cb36-571"><a href="#cb36-571"></a>  <span class="co"># exclude any remaining counts from species that shouldnt be there using again</span></span>
<span id="cb36-572"><a href="#cb36-572"></a>  <span class="co"># the fact that evo_hist should be NA for these species</span></span>
<span id="cb36-573"><a href="#cb36-573"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(evo_hist)) <span class="sc">%&gt;%</span> </span>
<span id="cb36-574"><a href="#cb36-574"></a>  <span class="co"># because we set 3% as our limit of detection we set read counts of species</span></span>
<span id="cb36-575"><a href="#cb36-575"></a>  <span class="co"># less than 1% to 0</span></span>
<span id="cb36-576"><a href="#cb36-576"></a>  <span class="fu">mutate</span>(<span class="at">count_correct_thresh =</span> <span class="fu">if_else</span>(f_raw_targetsp <span class="sc">&lt;=</span> <span class="fl">0.03</span>, <span class="dv">0</span>, count_correct)) <span class="sc">%&gt;%</span> </span>
<span id="cb36-577"><a href="#cb36-577"></a>  <span class="co"># now calculate a new relative abundance based only on the species that should</span></span>
<span id="cb36-578"><a href="#cb36-578"></a>  <span class="co"># be present and that are &gt; 3% relative abundance</span></span>
<span id="cb36-579"><a href="#cb36-579"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb36-580"><a href="#cb36-580"></a>  <span class="fu">mutate</span>(<span class="at">f_thresh =</span> count_correct_thresh<span class="sc">/</span><span class="fu">sum</span>(count_correct_thresh)) <span class="sc">%&gt;%</span> </span>
<span id="cb36-581"><a href="#cb36-581"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb36-582"><a href="#cb36-582"></a>  <span class="co"># for marking which sequencing batch these came from </span></span>
<span id="cb36-583"><a href="#cb36-583"></a>  <span class="fu">mutate</span>(<span class="at">batch =</span> <span class="st">"run20240711"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb36-584"><a href="#cb36-584"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(sample, strainID, evo_hist, count_correct_thresh, f_thresh,</span>
<span id="cb36-585"><a href="#cb36-585"></a>                <span class="at">target_f_masterplate =</span> target_f, replicate, strep_conc, </span>
<span id="cb36-586"><a href="#cb36-586"></a>                transfers, n_species, community_type, plate_well, batch)</span>
<span id="cb36-587"><a href="#cb36-587"></a><span class="in">```</span></span>
<span id="cb36-588"><a href="#cb36-588"></a></span>
<span id="cb36-589"><a href="#cb36-589"></a><span class="fu">## pos/neg control samples</span></span>
<span id="cb36-590"><a href="#cb36-590"></a></span>
<span id="cb36-591"><a href="#cb36-591"></a>Separate out the pos/neg control samples</span>
<span id="cb36-592"><a href="#cb36-592"></a></span>
<span id="cb36-595"><a href="#cb36-595"></a><span class="in">```{r}</span></span>
<span id="cb36-596"><a href="#cb36-596"></a>finaltable <span class="sc">%&gt;%</span> </span>
<span id="cb36-597"><a href="#cb36-597"></a>  <span class="fu">filter</span>(<span class="fu">str_detect</span>(community_type, <span class="st">"^pos|^neg"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb36-598"><a href="#cb36-598"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>transfers, <span class="sc">-</span>strep_conc, <span class="sc">-</span>evo_hist, <span class="sc">-</span>target_f, <span class="sc">-</span>count) <span class="sc">%&gt;%</span> </span>
<span id="cb36-599"><a href="#cb36-599"></a>  <span class="fu">mutate</span>(<span class="at">batch =</span> <span class="st">"run20240711"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb36-600"><a href="#cb36-600"></a>  <span class="fu">write_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(data, <span class="st">"pos_neg_ctrl_counts.tsv"</span>))</span>
<span id="cb36-601"><a href="#cb36-601"></a><span class="in">```</span></span>
<span id="cb36-602"><a href="#cb36-602"></a></span>
<span id="cb36-603"><a href="#cb36-603"></a><span class="fu">## Pairs</span></span>
<span id="cb36-604"><a href="#cb36-604"></a></span>
<span id="cb36-605"><a href="#cb36-605"></a>Write the pair samples</span>
<span id="cb36-606"><a href="#cb36-606"></a></span>
<span id="cb36-609"><a href="#cb36-609"></a><span class="in">```{r}</span></span>
<span id="cb36-610"><a href="#cb36-610"></a>finaltable_exp_mstr_export <span class="sc">%&gt;%</span> </span>
<span id="cb36-611"><a href="#cb36-611"></a>  <span class="fu">filter</span>(n_species <span class="sc">==</span> <span class="dv">2</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb36-612"><a href="#cb36-612"></a>  <span class="fu">arrange</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb36-613"><a href="#cb36-613"></a>  <span class="fu">write_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(data, <span class="st">"pairs_counts.tsv"</span>))</span>
<span id="cb36-614"><a href="#cb36-614"></a><span class="in">```</span></span>
<span id="cb36-615"><a href="#cb36-615"></a></span>
<span id="cb36-616"><a href="#cb36-616"></a><span class="fu">## Trios</span></span>
<span id="cb36-617"><a href="#cb36-617"></a></span>
<span id="cb36-618"><a href="#cb36-618"></a>Write the trio samples</span>
<span id="cb36-619"><a href="#cb36-619"></a></span>
<span id="cb36-622"><a href="#cb36-622"></a><span class="in">```{r}</span></span>
<span id="cb36-623"><a href="#cb36-623"></a>finaltable_exp_mstr_export <span class="sc">%&gt;%</span> </span>
<span id="cb36-624"><a href="#cb36-624"></a>  <span class="fu">filter</span>(n_species <span class="sc">==</span> <span class="dv">3</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb36-625"><a href="#cb36-625"></a>  <span class="fu">arrange</span>(sample) <span class="sc">%&gt;%</span> </span>
<span id="cb36-626"><a href="#cb36-626"></a>  <span class="fu">write_tsv</span>(here<span class="sc">::</span><span class="fu">here</span>(data, <span class="st">"trios_counts.tsv"</span>))</span>
<span id="cb36-627"><a href="#cb36-627"></a><span class="in">```</span></span>
</code><button title="Copy to Clipboard" class="code-copy-button" data-in-quarto-modal=""><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->




</body></html>